{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1e8a1726-9e8e-4087-9e48-1b7f69665b5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Standard Library\n",
    "import os\n",
    "import time\n",
    "import re\n",
    "import json\n",
    "import gzip\n",
    "import shutil\n",
    "import tarfile\n",
    "import subprocess\n",
    "from pathlib import Path\n",
    "from typing import TypedDict, List, Dict, Any\n",
    "from urllib.parse import urljoin, urlparse, parse_qs, unquote\n",
    "import gzip\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import json\n",
    "import re\n",
    "from collections import defaultdict\n",
    "from pathlib import Path\n",
    "\n",
    "# Third-Party Libraries\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "from openai import OpenAI\n",
    "from IPython.display import display, Markdown\n",
    "from geofetch import Geofetcher\n",
    "\n",
    "# Bioinformatics\n",
    "from pydeseq2.dds import DeseqDataSet\n",
    "from pydeseq2.ds import DeseqStats\n",
    "from pydeseq2.default_inference import DefaultInference\n",
    "import gseapy as gp\n",
    "from geofetch import Geofetcher\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "from matplotlib.patches import Patch\n",
    "import scanpy as sc\n",
    "from sanbomics.plots import volcano\n",
    "\n",
    "# LangChain/LangGraph\n",
    "from langgraph.graph import StateGraph\n",
    "from langchain_core.runnables import RunnableLambda"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "57ede832-246b-46bd-bc0d-ef48324e1c71",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAKkAAAITCAIAAADzeatCAAAAAXNSR0IArs4c6QAAIABJREFUeJztnWdcFEcfgGevN47epQuCgoCg2AHBrlGsqIC9a6wxllSjiYk9xhJLLDEGjQVbVFREBWxIERRFKSq9H1yv74fLeyEIiHrHHM48Pz7szs7O/u8eZnZ2b3aHUKlUAIMkJNgBYKCB3aMLdo8u2D26YPfogt2jCwV2AM0hk6jKCyWCWrmwVq6QA5lECTuit0Nnkig0gm1IYXMpFnZ02OE0B6GH1/digfLZw9q8TEFVidTQgsrmUthcCteEIhW3Afc0BrmqVCKoVVBpxMssobMn29mL4+zFhh1XI+id+6QLlcV5Ygs7ulMndjtXJuxwPgipWJmbISh4Lix4Ieo13MzVlwM7ov+gR+6fJtddO1baY6ipX4gx7Fi0TF21POlChYivHBBhyTIgww7nH/TF/e2YChIJ9PrEDHYgOqSqVBazq6D/REu7DizYsQB9cR9/stzInOoTaAQ7kNbg7J7CnsPNzG3hdwPhuz+/r8jBnd25jyHcMFqTmN1FHQMM3LoYwA0D8vV90oVKG2cmUuIBACPn2iRfq64slsINA6b7F+kCAMDH17NrCRNX2N88XQ63zYXp/uapMt8gJM7xjeLciZ14rgJiANDcp92s6eBnwOToywVP6+MTZPQ0uU7EV8AKAJr7/MeCXsM+5iu6lhA4yjz9Zg2so8Nx/zJLSKYQROvW+ZUrV549e/Y9duzfv39hYaEOIgL2HVgZSTxdlNwS4LjPeyxw8mztW9xPnjx5j72Ki4urq6t1EA4AANBZJBNrWlGuWEflNw+c6/szOwsHRVkxdXN3MzEx8ciRI48fPzYzM/P29l64cKGZmZm/v796K4fDiY+P5/P5R48evXPnTk5OjpmZWWBg4Ny5cxkMBgAgJCRkxowZcXFxqampmzdvXrZsmXrHwMDAzZs3az3azDu1Qp682yATrZf8dlStjkyi3L3ihY4Kz8rK8vPz27dvX3FxcWJiYnh4+Pz581UqlVgs9vPzi4mJUWfbt29fQEDA1atXHzx4EBcXN3jw4O3bt6s3DRw4cOzYsRs3brx7965MJrt9+7afn19BQYGOAs7N5J/fX6SjwpsHwu/3glo521BXx01LS2MwGNOmTSORSFZWVh07dnzx4sWb2SIiIkJCQpycnNSr6enpSUlJn376KQCAIAhDQ8Ply5frKMIGsLkUAU/eOsdqABT3ChZXV908Hx8fsVi8ePHigICAvn372tnZaVr7+lCp1Dt37nz99dfZ2dlyuRwAYGLyb6vbsWNHHYX3JmwuRVgLxz2Evp5KCegMXbl3d3f/+eefzc3Nd+zYERYWNm/evPT09Dez7dixY+/evWFhYTExMcnJyVOnTq2/lUaj6Si8NyFTAJUOp8cN4agsLplXocNb2T179vzyyy/Pnz//zTff8Hi8xYsXq2u2BpVKderUqfHjx4eFhVlZWQEA6urqdBdP8/B5CgqVgHJoCO7ZXLKgVlc3sx4+fJiUlAQAMDc3HzZs2LJly+rq6oqLi+vnkclkIpHIwsJCvSqVSm/duqWjeN6KoFbO4sIZNQnBPY1BsrRnyCU6ubZMT09fsWLF6dOnq6urMzMzo6Ojzc3Nra2t6XS6hYXF3bt3k5OTSSSSo6PjuXPnCgoKampq1q5d6+PjU1tbKxAI3izQ0dERAHD16tXMzExdBCwRKS3sGLoo+a3AOdMwOeScTL4uSo6IiAgLC9u0aVP//v1nzZrFZrP37t1LoVAAANOmTXvw4MGyZctEItH333/PYDDGjBkzcuTIbt26LViwgMFghIaGFhUVNSiwXbt2w4cP37Nnz44dO3QR8LPkWisHOO7h3Nt5nsrPzeAPjLJq/UPrFSol2PXZi/mb20M5Opx679SJDfH3K/3h1TOhVy9ov2LD6WVQaISFPePh9eqmBm6oVKrg4OBGNykUChKJRBCN941jYmKMjHTybaalpS1evLjRTVKplEqlNhqSu7v7nj17mioz8XzFwEhojR/M8Xq/LH2xYEuTzd2bp96WYGNj82FBNUdTIfH5fA6n8bH3VCrV3Ny80U3PHta9eirsP8lSqzG+AzDdZyTy5BKlbz8Ux2wBAC4eKA4aY8E2hDZ6BeaYLa9ehqWvJS/SdNLh13MuHijuGMCFKB7+ON1Bk63uXqosey2BG0Yrc+OvMgt7euuPYGgA/PH5QAVO7ijoPti0rT9910Juniq3dmJAH5wPv94DAAABxnzaLvla1eM7tbBD0S0qFYjZXWhoRtUH8fpR7//PvUtVORn8nsPMHDvqxeNq2iX5avWTe7XB4yzs3PSledMj9wCAymJp0oUKOpNs257p1ImtP4+svjdlryWvs4XJ16q8+xgFDDIl9KCd1aBf7tUU5YqfJdfmPRYYmlFNLGksLoXNJXMMqXJ5G3j3AplM8Cplglo5AMSz5FqOEcXZi+PdxxDWj/TNoI/uNZS9lvz/nSsKggDCOm3eBhaJRM+fP+/cubMWywQAcIzIABBsQ7KBMdXGmcnW2QilD0ev3euUnJyc1atXHz9+HHYg0NC7hgjTamD36ILdowt2jy7YPbpg9+iC3aMLdo8u2D26YPfogt2jC3aPLtg9umD36ILdowt2jy7YPbpg9+iC3aMLdo8u2D26YPfogt2jC7ruCYJo6o0YiICue5VKVV5eDjsKmKDrHoPdowt2jy7YPbpg9+iC3aMLdo8u2D26YPfogt2jC3aPLtg9umD36ILdowt2jy7IvVtx/PjxYrFYpVLJZLLKykpra2uVSiWRSK5cuQI7tNYGuXofFhZWXFxcVFRUXl6uVCoLCwuLioq4XC7suCCAnPvw8HA7O7v6KQRB9O7dG15E0EDOPQBg7NixdDpds2pvbz9+/HioEcEBRffh4eG2trbqZYIggoOD1bNhowaK7gEAEyZMUFd9e3v7MWPGwA4HDoi6DwsLs7W1JQgiKCgIzUoPbU7UBlSXSqvLZIrWnRZjRMjs69evB3Qa8Ty1rjWPS2dSzGxp+jAdDOTr+5dPhSlxNQKerJ0rW1grhxhJq0GmkgqeC2ycmQMmWZKpjc/q2zrAdF+QLUr6u3JARDsyFVYI0CjNFz+4Wj56gS2NAe20C+3AZa8lt89WDJ6KongAgKUjo+8oq+ObX0OMAZr7lLjq7sOgTQGtD3BNqY6dDCDOCAnNfX6WwMgMySpfD6YBuaxADOvocNyL+EpDUxqFBrOnow9wTWkSEbT+Fhz3BAEEPBmUQ+sVSoVKItTmpH/vBKL3djDYPdJg9+iC3aMLdo8u2D26YPfogt2jC3aPLtg9umD36NKW3AeH+B/78xDsKD4e2pJ7KISN7l9UXAg7Cp2A3TdHSUlxTU017Ch0hV6M030PRo4KnTplDo9Xc/jIXiaT2dW/x4L5y01NzQAAT55kbNu+oaDwlZeXb1TEjD17tzs7tV+yeBUAoKqqctfuLZmP08VicdeuPaIiZtjZOajfq3zq9J9Xrlx4XfDSwd7J37/7tKlzH2WkLl02BwAwKWJEr16B69ZuBgAc+X3/ldgLFRVlFhZWPt5+SxavIpFIAIARYSFRETNuJcQ9epR65VISjUaD/Q29nbZa76lU6vHjR0gkUsyZ64cPnsrITDt0+FcAgFgsXv3FEmNjk9/2n5g+bd7O3VvKy0sJggAAKBSKJctmp6U/XLJ49W/7jxsbmcybP7mwqAAAcPp09NE/fhszemL0sQvDh4+++HdM9PEjvj7+P6zfBgD44+hZtfiDh/bEnD0xd/bik39dmT5tXvzNq3+d/EMTz4W/z7Rv32HjTzsplLZRo9qqewCAra1dxKRpBhwDU1Ozrv49srOzAAB37yXweDWzZy2ysrJ2c3WfOWNBaWmJOn9GRtqrV/mrV30X0K2niYnp3DmLuYZGp04dAwCkP0rp0KHjwIHDjIyMhw0N2/nLoYBuvRocro5f92f04ciIGb17BxlwDIICQ8NGjj/6xwGZTKZ+tovLNVw4f7m/X4C6JdB/2kaUjeLm5qFZNjDgCgR8AEBe3gsOh+Ps3F6d7uvjb2Dwz/PVGZlpVCq1i29X9SpBED7efumPUgAAnp7eDx/e+2nj2stXzvNqebY27dq3d2twuNevX8pkMg8Pz/oB8Pn8wsJ/xtp2cOuo40+sZdpG69Qo6pa8AXX8OhaLXT/FyMhYvcDn18lksuAQ/ze3jhk9kcViJybd/PGnbykUSlBQ/9kzPzUz+8+sGlVVFQAABp2hSWEyWQAAkUioXm0T5/j6tGH3jcKgM6RSaf2Uysp/JscwNTVjMpnr122tv5VMIgMASCTSsKFhw4aG5efnpqTcP3Rkr0DA//6/OdlsDgBAJBZpUoRCAQDAxMRMx59JV3xs7m1t7WpqqquqKk1MTAEAqWnJQuE/9dLFxU0kEllYWNnatFOnFBUXGhkaAwCuXLng5ubh5OTi6Ojs6Ohcx6+7+PeZBiW7uLiRyeTHj9M93DupU7KyMg04BubmFq37EbVGGz7fN0r3gN5kMnnHLxsFAkFB4evff9+vcePXpVu3bj03bfqutLSEx6uJOfvXnLmRly+fAwBcj7v81TefJSXd4tXy7t5NuJ0Q59nJGwBgZ+8IAIiPv/okK5NrwO0fOuToH78lJd2qrauNjb14Jub4mDGT2krP7k0+tnpvamq2ZPGqA7/tGj12gKur++SoWTt+2Uih/PMQyA/rt507f2rtulVPnmTY2TmEhg4eNSocALBs6Re/7Ny05sulAAATE9NhQ8PGjokAANjatBs0cPjBQ3s8O3lv3fLr/HnLSCTSd+tXy+VyG5t2EydMnRA+GfYnfn/gPIspFiiP/pA//jNnXRReWFRgYMDlGnDVN22GfRI4bcrc0aMn6OJYH0jhC+GzBzUj5thAOfrHVu95vJp58ye3d3GbPn2+sbHJgQM7SQQpKKg/7Lj0kbZ6rmoKQ0OjDd9vV6lUX329fPbsSXV1tTt/OaS+14tpwMdW7wEAHh6eWzbvgR1FG+Bjq/eYloPdowt2jy7YPbpg9+iC3aMLdo8u2D26YPfogt2jCxz3JAowsqS3IOPHD9cE2ksG4bin0UmiOnltJeqvWSsvELO40F6oDa3N79CFW5InhHV0PYFXIXXqyG5BRp0AzX23QcY5GXWvn6GrPyGm1NaFYWEP7dwH8x3qKiU4se21vbsB25BiakVXKpGYqU8hV1UUigtzhE6dWJ17G0KMBP7ciJlJtYUvhEolqC6TtiC71lAoFHw+39Cwtb99QzMam0vu4Gdg7cRoQXYdAt89LHJyclavXn38+HHYgUADX9+jC3aPLtg9umD36ILdowt2jy7YPbpg9+iC3aMLdo8u2D26YPfogt2jC3aPLtg9umD36ILdowt2jy7YPbpg9+iC3aMLdo8u2D26oOueIAgHBwfYUcAEXfcqlerly5ewo4AJuu4x2D26YPfogt2jC3aPLtg9umD36ILdowt2jy7YPbpg9+iC3aMLdo8u2D26YPfogty7FaOiosrLywmCkMlkNTU1ZmZmBEHI5fLY2FjYobU2yNX7oKCg6urqsrKy6upqlUpVXl5eVlZGJkN7kTlEkHM/ZswYOzu7+ikqlSogIABeRNBAzj2Xyx02bBiF8u8k0JaWlhMnToQaFByQcw8ACAsLq1/1u3bt6ubmBjUiOKDonsvlDhkyhEqlqit9ZGQk7IjggKJ7AMCoUaPUVb9Lly7t27eHHQ4cKC3I01qogFSi5PMUoDUuO5n9A8MuCS+FDY2sKmmVORsIwsQS2pRYjaIv1/fPkuseJfKqSqQmlnSpWAE7HO1jZEnLy+C7+hr0Gm7KMdKLKqcX7h8l8F49E3UdaM4y+Kivs1WgqlR67Y+icUvbGeiBfvju02/zCnPEfcIs4YbRmvz5Y+7kLxzpLMidLciHl4iUeZkCpMQDAILH2SReqIQdBWz3lUUSmVQJN4bWh2tKyX/Chx0FbPc1FXJLBybcGFofFpdiaEqTiSH/00N2r1QoJQLk6j0AoKJQDAgCbgyI3tvBYPdIg92jC3aPLtg9umD36ILdowt2jy7YPbpg9+iC3aNLG3a/9rtVwSH+Z8+d/PCiTp2ODh2A3BD9tuqez+cnJt20t3e8dv0S3EjOxJz44cev4cbwfrRV9/E3r7JY7EWffp6ZmV5YVAAxkmfPnkA8+ocAf9TY+3H5yvlePQN9vP3MzS1iYy9MnTJHnZ6XlzNtxvhdOw8fO3YwITHe3NwiOGjArJkL1U/c3blzO+7GlUcZqbW1PA93z8jIGb4+/vWLXbRkJp1G/+nHXzQpX361vLKqYtcvh169yj94aE9a+kOVStWpU+fwcVFeXj6Ll85KT08BAMTGXrx65W79x330nzZZ7wuLCh4/fjSg/1ASidQ/dMjfl85qNqmfuNi8ZV1IyKDYy3fWrFp34q+jN+KvAgDEYvH6H76QSCQrP//2+/Xb7O0d13yxpKrqP2Onhgwa8TDlviZRLBbfvZcwoP9QqVS6eOksMpn844YdmzfuppApa75YIhaLt23Z6+HhOWDA0BvXk9uW+Lbq/uLFM9ZWNp07+wIAhg4Nq6goT0t7WD9DYN/QoMBQKpXq7d3Fxto2OzsLAMBgMPbvjV62dI2vj7+vj/+c2YtFIlFGZlr9HYODB7BYrLgbV9SrCYnxAIB+/Qa+fv2yurpq9KgJbq7uLi6uX3+14dtvN8rl8tb93Fqmjf2rqh+bjb168ZPhY9SrNta2np7eV2Iv+Pj4afK4uXloljkcAz6/Tr0sFAr2H/glLf1hZWWFOqWmprp+4TQaLTRk8LVrl8aMnggAuH07rlfPQK4Bl06jGxkZb/jpm/6hQ3y8/Tw9vRucLNoiba/e37ufVFlZcfDQnuAQf/VfZmb6zVvXJBKJJg+J1MjnKi0tWbRkhkwm+3LN97GX71y9crfR8ocNHfUsO6uwqEAsFt+7n9g/dAgAgE6nb9+6r3tA75Onji1cNH1S5MirV//W5adsDdpevb927W93906zZi7UpEil0lWrF91OuBEaMqiZHeNvXpVKpSs//5bJZL5Z4zW4uLh6eHheunTW1dWdyWQFBPRSp9vbO86ds3jqlDkpKfcvXT73/YavHByd3Vzdtf35Wo825l4sFt9OuDF71qIGTa6/X0Bs7IXm3dfW8gwMuGrxAICbt643lXPI4BHRx48UFLwKDRms7sG9epX/+MmjwYM+YTAYPXv2DQjoNWhIr+zsrDbtvo21+bduXZdKpYF9QxqkBwaGPky5X11d1cy+zs6ulZUV586fksvl9+4npaTcNzQ0KisreTNnv+CBlZXl9+4nDhk8Qp1SW8v7aePa3Xu2FRS+fv365R/HDsrlcs9O3gAAW1u7rKzMlNQHSmUbG3DcxtzfuHnVx9vP1NSsQXpQYH8AwJXYC83sG9JvYGTE9CO/7+s/sPupU8c+Xbiif+iQY38e2rL1+wY5WSyWn1+AvZ2jk5OLOsXT03vpktXXrl+KjAqLmjI6IyN1y+Y9jo7OAIDhQ0cRBPHZivkKRRt7hBTy83iZSbziPGn3YeYQY3gTqVQ6dvzgWTMXDh0yUkeHOPZDzrRvnal0mEP029j5XteUlBQXFr0+fSbawcFJ0+B/rLSxNl/XXI+7vPyzeVVVlWtWrSNgPzeja3C9/w+TJk6dNHEq7ChaCVzv0QW7RxfsHl2we3TB7tEFu0cX7B5dsHt0we7RBbtHF8juKVQSnY3i/5+FPQPA/rkA8vdubEkryhHCjaH1qauS1VXJqTS037FmaUen0UmwX+nb2lSXSl28OLCjgO0eEMA32OjyIZgPVbUy/Bp5QkxprxGmsAOBPW5HTUm+5Fp0adcB5oZmVLbhR/uzclWJtLZCevfvsmlrnfVhUja9cK/+Xh7GVRc8FykVKhG/jQ18awnWziypSO7kyek+2AR2LP+gL+7/RQVapwOck5OzevXq48ePt8bB9BLY5/s3gX3lgw765x7TWmD36ILdowt2jy7YPbpg9+iC3aMLdo8u2D26YPfogt2jC3aPLtg9umD36ILdowt2jy7YPbpg9+iC3aMLdo8u2D26YPfogq57Eonk7OwMOwqYoOteqVTm5ubCjgIm6LrHYPfogt2jC3aPLtg9umD36ILdowt2jy7YPbpg9+iC3aMLdo8u2D26YPfogt2ji/69W1HHzJo1i8fjEQQhFArLy8vt7e0JghCLxTExMbBDa20+2pfXNoWrq2t0dLRmrtucnBz1OA7YcUEAuTZ/0qRJNjY29VOUSmWPHj3gRQQN5Nzb2NgEBQXVTzEyMpo2bRq8iKCBnHsAQHh4uK2trWa1Y8eOfn5+UCOCA4rubW1te/furV42NTVFs9Ij6h4AMHHixHbt2gEAPDw8unTpAjscOLS0n6/6uDrCNta2PXv0unz5cmRE1Ef20YAKEC2blOMt1/cFz0Wp8TVlr8ViwUc4l8VHiaUDU8SXO3qwe48wI5pt1ptz/zyVn57A8+5jYmxJp7MQPTu0RWorZbxK2Y3oounfOTOaFtek+0cJvJdZoqBxVroMEqNbjm3InfKVI53ZuP7GU/k8ef4TIRbf1hkQYXPrTEVTWxt3X/pSQuB5a9o+xlb056l1TW1t3H1tpczSkaXLqDCtAZlC2LuzqstkjW5t/BpPKlbKGs+PaWPUlElBE1063HtHF+weXbB7dMHu0QW7RxfsHl2we3TB7tEFu0cX7B5dsHt00fKzGXfvJsTFx7548ayoqMDKysbL02fsmEn29o6aDEKh8NTpP+/eS8jLe0Gj0e3tHQP7hoaNHEcikQAAX3y1LDHx5pvF9u3T79tvfgIADB8RNGDAsIXzlzfIkJv7YvrM8O1b93Xu7NvCUIePCOLz+W+mL5i/fPSo8Hf83G0SrbmXSqVr161KTLw5csTY8WMj2RxOenrKnbu3b8THrlm1rkePPupsX329PP9l7qwZC80tLAEA9+8n/bJzU17ei+XLvlBnsLVpt+z/yxoMuUbNH93IyDgqcoaFRXMDDvLyclatWRR97IImpW+ffiNHjmuQzdbG7p0K+UDOxJx4+uzxqs+/1VaBLUdr7qOPH0lMvLlm9brQkEHqlN69gqZPmzdnXuT+33aq3RcUvn6Ycv+H77d3D+ilzuPr489ksi5fPicQCNhsNgCAwWT6+vi/69FNTEynTpnTfJ5n2U8apJiZW7zrsd4s5AN59kzLBbYcrbm/eeual5ePRrwaBoOxeeNuIyNj9SqvphoA0OAnxajIGVGRMz7w6PXb/Dp+3cFDe+7dTaiuqerg1jE0dPDQISMPHtpz5Pf9AIDgEP95c5eMHTOpmdIKiwqmThs7Z9aiUaPCAQACgWBS5Ih+/QYacAzqF+LXJWD6zPAf1m/btGWdkZHx/r1/5uXlnDt/MiX1QUlJkaOD85AhI0d8MkZdpkKh+OvkH4eP7AUAdPTwmjJ5tpeXz+Kls9LTUwAAsbEXf91z1M3V/dWr/G3bN2Q/zyKTKY6OzlMmz1b/d379zQoymWxpaR19/Mh3327q3TuomfhbiHbc8/n83NwXM2cseHOTqamZZtnZ2ZXFYm3/+UehSOjduUv9TVrkp5++LS8vXbx4lYO9U8zZE1u3/eDo4Dx1yhypVHojPrYlzbWtTbvJUbMOHNzVr99AIyPjAwd3cdic2TM/pdPp9Qt5/folAODI0f3jx0V6evoAAHbu2lxSUrR06RqCIF69yt/+84+WltbqRm7vvh23bl1f++0mqURyO+HG56sW7tn1+7Yte+ctmGJn56Bu86urqxYsnNqzZ+Dy5V8qFYr9B3Z+t2710SMxLBaLSqW+yMkWCAXrv9vSqVNnrXxR2nFfWVkOADA3s2g+G5PJ/Hnbge83fPndutUAAEtLKx9v/6iomTbW/z4hlZPzPDikYTu8a+dhD/dOLQwm/VFK+Piorv7dAQCzZi4MDAxtqrtw+nT06dPR9VMYDMaliwkAgPDxUdeuX9r967aJ4VPOnTv58/YDdDq9we7qh3m7+nfXtCJffvmDUCiwtrJRn84uXz53/0FS94BevFreib+OLl60Uh1VQEAvoVBQWVVRvxcMAPjr5B80On35si8oFAoA4LPlX40ZN/Dsub8mhE8mCKKkpGjPrt8ZDEYLv4e3os1+fv0nmc+eO7lt+wbN6pbNe9Rtl4uL6949f6SkPkhLS87KyrydEHcl9sKggcM/X/G1OmejfT17O0fQYry8fE78dZTHq/Hu3KVr1x4d3DyayvlmX4/0/xHtZDL58xXfzJ0XlZWVOXbMpI4enk0V4uZar3yV6vTp6Hv3E9VNAgDA2toWAJCflwMAcP//vy+FQln77cY3i8rNe+Hq6q4WDwBgs9l27Ryys7PUqw72TloUrzX35uaWAIDSshJNSs8efdX/1JWVFeu//49LMpnc1b+7ugbw+fyduzZfvnJ++PDR6u/3/fp69fl8xTfnzp2Mu3HlxF9HOWxOWNj4qMiZmi+0Ps339dw7dOzq3/1B8t2ePfo2czja/9sDpVK5cvUimUw6c8YCHx9/A47BwkXT1Zv4/DoAAIP+FnNVlRW2tv+5ymAwmUKRsMGBtIV27u2wWKz2Lm5JSf9empubW/j6+Pv6+Hfs6KVJFIlEmgqhhsPhzJ71KQBA89/94XANuBGTph3YF/3ztv2DB4/4/eiB02eiW7BfQzIy0h5lpPbs2XfbzxsUirc/lpT9/OnTp4/nzlnSp3ewAcdAoxwAwGZzAABCoaD5Elhstlgirp8iEgpNTXTSK9Lmfb3Royc8y846e+5kg/Ti4kLN8m8Hdy9cNL2kpPg/GUqK1BdpWgmDV8s7fea4WCwmCMLLy2fe3CW+Pv7Zz5++azkSieTHn76JjJjx+YpvykpL/ow+/PZD82rqd3ry83Pz8/+Zk6V9+w4UCiX9UYp6VaVSrVy96MqVhr3ODm4ds7IyZf8fJltbV/vyVZ6Tk8u7Bt9CtHa+HzRweE7O823bN2RnZwU5JXupAAATc0lEQVQF9adQKCKh8PKV83fvJfTt08/D3RMAMHbMpBvxsStWLpg6ZY76wi8398XhI3s9Pb017apYJEpNS36zfE3jXFFe1iBDR49/mxYKmXL4yN77D5IiJk6zsWmXnHz3+YunMwIXAADatbOvrKxISIh3cHCys3NotCgAAIdj4Nq+w979O0hk8vhxkRQKZdasT3/e8VO/fgNtrG3rF0L89xkGRwdnCoVy/MTvs2cvqqmu2vHLxq7+3UtKi9XNW//QIWfP/mVoaGRlZXP7dtzDh/fmzVkCALC1tcvKykxJfeDk6DJ8+OjTZ6I3b1k/dcockUj4676fGXTGkMEjteWoAY0/k3XvUpVMBrwDTd61uMTEmzdvX8/OziouLrSzczA2MgkbOb5nz3/Pl8UlRTExJ1JTH7wueCkWi21t7Xr1DJwyeTaTyWzmni6JRLp+9X5TN2L/OHpWLBJpru/T01N27NyYk/McAODk5DJ61ITBgz4hkUjqnkdqWvLkqFlTJs9q6p5uF9+u06fPn79gys/b9nt5+agT58yNpDMY27fuq19IaMigyMmjfvrxF3XfBQAQf/Pa4SN78/NzbW3t1qz6rrKq4suvltvbOx4+eFIikWzbvuHqtb8VCkV7F7dpU+eq73c9epS6eev6goJXP27Y4e8XkJAY//vv+7OfPzU0NPLw8Jw1Y6G63q///ouS0uId2w+8q5Gzu14OnWZtbEl7c5OW3WP0jWbc49/x0AW7RxfsHl2we3TB7tEFu0cX7B5dsHt0we7RBbtHF+weXbB7dMHu0aXx3++pdFIL38eL0XOMLJoc6dV4vecYkSsKxY1uwrQt8h/zG/0Bt0n35u3oiE2f9XHCK5e6eHOa2tq4e2MLmrElNTm2yVexYtoEccdLug1scgBOc+9Qv3Oxqq5a0bmvMdsQuanU2jq8clnc8aLBU6zNbRtv8N8+d0JGIu9RAk9YJ2eyPra+nwoApVJJJn1sVzqGFvSXT/gu3pyAgSYmVk2Kb9m8mCoglSgFPLmWY4TNq1evtm7dunXrVtiBaBkVACZNdO4a0ILGnAA0BonGaFFxbYgqPiFSVDTVB0aBj63Fw7Qc7B5dsHt0we7RBbtHF+weXbB7dMHu0QW7RxfsHl2we3TB7tEFu0cX7B5dsHt0we7RBbtHF+weXbB7dMHu0QW7RxfsHl3QdU8QhIODA+woYIKue5VK9fLlyxZk/GhB1z0Gu0cX7B5dsHt0we7RBbtHF+weXbB7dMHu0QW7RxfsHl2we3TB7tEFu0cX7B5dsHt0acF7NT8uVqxYce3aNfWyegJ7lUqlUqlSU1Nhh9baIFfvp02bZm1tTSKRSCQSQRAEQahUKn9/f9hxQQA59+7u7n5+fvVTTExMJk2aBC8iaCDnHgAQGRlpZWWlWXVycgoKCoIaERxQdO/q6urr66teZrPZaFZ6RN0DACZOnGhubg4AcHZ2Dg4Ohh0OHBB17+Hh0blzZzqdHhERATsWaOjkGq/guSg7pU7EV9SUS7VeuLaQyeS1tTxTU1PYgTQJjUmm0ggrR2a3gSa6mN5D++4fxtUU54ttnFimNgzyxzbRSutCIgQ1srpq+b1LZRErHQzNqNotXsvu78dW15TJegy30GKZGADAhb2vB0ZaNj//zbuizaakOE9cUSjF4nVBv3CbW2e0PGWdNt3nZwkMzdCdfkansLhkfo28qkSb/SdtuhfxFWbtGFosEFMfWzdWZbG+uq+tlAOA1i9DrYlUrJRKlFosENHrewx2jzTYPbpg9+iC3aMLdo8u2D26YPfogt2jC3aPLtg9umD36EKBe/jhI4L4fL562dDQyNzMwt+/e1TkTCaTqU786+Qfu3ZvfXNHLtfw7JnrmtW7dxPi4mNfvHhWVFRgZWXj5ekzdswke3tH9dbs509nz/nPuDwOh+Ps7Dpm9MQ+vREdqAnfPQCgb59+I0eOU6lUpaXFL1/mXYm9kJr6YOPGXQYcA02edWs3s9js+ntRyP9ELpVK165blZh4c+SIsePHRrI5nPT0lDt3b9+Ij12zal2PHn00u0ydMsfLy0e9nJ+feyM+9quvP9vww88B3Xq21mfVL+C7NzO38PX595Go0aMmTJ8xfsOPX6//bosm0auzL9eA2+ju0cePJCbeXLN6XWjIIHVK715B06fNmzMvcv9vO+u7d3R01hzI18c/bOS4qdPHnTp1DFn3ene+Nze3mDp1blLSrfz83Jbkv3nrmpeXj0a8GgaDsXnj7r17/mh+X2en9sUlRerlwUN7Rx8/otn008a1mtPEyFGhZ8+dPPL7/pD+3YZ9Evjt2pWVlRUAgLy8nOAQ/6ynj7/8anlwiP+48CG792xTKBTqvaqqKtetXxM+cdjIUaHrf/jy9et/3tl96nT06LEDExLjQ/p3OxNz4l2+Gy2jd+4BAL16BgIA0h+lvDUnn8/PzX3RPaD3m5tMTc3IbxsmXFRUYGZq/tajUKnU48ePkEikmDPXDx88lZGZdujwr+p0AMDmLetCQgbFXr6zZtW6E38dvRF/FQCgUCiWLJudlv5wyeLVv+0/bmxkMm/+5MKiAgAAjUYTCgXnzp1ctXJt714wnwWD3+a/iYWFJUEQlZXlmpQRI/s1yDN71qfh46PUeczN3nl0aB2/7tDhX58+e/LZ8i9bkt/W1i5i0jQAAOAYdPXvkZ2dpdkU2Dc0KDAUAODt3cXG2jY7Oys0ZFBGRtqrV/mbN+3u4tsVADB3zuLEpJunTh37dOEKgiDEYnF4+GT1Jojoo/s3ebOvZ2PdTrOsVP47kunsuZPbtm/QrG7ZvEdzjv/6mxX1S7C0tJo3d8mQwSNaEoCbm4dm2cCAKxDwG93E4Rjw+XUAgIzMNCqVqrFLEISPt1/9lsy9Q6eWHFen6KP70tISlUplVq82N9XXMze3BACUlpVoUnr26Ku+tKusrFj//Rf1M2v6+QI+/5u1nw8eNGLsmJY+hal+S0OjkBp7ZIbPr5PJZMEh/3ms38jIWLNMo8Ef0KyP7i/+fQYAENCt11tzslis9i5uSUk3oyJnqFPMzS3MzS0AAEXFhQ0y1+/nTwif/Mex30JDB9vatHujVAAAUCgVH/IRTE3NmEzm+nX/uTNBJunXY0p619d7kpX5Z/ThgQOGWVpatSA7GD16wrPsrLPnTjZIL37DfX0iI2YYG5ts2vSdJoVGo4tEQs2qplv+fri4uIlEIgsLK18ff/WfpaV1+/YdPqRMrQO/3leUl6WmJauX09NT/jj2m4W55ZzZi+rnyXiU2uB8DwBo376DAcdg0MDhOTnPt23fkJ2dFRTUn0KhiITCy1fO372X0LdPPw93z0YPSqPRFsxf/tXXn126fG7woE8AAB07et28dX3smAgOh/P70QMVFWXGRibv/aH8unTr1q3npk3frfz8WwaDcSP+6m8Hd0+JmjVqVPh7l6l14Lu/dTvu1u04AIABx6CTp/eM6fP7hw6pf2oEAHzx1bI3d9y+dV/nzr4AgPnzlvp4+928fX3nrs3FxYV2dg7GRibffPVjz559mzlun97BXXy77vl1e69eQVwD7oL5yzdvXjd8RBCFQhk/LjKk36CUlPsf8rl+WL/t3PlTa9etevIkw87OITR0sF6J1/KzmDG7izy6G9k4s7RVIKY+SRfK2rkwOnVv/P7me6B353tMq4Hdowt2jy7YPbpg9+iC3aMLdo8u2D26YPfogt2jC3aPLtg9umjTPYNFIpGaHN+C+UDo9LcOPn03tOmexiDVVcm0WCCmPpXFYo6RNl+pq033FnZ0YZ1ciwVi6qNSqUyt6FosUJvuPXsa5jyqravGVV/7PLxWaefKYhpo05eW36MtESlP7SjoNsjC0gG/XFVrJF+toNKI3p9o+VX/2n9/vlymunastOC50K4DWy7V31esqgBQKpVkXUxKoCWoNFJVmUQhV7l14XTt//6DB5tCV3MjigWK8kKpRPRBI511SklJybFjx5YuXQo7kKYhANeIamJNo1B1cvWkq7GaDDbZzo2po8K1AsEhKsWP23tzYAcCDf1t8TC6BrtHF+weXbB7dMHu0QW7RxfsHl2we3TB7tEFu0cX7B5dsHt0we7RBbtHF+weXbB7dMHu0QW7RxfsHl2we3TB7tEFu0cXdN0TBGFmZgY7Cpig616lUlVUVMCOAibousdg9+iC3aMLdo8u2D26YPfogt2jC3aPLtg9umD36ILdowt2jy7YPbpg9+iC3aMLdo8uunqvpt4yY8aM5ORkEolEEP/57CkpKVDjggBy9X7u3LmWlpZq9yQSSb3g4uICOy4IIOfez8/Pw8OjfgqdTo+IiIAXETSQcw8AiIqKqj9K08HBYcSIEVAjggOK7rt06eLu7q5eptFo48ePhx0RHFB0DwCIiIhQV30nJ6eRI0fCDgcOiLr39/fv1KkTk8kcN24c7Fig0Qau8arLZKUvRTUVcn6NAgAgqNXOdExCofDVq1eaxv/DYbDJdAaJY0QxtqDad2BS6fper/TXfW2FLCWel/dYoFAAjimLIBFUOpnKoOhtwCoVIRPL5BIFQYCqgloTK7q7v4F3X0PYcTWJProXCxQ3TlYU54m5VgZccxaNpc1J4VoNQZVYxJOU5Vd3H2zWJVgf/wP0zv2Da7yH1yot25sY2xrAjkULKBWq0udVhEo+MNLC2FxXM9S8H/rl/tqf5eUlSmv3j+0RSblEmXu/IHSChbMXG3Ys/6JH7uNOVNZUEyb2+tg8aoVXacWh4eY2Ttqc2/JD0Bf3F38rEUtppg4frXg1r9KKew8z1pParxfXIfdjqwVC8kcvHgBg72N9/XgZv0YvZg2G774oV/wqW2bhov15H/UT5652l38vgx0F0Av3N0+Vs80QmqCQTCOUBCU1vhp2ILDdv0jnKwGZaagv3Z/WwdzZNOl8JewoYLvPSOSbOelva79xx4RT53/SerEEASxdTR5eq9F6ye8ETPe1lbLKYjGd3SZv230gTEP604d8uDHAdJ+bKeCYsSAGABG2EUNQIxPxYc4TDvMuY8lLKddCV708hUJ+6dqerOzEmpoSJwfvngFjO3bopd709Q8DB4bMEghrYuP202nMDq7dRwxeyuWaAQBKynKjT60tLc9r7+wXGjhNR7GpMbE3eJUt7NAF2q1rmPW+OE9EoZF1VPiZC5tu3/mzd8DY1ctivDr1OxK98lFmnHoTmUyNTzhKEKS1q2JXfHoi72X6lRv7AAByuWz/kcVGhhYrPj0+dMCC+ISjdXU6fAmbUgFqymS6K/+twHQvFsgpdJ24l8kkyWkX+/WZ3KPbKDbLMMDvE9/OA6/GH9BkMDNpFxo4lck04HLNOrTvXlD4FACQ8eRGDa/0k8FLjI2srCycw4YtF4nrdBGeGgqdUlcN8yYPNPdSsYrGJJPIhC4Kf12UJZdL3doHaFJcHLsUl74QCHnq1Xa2/w7VZTK5YgkfAFBR+ZpGZZgYW6vTuQZmRoaWughPDZVBkYiUuiv/rUA735PIQCLUVU9HLOIDAHbun9UgvY5fyWap7xw38j8nFNXS6P/pe1IpDB1FCABQKZVKBcwfU6C5p1AJEplQyJVkivbbHnXHbcyIVWYmdvXTjQ2tmtmLxeRKJML6KWKJQOuxaZCJFSYmMPvaMI/N5JDlEoUu3Jub2lOpdABAe2c/dUodv0qlUtHpzV1SGhtZy2Ti4tIX1pbtAQCFxdm1deVaj02DXKowMNJVV7clwOzr2TgxpSKddHbodNaA4JlXbxzIfZkmk0sfZcbtPbTw9IW33KHr5NGXQqH9FfODVCrm1ZYfPfEFi6XDnxZJhNLEGubNbJj1vp0rIz2Jb2DG1EXhwX0ibazdbtw+8jznAYPBcbTzGjtidfO7MBmc6RFbLsb+8sX6fjQqY+iABSmPruikLwoAAKAsv9bRw1xnxb8dmGM3JELlwbX57oEOsAKACL9SJOXVjppvAzEGmG0+nUWyd2cLayQQY4CFqFbSsRvkwaiQR476BRte+r3c0a/Jf/+d+2cXl754M12pVKhUKjK58fhXLj7FYRtpK8i4W4fjbh9pYiMBQOMN54pPj3MNGh90KhXJ68rq3Ls6aivC9wP+eL2YPcUkJodr0XgPnFdbrlA0fuNTKpPQqI33lUyMtdmWikR1Td3gEwhr2Sxuo5sMuZZkcuPd+MLHZV1DOG6+kOs9fPe1lfLLR8st3CzghtFqSPhSeV3t0Gk6vGPYQuCP2eKaUvxDuIWZpbADaQ2UClXugyJ9EK8X7gEAzp5s186Moic6vJGiJ+TeK5i0Ul+ua+C3+RqeJfNTb/Ot3GFe8uoOuUSRc79g8hcODBbMe3n10SP3AIDHd2sfXK2x7WRJZerXo2sfCL9CVPysPGKlPZOjL+L1zj0AoLxAcn5/McuYaeFsQtLBrf5WRlAtrsirsnVmhE7Qu/ZM79yryUjg3blUxTZmcEzZXAsWQdLdrVWdIBXKa8sFComUUMn7hplZOejwt+D3Rk/dq8lOqctOFbx6KjCyYCqVgEwjU1lUpQzmeIdmIeQSmVyqoDNJAp7ExYvj5sOxdtZH62r02r2GstcSQa1cWKuQy5Rwx7o0A41JZrBIbC5F/doV2OG8nbbhHqML2nxnCvPeYPfogt2jC3aPLtg9umD36PI/BRKYaLPmoGIAAAAASUVORK5CYII=",
      "text/plain": [
       "<IPython.core.display.Image object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "API_KEY = ''\n",
    "os.environ['OPENAI_API_KEY']=API_KEY\n",
    "client = OpenAI()\n",
    "class AgentState(TypedDict):\n",
    "    query_gene: str\n",
    "    pathway_info: Dict[str, Any]\n",
    "    metadata: Dict[str, Any]\n",
    "    gse_list: list\n",
    "    research_plan: str\n",
    "    selected_gse: str  # ⬅️ now added\n",
    "    selected_gses: List[str]  # ✅ new field for multiple GSEs\n",
    "    deg_results: Dict[str, str]  # ✅ map GSE → output CSV path\n",
    "\n",
    "# Utility to extract all GSE IDs from plan\n",
    "def extract_all_gse_ids(plan_text: str) -> List[str]:\n",
    "    return re.findall(r\"GSE\\d{5,}\", plan_text)\n",
    "\n",
    "def extract_gse_from_research_plan(state: AgentState) -> AgentState:\n",
    "    plan = state.get(\"research_plan\", \"\")\n",
    "    match = re.search(r\"GSE\\d{5,}\", plan)\n",
    "    if match:\n",
    "        selected_gse = match.group(0)\n",
    "        print(f\"🔍 Extracted GSE ID: {selected_gse}\")\n",
    "        return {**state, \"selected_gse\": selected_gse}\n",
    "    else:\n",
    "        print(\"⚠️ No GSE ID found in research plan.\")\n",
    "        return state\n",
    "\n",
    "# LangGraph node: Extract GSEs from research plan and update state\n",
    "def extract_gse_list_from_plan(state: AgentState) -> AgentState:\n",
    "    plan = state.get(\"research_plan\", \"\")\n",
    "    gse_ids = extract_all_gse_ids(plan)\n",
    "    print(f\"🔍 Extracted GSE IDs: {gse_ids}\")\n",
    "    return {**state, \"selected_gses\": gse_ids}\n",
    "\n",
    "# ----------------------------------------\n",
    "# Utility functions (LLM + GEO search)\n",
    "# ----------------------------------------\n",
    "\n",
    "\n",
    "\n",
    "def generate_pathway_info(query_gene, model=\"gpt-4o-mini\", temperature=0.3):\n",
    "    prompt = f\"\"\"\n",
    "You are a biomedical assistant.\n",
    "\n",
    "Given the gene {query_gene}, return:\n",
    "1. Key gene symbols based on query.\n",
    "2. Drugs or compounds that inhibit this pathway or {query_gene}'s activity.\n",
    "3. The disease areas or biological processes this pathway is involved in.\n",
    "\n",
    "Provide answers in JSON format with keys: \"genes\", \"drugs\", \"pathways\"\n",
    "    \"\"\"\n",
    "    completion = client.chat.completions.create(\n",
    "        model=model,\n",
    "        temperature=temperature,\n",
    "        messages=[{\"role\": \"user\", \"content\": prompt}]\n",
    "    )\n",
    "    content = completion.choices[0].message.content\n",
    "    return extract_json_block(content)\n",
    "\n",
    "\n",
    "\n",
    "def scrape_organism_from_geo_html(geo_accession):\n",
    "    url = f\"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc={geo_accession}\"\n",
    "    try:\n",
    "        response = requests.get(url)\n",
    "        response.raise_for_status()\n",
    "        soup = BeautifulSoup(response.text, \"html.parser\")\n",
    "        organism_row = soup.find(\"td\", string=\"Organism\")\n",
    "        if organism_row and organism_row.find_next_sibling(\"td\"):\n",
    "            return organism_row.find_next_sibling(\"td\").text.strip()\n",
    "    except Exception as e:\n",
    "        print(f\"⚠️ Failed to scrape organism for {geo_accession}: {e}\")\n",
    "    return \"Unknown\"\n",
    "\n",
    "def search_geo_datasets(keyword, retmax=10):\n",
    "    base_url = \"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi\"\n",
    "    params = {\n",
    "        \"db\": \"gds\",\n",
    "        \"term\": keyword,\n",
    "        \"retmode\": \"json\",\n",
    "        \"retmax\": retmax\n",
    "    }\n",
    "    response = requests.get(base_url, params=params)\n",
    "    response.raise_for_status()\n",
    "    return response.json().get(\"esearchresult\", {}).get(\"idlist\", [])\n",
    "\n",
    "def fetch_gse_accessions(id_list):\n",
    "    if not id_list:\n",
    "        return {}\n",
    "    base_url = \"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi\"\n",
    "    params = {\n",
    "        \"db\": \"gds\",\n",
    "        \"id\": \",\".join(id_list),\n",
    "        \"retmode\": \"json\"\n",
    "    }\n",
    "    response = requests.get(base_url, params=params)\n",
    "    response.raise_for_status()\n",
    "    summaries = response.json().get(\"result\", {})\n",
    "    summaries.pop(\"uids\", None)\n",
    "    gse_dict = {}\n",
    "    for uid, info in summaries.items():\n",
    "        accession = info.get(\"accession\")\n",
    "        title = info.get(\"title\")\n",
    "        link = f\"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc={accession}\"\n",
    "        organism = scrape_organism_from_geo_html(accession)\n",
    "        gse_dict[accession] = {\n",
    "            \"title\": title,\n",
    "            \"organism\": organism,\n",
    "            \"link\": link\n",
    "        }\n",
    "    return gse_dict\n",
    "\n",
    "def search_geo_items(item_list, label=\"gene\", max_results=5):\n",
    "    results = {}\n",
    "    for item in item_list:\n",
    "        query = f\"{item} AND rna-seq\"\n",
    "        print(f\"🔍 Searching GEO for {label}: {item}\")\n",
    "        ids = search_geo_datasets(query, retmax=max_results)\n",
    "        datasets = fetch_gse_accessions(ids)\n",
    "        results[item] = datasets\n",
    "        time.sleep(0.3)\n",
    "    return results\n",
    "\n",
    "def get_geofetch_projects(gse_list, metadata_folder=\"geofetch_metadata\"):\n",
    "    geof = Geofetcher(\n",
    "        processed=True,\n",
    "        acc_anno=True,\n",
    "        discard_soft=True,\n",
    "        metadata_folder=metadata_folder\n",
    "    )\n",
    "    projects = {}\n",
    "    for gse in gse_list:\n",
    "        try:\n",
    "            print(f\"📥 Fetching metadata for {gse}\")\n",
    "            result = geof.get_projects(gse)\n",
    "            projects.update(result)\n",
    "        except Exception as e:\n",
    "            print(f\"❌ Failed to fetch {gse}: {e}\")\n",
    "    return projects\n",
    "\n",
    "import subprocess\n",
    "def download_processed_files_via_cli(gse_list, output_dir=\"geofetch_metadata\", overwrite=False):\n",
    "    for gse in gse_list:\n",
    "        gse_path = os.path.join(output_dir, gse)\n",
    "        if os.path.exists(gse_path) and not overwrite:\n",
    "            print(f\"✅ {gse}: already exists at {gse_path}, skipping.\")\n",
    "            continue\n",
    "        try:\n",
    "            subprocess.run(\n",
    "                [\"geofetch\", \"-i\", gse, \"--processed\", \"-m\", output_dir],\n",
    "                check=True\n",
    "            )\n",
    "            print(f\"✅ Finished downloading for {gse}\")\n",
    "        except subprocess.CalledProcessError as e:\n",
    "            print(f\"❌ geofetch failed for {gse}: {e}\")\n",
    "\n",
    "def save_combined_metadata_csv_from_state(state: Dict, csv_path: str = \"geofetch_metadata/combined_metadata.csv\") -> pd.DataFrame:\n",
    "    metadata = state.get(\"metadata\", {})\n",
    "    if not metadata:\n",
    "        raise ValueError(\"No metadata found in the agent state.\")\n",
    "\n",
    "    all_dfs = []\n",
    "    for gse, project in metadata.items():\n",
    "        try:\n",
    "            df = project.sample_table.copy()\n",
    "            df[\"source_gse\"] = gse\n",
    "            all_dfs.append(df)\n",
    "        except Exception as e:\n",
    "            print(f\"⚠️ Failed to extract sample_table from {gse}: {e}\")\n",
    "\n",
    "    if not all_dfs:\n",
    "        raise ValueError(\"No sample tables to save.\")\n",
    "\n",
    "    combined_df = pd.concat(all_dfs, ignore_index=True)\n",
    "    os.makedirs(os.path.dirname(csv_path), exist_ok=True)\n",
    "    combined_df.to_csv(csv_path, index=False)\n",
    "    print(f\"✅ Combined metadata saved to: {os.path.abspath(csv_path)}\")\n",
    "    return combined_df\n",
    "\n",
    "\n",
    "\n",
    "### scrapesupplementary function tested ###\n",
    "\n",
    "def scrape_geo_supplementary_downloads(gse_id: str):\n",
    "    \"\"\"\n",
    "    Scrape the GEO page for a given GSE ID and return a list of supplementary\n",
    "    files that are .tar, .gz, or .tar.gz with valid HTTP download links.\n",
    "\n",
    "    Returns:\n",
    "        List of tuples: (file_name, http_download_url)\n",
    "    \"\"\"\n",
    "    base_url = \"https://www.ncbi.nlm.nih.gov\"\n",
    "    url = f\"{base_url}/geo/query/acc.cgi?acc={gse_id}\"\n",
    "\n",
    "    try:\n",
    "        response = requests.get(url)\n",
    "        response.raise_for_status()\n",
    "\n",
    "        soup = BeautifulSoup(response.text, \"html.parser\")\n",
    "        tables = soup.find_all(\"table\")\n",
    "\n",
    "        for table in tables:\n",
    "            if \"Supplementary file\" in table.text:\n",
    "                rows = table.find_all(\"tr\")\n",
    "                files = []\n",
    "\n",
    "                for row in rows[1:]:  # skip header\n",
    "                    cols = row.find_all(\"td\")\n",
    "                    if len(cols) >= 3:\n",
    "                        file_name = cols[0].text.strip()\n",
    "                        if any(ext in file_name for ext in [\".tar\", \".gz\"]):\n",
    "                            all_links = cols[2].find_all(\"a\", href=True)\n",
    "                            for a in all_links:\n",
    "                                href = a[\"href\"]\n",
    "                                if href.startswith(\"/geo/download\"):\n",
    "                                    files.append((file_name, urljoin(base_url, href)))\n",
    "                                    break\n",
    "                                elif href.startswith(\"http\"):\n",
    "                                    files.append((file_name, href))\n",
    "                                    break\n",
    "\n",
    "                return files\n",
    "\n",
    "        return []\n",
    "\n",
    "    except Exception:\n",
    "        return []\n",
    "def ingest_and_prepare(state: dict) -> dict:\n",
    "    query_gene = state[\"query_gene\"]\n",
    "    pathway_info = generate_pathway_info(query_gene)\n",
    "    gene_list = pathway_info.get(\"genes\", [])\n",
    "    drug_list = pathway_info.get(\"drugs\", [])\n",
    "\n",
    "    gene_results = search_geo_items(gene_list, label=\"gene\")\n",
    "    drug_results = search_geo_items(drug_list, label=\"drug\")\n",
    "\n",
    "    all_gse = set()\n",
    "    for r in [gene_results, drug_results]:\n",
    "        for v in r.values():\n",
    "            all_gse.update(v.keys())\n",
    "    all_gse = list(all_gse)\n",
    "\n",
    "    # ----------------------------------------\n",
    "    # Download each GSE using embedded logic\n",
    "    # ----------------------------------------\n",
    "    base_dir = Path(\"rna_seq_analysis\")\n",
    "    for gse_id in all_gse:\n",
    "        download_dir = base_dir / gse_id\n",
    "\n",
    "        if download_dir.exists() and any(download_dir.iterdir()):\n",
    "            print(f\"📁 Skipping {gse_id}: folder already exists.\")\n",
    "            continue\n",
    "\n",
    "        download_dir.mkdir(parents=True, exist_ok=True)\n",
    "        files = scrape_geo_supplementary_downloads(gse_id)\n",
    "        if not files:\n",
    "            print(f\"⚠️ No supplementary files found for {gse_id}\")\n",
    "            continue\n",
    "\n",
    "        file_name, url = files[0]\n",
    "        if not file_name:\n",
    "            parsed = urlparse(url)\n",
    "            qs = parse_qs(parsed.query)\n",
    "            file_name = unquote(qs.get(\"file\", [\"unnamed_file\"])[0])\n",
    "\n",
    "        file_path = download_dir / file_name\n",
    "        print(f\"📥 Downloading {file_name} → {file_path.as_posix()}\")\n",
    "\n",
    "        try:\n",
    "            subprocess.run([\"curl\", \"-L\", \"-o\", file_path.as_posix(), url], check=True)\n",
    "        except subprocess.CalledProcessError as e:\n",
    "            print(f\"❌ Download failed for {gse_id}: {e}\")\n",
    "            continue\n",
    "\n",
    "        try:\n",
    "            if tarfile.is_tarfile(file_path):\n",
    "                print(\"🗂️ Extracting TAR...\")\n",
    "                with tarfile.open(file_path, \"r:*\") as tar:\n",
    "                    tar.extractall(path=download_dir)\n",
    "            elif file_path.suffix == \".gz\" and not file_path.name.endswith(\".tar.gz\"):\n",
    "                unzipped_path = file_path.with_suffix(\"\")\n",
    "                print(\"🗂️ Extracting GZ...\")\n",
    "                with gzip.open(file_path, 'rb') as f_in:\n",
    "                    with open(unzipped_path, 'wb') as f_out:\n",
    "                        shutil.copyfileobj(f_in, f_out)\n",
    "        except Exception as e:\n",
    "            print(f\"⚠️ Extraction failed for {gse_id}: {e}\")\n",
    "            continue\n",
    "\n",
    "    # ----------------------------------------\n",
    "    # Continue with metadata and return\n",
    "    # ----------------------------------------\n",
    "    metadata = get_geofetch_projects(all_gse, metadata_folder=\"geofetch_metadata\")\n",
    "    download_processed_files_via_cli(all_gse, output_dir=\"geofetch_metadata\")\n",
    "\n",
    "    try:\n",
    "        save_combined_metadata_csv_from_state({\"metadata\": metadata})\n",
    "    except Exception as e:\n",
    "        print(f\"⚠️ Failed to save metadata CSV: {e}\")\n",
    "\n",
    "    return {\n",
    "        \"query_gene\": query_gene,\n",
    "        \"pathway_info\": pathway_info,\n",
    "        \"metadata\": metadata,\n",
    "        \"gse_list\": all_gse\n",
    "    }\n",
    "\n",
    "\n",
    "# ----------------------------------------\n",
    "# Agent 2: Analyst node\n",
    "# ----------------------------------------\n",
    "\n",
    "def analyze_metadata_and_plan(state: AgentState) -> AgentState:\n",
    "    metadata = state[\"metadata\"]\n",
    "    drug_list = state[\"pathway_info\"].get(\"drugs\", [])\n",
    "    query_gene = state[\"query_gene\"]\n",
    "    selected = []\n",
    "\n",
    "    for gse, project in metadata.items():\n",
    "        df = project.sample_table\n",
    "        if \"processed_file_ftp\" in df.columns and df[\"processed_file_ftp\"].notna().any():\n",
    "            if any(drug.lower() in df.to_string().lower() for drug in drug_list):\n",
    "                selected.append((gse, df.shape[0]))\n",
    "\n",
    "    plan = f\"🧬 Research Plan for {query_gene} and drugs {drug_list}:\\n\"\n",
    "    if not selected:\n",
    "        plan += \"No relevant processed datasets were found.\\n\"\n",
    "    else:\n",
    "        plan += f\"{len(selected)} datasets selected:\\n\"\n",
    "        for gse, n in selected:\n",
    "            plan += f\"  - {gse} ({n} samples)\\n\"\n",
    "        plan += \"\\nNext: perform differential expression and gene signature clustering.\"\n",
    "\n",
    "    return {**state, \"research_plan\": plan}\n",
    "\n",
    "\n",
    "def analyze_metadata_and_plan(state: AgentState) -> AgentState:\n",
    "    import pandas as pd\n",
    "    import os\n",
    "\n",
    "    csv_path = \"geofetch_metadata/combined_metadata.csv\"\n",
    "    if not os.path.exists(csv_path):\n",
    "        raise FileNotFoundError(f\"❌ Metadata CSV not found at {csv_path}\")\n",
    "\n",
    "    df = pd.read_csv(csv_path)\n",
    "\n",
    "    # Ensure required columns exist\n",
    "    required_columns = {\"sample_name\", \"sample_source_name_ch1\", \"sample_title\"}\n",
    "    if not required_columns.issubset(df.columns):\n",
    "        raise ValueError(f\"❌ Metadata CSV must include columns: {required_columns}\")\n",
    "\n",
    "    # Take first 50 rows for LLM context\n",
    "    selected_df = df[list(required_columns)].fillna(\"\").head(50)\n",
    "    table_preview = selected_df.to_markdown(index=False)\n",
    "\n",
    "    query_gene = state[\"query_gene\"]\n",
    "    drug_list = state[\"pathway_info\"].get(\"drugs\", [])\n",
    "    selected_df = df[[\"sample_name\", \"sample_source_name_ch1\", \"sample_title\"]].fillna(\"\").head(50)\n",
    "    table_preview = selected_df.to_markdown(index=False)\n",
    "    print(\"🧪 Table preview sent to LLM:\\n\", table_preview)  # ✅ Add this line\n",
    "\n",
    "    # Build LLM prompt\n",
    "    prompt = f\"\"\"\n",
    "You are a biomedical research assistant.\n",
    "\n",
    "The target gene is **{query_gene}**, and the related drugs of interest are: {', '.join(drug_list)}.\n",
    "\n",
    "Below is a preview of sample metadata (first 50 rows) from multiple GEO studies. Each row includes:\n",
    "- sample name\n",
    "- sample source (cell line, tissue)\n",
    "- sample title (may indicate treatment or condition)\n",
    "\n",
    "Your task:\n",
    "1. Identify which studies include drug-treated samples.\n",
    "2. Identify the control groups if available.\n",
    "3. Determine the sample types (e.g., cell lines or tissues).\n",
    "4. Recommend studies and sample comparisons suitable for differential gene expression and drug-response signature analysis.\n",
    "\n",
    "Respond with:\n",
    "- GSE or study names (if known)\n",
    "- The experimental comparison design\n",
    "- Why the dataset is suitable (or not)\n",
    "- Bullet points summarizing each recommended comparison\n",
    "\n",
    "Sample Metadata Table:\n",
    "\n",
    "\"\"\"\n",
    "\n",
    "    # Call LLM\n",
    "    response = client.chat.completions.create(\n",
    "        model=\"gpt-4o-mini\",\n",
    "        temperature=0.3,\n",
    "        messages=[{\"role\": \"user\", \"content\": prompt}]\n",
    "    )\n",
    "\n",
    "    research_plan = response.choices[0].message.content.strip()\n",
    "\n",
    "    return {\n",
    "        **state,\n",
    "        \"research_plan\": research_plan\n",
    "    }\n",
    "\n",
    "###\n",
    "def analyze_metadata_and_plan(state: AgentState) -> AgentState:\n",
    "    import pandas as pd\n",
    "    import os\n",
    "\n",
    "    csv_path = \"geofetch_metadata/combined_metadata.csv\"\n",
    "    if not os.path.exists(csv_path):\n",
    "        raise FileNotFoundError(f\"❌ Metadata CSV not found at {csv_path}\")\n",
    "\n",
    "    df = pd.read_csv(csv_path)\n",
    "\n",
    "    # Ensure required columns are available\n",
    "    required_cols = [\"gse\", \"sample_name\", \"sample_title\", \"sample_source_name_ch1\", \"sample_geo_accession\"]\n",
    "    if not all(col in df.columns for col in required_cols):\n",
    "        raise ValueError(f\"❌ Metadata CSV must contain the following columns: {required_cols}\")\n",
    "\n",
    "    # Clean and preview first 30 rows\n",
    "    #preview_df = df[required_cols].fillna(\"\").head(30)\n",
    "    preview_df = df[required_cols].fillna(\"\")\n",
    "    preview_text = preview_df.to_string(index=False)\n",
    "\n",
    "    query_gene = state[\"query_gene\"]\n",
    "    drug_list = state[\"pathway_info\"].get(\"drugs\", [])\n",
    "\n",
    "    prompt = f\"\"\"\n",
    "You are a biomedical research assistant.\n",
    "\n",
    "The target gene is **{query_gene}** and the related drugs of interest are: {', '.join(drug_list)}.\n",
    "\n",
    "Below is a preview of sample metadata from several GEO datasets.\n",
    "Each row includes:\n",
    "- GSE accession\n",
    "- Sample name\n",
    "- Sample title (may contain treatment or control info)\n",
    "- Sample source (cell type or tissue)\n",
    "- Sample GEO accession\n",
    "\n",
    "Sample Metadata Table:\n",
    "{preview_text}\n",
    "\n",
    "Based on the sample names, titles, and sources:\n",
    "1. Which GSE studies contain drug-treated samples and matching control groups?\n",
    "2. What cell types or tissues are used?\n",
    "3. Which treatments are applied? What are the controls?\n",
    "4. Recommend GSEs and sample pairs suitable for differential gene expression to identify drug-response gene signatures.\n",
    "\n",
    "Be specific, refer to GSE and sample names where possible, and explain why you recommend them.\n",
    "\"\"\"\n",
    "\n",
    "    # Call OpenAI LLM\n",
    "    response = client.chat.completions.create(\n",
    "        model=\"gpt-4o-mini\",\n",
    "        temperature=0.3,\n",
    "        messages=[{\"role\": \"user\", \"content\": prompt}]\n",
    "    )\n",
    "\n",
    "    research_plan = response.choices[0].message.content.strip()\n",
    "\n",
    "    return {\n",
    "        **state,\n",
    "        \"research_plan\": research_plan\n",
    "    }\n",
    "\n",
    "\n",
    "# LangGraph node: Extract GSEs from research plan and update state\n",
    "# Utility to extract all GSE IDs from plan\n",
    "def extract_all_gse_ids(plan_text: str) -> List[str]:\n",
    "    return re.findall(r\"GSE\\d{5,}\", plan_text)\n",
    "def extract_gse_list_from_plan(state: AgentState) -> AgentState:\n",
    "    plan = state.get(\"research_plan\", \"\")\n",
    "    gse_ids = extract_all_gse_ids(plan)\n",
    "    print(f\"🔍 Extracted GSE IDs: {gse_ids}\")\n",
    "    return {**state, \"selected_gses\": gse_ids}\n",
    "\n",
    "# DEG Agent with Enhanced LLM Classification from GEO Series Matrix\n",
    "\n",
    "\n",
    "def decompress_all_in_dir(root_dir: str):\n",
    "    \"\"\"\n",
    "    Decompress .gz and .tar files only if no uncompressed files exist in each GSE subdirectory.\n",
    "    \"\"\"\n",
    "    print(f\"🔍 Scanning for compressed files under: {root_dir}\")\n",
    "    for dirpath, _, filenames in os.walk(root_dir):\n",
    "        # Skip decompression if the folder already has non-compressed data files\n",
    "        if any(f.lower().endswith(('.csv', '.tsv', '.txt', '.xlsx')) for f in filenames):\n",
    "            continue\n",
    "\n",
    "        for fname in filenames:\n",
    "            full_path = os.path.join(dirpath, fname)\n",
    "            if tarfile.is_tarfile(full_path):\n",
    "                try:\n",
    "                    print(f\"📦 Extracting TAR: {full_path}\")\n",
    "                    with tarfile.open(full_path, \"r:*\") as tar:\n",
    "                        tar.extractall(path=dirpath)\n",
    "                except Exception as e:\n",
    "                    print(f\"❌ Failed to extract TAR {fname}: {e}\")\n",
    "            elif fname.endswith(\".gz\") and not fname.endswith(\".tar.gz\"):\n",
    "                out_path = os.path.join(dirpath, fname[:-3])\n",
    "                if os.path.exists(out_path):\n",
    "                    print(f\"⚠️ Skipping (already exists): {out_path}\")\n",
    "                    continue\n",
    "                try:\n",
    "                    print(f\"🗜️ Decompressing GZ: {full_path} → {out_path}\")\n",
    "                    with gzip.open(full_path, 'rb') as f_in:\n",
    "                        with open(out_path, 'wb') as f_out:\n",
    "                            shutil.copyfileobj(f_in, f_out)\n",
    "                except Exception as e:\n",
    "                    print(f\"❌ Failed to decompress GZ {fname}: {e}\")\n",
    "def decompress_gse_in_dir(gse_id: str, root_dir: str = \"rna_seq_analysis\"):\n",
    "    \"\"\"\n",
    "    Decompress .gz and .tar files only if no uncompressed files exist\n",
    "    in the subdirectory for the specific GSE.\n",
    "    \"\"\"\n",
    "    gse_path = os.path.join(root_dir, gse_id)\n",
    "    print(f\"🔍 Scanning for compressed files in: {gse_path}\")\n",
    "    if not os.path.exists(gse_path):\n",
    "        print(f\"⚠️ Directory does not exist: {gse_path}\")\n",
    "        return\n",
    "\n",
    "    filenames = os.listdir(gse_path)\n",
    "    if any(f.lower().endswith(('.csv', '.tsv', '.txt', '.xlsx')) for f in filenames):\n",
    "        return\n",
    "\n",
    "    for fname in filenames:\n",
    "        full_path = os.path.join(gse_path, fname)\n",
    "        if tarfile.is_tarfile(full_path):\n",
    "            try:\n",
    "                print(f\"📦 Extracting TAR: {full_path}\")\n",
    "                with tarfile.open(full_path, \"r:*\") as tar:\n",
    "                    tar.extractall(path=gse_path)\n",
    "            except Exception as e:\n",
    "                print(f\"❌ Failed to extract TAR {fname}: {e}\")\n",
    "        elif fname.endswith(\".gz\") and not fname.endswith(\".tar.gz\"):\n",
    "            out_path = os.path.join(gse_path, fname[:-3])\n",
    "            if os.path.exists(out_path):\n",
    "                print(f\"⚠️ Skipping (already exists): {out_path}\")\n",
    "                continue\n",
    "            try:\n",
    "                print(f\"🗜️ Decompressing GZ: {full_path} → {out_path}\")\n",
    "                with gzip.open(full_path, 'rb') as f_in:\n",
    "                    with open(out_path, 'wb') as f_out:\n",
    "                        shutil.copyfileobj(f_in, f_out)\n",
    "            except Exception as e:\n",
    "                print(f\"❌ Failed to decompress GZ {fname}: {e}\")\n",
    "\n",
    "\n",
    "def get_series_matrix_url(gse_id: str) -> str:\n",
    "    \"\"\"\n",
    "    Find the final HTTP URL of the actual .series_matrix.txt.gz file\n",
    "    for a given GEO study by first finding the FTP matrix directory,\n",
    "    then scraping the listing page for the correct file.\n",
    "\n",
    "    Args:\n",
    "        gse_id (str): e.g. \"GSE94840\"\n",
    "\n",
    "    Returns:\n",
    "        str: Final download URL, or None if not found.\n",
    "    \"\"\"\n",
    "    base_geo_url = \"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=\"\n",
    "    response = requests.get(base_geo_url + gse_id)\n",
    "    response.raise_for_status()\n",
    "\n",
    "    soup = BeautifulSoup(response.text, \"html.parser\")\n",
    "    ftp_matrix_url = None\n",
    "\n",
    "    # Step 1: find ftp:// link for Series Matrix File(s)\n",
    "    for a in soup.find_all(\"a\", href=True):\n",
    "        href = a['href']\n",
    "        if href.startswith(\"ftp://\") and \"matrix\" in href and gse_id in href:\n",
    "            ftp_matrix_url = href\n",
    "            break\n",
    "\n",
    "    if not ftp_matrix_url:\n",
    "        return None\n",
    "\n",
    "    # Step 2: Convert FTP to HTTPS to scrape the file listing page\n",
    "    https_matrix_dir = ftp_matrix_url.replace(\"ftp://\", \"https://\")\n",
    "\n",
    "    response = requests.get(https_matrix_dir)\n",
    "    response.raise_for_status()\n",
    "\n",
    "    soup = BeautifulSoup(response.text, \"html.parser\")\n",
    "    for a in soup.find_all(\"a\", href=True):\n",
    "        if a['href'].endswith(\"series_matrix.txt.gz\"):\n",
    "            # Return full HTTPS URL to the file\n",
    "            return https_matrix_dir + a['href']\n",
    "\n",
    "    return None\n",
    "\n",
    "\n",
    "\n",
    "def download_series_matrix(gse_id, output_dir=\"geofetch_metadata\"):\n",
    "    url = get_series_matrix_url(gse_id)\n",
    "    output_dir = Path(output_dir)\n",
    "    output_dir.mkdir(parents=True, exist_ok=True)\n",
    "    output_path = output_dir / f\"{gse_id}_series_matrix.txt.gz\"\n",
    "    if output_path.exists():\n",
    "        return str(output_path)\n",
    "    response = requests.get(url, stream=True)\n",
    "    if response.status_code == 200:\n",
    "        with open(output_path, \"wb\") as f:\n",
    "            for chunk in response.iter_content(chunk_size=8192):\n",
    "                f.write(chunk)\n",
    "        return str(output_path)\n",
    "    else:\n",
    "        raise FileNotFoundError(f\"Could not download series matrix for {gse_id}\")\n",
    "\n",
    "\n",
    "\n",
    "def infer_conditions_with_labels_from_series_matrix(series_matrix_path, gse_id, max_samples=500):\n",
    "    characteristics_matrix, sample_accessions = [], []\n",
    "    with gzip.open(series_matrix_path, \"rt\", encoding=\"utf-8\") as f:\n",
    "        for line in f:\n",
    "            if line.startswith(\"!Sample_characteristics_ch1\"):\n",
    "                values = line.strip().split(\"\\t\")[1:]\n",
    "                characteristics_matrix.append(values)\n",
    "            elif line.startswith(\"!Sample_geo_accession\"):\n",
    "                sample_accessions = line.strip().split(\"\\t\")[1:]\n",
    "    if not characteristics_matrix or not sample_accessions:\n",
    "        raise ValueError(\"Missing sample characteristics or accessions.\")\n",
    "    characteristics_matrix = list(zip(*characteristics_matrix))\n",
    "    sample_data = [\n",
    "        {\"GSM\": gsm.strip('\"'), \"characteristics\": [c.strip('\"') for c in char_fields if c.strip()]}\n",
    "        for gsm, char_fields in zip(sample_accessions, characteristics_matrix)\n",
    "    ]\n",
    "    metadata_json = json.dumps(sample_data[:max_samples], indent=2)\n",
    "    prompt = f\"\"\"\n",
    "You are a biomedical data curator.\n",
    "\n",
    "Below is a list of samples from GEO study **{gse_id}**. Each sample includes multiple characteristics such as diagnosis, sex, age, and treatment.\n",
    "\n",
    "Your task:\n",
    "1. Classify each sample as either part of the **control** group or the **treated/disease** group.\n",
    "2. Extract a **short biological label** for each sample from its characteristics (e.g., \\\"ALS\\\", \\\"healthy\\\", \\\"treated with ..\\\").\n",
    "\n",
    "Here is the list:\n",
    "```json\n",
    "{metadata_json}\n",
    "```\n",
    "Return only a **valid JSON object** like:\n",
    "```json\n",
    "{{\n",
    "  \"GSM8077414\": {{ \"group\": \"control\", \"label\": \"healthy\" }},\n",
    "  \"GSM8077415\": {{ \"group\": \"treated\", \"label\": \"ALS\" }}\n",
    "}}\n",
    "---\n",
    "\"\"\"\n",
    "    response = client.chat.completions.create(\n",
    "        model=\"gpt-4o\",\n",
    "        temperature=0.2,\n",
    "        messages=[{\"role\": \"user\", \"content\": prompt}]\n",
    "    )\n",
    "    content = response.choices[0].message.content.strip()\n",
    "    mapping = extract_json_block(content)\n",
    "    if not isinstance(mapping, dict) or not all(isinstance(v, dict) and \"group\" in v and \"label\" in v for v in mapping.values()):\n",
    "        raise ValueError(\"Invalid response from LLM\")\n",
    "    return mapping\n",
    "\n",
    "\n",
    "\n",
    "def extract_json_block(text):\n",
    "    \"\"\"\n",
    "    Extract the first valid JSON object block from text.\n",
    "    Assumes the block is surrounded by ```json ... ``` or just {...}\n",
    "    \"\"\"\n",
    "    import re\n",
    "    import ast\n",
    "\n",
    "    try:\n",
    "        if \"```json\" in text:\n",
    "            json_block = re.findall(r\"```json\\s*(\\{.*?\\})\\s*```\", text, re.DOTALL)\n",
    "        else:\n",
    "            json_block = re.findall(r\"(\\{.*\\})\", text, re.DOTALL)\n",
    "        if json_block:\n",
    "            return json.loads(json_block[0])\n",
    "        else:\n",
    "            return ast.literal_eval(text.strip())  # fallback\n",
    "    except Exception as e:\n",
    "        raise ValueError(f\"Failed to extract valid JSON: {e}\")\n",
    "\n",
    "\n",
    "def extract_flexible_labels_from_series_matrix(series_matrix_path, gse_id, max_samples=600):\n",
    "    \"\"\"\n",
    "    Parse a GEO Series Matrix file and use an LLM to extract biologically meaningful sample labels.\n",
    "\n",
    "    Args:\n",
    "        series_matrix_path (str): Path to the .txt.gz series matrix file.\n",
    "        gse_id (str): GEO study ID.\n",
    "        client: An OpenAI-compatible API client object.\n",
    "        max_samples (int): Max number of samples to send to LLM (for token budget).\n",
    "\n",
    "    Returns:\n",
    "        dict: Mapping from GSM ID to flexible labels (as key-value fields).\n",
    "    \"\"\"\n",
    "    characteristics_matrix, sample_accessions = [], []\n",
    "\n",
    "    with gzip.open(series_matrix_path, \"rt\", encoding=\"utf-8\") as f:\n",
    "        for line in f:\n",
    "            if line.startswith(\"!Sample_characteristics_ch1\"):\n",
    "                values = line.strip().split(\"\\t\")[1:]\n",
    "                characteristics_matrix.append(values)\n",
    "            elif line.startswith(\"!Sample_geo_accession\"):\n",
    "                sample_accessions = line.strip().split(\"\\t\")[1:]\n",
    "\n",
    "    if not characteristics_matrix or not sample_accessions:\n",
    "        raise ValueError(\"Missing sample characteristics or accessions.\")\n",
    "\n",
    "    # Transpose to per-sample view\n",
    "    characteristics_matrix = list(zip(*characteristics_matrix))\n",
    "    sample_data = [\n",
    "        {\n",
    "            \"GSM\": gsm.strip('\"'),\n",
    "            \"characteristics\": [c.strip('\"') for c in char_fields if c.strip()]\n",
    "        }\n",
    "        for gsm, char_fields in zip(sample_accessions, characteristics_matrix)\n",
    "    ]\n",
    "\n",
    "    metadata_json = json.dumps(sample_data[:max_samples], indent=2)\n",
    "\n",
    "    prompt = f\"\"\"\n",
    "You are a biomedical data curator.\n",
    "\n",
    "Below is a list of samples from GEO study **{gse_id}**. Each sample includes multiple characteristics such as diagnosis, sex, age, and treatment.\n",
    "\n",
    "Your task:\n",
    "1. Classify each sample as either part of the **control** group or the **treated/disease** group. This classification should be stored in the field called `condition`.\n",
    "2. Extract all **biologically meaningful labels** from each sample (such as treatment, dose, time point, cell type, genotype, sex, etc.).\n",
    "3. Return a dictionary mapping each sample's GSM ID to a dictionary of extracted fields and values.\n",
    "4. The field names should be concise and biologically relevant (e.g., \"treatment\", \"time\", \"dose\").\n",
    "5. Do not include null or irrelevant fields.\n",
    "\n",
    "Here is the sample list:\n",
    "```json\n",
    "{metadata_json}\n",
    "```\n",
    "\n",
    "Please return only the final **valid JSON object**. Example format:\n",
    "```json\n",
    "{{\n",
    "  \"GSM123456\": {{\n",
    "    \"condition\": \"treated\",\n",
    "    \"treatment\": \"PGE2\",\n",
    "    \"dose\": \"30nM\",\n",
    "    \"time\": \"24 hr\"\n",
    "  }},\n",
    "  \"GSM123457\": {{\n",
    "    \"condition\": \"control\",\n",
    "    \"treatment\": \"Vehicle (DMSO)\"\n",
    "  }}\n",
    "}}\n",
    "\"\"\"\n",
    "\n",
    "    # Call the LLM\n",
    "    response = client.chat.completions.create(\n",
    "        model=\"gpt-4o\",\n",
    "        temperature=0.2,\n",
    "        messages=[{\"role\": \"user\", \"content\": prompt}]\n",
    "    )\n",
    "    content = response.choices[0].message.content.strip()\n",
    "\n",
    "    # Extract JSON\n",
    "    mapping = extract_json_block(content)\n",
    "\n",
    "    if not isinstance(mapping, dict) or not all(isinstance(v, dict) for v in mapping.values()):\n",
    "        raise ValueError(\"Invalid response format from LLM.\")\n",
    "\n",
    "    return mapping\n",
    "\n",
    "\n",
    "\n",
    "def get_group_label_mapping_from_gse(gse_id, output_dir=\"geofetch_metadata\", max_samples=30):\n",
    "    path = download_series_matrix(gse_id, output_dir=output_dir)\n",
    "    #return infer_conditions_with_labels_from_series_matrix(path, gse_id, max_samples=max_samples)\n",
    "    return extract_flexible_labels_from_series_matrix(path, gse_id, max_samples=max_samples)\n",
    "\n",
    "\n",
    "def download_and_extract_gse(gse_id, base_dir=\"rna_seq_analysis\"):\n",
    "    \"\"\"\n",
    "    Downloads and extracts supplementary files for a given GSE.\n",
    "    \"\"\"\n",
    "    gse_dir = Path(base_dir) / gse_id\n",
    "    gse_dir.mkdir(parents=True, exist_ok=True)\n",
    "\n",
    "    files = scrape_geo_supplementary_downloads(gse_id)\n",
    "    if not files:\n",
    "        print(f\"⚠️ No supplementary files found for {gse_id}\")\n",
    "        return str(gse_dir)\n",
    "\n",
    "    file_name, url = files[0]\n",
    "    if not file_name:\n",
    "        parsed = urlparse(url)\n",
    "        qs = parse_qs(parsed.query)\n",
    "        file_name = unquote(qs.get(\"file\", [\"unnamed_file\"])[0])\n",
    "\n",
    "    file_path = gse_dir / file_name\n",
    "    print(f\"📥 Downloading {file_name} → {file_path.as_posix()}\")\n",
    "\n",
    "    try:\n",
    "        subprocess.run([\"curl\", \"-L\", \"-o\", file_path.as_posix(), url], check=True)\n",
    "    except subprocess.CalledProcessError as e:\n",
    "        print(f\"❌ Download failed for {gse_id}: {e}\")\n",
    "        return str(gse_dir)\n",
    "\n",
    "    try:\n",
    "        if tarfile.is_tarfile(file_path):\n",
    "            print(\"🗂️ Extracting TAR...\")\n",
    "            with tarfile.open(file_path, \"r:*\") as tar:\n",
    "                tar.extractall(path=gse_dir)\n",
    "        elif file_path.suffix == \".gz\" and not file_path.name.endswith(\".tar.gz\"):\n",
    "            unzipped_path = file_path.with_suffix(\"\")\n",
    "            print(\"🗂️ Extracting GZ...\")\n",
    "            with gzip.open(file_path, 'rb') as f_in:\n",
    "                with open(unzipped_path, 'wb') as f_out:\n",
    "                    shutil.copyfileobj(f_in, f_out)\n",
    "    except Exception as e:\n",
    "        print(f\"⚠️ Extraction failed for {gse_id}: {e}\")\n",
    "\n",
    "def run_deg_for_multiple_gses_debug(state):\n",
    "    gse_ids = state.get(\"selected_gses\", [])\n",
    "    if not gse_ids:\n",
    "        print(\"No selected GSEs found.\")\n",
    "        return state\n",
    "    deg_paths = {}\n",
    "\n",
    "    # Pre-decompress all GSE folders\n",
    "    decompress_all_in_dir(\"rna_seq_analysis\")\n",
    "\n",
    "    for gse_id in gse_ids:\n",
    "        try:\n",
    "            print(f\"Running DESeq2 for: {gse_id}\")\n",
    "            download_dir = download_and_extract_gse(gse_id)\n",
    "            counts_df = find_and_merge_count_files(gse_id)\n",
    "            if counts_df is None or counts_df.empty:\n",
    "                continue\n",
    "            group_label_mapping = get_group_label_mapping_from_gse(gse_id)\n",
    "            metadata_df = pd.DataFrame.from_dict(group_label_mapping, orient=\"index\")\n",
    "            metadata_df.index.name = \"sample_geo_accession\"\n",
    "            metadata_df = metadata_df[[\"group\"]].rename(columns={\"group\": \"condition\"})\n",
    "            print(metadata_df)\n",
    "            counts_df.columns = [gsm if gsm in metadata_df.index else gsm for gsm in counts_df.columns]\n",
    "            valid_samples = metadata_df.index.intersection(counts_df.columns)\n",
    "            counts_df = counts_df[valid_samples]\n",
    "            metadata_df = metadata_df.loc[valid_samples]\n",
    "            if counts_df.empty or metadata_df.empty:\n",
    "                continue\n",
    "            dds = DeseqDataSet(\n",
    "                counts=counts_df.T,\n",
    "                metadata=metadata_df,\n",
    "                design_factors=\"condition\",\n",
    "                refit_cooks=True,\n",
    "            )\n",
    "            dds.deseq2()\n",
    "            ds = DeseqStats(dds, contrast=[\"condition\", \"treated\", \"control\"])\n",
    "            ds.summary()\n",
    "            output_csv = f\"rna_seq_analysis/{gse_id}_deseq2_results.csv\"\n",
    "            ds.results_df.to_csv(output_csv, index=False)\n",
    "            deg_paths[gse_id] = output_csv\n",
    "        except Exception as e:\n",
    "            print(f\"Error running DESeq2 for {gse_id}: {e}\")\n",
    "    return {**state, \"deg_results\": deg_paths}\n",
    "\n",
    "\n",
    "\n",
    "def find_and_merge_count_files(gse_id: str, base_dir=\"rna_seq_analysis\") -> pd.DataFrame:\n",
    "    gse_dir = Path(base_dir) / gse_id\n",
    "    if not gse_dir.exists():\n",
    "        print(f\"❌ GSE folder not found: {gse_dir}\")\n",
    "        return None\n",
    "\n",
    "    patterns = [\"*.csv\", \"*.tsv\", \"*.txt\", \"*.xlsx\", \"*.gz\"]\n",
    "    files = []\n",
    "    seen_uncompressed = set()\n",
    "    for pattern in patterns:\n",
    "        for f in gse_dir.glob(pattern):\n",
    "            if f.suffix == \".gz\":\n",
    "                uncompressed_name = f.with_suffix(\"\")\n",
    "                if uncompressed_name.name in seen_uncompressed:\n",
    "                    print(f\"⚠️ Skipping compressed duplicate: {f.name}\")\n",
    "                    continue\n",
    "            else:\n",
    "                seen_uncompressed.add(f.name)\n",
    "            files.append(f)\n",
    "\n",
    "    if not files:\n",
    "        print(f\"⚠️ No count files found in {gse_dir}\")\n",
    "        return None\n",
    "\n",
    "    print(f\"🔍 Found {len(files)} count files for {gse_id}\")\n",
    "\n",
    "    # Handle single uncompressed file\n",
    "    uncompressed_files = [f for f in files if f.suffix != \".gz\"]\n",
    "    if len(uncompressed_files) == 1:\n",
    "        file = uncompressed_files[0]\n",
    "        print(f\"📄 Only one uncompressed count file found: {file.name} — attempting to load without merging.\")\n",
    "        try:\n",
    "            if file.suffix == \".csv\":\n",
    "                df = pd.read_csv(file)\n",
    "            elif file.suffix == \".xlsx\":\n",
    "                df = pd.read_excel(file)\n",
    "            else:  # .txt or .tsv\n",
    "                df = pd.read_csv(file, sep=\"\\t\")\n",
    "\n",
    "            #df.columns = df.columns.astype(str)\n",
    "\n",
    "            # Set first column as gene index\n",
    "            #gene_col = df.columns[0]\n",
    "            #df.set_index(gene_col, inplace=True)\n",
    "\n",
    "            # Keep only numeric sample columns\n",
    "            #numeric_df = df.apply(pd.to_numeric, errors='coerce')\n",
    "            #numeric_df = numeric_df.dropna(axis=1, how='all')  # Drop fully non-numeric\n",
    "            #numeric_df = numeric_df.fillna(0).astype(int)\n",
    "  \n",
    "            df.columns = df.columns.astype(str)\n",
    "\n",
    "            # Set the first column (likely gene names) as index\n",
    "            gene_col = df.columns[0]\n",
    "            df.set_index(gene_col, inplace=True)\n",
    "\n",
    "                        \n",
    "            annotation_keywords = {\"gene_length\", \"length\", \"start\", \"end\", \"strand\", \"chromosome\", \"chr\"}\n",
    "            \n",
    "            # Identify numeric sample columns and exclude known annotation fields\n",
    "            numeric_cols = []\n",
    "            for col in df.columns:\n",
    "                col_lower = col.strip().lower()\n",
    "                if any(key in col_lower for key in annotation_keywords):\n",
    "                    continue\n",
    "                try:\n",
    "                    pd.to_numeric(df[col], errors='raise')\n",
    "                    numeric_cols.append(col)\n",
    "                except Exception:\n",
    "                    continue\n",
    "            \n",
    "            # Filter to only good sample columns\n",
    "            df = df[numeric_cols]\n",
    "            \n",
    "            # Final numeric conversion and zero fill\n",
    "            numeric_df = df.apply(pd.to_numeric, errors='coerce').fillna(0).astype(int)          \n",
    "             \n",
    "            \n",
    "            \n",
    "            # Conditionally apply +1 normalization if any zeros are present\n",
    "            if (numeric_df == 0).any().any():\n",
    "                print(\"⚠️ Zero counts detected — applying +1 normalization.\")\n",
    "                numeric_df += 1\n",
    "            else:\n",
    "                print(\"✅ No zero counts — no normalization applied.\")\n",
    "\n",
    "            print(f\"✅ Single count file processed. Shape: {numeric_df.shape}\")\n",
    "            return numeric_df\n",
    "\n",
    "        except Exception as e:\n",
    "            print(f\"❌ Failed to load single count file: {e}\")\n",
    "            return None\n",
    "\n",
    "    # Proceed with merging multiple files\n",
    "    merged_df = None\n",
    "    used_column_names = set()\n",
    "\n",
    "    for file in files:\n",
    "        try:\n",
    "            sample_base = file.stem.split(\"_\")[0] if \"_\" in file.stem else file.stem\n",
    "\n",
    "            if file.suffix == \".gz\":\n",
    "                with gzip.open(file, 'rt') as f:\n",
    "                    df = pd.read_csv(f, sep=\"\\t\", header=0)\n",
    "            elif file.suffix == \".csv\":\n",
    "                df = pd.read_csv(file, header=0)\n",
    "            elif file.suffix == \".xlsx\":\n",
    "                df = pd.read_excel(file, header=0)\n",
    "            else:\n",
    "                df = pd.read_csv(file, sep=\"\\t\", header=0)\n",
    "\n",
    "            if df.shape[1] < 2:\n",
    "                raise ValueError(f\"{file.name} has fewer than 2 columns, cannot extract count data.\")\n",
    "\n",
    "            df.columns = ['Gene'] + [f\"V{i}\" for i in range(1, df.shape[1])]\n",
    "            df.set_index('Gene', inplace=True)\n",
    "            df = df.apply(pd.to_numeric, errors=\"coerce\").fillna(0)\n",
    "            df = df.iloc[:, [0]]  # First data column only\n",
    "\n",
    "            col_name = sample_base\n",
    "            counter = 1\n",
    "            while col_name in used_column_names:\n",
    "                col_name = f\"{sample_base}_{counter}\"\n",
    "                counter += 1\n",
    "            used_column_names.add(col_name)\n",
    "            df.columns = [col_name]\n",
    "\n",
    "            if merged_df is None:\n",
    "                merged_df = df\n",
    "            else:\n",
    "                merged_df = merged_df.join(df, how=\"outer\")\n",
    "\n",
    "        except Exception as e:\n",
    "            print(f\"❌ Failed to read {file.name}: {e}\")\n",
    "\n",
    "    if merged_df is not None:\n",
    "        merged_df = merged_df.fillna(0).astype(int)\n",
    "\n",
    "        if (merged_df == 0).any().any():\n",
    "            print(\"⚠️ Zero counts detected in merged matrix — applying +1 normalization.\")\n",
    "            merged_df += 1\n",
    "        else:\n",
    "            print(\"✅ No zero counts — no normalization applied.\")\n",
    "\n",
    "        print(f\"✅ Final merged count matrix shape: {merged_df.shape}\")\n",
    "        return merged_df\n",
    "    else:\n",
    "        print(\"❌ No valid count data could be merged.\")\n",
    "        return None\n",
    "\n",
    "\n",
    "def load_de_result_file(file_path: Path) -> pd.DataFrame:\n",
    "    \"\"\"\n",
    "    Attempts to load a DE result table and validate expected DE columns.\n",
    "\n",
    "    Args:\n",
    "        file_path (Path): Path to the DE results file.\n",
    "\n",
    "    Returns:\n",
    "        pd.DataFrame: Parsed DE results with gene index, or None.\n",
    "    \"\"\"\n",
    "    try:\n",
    "        if file_path.suffix == \".gz\":\n",
    "            opener = gzip.open\n",
    "            mode = 'rt'\n",
    "        else:\n",
    "            opener = open\n",
    "            mode = 'r'\n",
    "\n",
    "        with opener(file_path, mode) as f:\n",
    "            # Try different separators\n",
    "            for sep in [\"\\t\", \",\"]:\n",
    "                f.seek(0)\n",
    "                df = pd.read_csv(f, sep=sep)\n",
    "                if any(col in df.columns for col in [\"log2FoldChange\", \"pvalue\", \"padj\", \"stat\", \"baseMean\"]):\n",
    "                    df.columns = df.columns.str.strip()\n",
    "                    gene_col = df.columns[0]\n",
    "                    df.set_index(gene_col, inplace=True)\n",
    "                    print(f\"✅ Loaded DE result file: {file_path.name} — shape: {df.shape}\")\n",
    "                    return df\n",
    "        print(f\"⚠️ No DE columns found in file: {file_path.name}\")\n",
    "        return None\n",
    "    except Exception as e:\n",
    "        print(f\"❌ Failed to read DE result file {file_path.name}: {e}\")\n",
    "        return None\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "def annotate_clustermap_with_metadata(grapher, metadata_df):\n",
    "    print(\"🔍 Starting clustermap annotation...\")\n",
    "\n",
    "    # 1. Build treatment label to GSM mapping\n",
    "    label_map = {}\n",
    "    for gsm, row in metadata_df.iterrows():\n",
    "        label = str(row['treatment']).strip()\n",
    "        if label not in label_map:\n",
    "            label_map[label] = []\n",
    "        label_map[label].append(gsm)\n",
    "    print(f\"✅ Built label_map with labels: {list(label_map.keys())}\")\n",
    "\n",
    "    # 2. Check if column names in grapher need remapping\n",
    "    if not all(label in metadata_df.index for label in grapher.columns):\n",
    "        print(\"⚠️ Column names in grapher are not pure GSM IDs. Attempting to remap based on treatment labels...\")\n",
    "        new_cols = []\n",
    "        for label in grapher.columns:\n",
    "            # Assume format: \"Vehicle (DMSO)_GSM7757588\"\n",
    "            try:\n",
    "                treatment_label = \"_\".join(label.split('_')[:-1])  # everything except last piece\n",
    "            except Exception as e:\n",
    "                raise ValueError(f\"❌ Failed to parse treatment label from: {label} — {e}\")\n",
    "\n",
    "            matches = label_map.get(treatment_label)\n",
    "            if not matches:\n",
    "                raise ValueError(f\"❌ No GSM match found for treatment label: {treatment_label} (from full label: {label})\")\n",
    "            new_cols.append(matches[0])  # Use only the first match\n",
    "            print(f\"🔄 Remapped {label} -> {matches[0]}\")\n",
    "        grapher.columns = new_cols\n",
    "    else:\n",
    "        print(\"✅ All grapher columns are valid GSM IDs\")\n",
    "\n",
    "    # 3. Extract annotations\n",
    "    annotation_cols = [\"condition\", \"treatment\", \"time\"]\n",
    "    missing_cols = [col for col in annotation_cols if col not in metadata_df.columns]\n",
    "    if missing_cols:\n",
    "        raise KeyError(f\"❌ Metadata is missing required columns: {missing_cols}\")\n",
    "\n",
    "    try:\n",
    "        col_annotation_df = metadata_df.loc[grapher.columns, annotation_cols].copy()\n",
    "    except KeyError as e:\n",
    "        raise KeyError(f\"❌ Some GSM IDs in grapher not found in metadata: {e}\")\n",
    "\n",
    "    col_annotation_df = col_annotation_df.applymap(lambda x: str(x).strip() if pd.notna(x) else \"NA\")\n",
    "    print(\"✅ Extracted column annotations.\")\n",
    "\n",
    "    # 4. Create annotation color maps\n",
    "    col_colors = pd.DataFrame(index=col_annotation_df.index)\n",
    "    palette_cache = {}\n",
    "\n",
    "    palette_map = {\n",
    "        \"condition\": \"Set1\",\n",
    "        \"treatment\": \"Dark2\",\n",
    "        \"time\": \"Paired\"\n",
    "    }\n",
    "\n",
    "    for col in annotation_cols:\n",
    "        unique_vals = sorted(col_annotation_df[col].unique())\n",
    "        real_vals = [v for v in unique_vals if v != \"NA\"]\n",
    "        print(f\"🎨 Mapping colors for {col}: {real_vals}\")\n",
    "        palette_name = palette_map.get(col, \"Set2\")\n",
    "        palette = sns.color_palette(palette_name, len(real_vals))\n",
    "        lut = dict(zip(real_vals, palette))\n",
    "        lut[\"NA\"] = \"#ffffff\"\n",
    "        col_colors[col] = col_annotation_df[col].map(lut)\n",
    "        palette_cache[col] = lut\n",
    "\n",
    "    print(\"✅ Created color palettes for annotations.\")\n",
    "\n",
    "    # 5. Plot the clustermap\n",
    "    print(\"📊 Generating clustermap...\")\n",
    "    g = sns.clustermap(\n",
    "        grapher,\n",
    "        z_score=0,\n",
    "        cmap=\"RdYlBu_r\",\n",
    "        col_cluster=False,\n",
    "        col_colors=col_colors\n",
    "    )\n",
    "\n",
    "    # 6. Add legend\n",
    "    legend_handles = []\n",
    "    for col, lut in palette_cache.items():\n",
    "        for label, color in lut.items():\n",
    "            legend_handles.append(Patch(facecolor=color, label=f\"{col}: {label}\"))\n",
    "\n",
    "    plt.gcf().legend(handles=legend_handles, bbox_to_anchor=(1.02, 1), loc=\"upper left\", title=\"Annotations\")\n",
    "    plt.show()\n",
    "    print(\"✅ Clustermap rendered successfully.\")\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "########## version 2\n",
    "# Step 1: Extract Metadata\n",
    "# -------------------------------\n",
    "\n",
    "# -------------------------------\n",
    "# Step 2: Remap Expression Columns\n",
    "# -------------------------------\n",
    "def extract_json_block(text):\n",
    "    try:\n",
    "        start = text.index(\"{\")\n",
    "        end = text.rindex(\"}\") + 1\n",
    "        return json.loads(text[start:end])\n",
    "    except Exception as e:\n",
    "        print(f\"\\u274c Failed to extract JSON: {e}\")\n",
    "        return {}\n",
    "def parse_series_matrix_to_sample_table(series_matrix_path):\n",
    "    \"\"\"\n",
    "    Parse Series Matrix file into a structured table per sample.\n",
    "    Returns a list of dicts: each dict = all metadata for one sample.\n",
    "    \"\"\"\n",
    "    metadata_fields = {}\n",
    "    with gzip.open(series_matrix_path, \"rt\", encoding=\"utf-8\") as f:\n",
    "        for line in f:\n",
    "            if line.startswith(\"!Sample_\"):\n",
    "                key, *values = line.strip().split(\"\\t\")\n",
    "                key = key.replace(\"!Sample_\", \"\")\n",
    "                metadata_fields.setdefault(key, []).extend(values)\n",
    "\n",
    "    # Transpose: per sample\n",
    "    num_samples = len(next(iter(metadata_fields.values())))\n",
    "    sample_table = []\n",
    "    for i in range(num_samples):\n",
    "        sample_entry = {}\n",
    "        for key, value_list in metadata_fields.items():\n",
    "            if i < len(value_list):\n",
    "                sample_entry[key] = value_list[i].strip('\"')\n",
    "        sample_table.append(sample_entry)\n",
    "\n",
    "    return sample_table\n",
    "\n",
    "\n",
    "def construct_prompt_to_match_samples_and_generate_metadata(sample_table, count_labels, gse_id):\n",
    "    \"\"\"\n",
    "    Construct an LLM prompt to:\n",
    "    - Identify which metadata field contains the count sample labels\n",
    "    - Extract biologically meaningful metadata from each sample\n",
    "    \"\"\"\n",
    "    prompt_data = {\n",
    "        \"GEO\": gse_id,\n",
    "        \"sample_labels_from_counts\": count_labels,\n",
    "        \"samples_metadata\": sample_table[:300]  # send only a subset to stay within token limits\n",
    "    }\n",
    "\n",
    "    prompt = f\"\"\"\n",
    "You are a biomedical data curator.\n",
    "\n",
    "Below is a list of samples from GEO study **{gse_id}**. Each sample contains metadata fields like `geo_accession`, `title`, `description`, and `characteristics`.\n",
    "\n",
    "Also included is a list of **sample labels from the count matrix**, such as:\n",
    "```json\n",
    "#{json.dumps(list(count_labels[:15]), indent=2)}\n",
    "{json.dumps(list(count_labels[:]), indent=2)}\n",
    "    Your tasks:\n",
    "\n",
    "Identify which metadata field (e.g., title, description, characteristics) contains the matching sample labels.\n",
    "\n",
    "Use the matched field to define sample identity (e.g., \"K401\", \"K402\").\n",
    "\n",
    "For each matched sample, generate a JSON object containing:\n",
    "\n",
    "\"condition\": Must be either \"control\" or \"treated\".\n",
    "\n",
    "Any other relevant fields: \"treatment\", \"time\", \"dose\", etc.\n",
    "\n",
    "Return a dictionary mapping the sample labels (e.g., \"K401\") to structured metadata.\n",
    "\n",
    "Here is the metadata table:\n",
    "\n",
    "{json.dumps(prompt_data[\"samples_metadata\"], indent=2)}\n",
    "Only return the final JSON dictionary — no explanations.\n",
    "\"\"\"\n",
    "    return prompt\n",
    "\n",
    "\n",
    "def split_sample_table(sample_table, batch_size=100):\n",
    "    \"\"\"Yield batches of the sample_table.\"\"\"\n",
    "    for i in range(0, len(sample_table), batch_size):\n",
    "        yield sample_table[i:i + batch_size]\n",
    "\n",
    "\n",
    "\n",
    "def extract_labels_using_llm(series_matrix_path, gse_id, client,sample_labels_from_counts):\n",
    "#Full pipeline: parse matrix, build prompt, call LLM, return metadata per count sample label.\n",
    "\n",
    "    print(f\"📂 Parsing series matrix: {series_matrix_path}\")\n",
    "    sample_table = parse_series_matrix_to_sample_table(series_matrix_path)\n",
    "    print(f\"✅ Parsed {len(sample_table)} samples\")\n",
    "    print(f\"🧠 Building prompt for {len(sample_labels_from_counts)} sample labels...\")\n",
    "    prompt = construct_prompt_to_match_samples_and_generate_metadata(sample_table, sample_labels_from_counts, gse_id)\n",
    "    \n",
    "    print(\"💬 Sending prompt to LLM...\")\n",
    "    response = client.chat.completions.create(\n",
    "        model=\"gpt-4o\",\n",
    "        temperature=0.2,\n",
    "        messages=[{\"role\": \"user\", \"content\": prompt}]\n",
    "    )\n",
    "    \n",
    "    content = response.choices[0].message.content.strip()\n",
    "    print(f\"🧾 LLM Response preview:\\n{content[:300]}...\")\n",
    "    \n",
    "    metadata_by_label = extract_json_block(content)\n",
    "    print(f\"✅ Extracted metadata for {len(metadata_by_label)} sample labels\")\n",
    "    \n",
    "    return metadata_by_label\n",
    "\n",
    "#################\n",
    "def extract_labels_using_llm(series_matrix_path, gse_id, client, sample_labels_from_counts, batch_size=100):\n",
    "    \"\"\"\n",
    "    Full pipeline: parse matrix, build prompt, call LLM for batches, return combined metadata.\n",
    "    \"\"\"\n",
    "    print(f\"📂 Parsing series matrix: {series_matrix_path}\")\n",
    "    sample_table = parse_series_matrix_to_sample_table(series_matrix_path)\n",
    "    print(f\"✅ Parsed {len(sample_table)} samples\")\n",
    "\n",
    "    all_metadata = {}\n",
    "    for batch_num, batch_table in enumerate(split_sample_table(sample_table, batch_size=batch_size), 1):\n",
    "        print(f\"🧠 Building prompt for batch {batch_num} with {len(batch_table)} samples...\")\n",
    "        prompt = construct_prompt_to_match_samples_and_generate_metadata(batch_table, sample_labels_from_counts, gse_id)\n",
    "\n",
    "        print(f\"💬 Sending batch {batch_num} prompt to LLM...\")\n",
    "        response = client.chat.completions.create(\n",
    "            model=\"gpt-4o\",\n",
    "            temperature=0.2,\n",
    "            messages=[{\"role\": \"user\", \"content\": prompt}]\n",
    "        )\n",
    "\n",
    "        content = response.choices[0].message.content.strip()\n",
    "        print(f\"🧾 Batch {batch_num} LLM Response preview:\\n{content[:300]}...\")\n",
    "\n",
    "        batch_metadata = extract_json_block(content)\n",
    "        print(f\"✅ Extracted metadata for {len(batch_metadata)} sample labels in batch {batch_num}\")\n",
    "\n",
    "        all_metadata.update(batch_metadata)\n",
    "\n",
    "    print(f\"🎉 Completed extraction for {len(all_metadata)} sample labels across all batches\")\n",
    "    return all_metadata\n",
    "\n",
    "#############\n",
    "\n",
    "\n",
    "######## new #########\n",
    "\n",
    "\n",
    "###############\n",
    "\n",
    "\n",
    "########## test \n",
    "def run_deg_for_single_gse_debug(state):\n",
    "    gse_id = state.get(\"selected_gse\")\n",
    "    if not gse_id:\n",
    "        print(\"No selected GSE found.\")\n",
    "        return state\n",
    "\n",
    "    deg_paths = {}\n",
    "    decompress_gse_in_dir(gse_id)\n",
    "    series_matrix_path = f\"geofetch_metadata/{gse_id}_series_matrix.txt.gz\"\n",
    "\n",
    "    try:\n",
    "        print(f\"Running DESeq2 for: {gse_id}\")\n",
    "        download_dir = download_and_extract_gse(gse_id)\n",
    "        counts_df = find_and_merge_count_files(gse_id)\n",
    "\n",
    "        if counts_df is None:\n",
    "            print(f\"⚠️ No count matrix found for {gse_id}. Trying to load DE results directly...\")\n",
    "            gse_dir = Path(\"rna_seq_analysis\") / gse_id\n",
    "            for file in gse_dir.glob(\"*\"):\n",
    "                if file.suffix in [\".txt\", \".tsv\", \".csv\", \".gz\"]:\n",
    "                    de_df = load_de_result_file(file)\n",
    "                    if de_df is not None:\n",
    "                        output_csv = f\"rna_seq_analysis/{gse_id}_de_results_direct.csv\"\n",
    "                        de_df.to_csv(output_csv)\n",
    "                        deg_paths[gse_id] = output_csv\n",
    "                        print(f\"✅ DE result loaded and saved: {output_csv}\")\n",
    "                        return {**state, \"deg_results\": deg_paths}\n",
    "            print(f\"❌ No DE result files found in {gse_dir}\")\n",
    "            return state\n",
    "\n",
    "        print(counts_df)\n",
    "\n",
    "        sample_labels_from_counts = counts_df.columns\n",
    "        group_label_mapping, name_map = extract_labels_using_llm(\n",
    "            series_matrix_path, gse_id, client, sample_labels_from_counts\n",
    "        )\n",
    "\n",
    "        # Rename counts_df columns to unique names\n",
    "        col_map = {orig: name_map.get(orig, orig) for orig in counts_df.columns}\n",
    "        counts_df = counts_df.rename(columns=col_map)\n",
    "\n",
    "        # Build metadata DataFrame and rename to unique names\n",
    "        metadata_df = pd.DataFrame.from_dict(group_label_mapping, orient=\"index\")\n",
    "        metadata_df.index = [name_map.get(idx, idx) for idx in metadata_df.index]\n",
    "        metadata_df.index.name = \"sample_geo_accession\"\n",
    "\n",
    "        # Align counts + metadata\n",
    "        valid_samples = metadata_df.index.intersection(counts_df.columns)\n",
    "        counts_df = counts_df[valid_samples]\n",
    "        metadata_df = metadata_df.loc[valid_samples]\n",
    "\n",
    "        if counts_df.shape[1] == 0:\n",
    "            raise ValueError(\"❌ No valid samples left after alignment — cannot run DESeq2.\")\n",
    "\n",
    "        tcounts = counts_df.T\n",
    "        inference = DefaultInference(n_cpus=4)\n",
    "        dds = DeseqDataSet(\n",
    "            counts=tcounts,\n",
    "            metadata=metadata_df,\n",
    "            design_factors=\"condition\",\n",
    "            refit_cooks=True,\n",
    "            inference=inference,\n",
    "        )\n",
    "\n",
    "        dds.fit_size_factors()\n",
    "        dds.fit_genewise_dispersions()\n",
    "        dds.fit_dispersion_trend()\n",
    "        dds.fit_dispersion_prior()\n",
    "        print(f\"logres_prior={dds.uns['_squared_logres']}, sigma_prior={dds.uns['prior_disp_var']}\")\n",
    "        dds.fit_MAP_dispersions()\n",
    "        dds.fit_LFC()\n",
    "        dds.calculate_cooks()\n",
    "\n",
    "        if dds.refit_cooks:\n",
    "            dds.refit()\n",
    "\n",
    "        stat_res = DeseqStats(\n",
    "            dds, alpha=0.05, contrast=[\"condition\", \"treated\", \"control\"],\n",
    "            cooks_filter=True, independent_filter=True\n",
    "        )\n",
    "        stat_res.summary()\n",
    "        res = stat_res.results_df\n",
    "        output_csv = f\"rna_seq_analysis/{gse_id}_deseq2_results.csv\"\n",
    "        res.to_csv(output_csv, index=False)\n",
    "        deg_paths[gse_id] = output_csv\n",
    "        print(f\"✅ DESeq2 results saved: {output_csv}\")\n",
    "\n",
    "        filtered_df = res[\n",
    "            (res[\"padj\"] < 0.05) & (res[\"log2FoldChange\"].abs() > 1)\n",
    "        ]\n",
    "        print(f\"🧬 {filtered_df.shape[0]} significant genes found\")\n",
    "\n",
    "        # Additional visualization + analysis\n",
    "        dds.layers[\"log1p\"] = np.log1p(dds.layers[\"normed_counts\"])\n",
    "        sigs = res[(res.padj < 0.05) & (abs(res.log2FoldChange) > 1.5)]\n",
    "        dds_sigs = dds[:, sigs.index]\n",
    "\n",
    "        grapher = pd.DataFrame(\n",
    "            dds_sigs.layers['log1p'],\n",
    "            index=dds_sigs.obs_names,\n",
    "            columns=dds_sigs.var_names\n",
    "        ).T\n",
    "        grapher.index.name = \"gene\"\n",
    "        grapher_cp = grapher.copy()\n",
    "\n",
    "        sample_names = list(dds_sigs.obs_names)\n",
    "        sample_labels = [group_label_mapping[sample][\"treatment\"] for sample in sample_names]\n",
    "        labeled_columns = [f\"{sample}_{label}\" for sample, label in zip(sample_labels, sample_names)]\n",
    "        grapher.columns = labeled_columns\n",
    "\n",
    "        sc.tl.pca(dds)\n",
    "        stat_res.lfc_shrink(coeff=\"condition[T.treated]\")\n",
    "        sc.pl.pca(dds, color=\"condition\", size=200)\n",
    "\n",
    "        res[\"symbol\"] = res.index\n",
    "        volcano(\n",
    "            res, symbol='symbol', log2fc='log2FoldChange',\n",
    "            pval_thresh=0.05, to_label=10, log2fc_thresh=0.75,\n",
    "            colors=['dimgrey', 'lightgrey', 'green']\n",
    "        )\n",
    "        plt.show()\n",
    "\n",
    "        annotate_clustermap_with_metadata(grapher_cp, metadata_df)\n",
    "\n",
    "    except Exception as e:\n",
    "        print(f\"Error running DESeq2 for {gse_id}: {e}\")\n",
    "\n",
    "    return {**state, \"deg_results\": deg_paths}\n",
    "\n",
    "\n",
    "\n",
    "def parse_series_matrix_to_dataframe(series_matrix_path):\n",
    "    data = {}\n",
    "    sample_ids = None\n",
    "    field_counter = defaultdict(int)\n",
    "\n",
    "    open_func = gzip.open if series_matrix_path.endswith(\".gz\") else open\n",
    "    with open_func(series_matrix_path, 'rt', encoding='utf-8') as f:\n",
    "        for line in f:\n",
    "            if line.startswith(\"!Sample_\"):\n",
    "                parts = line.strip().split(\"\\t\")\n",
    "                raw_field = parts[0].lstrip(\"!\")\n",
    "                count = field_counter[raw_field]\n",
    "                field_name = raw_field if count == 0 else f\"{raw_field}_{count}\"\n",
    "                field_counter[raw_field] += 1\n",
    "                values = [p.strip('\"') for p in parts[1:]]\n",
    "                data[field_name] = values\n",
    "                if raw_field == \"Sample_geo_accession\":\n",
    "                    sample_ids = values\n",
    "\n",
    "    if sample_ids is None:\n",
    "        raise ValueError(\"❌ No Sample_geo_accession found.\")\n",
    "\n",
    "    df = pd.DataFrame(data)\n",
    "    df.columns = df.columns.str.replace(r'\\\\', '', regex=True).str.strip()\n",
    "    df.index = sample_ids\n",
    "    df.index.name = \"geo_accession\"\n",
    "    print(f\"✅ Series matrix parsed: {df.shape}\")\n",
    "    return df\n",
    "\n",
    "def send_mini_table_to_llm(client, gse_id, mini_df):\n",
    "    records = mini_df.reset_index().to_dict(orient=\"records\")\n",
    "    prompt = f\"\"\"\n",
    "You are a biomedical data curator assisting with RNA-seq DESeq2 analysis.\n",
    "\n",
    "Below is a metadata table for GEO study {gse_id}. Each row represents a sample.\n",
    "\n",
    "Your task:\n",
    "- Identify the columns that contain key information useful for defining groups for differential expression analysis.\n",
    "- Focus on metadata about: *treatment*, *time*, *gender*, *dose*, *genotype*, *WT/KO*.\n",
    "- Return ONLY a JSON list of useful column names. No explanation.\n",
    "\n",
    "Here is the metadata:\n",
    "{json.dumps(records, indent=2)}\n",
    "\"\"\"\n",
    "    print(f\"📝 Sending LLM prompt (first 1000 chars): {prompt[:1000]}...\")\n",
    "    response = client.chat.completions.create(\n",
    "        model=\"gpt-4o\",\n",
    "        temperature=0.2,\n",
    "        messages=[{\"role\": \"user\", \"content\": prompt}]\n",
    "    )\n",
    "    content = response.choices[0].message.content.strip()\n",
    "    content = re.sub(r\"^```json\", \"\", content, flags=re.IGNORECASE).strip()\n",
    "    content = re.sub(r\"```$\", \"\", content).strip()\n",
    "    return json.loads(content)\n",
    "\n",
    "def llm_curate_conditions(client, gse_id, metadata_df, batch_size=5):\n",
    "    records = metadata_df.reset_index().to_dict(orient=\"records\")\n",
    "    all_map = {}\n",
    "    for i in range(0, len(records), batch_size):\n",
    "        batch_records = records[i:i+batch_size]\n",
    "        prompt = f\"\"\"\n",
    "You are a biomedical data curator assisting with RNA-seq DESeq2 analysis.\n",
    "\n",
    "Below is a metadata table for GEO study {gse_id}. Each row represents a sample.\n",
    "\n",
    "Your task:\n",
    "- Determine if each sample represents a \"treated\" or \"control\" condition.\n",
    "- Use the metadata fields provided (treatment, time, genotype, cytokines like IL etc).\n",
    "- Return ONLY a JSON mapping geo_accession → \"treated\" or \"control\". No explanation.\n",
    "\n",
    "Here is the metadata:\n",
    "{json.dumps(batch_records, indent=2)}\n",
    "\"\"\"\n",
    "        print(f\"📝 Sending LLM prompt batch {i//batch_size+1} (first 500 chars): {prompt[:500]}...\")\n",
    "        response = client.chat.completions.create(\n",
    "            model=\"gpt-4o\",\n",
    "            temperature=0.2,\n",
    "            messages=[{\"role\": \"user\", \"content\": prompt}]\n",
    "        )\n",
    "        content = response.choices[0].message.content.strip()\n",
    "        content = re.sub(r\"^```json\", \"\", content, flags=re.IGNORECASE).strip()\n",
    "        content = re.sub(r\"```$\", \"\", content).strip()\n",
    "        batch_map = json.loads(content)\n",
    "        all_map.update(batch_map)\n",
    "    print(f\"✅ LLM returned conditions for {len(all_map)} samples\")\n",
    "    return all_map\n",
    "\n",
    "def run_deg_for_single_gse_run(state, client):\n",
    "    gse_id = state.get(\"selected_gse\")\n",
    "    if not gse_id:\n",
    "        print(\"❌ No selected GSE found.\")\n",
    "        return state\n",
    "\n",
    "    deg_paths = {}\n",
    "    series_matrix_path = f\"geofetch_metadata/{gse_id}_series_matrix.txt.gz\"\n",
    "    decompress_gse_in_dir(gse_id)\n",
    "\n",
    "    try:\n",
    "        series_df = parse_series_matrix_to_dataframe(series_matrix_path)\n",
    "        mini_df = series_df.sample(n=min(6, len(series_df)), random_state=42)\n",
    "        print(mini_df)\n",
    "\n",
    "        useful_cols = send_mini_table_to_llm(client, gse_id, mini_df)\n",
    "        metadata_df = series_df[useful_cols].copy()\n",
    "        metadata_df.index.name = \"sample_geo_accession\"\n",
    "\n",
    "        condition_map = llm_curate_conditions(client, gse_id, metadata_df)\n",
    "        metadata_df[\"condition\"] = metadata_df.index.map(condition_map)\n",
    "\n",
    "        download_dir = download_and_extract_gse(gse_id)\n",
    "        counts_df = find_and_merge_count_files(gse_id)\n",
    "        if counts_df is None:\n",
    "            print(f\"⚠️ No count matrix found for {gse_id}\")\n",
    "            return state\n",
    "\n",
    "        valid_samples = metadata_df.index.intersection(counts_df.columns)\n",
    "        counts_df = counts_df[valid_samples].T\n",
    "        metadata_df = metadata_df.loc[valid_samples]\n",
    "        metadata_df = metadata_df[metadata_df[\"condition\"].isin([\"treated\", \"control\"])]\n",
    "\n",
    "        if len(metadata_df[\"condition\"].unique()) < 2:\n",
    "            raise ValueError(\"❌ Need both treated and control groups for DESeq2.\")\n",
    "\n",
    "        print(f\"✅ Counts shape: {counts_df.shape}\")\n",
    "        print(f\"✅ Metadata shape: {metadata_df.shape}\")\n",
    "        print(f\"✅ Condition counts:\\n{metadata_df['condition'].value_counts()}\")\n",
    "\n",
    "        inference = DefaultInference(n_cpus=4)\n",
    "        dds = DeseqDataSet(\n",
    "            counts=counts_df,\n",
    "            metadata=metadata_df,\n",
    "            design_factors=\"condition\",\n",
    "            refit_cooks=True,\n",
    "            inference=inference\n",
    "        )\n",
    "        dds.fit_size_factors()\n",
    "        dds.fit_genewise_dispersions()\n",
    "        dds.fit_dispersion_trend()\n",
    "        dds.fit_dispersion_prior()\n",
    "        dds.fit_MAP_dispersions()\n",
    "        dds.fit_LFC()\n",
    "        dds.calculate_cooks()\n",
    "        if dds.refit_cooks:\n",
    "            dds.refit()\n",
    "\n",
    "        stat_res = DeseqStats(\n",
    "            dds, alpha=0.05, contrast=[\"condition\", \"treated\", \"control\"],\n",
    "            cooks_filter=True, independent_filter=True\n",
    "        )\n",
    "        stat_res.summary()\n",
    "\n",
    "        res = stat_res.results_df\n",
    "        output_csv = f\"rna_seq_analysis/{gse_id}_deseq2_results.csv\"\n",
    "        res.to_csv(output_csv, index=False)\n",
    "        deg_paths[gse_id] = output_csv\n",
    "        print(f\"✅ DESeq2 results saved: {output_csv}\")\n",
    "\n",
    "        sig_genes = res[(res[\"padj\"] < 0.05) & (abs(res[\"log2FoldChange\"]) > 1)]\n",
    "        print(f\"🧬 {sig_genes.shape[0]} significant genes found\")\n",
    "        # Additional visualization + analysis\n",
    "        dds.layers[\"log1p\"] = np.log1p(dds.layers[\"normed_counts\"])\n",
    "        sigs = res[(res.padj < 0.05) & (abs(res.log2FoldChange) > 1.5)]\n",
    "        dds_sigs = dds[:, sigs.index]\n",
    "\n",
    "        grapher = pd.DataFrame(\n",
    "            dds_sigs.layers['log1p'],\n",
    "            index=dds_sigs.obs_names,\n",
    "            columns=dds_sigs.var_names\n",
    "        ).T\n",
    "        grapher.index.name = \"gene\"\n",
    "        grapher_cp = grapher.copy()\n",
    "\n",
    "        sample_names = list(dds_sigs.obs_names)\n",
    "        sample_labels = [group_label_mapping[sample][\"treatment\"] for sample in sample_names]\n",
    "        labeled_columns = [f\"{sample}_{label}\" for sample, label in zip(sample_labels, sample_names)]\n",
    "        grapher.columns = labeled_columns\n",
    "\n",
    "        sc.tl.pca(dds)\n",
    "        stat_res.lfc_shrink(coeff=\"condition[T.treated]\")\n",
    "        sc.pl.pca(dds, color=\"condition\", size=200)\n",
    "\n",
    "        res[\"symbol\"] = res.index\n",
    "        volcano(\n",
    "            res, symbol='symbol', log2fc='log2FoldChange',\n",
    "            pval_thresh=0.05, to_label=10, log2fc_thresh=0.75,\n",
    "            colors=['dimgrey', 'lightgrey', 'green']\n",
    "        )\n",
    "        plt.show()\n",
    "\n",
    "        annotate_clustermap_with_metadata(grapher_cp, metadata_df)\n",
    "\n",
    "    except Exception as e:\n",
    "        print(f\"Error running DESeq2 for {gse_id}: {e}\")\n",
    "\n",
    "    return {**state, \"deg_results\": deg_paths}\n",
    "   \n",
    "\n",
    "############# test agent 1 and 2 connection\n",
    "# Build the pipeline\n",
    "workflow = StateGraph(state_schema=AgentState)\n",
    "\n",
    "# Add nodes\n",
    "workflow.add_node(\"Ingestor\", RunnableLambda(ingest_and_prepare))\n",
    "workflow.add_node(\"Analyst\", RunnableLambda(analyze_metadata_and_plan))\n",
    "workflow.add_node(\"GSEListExtractor\", RunnableLambda(extract_gse_list_from_plan))\n",
    "workflow.add_node(\"DEGRunner\", RunnableLambda(run_deg_for_multiple_gses_debug))\n",
    "# Connect nodes\n",
    "\n",
    "workflow.set_entry_point(\"Ingestor\")\n",
    "workflow.add_edge(\"Ingestor\", \"Analyst\")\n",
    "workflow.add_edge(\"Analyst\", \"GSEListExtractor\")\n",
    "workflow.add_edge(\"GSEListExtractor\", \"DEGRunner\")\n",
    "\n",
    "# Final output node\n",
    "workflow.set_finish_point(\"DEGRunner\")\n",
    "#workflow.set_finish_point(\"GSEListExtractor\")\n",
    "# Compile the graph\n",
    "graph = workflow.compile()\n",
    "from IPython.display import Image, display\n",
    "try:\n",
    "    display(Image(graph.get_graph().draw_mermaid_png()))\n",
    "except Exception:\n",
    "    # This requires some extra dependencies and is optional\n",
    "    pass\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "5b91160d-30e1-427b-a522-da3bf270161f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'genes': ['JMJD1C', 'JMJD1A', 'JMJD2A', 'KDM3A', 'KDM4A', 'KDM3B'],\n",
       " 'drugs': ['GSK-J4',\n",
       "  'GSK2879552',\n",
       "  'IOX1',\n",
       "  'KDM4A Inhibitor',\n",
       "  'KDM5A Inhibitor'],\n",
       " 'pathways': ['Chromatin remodeling',\n",
       "  'Transcriptional regulation',\n",
       "  'Cell growth and differentiation',\n",
       "  'Cancer',\n",
       "  'Metabolic disorders',\n",
       "  'Neurodegenerative diseases']}"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## test function\n",
    "generate_pathway_info(\"JMJD1C\",model=\"gpt-4o-mini\", temperature=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "61452609-fcbe-429e-91ee-284443cc5549",
   "metadata": {},
   "outputs": [],
   "source": [
    "initial_state = AgentState(\n",
    "    query_gene=\"JMJD1C\",\n",
    "    pathway_info={\"genes\": [], \"drugs\": [], \"pathways\": []},\n",
    "    metadata={},\n",
    "    gse_list=[],\n",
    "    research_plan=\"\",\n",
    "    series_matrix_path=\"\",\n",
    "    selected_gse=\"\",\n",
    "    deg_results={}\n",
    ")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "423bf721-c0d9-4ae5-9eaf-9e7f0b6c3b6e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "🔍 Searching GEO for gene: JMJD1C\n",
      "🔍 Searching GEO for gene: JMJD1A\n",
      "🔍 Searching GEO for gene: JMJD2A\n",
      "🔍 Searching GEO for gene: JMJD2B\n",
      "🔍 Searching GEO for gene: JMJD2C\n",
      "🔍 Searching GEO for gene: JMJD2D\n",
      "🔍 Searching GEO for gene: KDM3A\n",
      "🔍 Searching GEO for gene: KDM4A\n",
      "🔍 Searching GEO for drug: GSK-J4\n",
      "🔍 Searching GEO for drug: IOX1\n",
      "🔍 Searching GEO for drug: BAY-850\n",
      "🔍 Searching GEO for drug: BAY-872\n",
      "🔍 Searching GEO for drug: PXD101\n",
      "📥 Downloading GSE193726_RAW.tar → rna_seq_analysis/GSE193726/GSE193726_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "📥 Downloading GSE269325_Kdm4a_KD_vs_WT_RNA-seq.txt.gz → rna_seq_analysis/GSE269325/GSE269325_Kdm4a_KD_vs_WT_RNA-seq.txt.gz\n",
      "🗂️ Extracting GZ...\n",
      "📥 Downloading GSE236921_RAW.tar → rna_seq_analysis/GSE236921/GSE236921_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "📥 Downloading GSE192673_rsem_count.txt.gz → rna_seq_analysis/GSE192673/GSE192673_rsem_count.txt.gz\n",
      "🗂️ Extracting GZ...\n",
      "📥 Downloading GSE148055_RAW.tar → rna_seq_analysis/GSE148055/GSE148055_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "📁 Skipping GSE139437: folder already exists.\n",
      "📥 Downloading GSE96686_RAW.tar → rna_seq_analysis/GSE96686/GSE96686_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "📥 Downloading GSE143151_RAW.tar → rna_seq_analysis/GSE143151/GSE143151_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "📥 Downloading GSE146833_6419c56694c2Kdm3aKO.vs.WT.batch.salmon.GeneTPM.combined.txt.gz → rna_seq_analysis/GSE146833/GSE146833_6419c56694c2Kdm3aKO.vs.WT.batch.salmon.GeneTPM.combined.txt.gz\n",
      "🗂️ Extracting GZ...\n",
      "⚠️ No supplementary files found for GSM5754577\n",
      "⚠️ No supplementary files found for GSM5754578\n",
      "📥 Downloading GSE279335_Normalized_count_matrix_for_RH30_cells_YK_vs_control.txt.gz → rna_seq_analysis/GSE279335/GSE279335_Normalized_count_matrix_for_RH30_cells_YK_vs_control.txt.gz\n",
      "🗂️ Extracting GZ...\n",
      "📥 Downloading GSE173995_RAW.tar → rna_seq_analysis/GSE173995/GSE173995_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "📥 Downloading GSE135633_RAW.tar → rna_seq_analysis/GSE135633/GSE135633_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "📥 Downloading GSE202506_RAW.tar → rna_seq_analysis/GSE202506/GSE202506_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "📥 Downloading GSE148053_RAW.tar → rna_seq_analysis/GSE148053/GSE148053_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "📥 Downloading GSE269147_RAW.tar → rna_seq_analysis/GSE269147/GSE269147_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "📥 Downloading GSE106872_RAW.tar → rna_seq_analysis/GSE106872/GSE106872_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "📥 Downloading GSE108090_inhibitor_expr.csv.gz → rna_seq_analysis/GSE108090/GSE108090_inhibitor_expr.csv.gz\n",
      "🗂️ Extracting GZ...\n",
      "📥 Downloading GSE221336_gene_FPKM.annot.txt.gz → rna_seq_analysis/GSE221336/GSE221336_gene_FPKM.annot.txt.gz\n",
      "🗂️ Extracting GZ...\n",
      "📥 Downloading GSE219007_RAW.tar → rna_seq_analysis/GSE219007/GSE219007_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "📥 Downloading GSE217127_RAW.tar → rna_seq_analysis/GSE217127/GSE217127_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "📥 Downloading GSE256159_I_GC-all_feature_matrix.txt.gz → rna_seq_analysis/GSE256159/GSE256159_I_GC-all_feature_matrix.txt.gz\n",
      "🗂️ Extracting GZ...\n",
      "📥 Downloading GSE274905_HGSOC_Dini.et.al_UMAP_embeddings.csv.gz → rna_seq_analysis/GSE274905/GSE274905_HGSOC_Dini.et.al_UMAP_embeddings.csv.gz\n",
      "🗂️ Extracting GZ...\n",
      "📥 Downloading GSE208446_Gene_FPKM_uqnorm.txt.gz → rna_seq_analysis/GSE208446/GSE208446_Gene_FPKM_uqnorm.txt.gz\n",
      "🗂️ Extracting GZ...\n",
      "📥 Downloading GSE148054_RAW.tar → rna_seq_analysis/GSE148054/GSE148054_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "📥 Downloading GSE180233_RAW.tar → rna_seq_analysis/GSE180233/GSE180233_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "📥 Downloading GSE143148_Gene_expression.tsv.gz → rna_seq_analysis/GSE143148/GSE143148_Gene_expression.tsv.gz\n",
      "🗂️ Extracting GZ...\n",
      "📥 Downloading GSE199913_HCC1806_shKDM4C_QC6352_counts_matrix.csv.gz → rna_seq_analysis/GSE199913/GSE199913_HCC1806_shKDM4C_QC6352_counts_matrix.csv.gz\n",
      "🗂️ Extracting GZ...\n",
      "📥 Downloading GSE251728_Kasumi1_shRUNX1_count.txt.gz → rna_seq_analysis/GSE251728/GSE251728_Kasumi1_shRUNX1_count.txt.gz\n",
      "🗂️ Extracting GZ...\n",
      "⚠️ No supplementary files found for GSM5754576\n",
      "📥 Downloading GSE241247_RNA-seq_Zscore_all_0_delete.csv.gz → rna_seq_analysis/GSE241247/GSE241247_RNA-seq_Zscore_all_0_delete.csv.gz\n",
      "🗂️ Extracting GZ...\n",
      "⚠️ No supplementary files found for GSE290475\n",
      "📥 Downloading GSE135632_RAW.tar → rna_seq_analysis/GSE135632/GSE135632_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "📥 Downloading GSE224611_RAW.tar → rna_seq_analysis/GSE224611/GSE224611_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "📥 Downloading GSE166351_RAW.tar → rna_seq_analysis/GSE166351/GSE166351_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "📥 Downloading GSE134120_genes.tsv.gz → rna_seq_analysis/GSE134120/GSE134120_genes.tsv.gz\n",
      "🗂️ Extracting GZ...\n",
      "📥 Downloading GSE135634_RAW.tar → rna_seq_analysis/GSE135634/GSE135634_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "⚠️ No supplementary files found for GSE269020\n",
      "⚠️ No supplementary files found for GSM5754579\n",
      "⚠️ No supplementary files found for GSE149233\n",
      "⚠️ No supplementary files found for GSE200900\n",
      "📥 Downloading GSE117865_RAW.tar → rna_seq_analysis/GSE117865/GSE117865_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "📥 Downloading GSE256469_All_peakNum.xls.gz → rna_seq_analysis/GSE256469/GSE256469_All_peakNum.xls.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:30]\u001b[0m Metadata folder: C:\\Users\\difen\\POPPER\\geofetch_metadata\\project_name\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:30]\u001b[0m Trying GSE193726 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:30]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE193726'\u001b[0m\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:30]\u001b[0m Trying GSE193726 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:31]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:31]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE193726'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "🗂️ Extracting GZ...\n",
      "📥 Fetching metadata for GSE193726\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:34]\u001b[0m \n",
      "Total number of processed SAMPLES files found is: 16\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:34]\u001b[0m Total number of processed SERIES files found is: 0\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:34]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:34]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:34]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:34]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:34]\u001b[0m Unifying and saving of metadata... \n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:34]\u001b[0m Trying GSE269325 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:34]\u001b[0m Trying GSE269325 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:34]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:34]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE269325'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE269325\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:35]\u001b[0m Total number of processed SERIES files found is: 1\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:35]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:35]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:35]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:35]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:35]\u001b[0m No files found. No data to save. File geofetch_metadata\\project_name\\GSE269325_samples\\GSE269325_samples.csv won't be created\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:35]\u001b[0m Trying GSE236921 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:35]\u001b[0m Trying GSE236921 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:35]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:35]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE236921'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE236921\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:38]\u001b[0m \n",
      "Total number of processed SAMPLES files found is: 70\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:38]\u001b[0m Total number of processed SERIES files found is: 0\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:38]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:38]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:38]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:38]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:38]\u001b[0m Unifying and saving of metadata... \n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:38]\u001b[0m Trying GSE192673 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:38]\u001b[0m Trying GSE192673 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:38]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:38]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE192673'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE192673\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:40]\u001b[0m Total number of processed SERIES files found is: 1\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:40]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:40]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:40]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:40]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:40]\u001b[0m No files found. No data to save. File geofetch_metadata\\project_name\\GSE192673_samples\\GSE192673_samples.csv won't be created\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:40]\u001b[0m Trying GSE148055 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:40]\u001b[0m Trying GSE148055 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:40]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:40]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE148055'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE148055\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:41]\u001b[0m \n",
      "Total number of processed SAMPLES files found is: 18\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:41]\u001b[0m Total number of processed SERIES files found is: 0\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:41]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:41]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:41]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:41]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:41]\u001b[0m Unifying and saving of metadata... \n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:41]\u001b[0m Trying GSE139437 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:41]\u001b[0m Trying GSE139437 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:41]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:41]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE139437'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE139437\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:51]\u001b[0m \n",
      "Total number of processed SAMPLES files found is: 7\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:51]\u001b[0m Total number of processed SERIES files found is: 0\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:51]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:51]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:51]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:51]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:51]\u001b[0m Unifying and saving of metadata... \n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:51]\u001b[0m Trying GSE96686 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:51]\u001b[0m Trying GSE96686 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:51]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:51]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE96686'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE96686\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:53]\u001b[0m \n",
      "Total number of processed SAMPLES files found is: 12\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:53]\u001b[0m Total number of processed SERIES files found is: 0\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:53]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:53]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:53]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:53]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:53]\u001b[0m Unifying and saving of metadata... \n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:53]\u001b[0m Trying GSE143151 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:53]\u001b[0m Trying GSE143151 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:53]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:53]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE143151'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE143151\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:57]\u001b[0m \n",
      "Total number of processed SAMPLES files found is: 18\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:57]\u001b[0m Total number of processed SERIES files found is: 0\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:57]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:57]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:57]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:57]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:57]\u001b[0m Unifying and saving of metadata... \n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:57]\u001b[0m Trying GSE146833 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:57]\u001b[0m Trying GSE146833 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:57]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:36:57]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE146833'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE146833\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:07]\u001b[0m Total number of processed SERIES files found is: 1\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:07]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:07]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:07]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:07]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:07]\u001b[0m No files found. No data to save. File geofetch_metadata\\project_name\\GSE146833_samples\\GSE146833_samples.csv won't be created\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:07]\u001b[0m Trying GSM5754577 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:07]\u001b[0m Trying GSM5754577 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:07]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:07]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSM5754577'\u001b[0m\n",
      "\u001b[1;30m[WARNING]\u001b[0m \u001b[32m[22:37:07]\u001b[0m \u001b[33mThis does not appear to be a correctly formatted GSE accession! Continue anyway...\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSM5754577\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[WARNING]\u001b[0m \u001b[32m[22:37:07]\u001b[0m \u001b[33mError in checking size of soft file, continuing... 'content-length'\u001b[0m\n",
      "\u001b[1;30m[WARNING]\u001b[0m \u001b[32m[22:37:08]\u001b[0m \u001b[33mError in checking size of soft file, continuing... 'content-length'\u001b[0m\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:10]\u001b[0m Total number of processed SERIES files found is: 0\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:10]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:10]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:10]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:10]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:10]\u001b[0m No files found. No data to save. File geofetch_metadata\\project_name\\GSM5754577_samples\\GSM5754577_samples.csv won't be created\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:10]\u001b[0m Trying GSM5754578 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:10]\u001b[0m Trying GSM5754578 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:10]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:10]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSM5754578'\u001b[0m\n",
      "\u001b[1;30m[WARNING]\u001b[0m \u001b[32m[22:37:10]\u001b[0m \u001b[33mThis does not appear to be a correctly formatted GSE accession! Continue anyway...\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSM5754578\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[WARNING]\u001b[0m \u001b[32m[22:37:10]\u001b[0m \u001b[33mError in checking size of soft file, continuing... 'content-length'\u001b[0m\n",
      "\u001b[1;30m[WARNING]\u001b[0m \u001b[32m[22:37:10]\u001b[0m \u001b[33mError in checking size of soft file, continuing... 'content-length'\u001b[0m\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:11]\u001b[0m Total number of processed SERIES files found is: 0\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:11]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:11]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:11]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:11]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:11]\u001b[0m No files found. No data to save. File geofetch_metadata\\project_name\\GSM5754578_samples\\GSM5754578_samples.csv won't be created\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:11]\u001b[0m Trying GSE279335 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:11]\u001b[0m Trying GSE279335 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:11]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:11]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE279335'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE279335\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:15]\u001b[0m Total number of processed SERIES files found is: 1\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:15]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:15]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:15]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:15]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:15]\u001b[0m No files found. No data to save. File geofetch_metadata\\project_name\\GSE279335_samples\\GSE279335_samples.csv won't be created\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:15]\u001b[0m Trying GSE173995 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:15]\u001b[0m Trying GSE173995 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:15]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:15]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE173995'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE173995\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:17]\u001b[0m \n",
      "Total number of processed SAMPLES files found is: 12\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:17]\u001b[0m Total number of processed SERIES files found is: 1\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:17]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:17]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:17]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:17]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:17]\u001b[0m Unifying and saving of metadata... \n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:17]\u001b[0m Trying GSE135633 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:17]\u001b[0m Trying GSE135633 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:17]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:17]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE135633'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE135633\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:20]\u001b[0m \n",
      "Total number of processed SAMPLES files found is: 7\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:20]\u001b[0m Total number of processed SERIES files found is: 0\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:20]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:20]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:20]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:20]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:20]\u001b[0m Unifying and saving of metadata... \n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:20]\u001b[0m Trying GSE202506 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:20]\u001b[0m Trying GSE202506 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:20]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:20]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE202506'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE202506\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:22]\u001b[0m \n",
      "Total number of processed SAMPLES files found is: 8\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:22]\u001b[0m Total number of processed SERIES files found is: 0\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:22]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:22]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:22]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:22]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:22]\u001b[0m Unifying and saving of metadata... \n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:22]\u001b[0m Trying GSE148053 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:22]\u001b[0m Trying GSE148053 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:22]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:22]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE148053'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE148053\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:24]\u001b[0m \n",
      "Total number of processed SAMPLES files found is: 14\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:24]\u001b[0m Total number of processed SERIES files found is: 0\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:24]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:24]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:24]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:24]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:24]\u001b[0m Unifying and saving of metadata... \n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:24]\u001b[0m Trying GSE269147 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:24]\u001b[0m Trying GSE269147 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:24]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:24]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE269147'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE269147\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:26]\u001b[0m \n",
      "Total number of processed SAMPLES files found is: 15\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:26]\u001b[0m Total number of processed SERIES files found is: 0\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:26]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:26]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:26]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:26]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:26]\u001b[0m Unifying and saving of metadata... \n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:26]\u001b[0m Trying GSE106872 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:26]\u001b[0m Trying GSE106872 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:26]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:26]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE106872'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE106872\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:54]\u001b[0m \n",
      "Total number of processed SAMPLES files found is: 341\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:54]\u001b[0m Total number of processed SERIES files found is: 0\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:54]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:54]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:54]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:54]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:54]\u001b[0m Unifying and saving of metadata... \n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:55]\u001b[0m Trying GSE108090 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:55]\u001b[0m Trying GSE108090 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:55]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:37:55]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE108090'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE108090\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:00]\u001b[0m Total number of processed SERIES files found is: 2\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:00]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:00]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:00]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:00]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:00]\u001b[0m No files found. No data to save. File geofetch_metadata\\project_name\\GSE108090_samples\\GSE108090_samples.csv won't be created\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:00]\u001b[0m Trying GSE221336 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:00]\u001b[0m Trying GSE221336 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:01]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:01]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE221336'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE221336\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:02]\u001b[0m Total number of processed SERIES files found is: 2\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:02]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:02]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:02]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:02]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:02]\u001b[0m No files found. No data to save. File geofetch_metadata\\project_name\\GSE221336_samples\\GSE221336_samples.csv won't be created\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:02]\u001b[0m Trying GSE219007 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:02]\u001b[0m Trying GSE219007 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:02]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:02]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE219007'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE219007\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:04]\u001b[0m \n",
      "Total number of processed SAMPLES files found is: 4\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:04]\u001b[0m Total number of processed SERIES files found is: 0\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:04]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:04]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:04]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:04]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:04]\u001b[0m Unifying and saving of metadata... \n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:04]\u001b[0m Trying GSE217127 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:04]\u001b[0m Trying GSE217127 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:04]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:04]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE217127'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE217127\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:06]\u001b[0m \n",
      "Total number of processed SAMPLES files found is: 12\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:06]\u001b[0m Total number of processed SERIES files found is: 0\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:06]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:06]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:06]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:06]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:06]\u001b[0m Unifying and saving of metadata... \n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:06]\u001b[0m Trying GSE256159 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:06]\u001b[0m Trying GSE256159 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:06]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:06]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE256159'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE256159\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:10]\u001b[0m Total number of processed SERIES files found is: 2\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:10]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:10]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:10]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:10]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:10]\u001b[0m No files found. No data to save. File geofetch_metadata\\project_name\\GSE256159_samples\\GSE256159_samples.csv won't be created\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:10]\u001b[0m Trying GSE274905 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:10]\u001b[0m Trying GSE274905 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:10]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:10]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE274905'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE274905\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:10]\u001b[0m Total number of processed SERIES files found is: 9\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:10]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:10]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:10]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:10]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:10]\u001b[0m No files found. No data to save. File geofetch_metadata\\project_name\\GSE274905_samples\\GSE274905_samples.csv won't be created\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:10]\u001b[0m Trying GSE208446 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:10]\u001b[0m Trying GSE208446 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:11]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:11]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE208446'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE208446\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:15]\u001b[0m Total number of processed SERIES files found is: 1\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:15]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:15]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:15]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:15]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:15]\u001b[0m No files found. No data to save. File geofetch_metadata\\project_name\\GSE208446_samples\\GSE208446_samples.csv won't be created\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:15]\u001b[0m Trying GSE148054 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:15]\u001b[0m Trying GSE148054 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:15]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:15]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE148054'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE148054\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:17]\u001b[0m \n",
      "Total number of processed SAMPLES files found is: 4\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:17]\u001b[0m Total number of processed SERIES files found is: 0\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:17]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:17]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:17]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:17]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:17]\u001b[0m Unifying and saving of metadata... \n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:17]\u001b[0m Trying GSE180233 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:17]\u001b[0m Trying GSE180233 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:17]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:17]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE180233'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE180233\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:18]\u001b[0m \n",
      "Total number of processed SAMPLES files found is: 16\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:18]\u001b[0m Total number of processed SERIES files found is: 0\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:18]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:18]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:18]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:18]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:18]\u001b[0m Unifying and saving of metadata... \n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:18]\u001b[0m Trying GSE143148 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:18]\u001b[0m Trying GSE143148 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:18]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:18]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE143148'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE143148\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:20]\u001b[0m \n",
      "Total number of processed SAMPLES files found is: 6\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:20]\u001b[0m Total number of processed SERIES files found is: 4\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:20]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:20]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:20]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:20]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:20]\u001b[0m Unifying and saving of metadata... \n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:20]\u001b[0m Trying GSE199913 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:20]\u001b[0m Trying GSE199913 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:20]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:20]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE199913'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE199913\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:32]\u001b[0m \n",
      "Total number of processed SAMPLES files found is: 212\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:32]\u001b[0m Total number of processed SERIES files found is: 10\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:32]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:32]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:32]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:32]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:32]\u001b[0m Unifying and saving of metadata... \n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:32]\u001b[0m Trying GSE251728 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:32]\u001b[0m Trying GSE251728 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:32]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:32]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE251728'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE251728\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:33]\u001b[0m Total number of processed SERIES files found is: 3\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:33]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:33]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:33]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:33]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:33]\u001b[0m No files found. No data to save. File geofetch_metadata\\project_name\\GSE251728_samples\\GSE251728_samples.csv won't be created\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:33]\u001b[0m Trying GSM5754576 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:33]\u001b[0m Trying GSM5754576 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:33]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:33]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSM5754576'\u001b[0m\n",
      "\u001b[1;30m[WARNING]\u001b[0m \u001b[32m[22:38:33]\u001b[0m \u001b[33mThis does not appear to be a correctly formatted GSE accession! Continue anyway...\u001b[0m\n",
      "\u001b[1;30m[WARNING]\u001b[0m \u001b[32m[22:38:33]\u001b[0m \u001b[33mError in checking size of soft file, continuing... 'content-length'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSM5754576\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[WARNING]\u001b[0m \u001b[32m[22:38:36]\u001b[0m \u001b[33mError in checking size of soft file, continuing... 'content-length'\u001b[0m\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:37]\u001b[0m Total number of processed SERIES files found is: 0\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:37]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:37]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:37]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:37]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:37]\u001b[0m No files found. No data to save. File geofetch_metadata\\project_name\\GSM5754576_samples\\GSM5754576_samples.csv won't be created\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:37]\u001b[0m Trying GSE241247 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:37]\u001b[0m Trying GSE241247 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:37]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:37]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE241247'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE241247\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:38]\u001b[0m Total number of processed SERIES files found is: 1\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:38]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:38]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:38]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:38]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:38]\u001b[0m No files found. No data to save. File geofetch_metadata\\project_name\\GSE241247_samples\\GSE241247_samples.csv won't be created\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:38]\u001b[0m Trying GSE290475 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:38]\u001b[0m Trying GSE290475 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:38]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:38]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE290475'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE290475\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:43]\u001b[0m Total number of processed SERIES files found is: 1\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:43]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:43]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:43]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:43]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:43]\u001b[0m No files found. No data to save. File geofetch_metadata\\project_name\\GSE290475_samples\\GSE290475_samples.csv won't be created\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:43]\u001b[0m Trying GSE135632 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:43]\u001b[0m Trying GSE135632 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:43]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:43]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE135632'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE135632\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:45]\u001b[0m \n",
      "Total number of processed SAMPLES files found is: 44\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:45]\u001b[0m Total number of processed SERIES files found is: 0\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:45]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:45]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:45]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:45]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:45]\u001b[0m Unifying and saving of metadata... \n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:45]\u001b[0m Trying GSE224611 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:45]\u001b[0m Trying GSE224611 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:45]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:45]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE224611'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE224611\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:49]\u001b[0m \n",
      "Total number of processed SAMPLES files found is: 24\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:49]\u001b[0m Total number of processed SERIES files found is: 0\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:49]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:49]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:49]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:49]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:49]\u001b[0m Unifying and saving of metadata... \n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:49]\u001b[0m Trying GSE166351 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:49]\u001b[0m Trying GSE166351 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:49]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:49]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE166351'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE166351\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:53]\u001b[0m \n",
      "Total number of processed SAMPLES files found is: 6\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:53]\u001b[0m Total number of processed SERIES files found is: 0\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:53]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:53]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:53]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:53]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:53]\u001b[0m Unifying and saving of metadata... \n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:53]\u001b[0m Trying GSE134120 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:53]\u001b[0m Trying GSE134120 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:53]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:53]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE134120'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE134120\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:55]\u001b[0m Total number of processed SERIES files found is: 1\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:55]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:55]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:55]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:55]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:55]\u001b[0m No files found. No data to save. File geofetch_metadata\\project_name\\GSE134120_samples\\GSE134120_samples.csv won't be created\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:55]\u001b[0m Trying GSE135634 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:55]\u001b[0m Trying GSE135634 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:55]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:38:55]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE135634'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE135634\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:01]\u001b[0m \n",
      "Total number of processed SAMPLES files found is: 153\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:01]\u001b[0m Total number of processed SERIES files found is: 0\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:01]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:01]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:01]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:01]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:01]\u001b[0m Unifying and saving of metadata... \n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:01]\u001b[0m Trying GSE269020 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:01]\u001b[0m Trying GSE269020 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:01]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:01]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE269020'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE269020\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:03]\u001b[0m Total number of processed SERIES files found is: 1\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:03]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:03]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:03]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:03]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:03]\u001b[0m No files found. No data to save. File geofetch_metadata\\project_name\\GSE269020_samples\\GSE269020_samples.csv won't be created\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:03]\u001b[0m Trying GSM5754579 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:03]\u001b[0m Trying GSM5754579 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:03]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:03]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSM5754579'\u001b[0m\n",
      "\u001b[1;30m[WARNING]\u001b[0m \u001b[32m[22:39:03]\u001b[0m \u001b[33mThis does not appear to be a correctly formatted GSE accession! Continue anyway...\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSM5754579\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[WARNING]\u001b[0m \u001b[32m[22:39:03]\u001b[0m \u001b[33mError in checking size of soft file, continuing... 'content-length'\u001b[0m\n",
      "\u001b[1;30m[WARNING]\u001b[0m \u001b[32m[22:39:04]\u001b[0m \u001b[33mError in checking size of soft file, continuing... 'content-length'\u001b[0m\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:05]\u001b[0m Total number of processed SERIES files found is: 0\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:05]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:05]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:05]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:05]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:05]\u001b[0m No files found. No data to save. File geofetch_metadata\\project_name\\GSM5754579_samples\\GSM5754579_samples.csv won't be created\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:05]\u001b[0m Trying GSE149233 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:05]\u001b[0m Trying GSE149233 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:05]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:05]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE149233'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE149233\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:06]\u001b[0m Total number of processed SERIES files found is: 3\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:06]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:06]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:06]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:06]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:06]\u001b[0m No files found. No data to save. File geofetch_metadata\\project_name\\GSE149233_samples\\GSE149233_samples.csv won't be created\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:06]\u001b[0m Trying GSE200900 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:06]\u001b[0m Trying GSE200900 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:06]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:06]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE200900'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE200900\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:08]\u001b[0m \n",
      "Total number of processed SAMPLES files found is: 8\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:08]\u001b[0m Total number of processed SERIES files found is: 1\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:08]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:08]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:08]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:08]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:08]\u001b[0m Unifying and saving of metadata... \n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:08]\u001b[0m Trying GSE117865 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:08]\u001b[0m Trying GSE117865 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:08]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:08]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE117865'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE117865\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:12]\u001b[0m \n",
      "Total number of processed SAMPLES files found is: 2\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:12]\u001b[0m Total number of processed SERIES files found is: 0\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:12]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:12]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:12]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:12]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:12]\u001b[0m Unifying and saving of metadata... \n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:12]\u001b[0m Trying GSE256469 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:12]\u001b[0m Trying GSE256469 (not a file) as accession...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:12]\u001b[0m Skipped 0 accessions. Starting now.\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:12]\u001b[0m \u001b[38;5;200mProcessing accession 1 of 1: 'GSE256469'\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Fetching metadata for GSE256469\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:15]\u001b[0m \n",
      "Total number of processed SAMPLES files found is: 4\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:15]\u001b[0m Total number of processed SERIES files found is: 1\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:15]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:15]\u001b[0m Expanding metadata list...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:15]\u001b[0m Finished processing 1 accession(s)\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:15]\u001b[0m Cleaning soft files ...\n",
      "\u001b[1;30m[INFO]\u001b[0m \u001b[32m[22:39:15]\u001b[0m Unifying and saving of metadata... \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "❌ geofetch failed for GSE193726: Command '['geofetch', '-i', 'GSE193726', '--processed', '-m', 'geofetch_metadata']' returned non-zero exit status 1.\n",
      "✅ Finished downloading for GSE269325\n",
      "❌ geofetch failed for GSE236921: Command '['geofetch', '-i', 'GSE236921', '--processed', '-m', 'geofetch_metadata']' returned non-zero exit status 1.\n",
      "✅ Finished downloading for GSE192673\n",
      "❌ geofetch failed for GSE148055: Command '['geofetch', '-i', 'GSE148055', '--processed', '-m', 'geofetch_metadata']' returned non-zero exit status 1.\n",
      "✅ GSE139437: already exists at geofetch_metadata\\GSE139437, skipping.\n",
      "❌ geofetch failed for GSE96686: Command '['geofetch', '-i', 'GSE96686', '--processed', '-m', 'geofetch_metadata']' returned non-zero exit status 1.\n",
      "❌ geofetch failed for GSE143151: Command '['geofetch', '-i', 'GSE143151', '--processed', '-m', 'geofetch_metadata']' returned non-zero exit status 1.\n",
      "✅ Finished downloading for GSE146833\n",
      "✅ Finished downloading for GSM5754577\n",
      "✅ Finished downloading for GSM5754578\n",
      "✅ Finished downloading for GSE279335\n",
      "❌ geofetch failed for GSE173995: Command '['geofetch', '-i', 'GSE173995', '--processed', '-m', 'geofetch_metadata']' returned non-zero exit status 1.\n",
      "❌ geofetch failed for GSE135633: Command '['geofetch', '-i', 'GSE135633', '--processed', '-m', 'geofetch_metadata']' returned non-zero exit status 1.\n",
      "❌ geofetch failed for GSE202506: Command '['geofetch', '-i', 'GSE202506', '--processed', '-m', 'geofetch_metadata']' returned non-zero exit status 1.\n",
      "❌ geofetch failed for GSE148053: Command '['geofetch', '-i', 'GSE148053', '--processed', '-m', 'geofetch_metadata']' returned non-zero exit status 1.\n",
      "❌ geofetch failed for GSE269147: Command '['geofetch', '-i', 'GSE269147', '--processed', '-m', 'geofetch_metadata']' returned non-zero exit status 1.\n",
      "❌ geofetch failed for GSE106872: Command '['geofetch', '-i', 'GSE106872', '--processed', '-m', 'geofetch_metadata']' returned non-zero exit status 1.\n",
      "✅ Finished downloading for GSE108090\n",
      "✅ Finished downloading for GSE221336\n",
      "❌ geofetch failed for GSE219007: Command '['geofetch', '-i', 'GSE219007', '--processed', '-m', 'geofetch_metadata']' returned non-zero exit status 1.\n",
      "❌ geofetch failed for GSE217127: Command '['geofetch', '-i', 'GSE217127', '--processed', '-m', 'geofetch_metadata']' returned non-zero exit status 1.\n",
      "✅ Finished downloading for GSE256159\n",
      "✅ Finished downloading for GSE274905\n",
      "✅ Finished downloading for GSE208446\n",
      "❌ geofetch failed for GSE148054: Command '['geofetch', '-i', 'GSE148054', '--processed', '-m', 'geofetch_metadata']' returned non-zero exit status 1.\n",
      "❌ geofetch failed for GSE180233: Command '['geofetch', '-i', 'GSE180233', '--processed', '-m', 'geofetch_metadata']' returned non-zero exit status 1.\n",
      "❌ geofetch failed for GSE143148: Command '['geofetch', '-i', 'GSE143148', '--processed', '-m', 'geofetch_metadata']' returned non-zero exit status 1.\n",
      "❌ geofetch failed for GSE199913: Command '['geofetch', '-i', 'GSE199913', '--processed', '-m', 'geofetch_metadata']' returned non-zero exit status 1.\n",
      "✅ Finished downloading for GSE251728\n",
      "✅ Finished downloading for GSM5754576\n",
      "✅ Finished downloading for GSE241247\n",
      "✅ Finished downloading for GSE290475\n",
      "❌ geofetch failed for GSE135632: Command '['geofetch', '-i', 'GSE135632', '--processed', '-m', 'geofetch_metadata']' returned non-zero exit status 1.\n",
      "❌ geofetch failed for GSE224611: Command '['geofetch', '-i', 'GSE224611', '--processed', '-m', 'geofetch_metadata']' returned non-zero exit status 1.\n",
      "❌ geofetch failed for GSE166351: Command '['geofetch', '-i', 'GSE166351', '--processed', '-m', 'geofetch_metadata']' returned non-zero exit status 1.\n",
      "✅ Finished downloading for GSE134120\n",
      "❌ geofetch failed for GSE135634: Command '['geofetch', '-i', 'GSE135634', '--processed', '-m', 'geofetch_metadata']' returned non-zero exit status 1.\n",
      "✅ Finished downloading for GSE269020\n",
      "✅ Finished downloading for GSM5754579\n",
      "✅ Finished downloading for GSE149233\n",
      "❌ geofetch failed for GSE200900: Command '['geofetch', '-i', 'GSE200900', '--processed', '-m', 'geofetch_metadata']' returned non-zero exit status 1.\n",
      "❌ geofetch failed for GSE117865: Command '['geofetch', '-i', 'GSE117865', '--processed', '-m', 'geofetch_metadata']' returned non-zero exit status 1.\n",
      "❌ geofetch failed for GSE256469: Command '['geofetch', '-i', 'GSE256469', '--processed', '-m', 'geofetch_metadata']' returned non-zero exit status 1.\n",
      "✅ Combined metadata saved to: C:\\Users\\difen\\POPPER\\geofetch_metadata\\combined_metadata.csv\n",
      "🔍 Extracted GSE IDs: ['GSE236921', 'GSE180233', 'GSE173995', 'GSE236921', 'GSE180233', 'GSE173995', 'GSE236921', 'GSE180233', 'GSE173995', 'GSE236921', 'GSE180233', 'GSE173995']\n",
      "🔍 Scanning for compressed files under: rna_seq_analysis\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE106872\\GSE106872_RAW.tar\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856250_CVB-H3K4me3.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856250_CVB-H3K4me3.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856251_IID4-H3K4me3.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856251_IID4-H3K4me3.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856252_IIA11-H3K4me3.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856252_IIA11-H3K4me3.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856253_IID12-H3K4me3.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856253_IID12-H3K4me3.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856254_IB12-H3K4me3.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856254_IB12-H3K4me3.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856255_IIA11_Q_S61.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856255_IIA11_Q_S61.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856256_ID8-H3K4me3.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856256_ID8-H3K4me3.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856257_IIIB12-H3K4me3.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856257_IIIB12-H3K4me3.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856258_IB7-H3K4me3_S20_L002_R1_001.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856258_IB7-H3K4me3_S20_L002_R1_001.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856259_CVB-CTCF.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856259_CVB-CTCF.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856260_IID4-CTCF.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856260_IID4-CTCF.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856261_IIA11-CTCF.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856261_IIA11-CTCF.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856262_IID12-CTCF.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856262_IID12-CTCF.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856263_IB12-CTCF.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856263_IB12-CTCF.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856264_IIA12-CTCF.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856264_IIA12-CTCF.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856265_ID8-CTCF.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856265_ID8-CTCF.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856266_IIIB12-CTCF.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856266_IIIB12-CTCF.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856267_IB7-CTCF.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856267_IB7-CTCF.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856268_CVB_CTCF_S63.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856268_CVB_CTCF_S63.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856269_IID4_CTCF_S64.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856269_IID4_CTCF_S64.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856270_IIA11_CTCF_S71.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856270_IIA11_CTCF_S71.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856271_IID12_CTCF_S65.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856271_IID12_CTCF_S65.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856272_IB12_CTCF_S68.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856272_IB12_CTCF_S68.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856273_IIA12_CTCF_S69.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856273_IIA12_CTCF_S69.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856274_ID8_CTCF_S66.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856274_ID8_CTCF_S66.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856275_IB7_CTCF_S70.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856275_IB7_CTCF_S70.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856276_CVB-Rad21_S11_L002_R1_001.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856276_CVB-Rad21_S11_L002_R1_001.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856277_IID4-Rad21_S12_L002_R1_001.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856277_IID4-Rad21_S12_L002_R1_001.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856278_IIA11-Rad21_S19_L002_R1_001.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856278_IIA11-Rad21_S19_L002_R1_001.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856279_IID12-Rad21_S13_L002_R1_001.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856279_IID12-Rad21_S13_L002_R1_001.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856280_IB12-Rad21_S16_L002_R1_001.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856280_IB12-Rad21_S16_L002_R1_001.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856281_IIA12-Rad21_S17_L002_R1_001.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856281_IIA12-Rad21_S17_L002_R1_001.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856282_ID8-Rad21_S14_L002_R1_001.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856282_ID8-Rad21_S14_L002_R1_001.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856283_IIIB12-Rad21_S15_L002_R1_001.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856283_IIIB12-Rad21_S15_L002_R1_001.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856284_IB7-Rad21_S18_L002_R1_001.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856284_IB7-Rad21_S18_L002_R1_001.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856285_CVB_Rad21_S84.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856285_CVB_Rad21_S84.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856286_IID4_Rad21_S85.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856286_IID4_Rad21_S85.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856287_IIA11_Rad21_S92.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856287_IIA11_Rad21_S92.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856288_IID12_Rad21_S86.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856288_IID12_Rad21_S86.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856289_IB12_Rad21_S89.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856289_IB12_Rad21_S89.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856290_IIA12_Rad21_S90.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856290_IIA12_Rad21_S90.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856291_ID8_Rad21_S87.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856291_ID8_Rad21_S87.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856292_IIIB12_Rad21_S88.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856292_IIIB12_Rad21_S88.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856293_IB7_Rad21_S91.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856293_IB7_Rad21_S91.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856294_IIA11_H3K4me2Q_S51.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856294_IIA11_H3K4me2Q_S51.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856295_IID12_H3K4me2_S48.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856295_IID12_H3K4me2_S48.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856296_IB12_H3K4me2_S49.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856296_IB12_H3K4me2_S49.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856297_IIA12_H3K4me2_Q_S50.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856297_IIA12_H3K4me2_Q_S50.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856298_IIA11_H3K9ac_Q_S47.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856298_IIA11_H3K9ac_Q_S47.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856299_IID12_H3K9ac_S44.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856299_IID12_H3K9ac_S44.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856300_IB12_H3K9ac_S45.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856300_IB12_H3K9ac_S45.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856301_IIA12_H3K9ac_Q_S46.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856301_IIA12_H3K9ac_Q_S46.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856302_IIA11_H3K9me3_Q_S31.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856302_IIA11_H3K9me3_Q_S31.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856303_IID12_H3K9me3_S28.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856303_IID12_H3K9me3_S28.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856304_IB12_H3K9me3_S29.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856304_IB12_H3K9me3_S29.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856305_IIA12_H3K9me3_Q_S30.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856305_IIA12_H3K9me3_Q_S30.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856306_IIA11_H2AZ_Q_S55.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856306_IIA11_H2AZ_Q_S55.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856307_IID12_H2AZ_S52.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856307_IID12_H2AZ_S52.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856308_IB12_H2AZ_S53.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856308_IB12_H2AZ_S53.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856309_IIA12_H2AZ_Q_S54.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856309_IIA12_H2AZ_Q_S54.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856310_IIA11_Rest_Q_S27.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856310_IIA11_Rest_Q_S27.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856311_IID12_Rest_S24.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856311_IID12_Rest_S24.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856312_IB12_Rest_S25.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856312_IB12_Rest_S25.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856313_IIA12_Rest_Q_S26.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856313_IIA12_Rest_Q_S26.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856314_IIA11_SIN3A_Q_S39.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856314_IIA11_SIN3A_Q_S39.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856315_IID12_SIN3A_S36.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856315_IID12_SIN3A_S36.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856316_IB12_SIN3A_S37.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856316_IB12_SIN3A_S37.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856317_IIA12_SIN3A_Q_S38.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856317_IIA12_SIN3A_Q_S38.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856318_IIA11_Smcdh1_Q_S43.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856318_IIA11_Smcdh1_Q_S43.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856319_IID12_Smcdh1_S40.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856319_IID12_Smcdh1_S40.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856320_IB12_Smcdh1_S41.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856320_IB12_Smcdh1_S41.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856321_IIA12_Smcdh1_Q_S42.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856321_IIA12_Smcdh1_Q_S42.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856322_IIA11_JMJD2A_Q_S59.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856322_IIA11_JMJD2A_Q_S59.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856323_IID12_JMJD2A_S56.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856323_IID12_JMJD2A_S56.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856324_IB12_JMJD2A_S57.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856324_IB12_JMJD2A_S57.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856325_IIA12_JMJD2A_Q_S58.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856325_IIA12_JMJD2A_Q_S58.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856326_CVB7-Me3_S1_L001_R1_001.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856326_CVB7-Me3_S1_L001_R1_001.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856327_CVB8-me3_S2_L001_R1_001.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856327_CVB8-me3_S2_L001_R1_001.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856328_CVB9-me3_S3_L001_R1_001.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856328_CVB9-me3_S3_L001_R1_001.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856329_CVB7_H3K4me3_S32.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856329_CVB7_H3K4me3_S32.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856330_CVB8_H3K4me3_S21.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856330_CVB8_H3K4me3_S21.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856331_CVB9_H3K4me3_S23.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856331_CVB9_H3K4me3_S23.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856332_CVI-me3_S4_L001_R1_001.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856332_CVI-me3_S4_L001_R1_001.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856333_CVI_H3K4me3_S16.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856333_CVI_H3K4me3_S16.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856334_CVI_CTCF_S13.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856334_CVI_CTCF_S13.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856335_CVI_Rad21_S12.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856335_CVI_Rad21_S12.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856336_CVI13-me3_S5_L001_R1_001.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856336_CVI13-me3_S5_L001_R1_001.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856337_CVI14-me3_S6_L001_R1_001.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856337_CVI14-me3_S6_L001_R1_001.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856338_CVI15-me3_S7_L001_R1_001.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856338_CVI15-me3_S7_L001_R1_001.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856339_CVI1-3_H3K4me3_S33.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856339_CVI1-3_H3K4me3_S33.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856340_CVI1-14_H3K4me3_Q_S35.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856340_CVI1-14_H3K4me3_Q_S35.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856341_CVI1-5_H3K4me3_S22.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856341_CVI1-5_H3K4me3_S22.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856342_Hues_S8_L001_R1_001.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856342_Hues_S8_L001_R1_001.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856343_Hues9_H3K4me3_S17.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856343_Hues9_H3K4me3_S17.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856344_Hues9_CTCF_S15.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856344_Hues9_CTCF_S15.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856345_Hues9_Rad21_S14.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856345_Hues9_Rad21_S14.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856346_Hues17_S9_L001.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856346_Hues17_S9_L001.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856347_Hues91-8_H3K4me3_ACTTGA_S19.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856347_Hues91-8_H3K4me3_ACTTGA_S19.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856348_Hues19_S10_L001_R1_001.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856348_Hues19_S10_L001_R1_001.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856349_Hues91-7_H3K4me3_S20.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856349_Hues91-7_H3K4me3_S20.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856350_Hues91-8_H3K4me3_CAGATC_S18.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856350_Hues91-8_H3K4me3_CAGATC_S18.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856351_Hues91-9_H3K4me3_Q_S34.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856351_Hues91-9_H3K4me3_Q_S34.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856352_mESC1_H3K4me3_ATCACG_S1.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856352_mESC1_H3K4me3_ATCACG_S1.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856353_mESC2_H3K4me3_CGATGT_S2.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856353_mESC2_H3K4me3_CGATGT_S2.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856354_mESC3_H3K4me3_TTAGGC_S3.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856354_mESC3_H3K4me3_TTAGGC_S3.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856355_mESC4_H3K4me3_TGACCA_S4.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856355_mESC4_H3K4me3_TGACCA_S4.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856356_mESC5_H3K4me3_ACAGTG_S5.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856356_mESC5_H3K4me3_ACAGTG_S5.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856357_mESC6_H3K4me3_GCCAAT_S6.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856357_mESC6_H3K4me3_GCCAAT_S6.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856358_mESC1_H3K4me3_CAGATC_S7.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856358_mESC1_H3K4me3_CAGATC_S7.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856359_mESC2_H3K4me3_ACTTGA_S8.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856359_mESC2_H3K4me3_ACTTGA_S8.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856360_mESC3_H3K4me3_GATCAG_S9.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856360_mESC3_H3K4me3_GATCAG_S9.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856361_mESC5_H3K4me3_GGCTAC_S10.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856361_mESC5_H3K4me3_GGCTAC_S10.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856362_mESC6_H3K4me3_CTTGTA_S11.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856362_mESC6_H3K4me3_CTTGTA_S11.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856363_AA05.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856363_AA05.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856364_AA13.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856364_AA13.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856365_AA21.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856365_AA21.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856366_AA06.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856366_AA06.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856367_AA14.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856367_AA14.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856368_AA22.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856368_AA22.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856369_AA03.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856369_AA03.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856370_AA11.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856370_AA11.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856371_AA19.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856371_AA19.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856372_AA04.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856372_AA04.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856373_AA12.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856373_AA12.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856374_AA20.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856374_AA20.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856375_AA02.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856375_AA02.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856376_AA10.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856376_AA10.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM2856376_AA10.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856377_AA18.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856377_AA18.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM2856377_AA18.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856378_AA08.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856378_AA08.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM2856378_AA08.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856379_AA16.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856379_AA16.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM2856379_AA16.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856380_AA24.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856380_AA24.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM2856380_AA24.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856381_AA07.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856381_AA07.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM2856381_AA07.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856382_AA15.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856382_AA15.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM2856382_AA15.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856383_AA23.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856383_AA23.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM2856383_AA23.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856384_AA01.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856384_AA01.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM2856384_AA01.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856385_AA09.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856385_AA09.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM2856385_AA09.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM2856386_AA17.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM2856386_AA17.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM2856386_AA17.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023826_directoryADAM_49_S1_L004_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023826_directoryADAM_49_S1_L004_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023826_directoryADAM_49_S1_L004_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023827_directoryADAM_58_S2_L004_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023827_directoryADAM_58_S2_L004_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023827_directoryADAM_58_S2_L004_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023828_directoryABCA7_108_S7_L004_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023828_directoryABCA7_108_S7_L004_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023828_directoryABCA7_108_S7_L004_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023829_directoryADAM_19_S15_L005_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023829_directoryADAM_19_S15_L005_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023829_directoryADAM_19_S15_L005_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023830_directoryADAM_98_S16_L005_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023830_directoryADAM_98_S16_L005_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023830_directoryADAM_98_S16_L005_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023831_directoryADAM_35_S17_L005_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023831_directoryADAM_35_S17_L005_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023831_directoryADAM_35_S17_L005_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023832_directoryHUES9-1_8_S8_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023832_directoryHUES9-1_8_S8_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023832_directoryHUES9-1_8_S8_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023833_directoryHUES_9_1_8_NAIVE_R3_S16_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023833_directoryHUES_9_1_8_NAIVE_R3_S16_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023833_directoryHUES_9_1_8_NAIVE_R3_S16_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023834_directoryHUES_9_1_8_NAIVE_R5_S21_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023834_directoryHUES_9_1_8_NAIVE_R5_S21_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023834_directoryHUES_9_1_8_NAIVE_R5_S21_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023835_directoryHUES_9_1_8_NAIVE_R5_HOX_S23_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023835_directoryHUES_9_1_8_NAIVE_R5_HOX_S23_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023835_directoryHUES_9_1_8_NAIVE_R5_HOX_S23_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023836_directoryHUES_9_1_8_RP_R3_S17_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023836_directoryHUES_9_1_8_RP_R3_S17_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023836_directoryHUES_9_1_8_RP_R3_S17_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023837_directoryHUES_9_1_8_RP_R5_S22_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023837_directoryHUES_9_1_8_RP_R5_S22_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023837_directoryHUES_9_1_8_RP_R5_S22_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023838_directoryHUES_9_1_8_RP_R5_HOX_S24_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023838_directoryHUES_9_1_8_RP_R5_HOX_S24_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023838_directoryHUES_9_1_8_RP_R5_HOX_S24_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023839_directoryHUES_9_1_8_RP_R3_SC3_S18_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023839_directoryHUES_9_1_8_RP_R3_SC3_S18_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023839_directoryHUES_9_1_8_RP_R3_SC3_S18_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023840_directoryHUES_9_1_8_RP_R3_SC9_S19_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023840_directoryHUES_9_1_8_RP_R3_SC9_S19_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023840_directoryHUES_9_1_8_RP_R3_SC9_S19_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023841_directoryHUES_9_1_8_RP_R3_SC10_S20_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023841_directoryHUES_9_1_8_RP_R3_SC10_S20_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023841_directoryHUES_9_1_8_RP_R3_SC10_S20_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023842_directoryRP_clone8_S25_L005_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023842_directoryRP_clone8_S25_L005_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023842_directoryRP_clone8_S25_L005_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023843_directoryRP_clone12_S26_L005_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023843_directoryRP_clone12_S26_L005_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023843_directoryRP_clone12_S26_L005_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023844_directoryRP_clone11_S14_L004_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023844_directoryRP_clone11_S14_L004_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023844_directoryRP_clone11_S14_L004_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023845_directoryHUES9_1_8_RP_SC13_S3_L004_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023845_directoryHUES9_1_8_RP_SC13_S3_L004_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023845_directoryHUES9_1_8_RP_SC13_S3_L004_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023846_directoryHUES9_1_8_RP_SC14_S4_L004_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023846_directoryHUES9_1_8_RP_SC14_S4_L004_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023846_directoryHUES9_1_8_RP_SC14_S4_L004_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023847_directoryIIA11_NAIVE_HOX_S10_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023847_directoryIIA11_NAIVE_HOX_S10_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023847_directoryIIA11_NAIVE_HOX_S10_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023848_directoryIIDV8_NAIVE_S27_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023848_directoryIIDV8_NAIVE_S27_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023848_directoryIIDV8_NAIVE_S27_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023849_directoryIIDV8_NAIVE_HOX_S28_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023849_directoryIIDV8_NAIVE_HOX_S28_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023849_directoryIIDV8_NAIVE_HOX_S28_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023850_directoryIB7_naive_S27_L005_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023850_directoryIB7_naive_S27_L005_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023850_directoryIB7_naive_S27_L005_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023851_directorymESC-parent_S1_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023851_directorymESC-parent_S1_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023851_directorymESC-parent_S1_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023852_directorymESC_SC_1_4_S13_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023852_directorymESC_SC_1_4_S13_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023852_directorymESC_SC_1_4_S13_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023853_directorymESC_SC_1_7_S14_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023853_directorymESC_SC_1_7_S14_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023853_directorymESC_SC_1_7_S14_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023854_directorymESC_SC_2_10_S5_L004_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023854_directorymESC_SC_2_10_S5_L004_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023854_directorymESC_SC_2_10_S5_L004_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023855_directorymESC_SC_2_13_S6_L004_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023855_directorymESC_SC_2_13_S6_L004_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023855_directorymESC_SC_2_13_S6_L004_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023856_directorymEPiSC_S12_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023856_directorymEPiSC_S12_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023856_directorymEPiSC_S12_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023857_directoryEpiSC-1_S2_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023857_directoryEpiSC-1_S2_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023857_directoryEpiSC-1_S2_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023858_directoryEpiSC-2_S3_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023858_directoryEpiSC-2_S3_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023858_directoryEpiSC-2_S3_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023859_directoryEpiSC-4_S4_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023859_directoryEpiSC-4_S4_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023859_directoryEpiSC-4_S4_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023860_directoryEpiSC-5_S5_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023860_directoryEpiSC-5_S5_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023860_directoryEpiSC-5_S5_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023861_directoryEpiSC-7_S6_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023861_directoryEpiSC-7_S6_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023861_directoryEpiSC-7_S6_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023862_directoryEpiSC-8_S7_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023862_directoryEpiSC-8_S7_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023862_directoryEpiSC-8_S7_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023863_directoryEpiSC-9_S8_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023863_directoryEpiSC-9_S8_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023863_directoryEpiSC-9_S8_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023864_directoryEpiSC-10_S9_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023864_directoryEpiSC-10_S9_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023864_directoryEpiSC-10_S9_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023865_directoryEpiSC-11_S10_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023865_directoryEpiSC-11_S10_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023865_directoryEpiSC-11_S10_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023866_directoryEpiSC-12_S11_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023866_directoryEpiSC-12_S11_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023866_directoryEpiSC-12_S11_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023867_directoryEpiSC-13_S12_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023867_directoryEpiSC-13_S12_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023867_directoryEpiSC-13_S12_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023868_directoryEpiSC-14_S13_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023868_directoryEpiSC-14_S13_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023868_directoryEpiSC-14_S13_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023869_directoryEpiSC-15_S14_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023869_directoryEpiSC-15_S14_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023869_directoryEpiSC-15_S14_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023870_directoryHues9_1_8_naive_S13_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023870_directoryHues9_1_8_naive_S13_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023870_directoryHues9_1_8_naive_S13_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023871_directoryHues9_1_8_primed_S12_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023871_directoryHues9_1_8_primed_S12_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023871_directoryHues9_1_8_primed_S12_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023872_directoryHues9_1_8_reprimed_S14_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023872_directoryHues9_1_8_reprimed_S14_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023872_directoryHues9_1_8_reprimed_S14_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023873_directoryCVB_organoids_S15_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023873_directoryCVB_organoids_S15_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023873_directoryCVB_organoids_S15_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023874_directoryIID4-cortical_S11_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023874_directoryIID4-cortical_S11_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023874_directoryIID4-cortical_S11_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023875_directoryID12-cortical_S10_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023875_directoryID12-cortical_S10_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023875_directoryID12-cortical_S10_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023876_directoryIIA12-cortical_S9_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023876_directoryIIA12-cortical_S9_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023876_directoryIIA12-cortical_S9_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023877_directory54a_S54_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023877_directory54a_S54_L007_R1_001.fastq.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023878_directory55a_S55_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023878_directory55a_S55_L008_R1_001.fastq.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023879_directory56a_S56_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023879_directory56a_S56_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023879_directory56a_S56_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023880_directory56a_S56_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023880_directory56a_S56_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023880_directory56a_S56_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023881_directory57a_S57_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023881_directory57a_S57_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023881_directory57a_S57_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023882_directory57a_S57_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023882_directory57a_S57_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023882_directory57a_S57_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023883_directory58a_S58_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023883_directory58a_S58_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023883_directory58a_S58_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023884_directory58a_S58_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023884_directory58a_S58_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023884_directory58a_S58_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023885_directory59a_S59_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023885_directory59a_S59_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023885_directory59a_S59_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023886_directory59a_S59_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023886_directory59a_S59_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023886_directory59a_S59_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023887_directory60a_S60_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023887_directory60a_S60_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023887_directory60a_S60_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023888_directory60a_S60_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023888_directory60a_S60_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023888_directory60a_S60_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023889_directory61a_S61_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023889_directory61a_S61_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023889_directory61a_S61_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023890_directory61a_S61_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023890_directory61a_S61_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023890_directory61a_S61_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023891_directory62a_S62_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023891_directory62a_S62_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023891_directory62a_S62_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023892_directory62a_S62_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023892_directory62a_S62_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023892_directory62a_S62_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023893_directory63a_S63_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023893_directory63a_S63_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023893_directory63a_S63_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023894_directory63a_S63_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023894_directory63a_S63_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023894_directory63a_S63_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023895_directory65a_S65_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023895_directory65a_S65_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023895_directory65a_S65_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023896_directory65a_S65_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023896_directory65a_S65_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023896_directory65a_S65_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023897_directory66a_S66_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023897_directory66a_S66_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023897_directory66a_S66_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023898_directory66a_S66_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023898_directory66a_S66_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023898_directory66a_S66_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023899_directory67a_S67_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023899_directory67a_S67_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023899_directory67a_S67_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023900_directory67a_S67_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023900_directory67a_S67_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023900_directory67a_S67_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023901_directory68a_S68_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023901_directory68a_S68_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023901_directory68a_S68_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023902_directory68a_S68_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023902_directory68a_S68_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023902_directory68a_S68_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023903_directory69a_S69_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023903_directory69a_S69_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023903_directory69a_S69_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023904_directory69a_S69_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023904_directory69a_S69_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023904_directory69a_S69_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023905_directory70a_S70_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023905_directory70a_S70_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023905_directory70a_S70_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023906_directory70a_S70_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023906_directory70a_S70_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023906_directory70a_S70_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023907_directory71a_S71_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023907_directory71a_S71_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023907_directory71a_S71_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023908_directory71a_S71_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023908_directory71a_S71_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023908_directory71a_S71_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023909_directory1a_S1_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023909_directory1a_S1_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023909_directory1a_S1_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023910_directory1a_S1_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023910_directory1a_S1_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023910_directory1a_S1_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023911_directory2a_S2_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023911_directory2a_S2_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023911_directory2a_S2_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023912_directory2a_S2_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023912_directory2a_S2_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023912_directory2a_S2_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023913_directory3a_S3_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023913_directory3a_S3_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023913_directory3a_S3_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023914_directory3a_S3_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023914_directory3a_S3_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023914_directory3a_S3_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023915_directory4a_S4_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023915_directory4a_S4_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023915_directory4a_S4_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023916_directory4a_S4_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023916_directory4a_S4_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023916_directory4a_S4_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023917_directory5a_S5_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023917_directory5a_S5_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023917_directory5a_S5_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023918_directory5a_S5_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023918_directory5a_S5_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023918_directory5a_S5_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023919_directory7a_S7_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023919_directory7a_S7_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023919_directory7a_S7_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023920_directory7a_S7_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023920_directory7a_S7_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023920_directory7a_S7_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023921_directory10a_S10_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023921_directory10a_S10_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023921_directory10a_S10_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023922_directory10a_S10_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023922_directory10a_S10_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023922_directory10a_S10_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023923_directory11a_S11_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023923_directory11a_S11_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023923_directory11a_S11_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023924_directory11a_S11_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023924_directory11a_S11_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023924_directory11a_S11_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023925_directory12a_S12_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023925_directory12a_S12_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023925_directory12a_S12_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023926_directory12a_S12_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023926_directory12a_S12_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023926_directory12a_S12_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023927_directory14a_S14_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023927_directory14a_S14_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023927_directory14a_S14_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023928_directory14a_S14_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023928_directory14a_S14_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023928_directory14a_S14_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023929_directory15a_S15_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023929_directory15a_S15_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023929_directory15a_S15_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023930_directory15a_S15_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023930_directory15a_S15_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023930_directory15a_S15_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023931_directory16a_S16_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023931_directory16a_S16_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023931_directory16a_S16_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023932_directory16a_S16_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023932_directory16a_S16_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023932_directory16a_S16_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023933_directory19a_S19_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023933_directory19a_S19_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023933_directory19a_S19_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023934_directory19a_S19_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023934_directory19a_S19_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023934_directory19a_S19_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023935_directory20a_S20_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023935_directory20a_S20_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023935_directory20a_S20_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023936_directory20a_S20_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023936_directory20a_S20_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023936_directory20a_S20_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023937_directory22a_S22_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023937_directory22a_S22_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023937_directory22a_S22_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023938_directory22a_S22_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023938_directory22a_S22_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023938_directory22a_S22_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023939_directory23a_S23_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023939_directory23a_S23_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023939_directory23a_S23_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023940_directory23a_S23_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023940_directory23a_S23_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023940_directory23a_S23_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023941_directory24a_S24_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023941_directory24a_S24_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023941_directory24a_S24_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023942_directory24a_S24_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023942_directory24a_S24_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023942_directory24a_S24_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023943_directory25a_S25_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023943_directory25a_S25_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023943_directory25a_S25_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023944_directory25a_S25_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023944_directory25a_S25_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023944_directory25a_S25_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023945_directory27a_S27_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023945_directory27a_S27_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023945_directory27a_S27_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023946_directory27a_S27_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023946_directory27a_S27_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023946_directory27a_S27_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023947_directory28a_S28_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023947_directory28a_S28_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023947_directory28a_S28_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023948_directory28a_S28_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023948_directory28a_S28_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023948_directory28a_S28_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023949_directory29a_S29_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023949_directory29a_S29_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023949_directory29a_S29_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023950_directory29a_S29_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023950_directory29a_S29_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023950_directory29a_S29_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023951_directory31a_S31_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023951_directory31a_S31_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023951_directory31a_S31_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023952_directory31a_S31_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023952_directory31a_S31_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023952_directory31a_S31_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023953_directory32a_S32_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023953_directory32a_S32_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023953_directory32a_S32_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023954_directory32a_S32_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023954_directory32a_S32_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023954_directory32a_S32_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023955_directory33a_S33_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023955_directory33a_S33_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023955_directory33a_S33_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023956_directory33a_S33_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023956_directory33a_S33_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023956_directory33a_S33_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023957_directory34a_S34_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023957_directory34a_S34_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023957_directory34a_S34_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023958_directory34a_S34_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023958_directory34a_S34_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023958_directory34a_S34_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023959_directory35a_S35_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023959_directory35a_S35_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023959_directory35a_S35_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023960_directory35a_S35_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023960_directory35a_S35_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023960_directory35a_S35_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023961_directory36a_S36_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023961_directory36a_S36_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023961_directory36a_S36_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023962_directory36a_S36_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023962_directory36a_S36_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023962_directory36a_S36_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023963_directory37a_S37_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023963_directory37a_S37_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023963_directory37a_S37_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023964_directory37a_S37_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023964_directory37a_S37_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023964_directory37a_S37_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023965_directory38a_S38_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023965_directory38a_S38_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023965_directory38a_S38_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023966_directory38a_S38_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023966_directory38a_S38_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023966_directory38a_S38_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023967_directory39a_S39_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023967_directory39a_S39_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023967_directory39a_S39_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023968_directory39a_S39_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023968_directory39a_S39_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023968_directory39a_S39_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023969_directory40a_S40_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023969_directory40a_S40_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023969_directory40a_S40_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023970_directory40a_S40_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023970_directory40a_S40_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023970_directory40a_S40_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023971_directory42a_S42_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023971_directory42a_S42_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023971_directory42a_S42_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023972_directory42a_S42_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023972_directory42a_S42_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023972_directory42a_S42_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023973_directory43a_S43_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023973_directory43a_S43_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023973_directory43a_S43_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023974_directory43a_S43_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023974_directory43a_S43_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023974_directory43a_S43_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023975_directory44a_S44_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023975_directory44a_S44_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023975_directory44a_S44_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023976_directory44a_S44_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023976_directory44a_S44_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023976_directory44a_S44_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023977_directory45a_S45_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023977_directory45a_S45_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023977_directory45a_S45_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023978_directory45a_S45_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023978_directory45a_S45_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023978_directory45a_S45_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023979_directory46a_S46_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023979_directory46a_S46_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023979_directory46a_S46_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023980_directory46a_S46_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023980_directory46a_S46_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023980_directory46a_S46_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023981_directory47a_S47_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023981_directory47a_S47_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023981_directory47a_S47_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023982_directory47a_S47_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023982_directory47a_S47_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023982_directory47a_S47_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023983_directory48a_S48_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023983_directory48a_S48_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023983_directory48a_S48_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023984_directory48a_S48_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023984_directory48a_S48_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023984_directory48a_S48_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023985_directory49a_S49_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023985_directory49a_S49_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023985_directory49a_S49_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023986_directory50a_S50_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023986_directory50a_S50_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023986_directory50a_S50_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023987_directory51a_S51_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023987_directory51a_S51_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023987_directory51a_S51_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023988_directory88a_S88_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023988_directory88a_S88_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023988_directory88a_S88_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023989_directory88a_S88_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023989_directory88a_S88_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023989_directory88a_S88_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023990_directory89a_S89_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023990_directory89a_S89_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023990_directory89a_S89_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023991_directory89a_S89_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023991_directory89a_S89_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023991_directory89a_S89_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023992_directory90a_S90_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023992_directory90a_S90_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023992_directory90a_S90_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023993_directory90a_S90_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023993_directory90a_S90_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023993_directory90a_S90_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023994_directory91a_S91_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023994_directory91a_S91_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023994_directory91a_S91_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023995_directory91a_S91_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023995_directory91a_S91_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023995_directory91a_S91_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023996_directory92a_S92_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023996_directory92a_S92_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023996_directory92a_S92_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023997_directory92a_S92_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023997_directory92a_S92_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023997_directory92a_S92_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023998_directory93a_S93_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023998_directory93a_S93_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023998_directory93a_S93_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4023999_directory93a_S93_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4023999_directory93a_S93_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023999_directory93a_S93_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024000_directory94a_S94_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024000_directory94a_S94_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024000_directory94a_S94_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024001_directory94a_S94_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024001_directory94a_S94_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024001_directory94a_S94_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024002_directory95a_S95_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024002_directory95a_S95_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024002_directory95a_S95_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024003_directory95a_S95_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024003_directory95a_S95_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024003_directory95a_S95_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024004_directory96a_S96_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024004_directory96a_S96_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024004_directory96a_S96_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024005_directory96a_S96_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024005_directory96a_S96_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024005_directory96a_S96_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024006_directory72a_S72_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024006_directory72a_S72_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024006_directory72a_S72_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024007_directory72a_S72_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024007_directory72a_S72_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024007_directory72a_S72_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024008_directory73a_S73_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024008_directory73a_S73_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024008_directory73a_S73_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024009_directory73a_S73_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024009_directory73a_S73_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024009_directory73a_S73_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024010_directory74a_S74_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024010_directory74a_S74_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024010_directory74a_S74_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024011_directory74a_S74_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024011_directory74a_S74_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024011_directory74a_S74_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024012_directory76a_S76_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024012_directory76a_S76_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024012_directory76a_S76_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024013_directory76a_S76_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024013_directory76a_S76_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024013_directory76a_S76_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024014_directory78a_S78_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024014_directory78a_S78_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024014_directory78a_S78_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024015_directory78a_S78_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024015_directory78a_S78_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024015_directory78a_S78_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024016_directory79a_S79_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024016_directory79a_S79_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024016_directory79a_S79_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024017_directory79a_S79_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024017_directory79a_S79_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024017_directory79a_S79_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024018_directory81a_S81_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024018_directory81a_S81_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024018_directory81a_S81_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024019_directory81a_S81_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024019_directory81a_S81_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024019_directory81a_S81_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024020_directory82a_S82_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024020_directory82a_S82_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024020_directory82a_S82_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024021_directory82a_S82_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024021_directory82a_S82_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024021_directory82a_S82_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024022_directory83a_S83_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024022_directory83a_S83_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024022_directory83a_S83_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024023_directory83a_S83_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024023_directory83a_S83_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024023_directory83a_S83_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024024_directory85a_S85_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024024_directory85a_S85_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024024_directory85a_S85_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024025_directory85a_S85_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024025_directory85a_S85_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024025_directory85a_S85_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024026_directory86a_S86_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024026_directory86a_S86_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024026_directory86a_S86_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024027_directory86a_S86_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024027_directory86a_S86_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024027_directory86a_S86_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024028_directory87a_S87_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024028_directory87a_S87_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024028_directory87a_S87_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE106872\\GSM4024029_directory87a_S87_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE106872\\GSM4024029_directory87a_S87_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024029_directory87a_S87_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE117865\\GSE117865_RAW.tar\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE117865\\GSM3681445_RS411_ATAC_DMSO_peaks.bed.gz → rna_seq_analysis\\GSE117865\\GSM3681445_RS411_ATAC_DMSO_peaks.bed\n",
      "❌ Failed to decompress GZ GSM3681445_RS411_ATAC_DMSO_peaks.bed.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE117865\\GSM3681446_RS411_ATAC_EPZ_peaks.bed.gz → rna_seq_analysis\\GSE117865\\GSM3681446_RS411_ATAC_EPZ_peaks.bed\n",
      "❌ Failed to decompress GZ GSM3681446_RS411_ATAC_EPZ_peaks.bed.gz: [Errno 28] No space left on device\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE11884\\GSE11884_RAW.tar\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE11884\\GSM300153.CEL\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE11884\\GSM300154.CEL\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE11884\\GSM300155.CEL\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE11884\\GSM300156.CEL\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE130592\\GSE130592_RAW.tar\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE130592\\GSM3744435_01_day2_m2_summits.bed\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE130592\\GSM3744436_02_day2_m2_summits.bed\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE130592\\GSM3744437_03_day2_m2_summits.bed\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE130592\\GSM3744438_04_day2_m2_summits.bed\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE130592\\GSM3744439_05_day2_m2_summits.bed\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE130592\\GSM3744440_06_day2_m2_summits.bed\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE135632\\GSE135632_RAW.tar\n",
      "❌ Failed to extract TAR GSE135632_RAW.tar: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023826_directoryADAM_49_S1_L004_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023826_directoryADAM_49_S1_L004_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023826_directoryADAM_49_S1_L004_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023827_directoryADAM_58_S2_L004_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023827_directoryADAM_58_S2_L004_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023827_directoryADAM_58_S2_L004_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023828_directoryABCA7_108_S7_L004_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023828_directoryABCA7_108_S7_L004_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023828_directoryABCA7_108_S7_L004_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023829_directoryADAM_19_S15_L005_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023829_directoryADAM_19_S15_L005_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023829_directoryADAM_19_S15_L005_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023830_directoryADAM_98_S16_L005_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023830_directoryADAM_98_S16_L005_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023830_directoryADAM_98_S16_L005_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023831_directoryADAM_35_S17_L005_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023831_directoryADAM_35_S17_L005_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023831_directoryADAM_35_S17_L005_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023832_directoryHUES9-1_8_S8_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023832_directoryHUES9-1_8_S8_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023832_directoryHUES9-1_8_S8_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023833_directoryHUES_9_1_8_NAIVE_R3_S16_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023833_directoryHUES_9_1_8_NAIVE_R3_S16_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023833_directoryHUES_9_1_8_NAIVE_R3_S16_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023834_directoryHUES_9_1_8_NAIVE_R5_S21_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023834_directoryHUES_9_1_8_NAIVE_R5_S21_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023834_directoryHUES_9_1_8_NAIVE_R5_S21_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023835_directoryHUES_9_1_8_NAIVE_R5_HOX_S23_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023835_directoryHUES_9_1_8_NAIVE_R5_HOX_S23_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023835_directoryHUES_9_1_8_NAIVE_R5_HOX_S23_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023836_directoryHUES_9_1_8_RP_R3_S17_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023836_directoryHUES_9_1_8_RP_R3_S17_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023836_directoryHUES_9_1_8_RP_R3_S17_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023837_directoryHUES_9_1_8_RP_R5_S22_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023837_directoryHUES_9_1_8_RP_R5_S22_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023837_directoryHUES_9_1_8_RP_R5_S22_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023838_directoryHUES_9_1_8_RP_R5_HOX_S24_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023838_directoryHUES_9_1_8_RP_R5_HOX_S24_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023838_directoryHUES_9_1_8_RP_R5_HOX_S24_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023839_directoryHUES_9_1_8_RP_R3_SC3_S18_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023839_directoryHUES_9_1_8_RP_R3_SC3_S18_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023839_directoryHUES_9_1_8_RP_R3_SC3_S18_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023840_directoryHUES_9_1_8_RP_R3_SC9_S19_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023840_directoryHUES_9_1_8_RP_R3_SC9_S19_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023840_directoryHUES_9_1_8_RP_R3_SC9_S19_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023841_directoryHUES_9_1_8_RP_R3_SC10_S20_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023841_directoryHUES_9_1_8_RP_R3_SC10_S20_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023841_directoryHUES_9_1_8_RP_R3_SC10_S20_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023842_directoryRP_clone8_S25_L005_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023842_directoryRP_clone8_S25_L005_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023842_directoryRP_clone8_S25_L005_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023843_directoryRP_clone12_S26_L005_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023843_directoryRP_clone12_S26_L005_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023843_directoryRP_clone12_S26_L005_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023844_directoryRP_clone11_S14_L004_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023844_directoryRP_clone11_S14_L004_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023844_directoryRP_clone11_S14_L004_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023845_directoryHUES9_1_8_RP_SC13_S3_L004_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023845_directoryHUES9_1_8_RP_SC13_S3_L004_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023845_directoryHUES9_1_8_RP_SC13_S3_L004_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023846_directoryHUES9_1_8_RP_SC14_S4_L004_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023846_directoryHUES9_1_8_RP_SC14_S4_L004_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023846_directoryHUES9_1_8_RP_SC14_S4_L004_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023847_directoryIIA11_NAIVE_HOX_S10_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023847_directoryIIA11_NAIVE_HOX_S10_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023847_directoryIIA11_NAIVE_HOX_S10_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023848_directoryIIDV8_NAIVE_S27_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023848_directoryIIDV8_NAIVE_S27_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023848_directoryIIDV8_NAIVE_S27_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023849_directoryIIDV8_NAIVE_HOX_S28_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023849_directoryIIDV8_NAIVE_HOX_S28_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023849_directoryIIDV8_NAIVE_HOX_S28_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023850_directoryIB7_naive_S27_L005_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023850_directoryIB7_naive_S27_L005_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023850_directoryIB7_naive_S27_L005_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023851_directorymESC-parent_S1_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023851_directorymESC-parent_S1_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023851_directorymESC-parent_S1_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023852_directorymESC_SC_1_4_S13_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023852_directorymESC_SC_1_4_S13_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023852_directorymESC_SC_1_4_S13_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023853_directorymESC_SC_1_7_S14_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023853_directorymESC_SC_1_7_S14_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023853_directorymESC_SC_1_7_S14_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023854_directorymESC_SC_2_10_S5_L004_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023854_directorymESC_SC_2_10_S5_L004_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023854_directorymESC_SC_2_10_S5_L004_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023855_directorymESC_SC_2_13_S6_L004_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023855_directorymESC_SC_2_13_S6_L004_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023855_directorymESC_SC_2_13_S6_L004_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023856_directorymEPiSC_S12_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023856_directorymEPiSC_S12_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023856_directorymEPiSC_S12_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023857_directoryEpiSC-1_S2_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023857_directoryEpiSC-1_S2_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023857_directoryEpiSC-1_S2_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023858_directoryEpiSC-2_S3_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023858_directoryEpiSC-2_S3_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023858_directoryEpiSC-2_S3_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023859_directoryEpiSC-4_S4_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023859_directoryEpiSC-4_S4_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023859_directoryEpiSC-4_S4_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023860_directoryEpiSC-5_S5_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023860_directoryEpiSC-5_S5_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023860_directoryEpiSC-5_S5_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023861_directoryEpiSC-7_S6_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023861_directoryEpiSC-7_S6_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023861_directoryEpiSC-7_S6_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023862_directoryEpiSC-8_S7_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023862_directoryEpiSC-8_S7_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023862_directoryEpiSC-8_S7_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023863_directoryEpiSC-9_S8_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023863_directoryEpiSC-9_S8_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023863_directoryEpiSC-9_S8_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023864_directoryEpiSC-10_S9_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023864_directoryEpiSC-10_S9_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023864_directoryEpiSC-10_S9_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023865_directoryEpiSC-11_S10_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023865_directoryEpiSC-11_S10_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023865_directoryEpiSC-11_S10_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023866_directoryEpiSC-12_S11_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023866_directoryEpiSC-12_S11_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023866_directoryEpiSC-12_S11_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023867_directoryEpiSC-13_S12_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023867_directoryEpiSC-13_S12_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023867_directoryEpiSC-13_S12_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023868_directoryEpiSC-14_S13_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023868_directoryEpiSC-14_S13_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023868_directoryEpiSC-14_S13_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135632\\GSM4023869_directoryEpiSC-15_S14_L002_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135632\\GSM4023869_directoryEpiSC-15_S14_L002_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023869_directoryEpiSC-15_S14_L002_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE135633\\GSE135633_RAW.tar\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135633\\GSM4023870_directoryHues9_1_8_naive_S13_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135633\\GSM4023870_directoryHues9_1_8_naive_S13_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023870_directoryHues9_1_8_naive_S13_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135633\\GSM4023871_directoryHues9_1_8_primed_S12_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135633\\GSM4023871_directoryHues9_1_8_primed_S12_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023871_directoryHues9_1_8_primed_S12_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135633\\GSM4023872_directoryHues9_1_8_reprimed_S14_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135633\\GSM4023872_directoryHues9_1_8_reprimed_S14_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023872_directoryHues9_1_8_reprimed_S14_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135633\\GSM4023873_directoryCVB_organoids_S15_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135633\\GSM4023873_directoryCVB_organoids_S15_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023873_directoryCVB_organoids_S15_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135633\\GSM4023874_directoryIID4-cortical_S11_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135633\\GSM4023874_directoryIID4-cortical_S11_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023874_directoryIID4-cortical_S11_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135633\\GSM4023875_directoryID12-cortical_S10_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135633\\GSM4023875_directoryID12-cortical_S10_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023875_directoryID12-cortical_S10_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135633\\GSM4023876_directoryIIA12-cortical_S9_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135633\\GSM4023876_directoryIIA12-cortical_S9_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023876_directoryIIA12-cortical_S9_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE135634\\GSE135634_RAW.tar\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023877_directory54a_S54_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023877_directory54a_S54_L007_R1_001.fastq.ucsc.bedGraph\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023878_directory55a_S55_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023878_directory55a_S55_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023878_directory55a_S55_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023879_directory56a_S56_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023879_directory56a_S56_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023879_directory56a_S56_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023880_directory56a_S56_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023880_directory56a_S56_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023880_directory56a_S56_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023881_directory57a_S57_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023881_directory57a_S57_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023881_directory57a_S57_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023882_directory57a_S57_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023882_directory57a_S57_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023882_directory57a_S57_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023883_directory58a_S58_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023883_directory58a_S58_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023883_directory58a_S58_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023884_directory58a_S58_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023884_directory58a_S58_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023884_directory58a_S58_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023885_directory59a_S59_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023885_directory59a_S59_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023885_directory59a_S59_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023886_directory59a_S59_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023886_directory59a_S59_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023886_directory59a_S59_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023887_directory60a_S60_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023887_directory60a_S60_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023887_directory60a_S60_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023888_directory60a_S60_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023888_directory60a_S60_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023888_directory60a_S60_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023889_directory61a_S61_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023889_directory61a_S61_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023889_directory61a_S61_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023890_directory61a_S61_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023890_directory61a_S61_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023890_directory61a_S61_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023891_directory62a_S62_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023891_directory62a_S62_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023891_directory62a_S62_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023892_directory62a_S62_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023892_directory62a_S62_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023892_directory62a_S62_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023893_directory63a_S63_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023893_directory63a_S63_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023893_directory63a_S63_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023894_directory63a_S63_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023894_directory63a_S63_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023894_directory63a_S63_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023895_directory65a_S65_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023895_directory65a_S65_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023895_directory65a_S65_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023896_directory65a_S65_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023896_directory65a_S65_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023896_directory65a_S65_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023897_directory66a_S66_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023897_directory66a_S66_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023897_directory66a_S66_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023898_directory66a_S66_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023898_directory66a_S66_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023898_directory66a_S66_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023899_directory67a_S67_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023899_directory67a_S67_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023899_directory67a_S67_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023900_directory67a_S67_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023900_directory67a_S67_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023900_directory67a_S67_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023901_directory68a_S68_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023901_directory68a_S68_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023901_directory68a_S68_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023902_directory68a_S68_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023902_directory68a_S68_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023902_directory68a_S68_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023903_directory69a_S69_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023903_directory69a_S69_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023903_directory69a_S69_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023904_directory69a_S69_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023904_directory69a_S69_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023904_directory69a_S69_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023905_directory70a_S70_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023905_directory70a_S70_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023905_directory70a_S70_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023906_directory70a_S70_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023906_directory70a_S70_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023906_directory70a_S70_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023907_directory71a_S71_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023907_directory71a_S71_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023907_directory71a_S71_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023908_directory71a_S71_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023908_directory71a_S71_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023908_directory71a_S71_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023909_directory1a_S1_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023909_directory1a_S1_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023909_directory1a_S1_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023910_directory1a_S1_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023910_directory1a_S1_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023910_directory1a_S1_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023911_directory2a_S2_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023911_directory2a_S2_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023911_directory2a_S2_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023912_directory2a_S2_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023912_directory2a_S2_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023912_directory2a_S2_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023913_directory3a_S3_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023913_directory3a_S3_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023913_directory3a_S3_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023914_directory3a_S3_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023914_directory3a_S3_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023914_directory3a_S3_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023915_directory4a_S4_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023915_directory4a_S4_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023915_directory4a_S4_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023916_directory4a_S4_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023916_directory4a_S4_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023916_directory4a_S4_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023917_directory5a_S5_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023917_directory5a_S5_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023917_directory5a_S5_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023918_directory5a_S5_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023918_directory5a_S5_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023918_directory5a_S5_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023919_directory7a_S7_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023919_directory7a_S7_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023919_directory7a_S7_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023920_directory7a_S7_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023920_directory7a_S7_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023920_directory7a_S7_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023921_directory10a_S10_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023921_directory10a_S10_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023921_directory10a_S10_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023922_directory10a_S10_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023922_directory10a_S10_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023922_directory10a_S10_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023923_directory11a_S11_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023923_directory11a_S11_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023923_directory11a_S11_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023924_directory11a_S11_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023924_directory11a_S11_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023924_directory11a_S11_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023925_directory12a_S12_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023925_directory12a_S12_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023925_directory12a_S12_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023926_directory12a_S12_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023926_directory12a_S12_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023926_directory12a_S12_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023927_directory14a_S14_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023927_directory14a_S14_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023927_directory14a_S14_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023928_directory14a_S14_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023928_directory14a_S14_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023928_directory14a_S14_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023929_directory15a_S15_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023929_directory15a_S15_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023929_directory15a_S15_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023930_directory15a_S15_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023930_directory15a_S15_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023930_directory15a_S15_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023931_directory16a_S16_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023931_directory16a_S16_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023931_directory16a_S16_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023932_directory16a_S16_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023932_directory16a_S16_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023932_directory16a_S16_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023933_directory19a_S19_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023933_directory19a_S19_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023933_directory19a_S19_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023934_directory19a_S19_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023934_directory19a_S19_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023934_directory19a_S19_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023935_directory20a_S20_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023935_directory20a_S20_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023935_directory20a_S20_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023936_directory20a_S20_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023936_directory20a_S20_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023936_directory20a_S20_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023937_directory22a_S22_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023937_directory22a_S22_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023937_directory22a_S22_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023938_directory22a_S22_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023938_directory22a_S22_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023938_directory22a_S22_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023939_directory23a_S23_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023939_directory23a_S23_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023939_directory23a_S23_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023940_directory23a_S23_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023940_directory23a_S23_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023940_directory23a_S23_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023941_directory24a_S24_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023941_directory24a_S24_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023941_directory24a_S24_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023942_directory24a_S24_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023942_directory24a_S24_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023942_directory24a_S24_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023943_directory25a_S25_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023943_directory25a_S25_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023943_directory25a_S25_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023944_directory25a_S25_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023944_directory25a_S25_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023944_directory25a_S25_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023945_directory27a_S27_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023945_directory27a_S27_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023945_directory27a_S27_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023946_directory27a_S27_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023946_directory27a_S27_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023946_directory27a_S27_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023947_directory28a_S28_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023947_directory28a_S28_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023947_directory28a_S28_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023948_directory28a_S28_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023948_directory28a_S28_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023948_directory28a_S28_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023949_directory29a_S29_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023949_directory29a_S29_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023949_directory29a_S29_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023950_directory29a_S29_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023950_directory29a_S29_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023950_directory29a_S29_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023951_directory31a_S31_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023951_directory31a_S31_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023951_directory31a_S31_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023952_directory31a_S31_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023952_directory31a_S31_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023952_directory31a_S31_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023953_directory32a_S32_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023953_directory32a_S32_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023953_directory32a_S32_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023954_directory32a_S32_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023954_directory32a_S32_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023954_directory32a_S32_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023955_directory33a_S33_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023955_directory33a_S33_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023955_directory33a_S33_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023956_directory33a_S33_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023956_directory33a_S33_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023956_directory33a_S33_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023957_directory34a_S34_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023957_directory34a_S34_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023957_directory34a_S34_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023958_directory34a_S34_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023958_directory34a_S34_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023958_directory34a_S34_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023959_directory35a_S35_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023959_directory35a_S35_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023959_directory35a_S35_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023960_directory35a_S35_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023960_directory35a_S35_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023960_directory35a_S35_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023961_directory36a_S36_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023961_directory36a_S36_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023961_directory36a_S36_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023962_directory36a_S36_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023962_directory36a_S36_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023962_directory36a_S36_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023963_directory37a_S37_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023963_directory37a_S37_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023963_directory37a_S37_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023964_directory37a_S37_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023964_directory37a_S37_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023964_directory37a_S37_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023965_directory38a_S38_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023965_directory38a_S38_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023965_directory38a_S38_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023966_directory38a_S38_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023966_directory38a_S38_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023966_directory38a_S38_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023967_directory39a_S39_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023967_directory39a_S39_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023967_directory39a_S39_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023968_directory39a_S39_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023968_directory39a_S39_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023968_directory39a_S39_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023969_directory40a_S40_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023969_directory40a_S40_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023969_directory40a_S40_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023970_directory40a_S40_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023970_directory40a_S40_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023970_directory40a_S40_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023971_directory42a_S42_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023971_directory42a_S42_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023971_directory42a_S42_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023972_directory42a_S42_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023972_directory42a_S42_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023972_directory42a_S42_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023973_directory43a_S43_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023973_directory43a_S43_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023973_directory43a_S43_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023974_directory43a_S43_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023974_directory43a_S43_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023974_directory43a_S43_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023975_directory44a_S44_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023975_directory44a_S44_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023975_directory44a_S44_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023976_directory44a_S44_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023976_directory44a_S44_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023976_directory44a_S44_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023977_directory45a_S45_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023977_directory45a_S45_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023977_directory45a_S45_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023978_directory45a_S45_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023978_directory45a_S45_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023978_directory45a_S45_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023979_directory46a_S46_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023979_directory46a_S46_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023979_directory46a_S46_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023980_directory46a_S46_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023980_directory46a_S46_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023980_directory46a_S46_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023981_directory47a_S47_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023981_directory47a_S47_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023981_directory47a_S47_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023982_directory47a_S47_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023982_directory47a_S47_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023982_directory47a_S47_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023983_directory48a_S48_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023983_directory48a_S48_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023983_directory48a_S48_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023984_directory48a_S48_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023984_directory48a_S48_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023984_directory48a_S48_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023985_directory49a_S49_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023985_directory49a_S49_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023985_directory49a_S49_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023986_directory50a_S50_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023986_directory50a_S50_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023986_directory50a_S50_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023987_directory51a_S51_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023987_directory51a_S51_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023987_directory51a_S51_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023988_directory88a_S88_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023988_directory88a_S88_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023988_directory88a_S88_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023989_directory88a_S88_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023989_directory88a_S88_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023989_directory88a_S88_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023990_directory89a_S89_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023990_directory89a_S89_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023990_directory89a_S89_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023991_directory89a_S89_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023991_directory89a_S89_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023991_directory89a_S89_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023992_directory90a_S90_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023992_directory90a_S90_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023992_directory90a_S90_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023993_directory90a_S90_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023993_directory90a_S90_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023993_directory90a_S90_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023994_directory91a_S91_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023994_directory91a_S91_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023994_directory91a_S91_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023995_directory91a_S91_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023995_directory91a_S91_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023995_directory91a_S91_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023996_directory92a_S92_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023996_directory92a_S92_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023996_directory92a_S92_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023997_directory92a_S92_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023997_directory92a_S92_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023997_directory92a_S92_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023998_directory93a_S93_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023998_directory93a_S93_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023998_directory93a_S93_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4023999_directory93a_S93_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4023999_directory93a_S93_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4023999_directory93a_S93_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024000_directory94a_S94_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024000_directory94a_S94_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024000_directory94a_S94_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024001_directory94a_S94_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024001_directory94a_S94_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024001_directory94a_S94_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024002_directory95a_S95_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024002_directory95a_S95_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024002_directory95a_S95_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024003_directory95a_S95_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024003_directory95a_S95_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024003_directory95a_S95_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024004_directory96a_S96_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024004_directory96a_S96_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024004_directory96a_S96_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024005_directory96a_S96_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024005_directory96a_S96_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024005_directory96a_S96_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024006_directory72a_S72_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024006_directory72a_S72_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024006_directory72a_S72_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024007_directory72a_S72_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024007_directory72a_S72_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024007_directory72a_S72_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024008_directory73a_S73_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024008_directory73a_S73_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024008_directory73a_S73_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024009_directory73a_S73_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024009_directory73a_S73_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024009_directory73a_S73_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024010_directory74a_S74_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024010_directory74a_S74_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024010_directory74a_S74_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024011_directory74a_S74_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024011_directory74a_S74_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024011_directory74a_S74_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024012_directory76a_S76_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024012_directory76a_S76_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024012_directory76a_S76_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024013_directory76a_S76_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024013_directory76a_S76_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024013_directory76a_S76_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024014_directory78a_S78_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024014_directory78a_S78_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024014_directory78a_S78_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024015_directory78a_S78_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024015_directory78a_S78_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024015_directory78a_S78_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024016_directory79a_S79_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024016_directory79a_S79_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024016_directory79a_S79_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024017_directory79a_S79_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024017_directory79a_S79_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024017_directory79a_S79_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024018_directory81a_S81_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024018_directory81a_S81_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024018_directory81a_S81_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024019_directory81a_S81_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024019_directory81a_S81_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024019_directory81a_S81_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024020_directory82a_S82_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024020_directory82a_S82_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024020_directory82a_S82_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024021_directory82a_S82_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024021_directory82a_S82_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024021_directory82a_S82_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024022_directory83a_S83_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024022_directory83a_S83_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024022_directory83a_S83_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024023_directory83a_S83_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024023_directory83a_S83_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024023_directory83a_S83_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024024_directory85a_S85_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024024_directory85a_S85_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024024_directory85a_S85_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024025_directory85a_S85_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024025_directory85a_S85_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024025_directory85a_S85_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024026_directory86a_S86_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024026_directory86a_S86_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024026_directory86a_S86_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024027_directory86a_S86_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024027_directory86a_S86_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024027_directory86a_S86_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024028_directory87a_S87_L007_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024028_directory87a_S87_L007_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024028_directory87a_S87_L007_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE135634\\GSM4024029_directory87a_S87_L008_R1_001.fastq.ucsc.bedGraph.gz → rna_seq_analysis\\GSE135634\\GSM4024029_directory87a_S87_L008_R1_001.fastq.ucsc.bedGraph\n",
      "❌ Failed to decompress GZ GSM4024029_directory87a_S87_L008_R1_001.fastq.ucsc.bedGraph.gz: [Errno 28] No space left on device\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE143151\\GSE143151_RAW.tar\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE143151\\GSM4250986_N1_day3_DM_SCR.CEL.gz → rna_seq_analysis\\GSE143151\\GSM4250986_N1_day3_DM_SCR.CEL\n",
      "❌ Failed to decompress GZ GSM4250986_N1_day3_DM_SCR.CEL.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE143151\\GSM4250987_N1_day3_DM_SiJMJD2B.CEL.gz → rna_seq_analysis\\GSE143151\\GSM4250987_N1_day3_DM_SiJMJD2B.CEL\n",
      "❌ Failed to decompress GZ GSM4250987_N1_day3_DM_SiJMJD2B.CEL.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE143151\\GSM4250988_N1_day3_FM_SCR.CEL.gz → rna_seq_analysis\\GSE143151\\GSM4250988_N1_day3_FM_SCR.CEL\n",
      "❌ Failed to decompress GZ GSM4250988_N1_day3_FM_SCR.CEL.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE143151\\GSM4250989_N1_day3_FM_SiJMJD2B.CEL.gz → rna_seq_analysis\\GSE143151\\GSM4250989_N1_day3_FM_SiJMJD2B.CEL\n",
      "❌ Failed to decompress GZ GSM4250989_N1_day3_FM_SiJMJD2B.CEL.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE143151\\GSM4250990_N2_day3_DM_SCR.CEL.gz → rna_seq_analysis\\GSE143151\\GSM4250990_N2_day3_DM_SCR.CEL\n",
      "❌ Failed to decompress GZ GSM4250990_N2_day3_DM_SCR.CEL.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE143151\\GSM4250991_N2_day3_DM_SiJMJD2B.CEL.gz → rna_seq_analysis\\GSE143151\\GSM4250991_N2_day3_DM_SiJMJD2B.CEL\n",
      "❌ Failed to decompress GZ GSM4250991_N2_day3_DM_SiJMJD2B.CEL.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE143151\\GSM4250992_N2_day3_FM_SCR.CEL.gz → rna_seq_analysis\\GSE143151\\GSM4250992_N2_day3_FM_SCR.CEL\n",
      "❌ Failed to decompress GZ GSM4250992_N2_day3_FM_SCR.CEL.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE143151\\GSM4250993_N2_day3_FM_SiJMJD2B.CEL.gz → rna_seq_analysis\\GSE143151\\GSM4250993_N2_day3_FM_SiJMJD2B.CEL\n",
      "❌ Failed to decompress GZ GSM4250993_N2_day3_FM_SiJMJD2B.CEL.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE143151\\GSM4250994_N3_day3_DM_SCR.CEL.gz → rna_seq_analysis\\GSE143151\\GSM4250994_N3_day3_DM_SCR.CEL\n",
      "❌ Failed to decompress GZ GSM4250994_N3_day3_DM_SCR.CEL.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE143151\\GSM4250995_N3_day3_DM_SiJMJD2B.CEL.gz → rna_seq_analysis\\GSE143151\\GSM4250995_N3_day3_DM_SiJMJD2B.CEL\n",
      "❌ Failed to decompress GZ GSM4250995_N3_day3_DM_SiJMJD2B.CEL.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE143151\\GSM4250996_N3_day3_FM_SCR.CEL.gz → rna_seq_analysis\\GSE143151\\GSM4250996_N3_day3_FM_SCR.CEL\n",
      "❌ Failed to decompress GZ GSM4250996_N3_day3_FM_SCR.CEL.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE143151\\GSM4250997_N3_day3_FM_SiJMJD2B.CEL.gz → rna_seq_analysis\\GSE143151\\GSM4250997_N3_day3_FM_SiJMJD2B.CEL\n",
      "❌ Failed to decompress GZ GSM4250997_N3_day3_FM_SiJMJD2B.CEL.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE143151\\GSM4254154_H3K9_n1_DM.bed.gz → rna_seq_analysis\\GSE143151\\GSM4254154_H3K9_n1_DM.bed\n",
      "❌ Failed to decompress GZ GSM4254154_H3K9_n1_DM.bed.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE143151\\GSM4254155_H3K9_n1_FM.bed.gz → rna_seq_analysis\\GSE143151\\GSM4254155_H3K9_n1_FM.bed\n",
      "❌ Failed to decompress GZ GSM4254155_H3K9_n1_FM.bed.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE143151\\GSM4254156_H3K9_n2_DM.bed.gz → rna_seq_analysis\\GSE143151\\GSM4254156_H3K9_n2_DM.bed\n",
      "❌ Failed to decompress GZ GSM4254156_H3K9_n2_DM.bed.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE143151\\GSM4254157_H3K9_n2_FM.bed.gz → rna_seq_analysis\\GSE143151\\GSM4254157_H3K9_n2_FM.bed\n",
      "❌ Failed to decompress GZ GSM4254157_H3K9_n2_FM.bed.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE143151\\GSM4254158_H3K9_n3_DM.bed.gz → rna_seq_analysis\\GSE143151\\GSM4254158_H3K9_n3_DM.bed\n",
      "❌ Failed to decompress GZ GSM4254158_H3K9_n3_DM.bed.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE143151\\GSM4254159_H3K9_n3_FM.bed.gz → rna_seq_analysis\\GSE143151\\GSM4254159_H3K9_n3_FM.bed\n",
      "❌ Failed to decompress GZ GSM4254159_H3K9_n3_FM.bed.gz: [Errno 28] No space left on device\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE146579\\GSE146579_RAW.tar\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE146581\\GSE146581_RAW.tar\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE146582\\GSE146582_RAW.tar\n",
      "❌ Failed to extract TAR GSE146582_RAW.tar: [Errno 28] No space left on device\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE146585\\GSE146585_RAW.tar\n",
      "❌ Failed to extract TAR GSE146585_RAW.tar: [Errno 28] No space left on device\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE148053\\GSE148053_RAW.tar\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148053\\GSM4453128_P3_Ctrl_rep1_RPKM.txt.gz → rna_seq_analysis\\GSE148053\\GSM4453128_P3_Ctrl_rep1_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453128_P3_Ctrl_rep1_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148053\\GSM4453129_P3_Ctrl_rep2_RPKM.txt.gz → rna_seq_analysis\\GSE148053\\GSM4453129_P3_Ctrl_rep2_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453129_P3_Ctrl_rep2_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148053\\GSM4453130_P3_DKO_rep1_RPKM.txt.gz → rna_seq_analysis\\GSE148053\\GSM4453130_P3_DKO_rep1_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453130_P3_DKO_rep1_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148053\\GSM4453131_P3_DKO_rep2_RPKM.txt.gz → rna_seq_analysis\\GSE148053\\GSM4453131_P3_DKO_rep2_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453131_P3_DKO_rep2_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148053\\GSM4453132_P3_Jmjd1aKO_rep1_RPKM.txt.gz → rna_seq_analysis\\GSE148053\\GSM4453132_P3_Jmjd1aKO_rep1_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453132_P3_Jmjd1aKO_rep1_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148053\\GSM4453133_P3_Jmjd1aKO_rep2_RPKM.txt.gz → rna_seq_analysis\\GSE148053\\GSM4453133_P3_Jmjd1aKO_rep2_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453133_P3_Jmjd1aKO_rep2_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148053\\GSM4453134_P3_Jmjd1bKO_rep1_RPKM.txt.gz → rna_seq_analysis\\GSE148053\\GSM4453134_P3_Jmjd1bKO_rep1_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453134_P3_Jmjd1bKO_rep1_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148053\\GSM4453135_P3_Jmjd1bKO_rep2_RPKM.txt.gz → rna_seq_analysis\\GSE148053\\GSM4453135_P3_Jmjd1bKO_rep2_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453135_P3_Jmjd1bKO_rep2_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148053\\GSM4453136_P3_WT_rep1_RPKM.txt.gz → rna_seq_analysis\\GSE148053\\GSM4453136_P3_WT_rep1_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453136_P3_WT_rep1_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148053\\GSM4453137_P3_WT_rep2_RPKM.txt.gz → rna_seq_analysis\\GSE148053\\GSM4453137_P3_WT_rep2_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453137_P3_WT_rep2_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148053\\GSM4453138_P7_Ctrl_rep1_RPKM.txt.gz → rna_seq_analysis\\GSE148053\\GSM4453138_P7_Ctrl_rep1_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453138_P7_Ctrl_rep1_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148053\\GSM4453139_P7_Ctrl_rep2_RPKM.txt.gz → rna_seq_analysis\\GSE148053\\GSM4453139_P7_Ctrl_rep2_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453139_P7_Ctrl_rep2_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148053\\GSM4453140_P7_DKO_rep1_RPKM.txt.gz → rna_seq_analysis\\GSE148053\\GSM4453140_P7_DKO_rep1_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453140_P7_DKO_rep1_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148053\\GSM4453141_P7_DKO_rep2_RPKM.txt.gz → rna_seq_analysis\\GSE148053\\GSM4453141_P7_DKO_rep2_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453141_P7_DKO_rep2_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE148054\\GSE148054_RAW.tar\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE148055\\GSE148055_RAW.tar\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148055\\GSM4453128_P3_Ctrl_rep1_RPKM.txt.gz → rna_seq_analysis\\GSE148055\\GSM4453128_P3_Ctrl_rep1_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453128_P3_Ctrl_rep1_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148055\\GSM4453129_P3_Ctrl_rep2_RPKM.txt.gz → rna_seq_analysis\\GSE148055\\GSM4453129_P3_Ctrl_rep2_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453129_P3_Ctrl_rep2_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148055\\GSM4453130_P3_DKO_rep1_RPKM.txt.gz → rna_seq_analysis\\GSE148055\\GSM4453130_P3_DKO_rep1_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453130_P3_DKO_rep1_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148055\\GSM4453131_P3_DKO_rep2_RPKM.txt.gz → rna_seq_analysis\\GSE148055\\GSM4453131_P3_DKO_rep2_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453131_P3_DKO_rep2_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148055\\GSM4453132_P3_Jmjd1aKO_rep1_RPKM.txt.gz → rna_seq_analysis\\GSE148055\\GSM4453132_P3_Jmjd1aKO_rep1_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453132_P3_Jmjd1aKO_rep1_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148055\\GSM4453133_P3_Jmjd1aKO_rep2_RPKM.txt.gz → rna_seq_analysis\\GSE148055\\GSM4453133_P3_Jmjd1aKO_rep2_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453133_P3_Jmjd1aKO_rep2_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148055\\GSM4453134_P3_Jmjd1bKO_rep1_RPKM.txt.gz → rna_seq_analysis\\GSE148055\\GSM4453134_P3_Jmjd1bKO_rep1_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453134_P3_Jmjd1bKO_rep1_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148055\\GSM4453135_P3_Jmjd1bKO_rep2_RPKM.txt.gz → rna_seq_analysis\\GSE148055\\GSM4453135_P3_Jmjd1bKO_rep2_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453135_P3_Jmjd1bKO_rep2_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148055\\GSM4453136_P3_WT_rep1_RPKM.txt.gz → rna_seq_analysis\\GSE148055\\GSM4453136_P3_WT_rep1_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453136_P3_WT_rep1_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148055\\GSM4453137_P3_WT_rep2_RPKM.txt.gz → rna_seq_analysis\\GSE148055\\GSM4453137_P3_WT_rep2_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453137_P3_WT_rep2_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148055\\GSM4453138_P7_Ctrl_rep1_RPKM.txt.gz → rna_seq_analysis\\GSE148055\\GSM4453138_P7_Ctrl_rep1_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453138_P7_Ctrl_rep1_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148055\\GSM4453139_P7_Ctrl_rep2_RPKM.txt.gz → rna_seq_analysis\\GSE148055\\GSM4453139_P7_Ctrl_rep2_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453139_P7_Ctrl_rep2_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148055\\GSM4453140_P7_DKO_rep1_RPKM.txt.gz → rna_seq_analysis\\GSE148055\\GSM4453140_P7_DKO_rep1_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453140_P7_DKO_rep1_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE148055\\GSM4453141_P7_DKO_rep2_RPKM.txt.gz → rna_seq_analysis\\GSE148055\\GSM4453141_P7_DKO_rep2_RPKM.txt\n",
      "❌ Failed to decompress GZ GSM4453141_P7_DKO_rep2_RPKM.txt.gz: [Errno 28] No space left on device\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE166351\\GSE166351_RAW.tar\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE166351\\GSM5068901_A1.fpkm.txt.gz → rna_seq_analysis\\GSE166351\\GSM5068901_A1.fpkm.txt\n",
      "❌ Failed to decompress GZ GSM5068901_A1.fpkm.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE166351\\GSM5068902_A2.fpkm.txt.gz → rna_seq_analysis\\GSE166351\\GSM5068902_A2.fpkm.txt\n",
      "❌ Failed to decompress GZ GSM5068902_A2.fpkm.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE166351\\GSM5068903_A3.fpkm.txt.gz → rna_seq_analysis\\GSE166351\\GSM5068903_A3.fpkm.txt\n",
      "❌ Failed to decompress GZ GSM5068903_A3.fpkm.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE166351\\GSM5068904_B1.fpkm.txt.gz → rna_seq_analysis\\GSE166351\\GSM5068904_B1.fpkm.txt\n",
      "❌ Failed to decompress GZ GSM5068904_B1.fpkm.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE166351\\GSM5068905_B2.fpkm.txt.gz → rna_seq_analysis\\GSE166351\\GSM5068905_B2.fpkm.txt\n",
      "❌ Failed to decompress GZ GSM5068905_B2.fpkm.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE166351\\GSM5068906_B3.fpkm.txt.gz → rna_seq_analysis\\GSE166351\\GSM5068906_B3.fpkm.txt\n",
      "❌ Failed to decompress GZ GSM5068906_B3.fpkm.txt.gz: [Errno 28] No space left on device\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE173995\\GSE173995_RAW.tar\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE173995\\GSM5284287_K9me2_30uM_IOX1.broadPeak.gz → rna_seq_analysis\\GSE173995\\GSM5284287_K9me2_30uM_IOX1.broadPeak\n",
      "❌ Failed to decompress GZ GSM5284287_K9me2_30uM_IOX1.broadPeak.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE173995\\GSM5284288_K9me2_50uM_IOX1.broadPeak.gz → rna_seq_analysis\\GSE173995\\GSM5284288_K9me2_50uM_IOX1.broadPeak\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE173995\\GSM5284289_K9me2_100uM_IOX1.broadPeak.gz → rna_seq_analysis\\GSE173995\\GSM5284289_K9me2_100uM_IOX1.broadPeak\n",
      "❌ Failed to decompress GZ GSM5284289_K9me2_100uM_IOX1.broadPeak.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE173995\\\\GSM5284289_K9me2_100uM_IOX1.broadPeak'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE173995\\GSM5284290_K9me2_DMSO.broadPeak.gz → rna_seq_analysis\\GSE173995\\GSM5284290_K9me2_DMSO.broadPeak\n",
      "❌ Failed to decompress GZ GSM5284290_K9me2_DMSO.broadPeak.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE173995\\\\GSM5284290_K9me2_DMSO.broadPeak'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE173995\\GSM5284291_K9me3_30uM_IOX1.broadPeak.gz → rna_seq_analysis\\GSE173995\\GSM5284291_K9me3_30uM_IOX1.broadPeak\n",
      "❌ Failed to decompress GZ GSM5284291_K9me3_30uM_IOX1.broadPeak.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE173995\\\\GSM5284291_K9me3_30uM_IOX1.broadPeak'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE173995\\GSM5284292_K9me3_50uM_IOX1.broadPeak.gz → rna_seq_analysis\\GSE173995\\GSM5284292_K9me3_50uM_IOX1.broadPeak\n",
      "❌ Failed to decompress GZ GSM5284292_K9me3_50uM_IOX1.broadPeak.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE173995\\\\GSM5284292_K9me3_50uM_IOX1.broadPeak'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE173995\\GSM5284293_K9me3_100uM_IOX1.broadPeak.gz → rna_seq_analysis\\GSE173995\\GSM5284293_K9me3_100uM_IOX1.broadPeak\n",
      "❌ Failed to decompress GZ GSM5284293_K9me3_100uM_IOX1.broadPeak.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE173995\\\\GSM5284293_K9me3_100uM_IOX1.broadPeak'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE173995\\GSM5284294_K9me3_DMSO.broadPeak.gz → rna_seq_analysis\\GSE173995\\GSM5284294_K9me3_DMSO.broadPeak\n",
      "❌ Failed to decompress GZ GSM5284294_K9me3_DMSO.broadPeak.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE173995\\\\GSM5284294_K9me3_DMSO.broadPeak'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE173995\\GSM5284295_K27me3_30uM_IOX1.broadPeak.gz → rna_seq_analysis\\GSE173995\\GSM5284295_K27me3_30uM_IOX1.broadPeak\n",
      "❌ Failed to decompress GZ GSM5284295_K27me3_30uM_IOX1.broadPeak.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE173995\\\\GSM5284295_K27me3_30uM_IOX1.broadPeak'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE173995\\GSM5284296_K27me3_50uM_IOX1.broadPeak.gz → rna_seq_analysis\\GSE173995\\GSM5284296_K27me3_50uM_IOX1.broadPeak\n",
      "❌ Failed to decompress GZ GSM5284296_K27me3_50uM_IOX1.broadPeak.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE173995\\\\GSM5284296_K27me3_50uM_IOX1.broadPeak'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE173995\\GSM5284297_K27me3_100uM_IOX1.broadPeak.gz → rna_seq_analysis\\GSE173995\\GSM5284297_K27me3_100uM_IOX1.broadPeak\n",
      "❌ Failed to decompress GZ GSM5284297_K27me3_100uM_IOX1.broadPeak.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE173995\\\\GSM5284297_K27me3_100uM_IOX1.broadPeak'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE173995\\GSM5284298_K27me3_DMSO.broadPeak.gz → rna_seq_analysis\\GSE173995\\GSM5284298_K27me3_DMSO.broadPeak\n",
      "❌ Failed to decompress GZ GSM5284298_K27me3_DMSO.broadPeak.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE173995\\\\GSM5284298_K27me3_DMSO.broadPeak'\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE180233\\GSE180233_RAW.tar\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE180233\\GSM5456377_ASC1.txt.gz → rna_seq_analysis\\GSE180233\\GSM5456377_ASC1.txt\n",
      "❌ Failed to decompress GZ GSM5456377_ASC1.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE180233\\\\GSM5456377_ASC1.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE180233\\GSM5456378_ASC2.txt.gz → rna_seq_analysis\\GSE180233\\GSM5456378_ASC2.txt\n",
      "❌ Failed to decompress GZ GSM5456378_ASC2.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE180233\\\\GSM5456378_ASC2.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE180233\\GSM5456379_ASC3.txt.gz → rna_seq_analysis\\GSE180233\\GSM5456379_ASC3.txt\n",
      "❌ Failed to decompress GZ GSM5456379_ASC3.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE180233\\\\GSM5456379_ASC3.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE180233\\GSM5456380_ASC4.txt.gz → rna_seq_analysis\\GSE180233\\GSM5456380_ASC4.txt\n",
      "❌ Failed to decompress GZ GSM5456380_ASC4.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE180233\\\\GSM5456380_ASC4.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE180233\\GSM5456381_ASC5.txt.gz → rna_seq_analysis\\GSE180233\\GSM5456381_ASC5.txt\n",
      "❌ Failed to decompress GZ GSM5456381_ASC5.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE180233\\\\GSM5456381_ASC5.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE180233\\GSM5456382_ASC6.txt.gz → rna_seq_analysis\\GSE180233\\GSM5456382_ASC6.txt\n",
      "❌ Failed to decompress GZ GSM5456382_ASC6.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE180233\\\\GSM5456382_ASC6.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE180233\\GSM5456383_ASC7.txt.gz → rna_seq_analysis\\GSE180233\\GSM5456383_ASC7.txt\n",
      "❌ Failed to decompress GZ GSM5456383_ASC7.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE180233\\\\GSM5456383_ASC7.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE180233\\GSM5456384_ASC8.txt.gz → rna_seq_analysis\\GSE180233\\GSM5456384_ASC8.txt\n",
      "❌ Failed to decompress GZ GSM5456384_ASC8.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE180233\\\\GSM5456384_ASC8.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE180233\\GSM5456385_ASC9.txt.gz → rna_seq_analysis\\GSE180233\\GSM5456385_ASC9.txt\n",
      "❌ Failed to decompress GZ GSM5456385_ASC9.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE180233\\\\GSM5456385_ASC9.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE180233\\GSM5456386_ASC10.txt.gz → rna_seq_analysis\\GSE180233\\GSM5456386_ASC10.txt\n",
      "❌ Failed to decompress GZ GSM5456386_ASC10.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE180233\\\\GSM5456386_ASC10.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE180233\\GSM5456387_ASC11.txt.gz → rna_seq_analysis\\GSE180233\\GSM5456387_ASC11.txt\n",
      "❌ Failed to decompress GZ GSM5456387_ASC11.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE180233\\\\GSM5456387_ASC11.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE180233\\GSM5456388_ASC12.txt.gz → rna_seq_analysis\\GSE180233\\GSM5456388_ASC12.txt\n",
      "❌ Failed to decompress GZ GSM5456388_ASC12.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE180233\\\\GSM5456388_ASC12.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE180233\\GSM5456389_ASC13.txt.gz → rna_seq_analysis\\GSE180233\\GSM5456389_ASC13.txt\n",
      "❌ Failed to decompress GZ GSM5456389_ASC13.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE180233\\\\GSM5456389_ASC13.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE180233\\GSM5456390_ASC14.txt.gz → rna_seq_analysis\\GSE180233\\GSM5456390_ASC14.txt\n",
      "❌ Failed to decompress GZ GSM5456390_ASC14.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE180233\\\\GSM5456390_ASC14.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE180233\\GSM5456391_ASC15.txt.gz → rna_seq_analysis\\GSE180233\\GSM5456391_ASC15.txt\n",
      "❌ Failed to decompress GZ GSM5456391_ASC15.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE180233\\\\GSM5456391_ASC15.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE180233\\GSM5456392_ASC16.txt.gz → rna_seq_analysis\\GSE180233\\GSM5456392_ASC16.txt\n",
      "❌ Failed to decompress GZ GSM5456392_ASC16.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE180233\\\\GSM5456392_ASC16.txt'\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE193496\\GSE193496_RAW.tar\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE199117\\GSE199117_Differential_gene_expression_RNA-Seq_-Jej.xls\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE202506\\GSE202506_RAW.tar\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE202506\\GSM6123291_scWAT_D0_si-Ctr.txt.gz → rna_seq_analysis\\GSE202506\\GSM6123291_scWAT_D0_si-Ctr.txt\n",
      "❌ Failed to decompress GZ GSM6123291_scWAT_D0_si-Ctr.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE202506\\GSM6123292_scWAT_D0_si-Mypt1.txt.gz → rna_seq_analysis\\GSE202506\\GSM6123292_scWAT_D0_si-Mypt1.txt\n",
      "❌ Failed to decompress GZ GSM6123292_scWAT_D0_si-Mypt1.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE202506\\GSM6123293_scWAT_D8_si-Ctr.txt.gz → rna_seq_analysis\\GSE202506\\GSM6123293_scWAT_D8_si-Ctr.txt\n",
      "❌ Failed to decompress GZ GSM6123293_scWAT_D8_si-Ctr.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE202506\\GSM6123294_scWAT_D8_si-Mypt1.txt.gz → rna_seq_analysis\\GSE202506\\GSM6123294_scWAT_D8_si-Mypt1.txt\n",
      "❌ Failed to decompress GZ GSM6123294_scWAT_D8_si-Mypt1.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE202506\\GSM6123295_im-scWAT_D3_si-Ctr.txt.gz → rna_seq_analysis\\GSE202506\\GSM6123295_im-scWAT_D3_si-Ctr.txt\n",
      "❌ Failed to decompress GZ GSM6123295_im-scWAT_D3_si-Ctr.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE202506\\GSM6123296_im-scWAT_D3_si-Mypt1.txt.gz → rna_seq_analysis\\GSE202506\\GSM6123296_im-scWAT_D3_si-Mypt1.txt\n",
      "❌ Failed to decompress GZ GSM6123296_im-scWAT_D3_si-Mypt1.txt.gz: [Errno 28] No space left on device\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE210740\\GSE210740_all_compare.xls\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE217127\\GSE217127_RAW.tar\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE217127\\GSM6705480_HuT78_DpVp50_1.sorted.rmdup.count.txt.gz → rna_seq_analysis\\GSE217127\\GSM6705480_HuT78_DpVp50_1.sorted.rmdup.count.txt\n",
      "❌ Failed to decompress GZ GSM6705480_HuT78_DpVp50_1.sorted.rmdup.count.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE217127\\GSM6705481_HuT78_DpVp50_2.sorted.rmdup.count.txt.gz → rna_seq_analysis\\GSE217127\\GSM6705481_HuT78_DpVp50_2.sorted.rmdup.count.txt\n",
      "❌ Failed to decompress GZ GSM6705481_HuT78_DpVp50_2.sorted.rmdup.count.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE217127\\\\GSM6705481_HuT78_DpVp50_2.sorted.rmdup.count.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE217127\\GSM6705482_HuT78_DpVp50_3.sorted.rmdup.count.txt.gz → rna_seq_analysis\\GSE217127\\GSM6705482_HuT78_DpVp50_3.sorted.rmdup.count.txt\n",
      "❌ Failed to decompress GZ GSM6705482_HuT78_DpVp50_3.sorted.rmdup.count.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE217127\\\\GSM6705482_HuT78_DpVp50_3.sorted.rmdup.count.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE217127\\GSM6705483_Hut78_c1.sorted.rmdup.count.txt.gz → rna_seq_analysis\\GSE217127\\GSM6705483_Hut78_c1.sorted.rmdup.count.txt\n",
      "❌ Failed to decompress GZ GSM6705483_Hut78_c1.sorted.rmdup.count.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE217127\\\\GSM6705483_Hut78_c1.sorted.rmdup.count.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE217127\\GSM6705484_Hut78_c2.sorted.rmdup.count.txt.gz → rna_seq_analysis\\GSE217127\\GSM6705484_Hut78_c2.sorted.rmdup.count.txt\n",
      "❌ Failed to decompress GZ GSM6705484_Hut78_c2.sorted.rmdup.count.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE217127\\\\GSM6705484_Hut78_c2.sorted.rmdup.count.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE217127\\GSM6705485_Hut78_c3.sorted.rmdup.count.txt.gz → rna_seq_analysis\\GSE217127\\GSM6705485_Hut78_c3.sorted.rmdup.count.txt\n",
      "❌ Failed to decompress GZ GSM6705485_Hut78_c3.sorted.rmdup.count.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE217127\\\\GSM6705485_Hut78_c3.sorted.rmdup.count.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE217127\\GSM6705486_MCF7_M300_1.sorted.rmdup.count.txt.gz → rna_seq_analysis\\GSE217127\\GSM6705486_MCF7_M300_1.sorted.rmdup.count.txt\n",
      "❌ Failed to decompress GZ GSM6705486_MCF7_M300_1.sorted.rmdup.count.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE217127\\\\GSM6705486_MCF7_M300_1.sorted.rmdup.count.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE217127\\GSM6705487_MCF7_M300_2.sorted.rmdup.count.txt.gz → rna_seq_analysis\\GSE217127\\GSM6705487_MCF7_M300_2.sorted.rmdup.count.txt\n",
      "❌ Failed to decompress GZ GSM6705487_MCF7_M300_2.sorted.rmdup.count.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE217127\\\\GSM6705487_MCF7_M300_2.sorted.rmdup.count.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE217127\\GSM6705488_MCF7_M300_3.sorted.rmdup.count.txt.gz → rna_seq_analysis\\GSE217127\\GSM6705488_MCF7_M300_3.sorted.rmdup.count.txt\n",
      "❌ Failed to decompress GZ GSM6705488_MCF7_M300_3.sorted.rmdup.count.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE217127\\\\GSM6705488_MCF7_M300_3.sorted.rmdup.count.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE217127\\GSM6705489_MCF7_C-1.sorted.rmdup.count.txt.gz → rna_seq_analysis\\GSE217127\\GSM6705489_MCF7_C-1.sorted.rmdup.count.txt\n",
      "❌ Failed to decompress GZ GSM6705489_MCF7_C-1.sorted.rmdup.count.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE217127\\\\GSM6705489_MCF7_C-1.sorted.rmdup.count.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE217127\\GSM6705490_MCF7_C-2.sorted.rmdup.count.txt.gz → rna_seq_analysis\\GSE217127\\GSM6705490_MCF7_C-2.sorted.rmdup.count.txt\n",
      "❌ Failed to decompress GZ GSM6705490_MCF7_C-2.sorted.rmdup.count.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE217127\\\\GSM6705490_MCF7_C-2.sorted.rmdup.count.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE217127\\GSM6705491_MCF7_C-3.sorted.rmdup.count.txt.gz → rna_seq_analysis\\GSE217127\\GSM6705491_MCF7_C-3.sorted.rmdup.count.txt\n",
      "❌ Failed to decompress GZ GSM6705491_MCF7_C-3.sorted.rmdup.count.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE217127\\\\GSM6705491_MCF7_C-3.sorted.rmdup.count.txt'\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE219007\\GSE219007_RAW.tar\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE220927\\GSE220927_RAW.tar\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE220927\\GSM6829413_A2WT.gtf\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE220927\\GSM6829414_B5WT.gtf\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE220927\\GSM6829415_C8WT.gtf\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE220927\\GSM6829416_D2WT.gtf\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE220927\\GSM6829417_E2WT.gtf\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE220927\\GSM6829418_A1OE705.gtf\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE220927\\GSM6829419_B4OE705.gtf\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE220927\\GSM6829420_C7OE705.gtf\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE220927\\GSM6829421_D1OE705.gtf\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE220927\\GSM6829422_E1OE705.gtf\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE224611\\GSE224611_RAW.tar\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE224611\\GSM7027442_9409_R1_norm_counts.txt.gz → rna_seq_analysis\\GSE224611\\GSM7027442_9409_R1_norm_counts.txt\n",
      "❌ Failed to decompress GZ GSM7027442_9409_R1_norm_counts.txt.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE224611\\GSM7027443_9410_R2_norm_counts.txt.gz → rna_seq_analysis\\GSE224611\\GSM7027443_9410_R2_norm_counts.txt\n",
      "❌ Failed to decompress GZ GSM7027443_9410_R2_norm_counts.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE224611\\\\GSM7027443_9410_R2_norm_counts.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE224611\\GSM7027444_9411_R3_norm_counts.txt.gz → rna_seq_analysis\\GSE224611\\GSM7027444_9411_R3_norm_counts.txt\n",
      "❌ Failed to decompress GZ GSM7027444_9411_R3_norm_counts.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE224611\\\\GSM7027444_9411_R3_norm_counts.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE224611\\GSM7027445_9412_R1_norm_counts.txt.gz → rna_seq_analysis\\GSE224611\\GSM7027445_9412_R1_norm_counts.txt\n",
      "❌ Failed to decompress GZ GSM7027445_9412_R1_norm_counts.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE224611\\\\GSM7027445_9412_R1_norm_counts.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE224611\\GSM7027446_9413_R2_norm_counts.txt.gz → rna_seq_analysis\\GSE224611\\GSM7027446_9413_R2_norm_counts.txt\n",
      "❌ Failed to decompress GZ GSM7027446_9413_R2_norm_counts.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE224611\\\\GSM7027446_9413_R2_norm_counts.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE224611\\GSM7027447_9414_R3_norm_counts.txt.gz → rna_seq_analysis\\GSE224611\\GSM7027447_9414_R3_norm_counts.txt\n",
      "❌ Failed to decompress GZ GSM7027447_9414_R3_norm_counts.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE224611\\\\GSM7027447_9414_R3_norm_counts.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE224611\\GSM7027448_9415_R1_norm_counts.txt.gz → rna_seq_analysis\\GSE224611\\GSM7027448_9415_R1_norm_counts.txt\n",
      "❌ Failed to decompress GZ GSM7027448_9415_R1_norm_counts.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE224611\\\\GSM7027448_9415_R1_norm_counts.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE224611\\GSM7027449_9416_R2_norm_counts.txt.gz → rna_seq_analysis\\GSE224611\\GSM7027449_9416_R2_norm_counts.txt\n",
      "❌ Failed to decompress GZ GSM7027449_9416_R2_norm_counts.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE224611\\\\GSM7027449_9416_R2_norm_counts.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE224611\\GSM7027450_9417_R3_norm_counts.txt.gz → rna_seq_analysis\\GSE224611\\GSM7027450_9417_R3_norm_counts.txt\n",
      "❌ Failed to decompress GZ GSM7027450_9417_R3_norm_counts.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE224611\\\\GSM7027450_9417_R3_norm_counts.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE224611\\GSM7027451_9418_R1_norm_counts.txt.gz → rna_seq_analysis\\GSE224611\\GSM7027451_9418_R1_norm_counts.txt\n",
      "❌ Failed to decompress GZ GSM7027451_9418_R1_norm_counts.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE224611\\\\GSM7027451_9418_R1_norm_counts.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE224611\\GSM7027452_9419_R2_norm_counts.txt.gz → rna_seq_analysis\\GSE224611\\GSM7027452_9419_R2_norm_counts.txt\n",
      "❌ Failed to decompress GZ GSM7027452_9419_R2_norm_counts.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE224611\\\\GSM7027452_9419_R2_norm_counts.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE224611\\GSM7027453_9420_R3_norm_counts.txt.gz → rna_seq_analysis\\GSE224611\\GSM7027453_9420_R3_norm_counts.txt\n",
      "❌ Failed to decompress GZ GSM7027453_9420_R3_norm_counts.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE224611\\\\GSM7027453_9420_R3_norm_counts.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE224611\\GSM7027454_8794_R1_norm_counts.txt.gz → rna_seq_analysis\\GSE224611\\GSM7027454_8794_R1_norm_counts.txt\n",
      "❌ Failed to decompress GZ GSM7027454_8794_R1_norm_counts.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE224611\\\\GSM7027454_8794_R1_norm_counts.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE224611\\GSM7027455_8795_R2_norm_counts.txt.gz → rna_seq_analysis\\GSE224611\\GSM7027455_8795_R2_norm_counts.txt\n",
      "❌ Failed to decompress GZ GSM7027455_8795_R2_norm_counts.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE224611\\\\GSM7027455_8795_R2_norm_counts.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE224611\\GSM7027456_8796_R3_norm_counts.txt.gz → rna_seq_analysis\\GSE224611\\GSM7027456_8796_R3_norm_counts.txt\n",
      "❌ Failed to decompress GZ GSM7027456_8796_R3_norm_counts.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE224611\\\\GSM7027456_8796_R3_norm_counts.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE224611\\GSM7027457_8797_R1_norm_counts.txt.gz → rna_seq_analysis\\GSE224611\\GSM7027457_8797_R1_norm_counts.txt\n",
      "❌ Failed to decompress GZ GSM7027457_8797_R1_norm_counts.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE224611\\\\GSM7027457_8797_R1_norm_counts.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE224611\\GSM7027458_8798_R2_norm_counts.txt.gz → rna_seq_analysis\\GSE224611\\GSM7027458_8798_R2_norm_counts.txt\n",
      "❌ Failed to decompress GZ GSM7027458_8798_R2_norm_counts.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE224611\\\\GSM7027458_8798_R2_norm_counts.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE224611\\GSM7027459_8799_R3_norm_counts.txt.gz → rna_seq_analysis\\GSE224611\\GSM7027459_8799_R3_norm_counts.txt\n",
      "❌ Failed to decompress GZ GSM7027459_8799_R3_norm_counts.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE224611\\\\GSM7027459_8799_R3_norm_counts.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE224611\\GSM7729531_BT549_count_veh1.txt.gz → rna_seq_analysis\\GSE224611\\GSM7729531_BT549_count_veh1.txt\n",
      "❌ Failed to decompress GZ GSM7729531_BT549_count_veh1.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE224611\\\\GSM7729531_BT549_count_veh1.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE224611\\GSM7729532_BT549_count_veh2.txt.gz → rna_seq_analysis\\GSE224611\\GSM7729532_BT549_count_veh2.txt\n",
      "❌ Failed to decompress GZ GSM7729532_BT549_count_veh2.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE224611\\\\GSM7729532_BT549_count_veh2.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE224611\\GSM7729533_BT549_count_veh3.txt.gz → rna_seq_analysis\\GSE224611\\GSM7729533_BT549_count_veh3.txt\n",
      "❌ Failed to decompress GZ GSM7729533_BT549_count_veh3.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE224611\\\\GSM7729533_BT549_count_veh3.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE224611\\GSM7729534_BT549_count_mb3_1.txt.gz → rna_seq_analysis\\GSE224611\\GSM7729534_BT549_count_mb3_1.txt\n",
      "❌ Failed to decompress GZ GSM7729534_BT549_count_mb3_1.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE224611\\\\GSM7729534_BT549_count_mb3_1.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE224611\\GSM7729535_BT549_count_mb3_2.txt.gz → rna_seq_analysis\\GSE224611\\GSM7729535_BT549_count_mb3_2.txt\n",
      "❌ Failed to decompress GZ GSM7729535_BT549_count_mb3_2.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE224611\\\\GSM7729535_BT549_count_mb3_2.txt'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE224611\\GSM7729536_BT549_count_mb3_3.txt.gz → rna_seq_analysis\\GSE224611\\GSM7729536_BT549_count_mb3_3.txt\n",
      "❌ Failed to decompress GZ GSM7729536_BT549_count_mb3_3.txt.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE224611\\\\GSM7729536_BT549_count_mb3_3.txt'\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE230635\\GSE230635_RAW.tar\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE230675\\GSE230675_RAW.tar\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231101_ago3_set54.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231102_ago3_set59.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231103_armiKD_set63.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231104_armiKD2_set63.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231105_aubKD_set61.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231106_aubKD1_set60.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231107_aubKD2_set60.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231108_nos_set61.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231109_nos_set63.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231110_nos1_set60.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231111_nos2_set60.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231112_set_93_Y332F.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231113_set92_Y332F.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231114_top3b10-2_set92.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231115_top3b10-2_set93.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231116_top3b26_set40.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231117_top3b26_set47.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231118_top3b26_set53.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231119_top3bago3_set54.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231120_top3bago3_set59.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231121_top3barmiKD1_set63.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231122_top3barmiKD2_set63.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231123_top3baubKD_set61.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231124_top3baubKD1_set60.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231125_top3baubKD2_set60.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231126_top3bnos_set60.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231127_top3bnos_set61.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231128_top3bnos_set63.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231129_top3bnos2_set60.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231130_top3bvas_set37.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231131_top3bvas_set40.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231132_top3bvas_set47.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231133_top3bzucKD_set60.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231134_top3bzucKD_set61.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231135_vas_set37.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231136_vas_set40.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231137_vas_set47.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231138_WT_set37.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231139_WT_set40.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231140_WT_set47.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231141_WT_set53.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231142_WT_set54.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231143_WT_set59.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231144_WT_set92.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231145_WT_set93.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231146_zucKD_set60.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231147_zucKD_set61.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231148_fly_sRNA_aubKD.bedgraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231149_fly_sRNA_control.bedgraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231150_fly_sRNA_Top3b_aubKD.bedgraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231151_fly_sRNA_Top3b.bedgraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231152_SRNA61D_NOAGAL4-minus.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231152_SRNA61D_NOAGAL4-plus.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231153_SRNA61D_NOSAUBKD1-minus.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231153_SRNA61D_NOSAUBKD1-plus.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231154_SRNA61D_NOSAUBKD2-minus.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231154_SRNA61D_NOSAUBKD2-plus.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231155_SRNA61D_TOP3B_NOSAUBKD1-minus.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231155_SRNA61D_TOP3B_NOSAUBKD1-plus.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231156_SRNA61D_TOP3B_NOSAUBKD2-minus.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231156_SRNA61D_TOP3B_NOSAUBKD2-plus.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231157_SRNA61D_TOP3B_NOSGAL4-minus.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE230675\\GSM7231157_SRNA61D_TOP3B_NOSGAL4-plus.bedGraph\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE233305\\GSE233305_RAW.tar\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421876_DSP-1001660004790-E-A04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421877_DSP-1001660014732-E-A05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421878_DSP-1001660014734-G-A06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421879_DSP-1001660014732-E-B02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421880_DSP-1001660004956-F-A06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421881_DSP-1001660004956-F-A09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421882_DSP-1001660014732-E-B07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421883_DSP-1001660004790-E-A08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421884_DSP-1001660004790-E-A11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421885_DSP-1001660004956-F-B02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421886_DSP-1001660014732-E-B10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421887_DSP-1001660004790-E-B02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421888_DSP-1001660004956-F-B05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421889_DSP-1001660014732-E-C01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421890_DSP-1001660004790-E-B04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421891_DSP-1001660004956-F-B07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421892_DSP-1001660014732-E-C06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421893_DSP-1001660014734-G-C03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421894_DSP-1001660004790-E-B08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421895_DSP-1001660004956-F-B11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421896_DSP-1001660014732-E-C09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421897_DSP-1001660004790-E-B11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421898_DSP-1001660014732-E-D02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421899_DSP-1001660014732-E-D05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421900_DSP-1001660004790-E-C01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421901_DSP-1001660004790-E-C04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421902_DSP-1001660004956-F-C06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421903_DSP-1001660014732-E-D08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421904_DSP-1001660014732-E-D11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421905_DSP-1001660014732-E-E02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421906_DSP-1001660014734-G-D06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421907_DSP-1001660004790-E-C10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421908_DSP-1001660004956-F-C12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421909_DSP-1001660014732-E-E04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421910_DSP-1001660004956-F-D03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421911_DSP-1001660014732-E-E07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421912_DSP-1001660014734-G-D10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421913_DSP-1001660004790-E-D05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421914_DSP-1001660004790-E-D08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421915_DSP-1001660004790-E-D10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421916_DSP-1001660004956-F-D11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421917_DSP-1001660014732-E-F02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421918_DSP-1001660014734-G-E07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421919_DSP-1001660004790-E-E03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421920_DSP-1001660004790-E-E06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421921_DSP-1001660014734-G-E12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421922_DSP-1001660014732-E-F09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421923_DSP-1001660014734-G-F03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421924_DSP-1001660004790-E-E11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421925_DSP-1001660004956-F-E08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421926_DSP-1001660014732-E-F11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421927_DSP-1001660014734-G-F05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421928_DSP-1001660004790-E-F02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421929_DSP-1001660014732-E-G02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421930_DSP-1001660004790-E-F05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421931_DSP-1001660004956-F-F01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421932_DSP-1001660004956-F-F04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421933_DSP-1001660004956-F-F06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421934_DSP-1001660014732-E-G09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421935_DSP-1001660004956-F-F10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421936_DSP-1001660014732-E-G12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421937_DSP-1001660004956-F-G01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421938_DSP-1001660014734-G-G06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421939_DSP-1001660004790-E-G03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421940_DSP-1001660014732-E-H07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421941_DSP-1001660014734-G-G09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421942_DSP-1001660004790-E-G06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421943_DSP-1001660004790-E-G09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421944_DSP-1001660014734-G-G12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421945_DSP-1001660014734-G-H03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421946_DSP-1001660014733-F-A05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421947_DSP-1001660014734-G-H10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421948_DSP-1001660004790-E-H04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421949_DSP-1001660014733-F-A09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421950_DSP-1001660014733-F-A12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421951_DSP-1001660014735-H-A06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421952_DSP-1001660014733-F-B02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421953_DSP-1001660004790-E-H09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421954_DSP-1001660014733-F-B05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421955_DSP-1001660014733-F-B08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421956_DSP-1001660004789-B-A03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421957_DSP-1001660014733-F-B11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421958_DSP-1001660014735-H-B01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421959_DSP-1001660014733-F-C02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421960_DSP-1001660004789-B-A06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421961_DSP-1001660014733-F-C05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421962_DSP-1001660014733-F-C08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421963_DSP-1001660014733-F-C11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421964_DSP-1001660004789-B-A11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421965_DSP-1001660014733-F-D02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421966_DSP-1001660004789-B-B01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421967_DSP-1001660004789-B-B07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421968_DSP-1001660004789-B-B11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421969_DSP-1001660004789-B-C01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421970_DSP-1001660004956-F-G12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421971_DSP-1001660014734-G-A02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421972_DSP-1001660004956-F-A03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421973_DSP-1001660004790-E-A03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421974_DSP-1001660014734-G-A04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421975_DSP-1001660014732-E-A03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421976_DSP-1001660014734-G-A07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421977_DSP-1001660014732-E-A06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421978_DSP-1001660014734-G-A09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421979_DSP-1001660014732-E-A08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421980_DSP-1001660014734-G-A11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421981_DSP-1001660014732-E-A10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421982_DSP-1001660014734-G-B01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421983_DSP-1001660014732-E-A12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421984_DSP-1001660004956-F-A05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421985_DSP-1001660014734-G-B03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421986_DSP-1001660004956-F-A08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421987_DSP-1001660014732-E-B03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421988_DSP-1001660004790-E-A07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421989_DSP-1001660014734-G-B05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421990_DSP-1001660004956-F-A11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421991_DSP-1001660014732-E-B05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421992_DSP-1001660004790-E-A10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421993_DSP-1001660014734-G-B07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421994_DSP-1001660004956-F-B01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421995_DSP-1001660014732-E-B08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421996_DSP-1001660004790-E-B01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421997_DSP-1001660014734-G-B09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421998_DSP-1001660004956-F-B04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7421999_DSP-1001660014732-E-B11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422000_DSP-1001660014734-G-B11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422001_DSP-1001660014732-E-C02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422002_DSP-1001660014734-G-C01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422003_DSP-1001660014732-E-C04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422004_DSP-1001660014734-G-C04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422005_DSP-1001660004790-E-B07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422006_DSP-1001660004956-F-B10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422007_DSP-1001660014732-E-C07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422008_DSP-1001660014734-G-C06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422009_DSP-1001660004790-E-B10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422010_DSP-1001660004956-F-C01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422011_DSP-1001660014732-E-C10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422012_DSP-1001660014734-G-C08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422013_DSP-1001660014732-E-C12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422014_DSP-1001660014734-G-C10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422015_DSP-1001660014732-E-D03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422016_DSP-1001660004956-F-C03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422017_DSP-1001660014734-G-C12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422018_DSP-1001660004790-E-C03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422019_DSP-1001660004956-F-C05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422020_DSP-1001660014732-E-D06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422021_DSP-1001660014734-G-D02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422022_DSP-1001660004956-F-C08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422023_DSP-1001660014732-E-D09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422024_DSP-1001660014734-G-D04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422025_DSP-1001660004790-E-C08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422026_DSP-1001660014732-E-D12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422027_DSP-1001660014734-G-D07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422028_DSP-1001660004790-E-C09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422029_DSP-1001660004956-F-C11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422030_DSP-1001660014732-E-E03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422031_DSP-1001660004790-E-C12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422032_DSP-1001660004956-F-D02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422033_DSP-1001660004790-E-D02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422034_DSP-1001660004956-F-D05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422035_DSP-1001660014734-G-D08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422036_DSP-1001660014732-E-E05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422037_DSP-1001660004790-E-D04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422038_DSP-1001660004956-F-D07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422039_DSP-1001660014732-E-E08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422040_DSP-1001660014734-G-D12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422041_DSP-1001660004790-E-D07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422042_DSP-1001660004956-F-D10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422043_DSP-1001660014732-E-E10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422044_DSP-1001660004790-E-D12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422045_DSP-1001660004956-F-E01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422046_DSP-1001660014734-G-E03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422047_DSP-1001660014732-E-E12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422048_DSP-1001660014734-G-E05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422049_DSP-1001660014734-G-E08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422050_DSP-1001660004790-E-E02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422051_DSP-1001660014732-E-F03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422052_DSP-1001660014734-G-E10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422053_DSP-1001660004790-E-E05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422054_DSP-1001660004956-F-E04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422055_DSP-1001660014732-E-F05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422056_DSP-1001660014734-G-F01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422057_DSP-1001660004790-E-E08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422058_DSP-1001660014732-E-F07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422059_DSP-1001660004790-E-E10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422060_DSP-1001660004956-F-E07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422061_DSP-1001660014734-G-F06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422062_DSP-1001660004790-E-F01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422063_DSP-1001660004956-F-E10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422064_DSP-1001660014732-E-F12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422065_DSP-1001660014734-G-F08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422066_DSP-1001660004790-E-F04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422067_DSP-1001660004956-F-E12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422068_DSP-1001660014732-E-G03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422069_DSP-1001660014734-G-F10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422070_DSP-1001660004956-F-F03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422071_DSP-1001660014732-E-G05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422072_DSP-1001660014734-G-F12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422073_DSP-1001660014732-E-G07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422074_DSP-1001660004790-E-F09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422075_DSP-1001660004956-F-F09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422076_DSP-1001660014732-E-G10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422077_DSP-1001660004956-F-F12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422078_DSP-1001660014734-G-G02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422079_DSP-1001660014732-E-H01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422080_DSP-1001660014734-G-G04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422081_DSP-1001660014732-E-H03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422082_DSP-1001660004790-E-F12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422083_DSP-1001660014732-E-H05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422084_DSP-1001660014734-G-G07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422085_DSP-1001660004790-E-G02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422086_DSP-1001660004956-F-G04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422087_DSP-1001660004790-E-G05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422088_DSP-1001660014732-E-H08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422089_DSP-1001660014734-G-G10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422090_DSP-1001660004790-E-G08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422091_DSP-1001660004956-F-G07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422092_DSP-1001660014734-G-H01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422093_DSP-1001660004790-E-G11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422094_DSP-1001660014732-E-H10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422095_DSP-1001660014734-G-H04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422096_DSP-1001660004790-E-H01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422097_DSP-1001660014732-E-H12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422098_DSP-1001660014734-G-H06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422099_DSP-1001660014733-F-A03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422100_DSP-1001660014734-G-H08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422101_DSP-1001660014733-F-A06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422102_DSP-1001660014734-G-H11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422103_DSP-1001660004790-E-H03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422104_DSP-1001660014733-F-A08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422105_DSP-1001660014735-H-A02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422106_DSP-1001660004790-E-H06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422107_DSP-1001660014733-F-A10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422108_DSP-1001660014735-H-A04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422109_DSP-1001660014733-F-B01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422110_DSP-1001660014733-F-B03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422111_DSP-1001660014735-H-A07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422112_DSP-1001660004790-E-H08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422113_DSP-1001660014733-F-B06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422114_DSP-1001660014735-H-A09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422115_DSP-1001660004790-E-H11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422116_DSP-1001660004789-B-A02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422117_DSP-1001660014733-F-B09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422118_DSP-1001660014735-H-A11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422119_DSP-1001660014733-F-B12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422120_DSP-1001660014735-H-B02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422121_DSP-1001660014733-F-C03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422122_DSP-1001660014735-H-B04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422123_DSP-1001660004789-B-A05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422124_DSP-1001660014733-F-C06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422125_DSP-1001660004789-B-A07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422126_DSP-1001660014733-F-C09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422127_DSP-1001660014733-F-C12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422128_DSP-1001660014735-H-B06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422129_DSP-1001660004789-B-A10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422130_DSP-1001660004789-B-A12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422131_DSP-1001660004789-B-B04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422132_DSP-1001660004789-B-B06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422133_DSP-1001660004789-B-B10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422134_DSP-1001660004956-F-G10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422135_DSP-1001660004790-E-A01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422136_DSP-1001660004956-F-A01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422137_DSP-1001660004790-E-A02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422138_DSP-1001660014734-G-A03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422139_DSP-1001660004956-F-A02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422140_DSP-1001660014732-E-A02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422141_DSP-1001660014734-G-A05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422142_DSP-1001660014732-E-A04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422143_DSP-1001660014734-G-A08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422144_DSP-1001660014732-E-A07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422145_DSP-1001660014734-G-A10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422146_DSP-1001660014732-E-A09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422147_DSP-1001660014734-G-A12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422148_DSP-1001660014732-E-A11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422149_DSP-1001660014734-G-B02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422150_DSP-1001660014732-E-B01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422151_DSP-1001660004956-F-A04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422152_DSP-1001660004790-E-A05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422153_DSP-1001660014734-G-B04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422154_DSP-1001660004956-F-A07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422155_DSP-1001660014732-E-B04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422156_DSP-1001660004790-E-A06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422157_DSP-1001660014734-G-B06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422158_DSP-1001660004956-F-A10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422159_DSP-1001660014732-E-B06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422160_DSP-1001660004790-E-A09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422161_DSP-1001660014734-G-B08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422162_DSP-1001660004956-F-A12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422163_DSP-1001660014732-E-B09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422164_DSP-1001660014734-G-B10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422165_DSP-1001660004790-E-A12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422166_DSP-1001660004956-F-B03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422167_DSP-1001660014732-E-B12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422168_DSP-1001660014732-E-C03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422169_DSP-1001660014734-G-B12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422170_DSP-1001660004790-E-B03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422171_DSP-1001660004956-F-B06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422172_DSP-1001660014732-E-C05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422173_DSP-1001660014734-G-C02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422174_DSP-1001660004790-E-B05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422175_DSP-1001660004956-F-B08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422176_DSP-1001660014732-E-C08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422177_DSP-1001660014734-G-C05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422178_DSP-1001660004790-E-B06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422179_DSP-1001660004956-F-B09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422180_DSP-1001660014732-E-C11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422181_DSP-1001660014734-G-C07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422182_DSP-1001660004790-E-B09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422183_DSP-1001660004956-F-B12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422184_DSP-1001660014732-E-D01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422185_DSP-1001660014734-G-C09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422186_DSP-1001660014732-E-D04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422187_DSP-1001660014734-G-C11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422188_DSP-1001660004790-E-B12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422189_DSP-1001660004956-F-C02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422190_DSP-1001660014732-E-D07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422191_DSP-1001660014734-G-D01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422192_DSP-1001660004790-E-C02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422193_DSP-1001660004956-F-C04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422194_DSP-1001660014732-E-D10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422195_DSP-1001660014734-G-D03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422196_DSP-1001660004790-E-C05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422197_DSP-1001660004956-F-C07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422198_DSP-1001660014732-E-E01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422199_DSP-1001660014734-G-D05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422200_DSP-1001660004790-E-C07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422201_DSP-1001660004956-F-C09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422202_DSP-1001660004956-F-C10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422203_DSP-1001660004790-E-C11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422204_DSP-1001660004956-F-D01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422205_DSP-1001660004790-E-D01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422206_DSP-1001660004956-F-D04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422207_DSP-1001660014732-E-E06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422208_DSP-1001660014734-G-D09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422209_DSP-1001660014732-E-E09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422210_DSP-1001660014734-G-D11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422211_DSP-1001660004790-E-D03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422212_DSP-1001660004956-F-D06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422213_DSP-1001660014734-G-E01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422214_DSP-1001660004790-E-D06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422215_DSP-1001660004956-F-D08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422216_DSP-1001660014732-E-E11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422217_DSP-1001660014734-G-E02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422218_DSP-1001660004790-E-D09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422219_DSP-1001660004956-F-D09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422220_DSP-1001660004790-E-D11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422221_DSP-1001660004956-F-D12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422222_DSP-1001660014732-E-F01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422223_DSP-1001660014734-G-E04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422224_DSP-1001660014734-G-E06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422225_DSP-1001660014732-E-F04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422226_DSP-1001660014734-G-E09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422227_DSP-1001660004790-E-E01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422228_DSP-1001660004956-F-E02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422229_DSP-1001660014732-E-F06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422230_DSP-1001660014734-G-E11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422231_DSP-1001660004790-E-E04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422232_DSP-1001660004956-F-E03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422233_DSP-1001660014732-E-F08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422234_DSP-1001660014734-G-F02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422235_DSP-1001660004790-E-E07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422236_DSP-1001660004956-F-E05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422237_DSP-1001660014732-E-F10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422238_DSP-1001660014734-G-F04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422239_DSP-1001660004790-E-E09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422240_DSP-1001660004956-F-E06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422241_DSP-1001660014732-E-G01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422242_DSP-1001660014734-G-F07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422243_DSP-1001660004790-E-E12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422244_DSP-1001660004956-F-E09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422245_DSP-1001660014732-E-G04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422246_DSP-1001660014734-G-F09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422247_DSP-1001660004790-E-F03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422248_DSP-1001660004956-F-E11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422249_DSP-1001660014732-E-G06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422250_DSP-1001660014734-G-F11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422251_DSP-1001660004790-E-F06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422252_DSP-1001660004956-F-F02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422253_DSP-1001660014732-E-G08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422254_DSP-1001660014734-G-G01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422255_DSP-1001660004790-E-F07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422256_DSP-1001660004956-F-F05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422257_DSP-1001660004790-E-F08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422258_DSP-1001660004956-F-F07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422259_DSP-1001660014732-E-G11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422260_DSP-1001660004956-F-F08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422261_DSP-1001660004790-E-F10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422262_DSP-1001660004956-F-F11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422263_DSP-1001660014732-E-H02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422264_DSP-1001660014734-G-G03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422265_DSP-1001660014732-E-H04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422266_DSP-1001660014734-G-G05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422267_DSP-1001660004790-E-F11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422268_DSP-1001660004956-F-G02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422269_DSP-1001660014732-E-H06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422270_DSP-1001660014734-G-G08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422271_DSP-1001660004790-E-G01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422272_DSP-1001660004956-F-G03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422273_DSP-1001660004790-E-G04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422274_DSP-1001660004956-F-G05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422275_DSP-1001660014732-E-H09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422276_DSP-1001660014734-G-G11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422277_DSP-1001660004790-E-G07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422278_DSP-1001660004956-F-G06.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422279_DSP-1001660014734-G-H02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422280_DSP-1001660004790-E-G10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422281_DSP-1001660014732-E-H11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422282_DSP-1001660014734-G-H05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422283_DSP-1001660004790-E-G12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422284_DSP-1001660004956-F-G08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422285_DSP-1001660014733-F-A02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422286_DSP-1001660014734-G-H07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422287_DSP-1001660014733-F-A04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422288_DSP-1001660014734-G-H09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422289_DSP-1001660014733-F-A07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422290_DSP-1001660014734-G-H12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422291_DSP-1001660004790-E-H02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422292_DSP-1001660014735-H-A03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422293_DSP-1001660004790-E-H05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422294_DSP-1001660014733-F-A11.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422295_DSP-1001660014735-H-A05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422296_DSP-1001660014733-F-B04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422297_DSP-1001660014735-H-A08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422298_DSP-1001660004790-E-H07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422299_DSP-1001660014733-F-B07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422300_DSP-1001660014735-H-A10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422301_DSP-1001660004790-E-H10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422302_DSP-1001660014733-F-B10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422303_DSP-1001660014735-H-A12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422304_DSP-1001660004790-E-H12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422305_DSP-1001660014733-F-C01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422306_DSP-1001660014735-H-B03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422307_DSP-1001660004789-B-A04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422308_DSP-1001660014733-F-C04.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422309_DSP-1001660014735-H-B05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422310_DSP-1001660014733-F-C07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422311_DSP-1001660014733-F-C10.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422312_DSP-1001660004789-B-A08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422313_DSP-1001660014733-F-D01.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422314_DSP-1001660014735-H-B07.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422315_DSP-1001660004789-B-A09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422316_DSP-1001660004789-B-B02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422317_DSP-1001660004789-B-B03.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422318_DSP-1001660004789-B-B05.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422319_DSP-1001660004789-B-B08.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422320_DSP-1001660004789-B-B09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422321_DSP-1001660004956-F-G09.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422322_DSP-1001660004789-B-B12.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422323_DSP-1001660004789-B-C02.dcc\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE233305\\GSM7422324_DSP-1001660004956-F-G11.dcc\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE235472\\GSE235472_RAW.tar\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE235472\\GSM7502702_18T_1_peaks.bedgraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE235472\\GSM7502703_19T_1_peaks.bedgraph\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE236921\\GSE236921_RAW.tar\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585578_705-1_S1_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585578_705-1_S1_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585579_705-2_S12_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585579_705-2_S12_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585580_709-1_S2_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585580_709-1_S2_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585581_709-2_S13_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585581_709-2_S13_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585582_719-1_S3_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585582_719-1_S3_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585583_719-2_S14_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585583_719-2_S14_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585584_720-1_S4_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585584_720-1_S4_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585585_720-2_S15_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585585_720-2_S15_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585586_721-1_S5_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585586_721-1_S5_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585587_721-2_S16_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585587_721-2_S16_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585588_722-1_S6_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585588_722-1_S6_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585589_722-2_S17_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585589_722-2_S17_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585590_728-1_S7_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585590_728-1_S7_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585591_728-2_S18_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585591_728-2_S18_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585592_729-1_S8_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585592_729-1_S8_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585593_729-2_S19_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585593_729-2_S19_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585594_739-1_S9_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585594_739-1_S9_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585595_739-2_S20_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585595_739-2_S20_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585596_740-1_S10_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585596_740-1_S10_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585597_740-2_S21_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585597_740-2_S21_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585598_748-1_S11_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585598_748-1_S11_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585599_748-2_S22_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585599_748-2_S22_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585600_614_S7_L001_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585600_614_S7_L001_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585600_614_S7_L002_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585600_614_S7_L002_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585601_615_S23_L001_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585601_615_S23_L001_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585601_615_S23_L002_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585601_615_S23_L002_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585602_621_S4_L001_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585602_621_S4_L001_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585602_621_S4_L002_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585602_621_S4_L002_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585603_622_S9_L001_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585603_622_S9_L001_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585603_622_S9_L002_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585603_622_S9_L002_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585604_626_S8_L001_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585604_626_S8_L001_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585604_626_S8_L002_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585604_626_S8_L002_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585605_627_S6_L001_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585605_627_S6_L001_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585605_627_S6_L002_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585605_627_S6_L002_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585606_635_S18_L001_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585606_635_S18_L001_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585606_635_S18_L002_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585606_635_S18_L002_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585607_636_S19_L001_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585607_636_S19_L001_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585607_636_S19_L002_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585607_636_S19_L002_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585608_637_S22_L001_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585608_637_S22_L001_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585608_637_S22_L002_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585608_637_S22_L002_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585609_638_S1_L001_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585609_638_S1_L001_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585609_638_S1_L002_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585609_638_S1_L002_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585610_648_S16_L001_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585610_648_S16_L001_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585610_648_S16_L002_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585610_648_S16_L002_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585611_649_S13_L001_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585611_649_S13_L001_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585611_649_S13_L002_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585611_649_S13_L002_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585612_614_tx_S5_L001_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585612_614_tx_S5_L001_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585612_614_tx_S5_L002_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585612_614_tx_S5_L002_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585613_615_tx_S24_L001_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585613_615_tx_S24_L001_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585613_615_tx_S24_L002_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585613_615_tx_S24_L002_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585614_621_tx_S21_L001_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585614_621_tx_S21_L001_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585614_621_tx_S21_L002_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585614_621_tx_S21_L002_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585615_622_tx_S10_L001_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585615_622_tx_S10_L001_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585615_622_tx_S10_L002_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585615_622_tx_S10_L002_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585616_626_tx_S12_L001_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585616_626_tx_S12_L001_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585616_626_tx_S12_L002_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585616_626_tx_S12_L002_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585617_627_tx_S15_L001_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585617_627_tx_S15_L001_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585617_627_tx_S15_L002_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585617_627_tx_S15_L002_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585618_635_tx_S14_L001_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585618_635_tx_S14_L001_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585618_635_tx_S14_L002_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585618_635_tx_S14_L002_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585619_636_tx_S11_L001_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585619_636_tx_S11_L001_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585619_636_tx_S11_L002_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585619_636_tx_S11_L002_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585620_637_tx_S2_L001_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585620_637_tx_S2_L001_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585620_637_tx_S2_L002_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585620_637_tx_S2_L002_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585621_638_tx_S20_L001_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585621_638_tx_S20_L001_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585621_638_tx_S20_L002_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585621_638_tx_S20_L002_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585622_648_tx_S3_L001_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585622_648_tx_S3_L001_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585622_648_tx_S3_L002_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585622_648_tx_S3_L002_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585623_649_tx_S17_L001_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585623_649_tx_S17_L001_quant.sf\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE236921\\GSM7585623_649_tx_S17_L002_quant.sf.gz → rna_seq_analysis\\GSE236921\\GSM7585623_649_tx_S17_L002_quant.sf\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE245681\\GSE245681_RAW.tar\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846864_p50_0.0h_R1.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846865_p50_0.5h_R1.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846866_p50_2.0h_R1.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846867_p50_6.0h_R1.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846868_p50_0.0h_R2.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846869_p50_0.5h_R2.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846870_p50_2.0h_R2.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846871_p50_6.0h_R2.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846872_p50_Input_R2.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846873_IkBz_0.0h_R1.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846874_IkBz_0.5h_R1.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846875_IkBz_1.0h_R1.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846876_IkBz_2.0h_R1.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846877_IkBz_6.0h_R1.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846878_IkBz_Input_R1.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846879_IkBz_0.0h_R2.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846880_IkBz_0.5h_R2.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846881_IkBz_1.0h_R2.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846882_IkBz_2.0h_R2.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846883_IkBz_6.0h_R2.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846884_IkBz_Input_R2.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846885_IkBz_2.0h_R3.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846886_IkBz_Input_R3.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846887_IkBz_in_p50_KO_2.0h_R1.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846888_IkBz_in_p50_KO_2.0h_R2.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846889_IkBz_in_p50_KO_2.0h_R3.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846890_IkBz_in_p50_KO_Input_R1_R2.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846891_IkBz_in_p50_KO_Input_R3.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846892_RelA_in_p50_KO.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846893_Sequential_RelA_2.0h.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846894_Sequential_RelA_p50_2.0h.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846895_Sequential_RelA_IkBz_2.0h.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846896_Sequential_RelA_p53_2.0h.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE245681\\GSM7846897_Sequential_Input.bedGraph\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE246326\\GSM7866761_seurat_object.rds\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE246326\\GSM7866761_seurat_object_CD4.rds\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE246326\\GSM7866761_seurat_object_CD8.rds\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE246326\\GSM7866761_seurat_object_Mono.rds\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE246326\\unnamed_file\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE247343\\GSE247343_RAW.tar\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887458_WJL2019_106.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887459_WJL2019_107.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887460_WJL2019_108.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887461_WJL2019_109.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887462_WJL2019_110.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887463_WJL2019_111.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887464_WJL2019_112.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887465_WJL2019_113.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887466_WJL2019_114.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887467_WJL2019_115.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887468_WJL2019_116.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887469_WJL2019_117.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887470_WJL2019_118.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887471_WJL2019_119.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887472_WJL2019_120.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887473_WJL2019_121.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887474_WJL2019_122.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887475_WJL2019_123.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887476_WJL2019_124.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887477_WJL2019_125.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887478_WJL2019_126.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887479_WJL2019_127.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887480_WJL2019_128.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887481_WJL2019_129.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887482_WJL2019_130.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887483_WJL2019_131.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887484_WJL2019_132.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887485_WJL2019_133.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887486_WJL2019_134.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887487_WJL2019_135.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887488_WJL2019_136.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887489_WJL2019_137.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887490_WJL2019_138.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887491_WJL2019_139.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887492_WJL2019_140.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887493_WJL2019_141.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887494_WJL2019_142.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887495_WJL2019_143.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887496_WJL2019_144.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887497_WJL2019_145.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887498_WJL2019_146.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887499_WJL2019_147.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887500_WJL2019_148.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887501_WJL2019_149.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887502_WJL2019_150.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887503_WJL2019_151.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887504_WJL2019_152.mm10.rpkm\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE247343\\GSM7887505_WJL2019_153.mm10.rpkm\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE254317\\GSE254317_RAW.tar\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE254317\\GSM8037653_TP_H3K27me3_rep1_peaks.narrowPeak\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE254317\\GSM8037654_TP_H3K27me3_rep2_peaks.narrowPeak\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE254317\\GSM8037655_TPE_H3K27me3_rep1_peaks.narrowPeak\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE254317\\GSM8037656_TPE_H3K27me3_rep2_peaks.narrowPeak\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE255683\\GSM8077414_SAM24436411_NXG14799322.counts_gene_exonic.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE255683\\GSM8077415_SAM24436412_NXG14799313.counts_gene_exonic.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE255683\\GSM8077416_SAM24436413_NXG14799317.counts_gene_exonic.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE255683\\GSM8077417_SAM24436414_NXG14799314.counts_gene_exonic.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE255683\\GSM8077418_SAM24436415_NXG14799318.counts_gene_exonic.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE255683\\GSM8077419_SAM24436416_NXG14799325.counts_gene_exonic.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE255683\\GSM8077420_SAM24436417_NXG14799309.counts_gene_exonic.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE255683\\GSM8077421_SAM24436418_NXG14799326.counts_gene_exonic.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE255683\\GSM8077422_SAM24436419_NXG14799308.counts_gene_exonic.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE255683\\GSM8077423_SAM24436420_NXG14799311.counts_gene_exonic.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE255683\\GSM8077424_SAM24436421_NXG14799307.counts_gene_exonic.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE255683\\GSM8077425_SAM24436422_NXG14799324.counts_gene_exonic.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE255683\\GSM8077426_SAM24436423_NXG14799319.counts_gene_exonic.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE255683\\GSM8077427_SAM24436424_NXG14799320.counts_gene_exonic.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE255683\\GSM8077428_SAM24436425_NXG14799312.counts_gene_exonic.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE255683\\GSM8077429_SAM24436426_NXG14799323.counts_gene_exonic.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE255683\\GSM8077430_SAM24436427_NXG14799316.counts_gene_exonic.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE255683\\GSM8077431_SAM24436428_NXG14799321.counts_gene_exonic.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE255683\\GSM8077432_SAM24436429_NXG14799315.counts_gene_exonic.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE255683\\GSM8077433_SAM24436430_NXG14799310.counts_gene_exonic.tab\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE255683\\unnamed_file\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE256469\\GSE256469_All_peakNum.xls\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE259361\\GSE259361_RAW.tar\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE260623\\GSE260623_RAW.tar\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121157_2214230_R1_80S_Fraction_WT_S41_L002ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121158_2214231_R1_Di-trisome_Fraction_WT_S42_L002ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121159_2214232_R1_Polysome_Fraction_WT_S80_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121160_2214233_R2_Total_RNA_WT_S81_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121161_2214234_R2_80S_Fraction_WT_S82_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121162_2214235_R2_Di-trisome_Fraction_WT_S83_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121163_2214236_R2_Polysome_Fraction_WT_S84_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121164_2214237_R3_Total_RNA_WT_S85_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121165_2214238_R3_80S_Fraction_WT_S86_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121166_2214239_R3_Di-trisome_Fraction_WT_S87_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121167_2214240_R3_Polysome_Fraction_WT_S88_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121168_2214241_R4_Total_RNA_S89_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121169_2214242_R4_80S_Fraction_WT_S90_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121170_2214243_R4_Di-trisome_Fraction_WT_S91_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121171_2214244_R4_Polysome_Fraction_WT_S92_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121172_2214245_R5_Total_RNA_WT_S93_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121173_2214246_R5_80S_Fraction_WT_S94_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121174_2214247_R5_Di-trisome_Fraction_WT_S95_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121175_2214248_R5_Polysome_Fraction_WT_S96_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121176_2214249_R1_Total_RNA_MUT_S97_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121177_2214250_R1_80S_Fraction_MUT_S98_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121178_2214251_R1_Di-trisome_Fraction_MUT_S99_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121179_2214252_R1_Polysome_Fraction_MUT_S100_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121180_2214254_R2_80S_Fraction_MUT_S102_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121181_2214255_R2_Di-trisome_Fraction_MUT_S103_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121182_2214256_R2_Polysome_Fraction_MUT_S104_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121183_2214257_R3_Total_RNA_MUT_S105_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121184_2214258_R3_80S_Fraction_MUT_S106_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121185_2214259_R3_Di-trisome_Fraction_MUT_S107_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121186_2214260_R3_Polysome_Fraction_MUT_S108_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121187_2214261_R4_Total_RNA_S109_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121188_2214262_R4_80S_Fraction_MUT_S110_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121189_2214263_R4_Di-trisome_Fraction_MUT_S111_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121190_2214264_R4_Polysome_Fraction_MUT_S112_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121191_2214265_R5_Total_RNA_MUT_S113_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121192_2214266_R5_80S_Fraction_MUT_S114_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121193_2214267_R5_Di-trisome_Fraction_MUT_S115_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121194_2214268_R5_Polysome_Fraction_MUT_S116_L003ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121195_2580496_JH35M_2207_S9ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121196_2580497_JH35M_2214_S10ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121197_2580498_JH35M_2217_S11ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121198_2580500_JH35M_2354_S13ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121199_2580501_JH35W_2227_S14ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121200_2580502_JH35W_2228_S15ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121201_2580503_JH35W_2229_S16ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121202_2580504_JH35W_2236_S17ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121203_2580505_JH35W_2467_S18ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121204_2604091_JH35_2234_S23ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121205_2604092_JH35_2235_S24ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121206_2604093_JH35_2241_S25ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121207_2604094_JH35_2245_S26ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121208_2604095_JH35_2464_S27ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121209_2604096_JH35_2466_S28ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121210_2604097_JH35_2470_S29ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121211_2604098_JH35_2361_S30ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121212_2604099_JH35_2365_S31ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121213_2604100_JH35_2453_S32ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121214_2604101_JH35_2456_S33ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121215_2604102_JH35_2209_S34ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121216_2604103_JH35_2217_S35ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121217_2626989_JH35_2461_S39ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121218_2626990_JH35_2465_S42ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121219_2626991_JH35_2452_S44ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121220_2626992_JH35_2237_S46ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121221_2626993_JH35_2243_S48ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121222_2626994_JH35_2351_S33ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121223_2626995_JH35_2458_S35ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121224_2657484_JH35_mut_2475_S60ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121225_1940132_JGL_4782_S107ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121226_1940133_JGL_4783_S108ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121227_1940134_JGL_4785_S109ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121228_1940135_JGL_4936_S110ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121229_1940136_JGL_4937_S111ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121230_1940137_JGL_4938_S112ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121231_1940138_JGL_4786_S113ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121232_1940139_JGL_4787_S114ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121233_1940140_JGL_4788_S115ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121234_1940141_JGL_4789_S116ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121235_1980254_4781_S20ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121236_1980255_4784_S21ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121237_1980256_7760_S22ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121238_1980257_7757_S23ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121239_1980259_7758_S25ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121240_1980260_7756_S26ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121241_1980261_7247_S27ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121242_1980262_7248_S28ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121243_1980263_7246_S29ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121244_1980264_4790_S30ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121245_1980265_7763_S31ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121246_1980266_7765_S32ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE260623\\GSM8121247_1980267_7762_S33ReadsPerGene.out.tab\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE261316\\GSE261316_RAW.tar\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE261316\\GSM8140423_JY-101.ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE261316\\GSM8140424_JY-103.ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE261316\\GSM8140425_ADDED-JY-113.ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE261316\\GSM8140426_JY-105.ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE261316\\GSM8140427_JY-106.ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE261316\\GSM8140428_ADDED-JY-108.ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE261316\\GSM8140429_JY-110.ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE261316\\GSM8140430_JY-111.ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE261316\\GSM8140431_ADDED-JY-112-1.ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE261316\\GSM8140432_JY-116.ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE261316\\GSM8140433_JY-117.ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE261316\\GSM8140434_JY-118.ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE261316\\GSM8140435_JY-120.ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE261316\\GSM8140436_JY-121.ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE261316\\GSM8140437_JY-122.ReadsPerGene.out.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE266919\\GSE266919_Bcell.rds\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE267856\\GSE267856_readcount.xls\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE269147\\GSE269147_RAW.tar\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE269147\\GSM8307004_shJmjd1a_RNAseq_010_43_shEmpty_Day2_Rep1_HISAT2_featureCounts_Counts.txt.gz → rna_seq_analysis\\GSE269147\\GSM8307004_shJmjd1a_RNAseq_010_43_shEmpty_Day2_Rep1_HISAT2_featureCounts_Counts.txt\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE269147\\GSM8307005_shJmjd1a_RNAseq_011_44_shEmpty_Day2_Rep2_HISAT2_featureCounts_Counts.txt.gz → rna_seq_analysis\\GSE269147\\GSM8307005_shJmjd1a_RNAseq_011_44_shEmpty_Day2_Rep2_HISAT2_featureCounts_Counts.txt\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE269147\\GSM8307006_shJmjd1a_RNAseq_012_45_shEmpty_Day2_Rep3_HISAT2_featureCounts_Counts.txt.gz → rna_seq_analysis\\GSE269147\\GSM8307006_shJmjd1a_RNAseq_012_45_shEmpty_Day2_Rep3_HISAT2_featureCounts_Counts.txt\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE269147\\GSM8307007_shJmjd1a_RNAseq_013_46_shJmjd1a_Day2_Rep1_HISAT2_featureCounts_Counts.txt.gz → rna_seq_analysis\\GSE269147\\GSM8307007_shJmjd1a_RNAseq_013_46_shJmjd1a_Day2_Rep1_HISAT2_featureCounts_Counts.txt\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE269147\\GSM8307008_shJmjd1a_RNAseq_014_47_shJmjd1a_Day2_Rep2_HISAT2_featureCounts_Counts.txt.gz → rna_seq_analysis\\GSE269147\\GSM8307008_shJmjd1a_RNAseq_014_47_shJmjd1a_Day2_Rep2_HISAT2_featureCounts_Counts.txt\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE269147\\GSM8307009_shJmjd1a_RNAseq_015_48_shJmjd1a_Day2_Rep3_HISAT2_featureCounts_Counts.txt.gz → rna_seq_analysis\\GSE269147\\GSM8307009_shJmjd1a_RNAseq_015_48_shJmjd1a_Day2_Rep3_HISAT2_featureCounts_Counts.txt\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE269147\\GSM8307010_shJmjd1a_RNAseq_001_10_shJmjd1aLacZ_Day2_Rep1_HISAT2_featureCounts_Counts.txt.gz → rna_seq_analysis\\GSE269147\\GSM8307010_shJmjd1a_RNAseq_001_10_shJmjd1aLacZ_Day2_Rep1_HISAT2_featureCounts_Counts.txt\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE269147\\GSM8307011_shJmjd1a_RNAseq_002_11_shJmjd1aLacZ_Day2_Rep2_HISAT2_featureCounts_Counts.txt.gz → rna_seq_analysis\\GSE269147\\GSM8307011_shJmjd1a_RNAseq_002_11_shJmjd1aLacZ_Day2_Rep2_HISAT2_featureCounts_Counts.txt\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE269147\\GSM8307012_shJmjd1a_RNAseq_003_12_shJmjd1aLacZ_Day2_Rep3_HISAT2_featureCounts_Counts.txt.gz → rna_seq_analysis\\GSE269147\\GSM8307012_shJmjd1a_RNAseq_003_12_shJmjd1aLacZ_Day2_Rep3_HISAT2_featureCounts_Counts.txt\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE269147\\GSM8307013_shJmjd1a_RNAseq_007_16_shJmjd1aMut_Day2_Rep1_HISAT2_featureCounts_Counts.txt.gz → rna_seq_analysis\\GSE269147\\GSM8307013_shJmjd1a_RNAseq_007_16_shJmjd1aMut_Day2_Rep1_HISAT2_featureCounts_Counts.txt\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE269147\\GSM8307014_shJmjd1a_RNAseq_008_17_shJmjd1aMut_Day2_Rep2_HISAT2_featureCounts_Counts.txt.gz → rna_seq_analysis\\GSE269147\\GSM8307014_shJmjd1a_RNAseq_008_17_shJmjd1aMut_Day2_Rep2_HISAT2_featureCounts_Counts.txt\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE269147\\GSM8307015_shJmjd1a_RNAseq_009_18_shJmjd1aMut_Day2_Rep3_HISAT2_featureCounts_Counts.txt.gz → rna_seq_analysis\\GSE269147\\GSM8307015_shJmjd1a_RNAseq_009_18_shJmjd1aMut_Day2_Rep3_HISAT2_featureCounts_Counts.txt\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE269147\\GSM8307016_shJmjd1a_RNAseq_004_13_shJmjd1aWT_Day2_Rep1_HISAT2_featureCounts_Counts.txt.gz → rna_seq_analysis\\GSE269147\\GSM8307016_shJmjd1a_RNAseq_004_13_shJmjd1aWT_Day2_Rep1_HISAT2_featureCounts_Counts.txt\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE269147\\GSM8307017_shJmjd1a_RNAseq_005_14_shJmjd1aWT_Day2_Rep2_HISAT2_featureCounts_Counts.txt.gz → rna_seq_analysis\\GSE269147\\GSM8307017_shJmjd1a_RNAseq_005_14_shJmjd1aWT_Day2_Rep2_HISAT2_featureCounts_Counts.txt\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE269147\\GSM8307018_shJmjd1a_RNAseq_006_15_shJmjd1aWT_Day2_Rep3_HISAT2_featureCounts_Counts.txt.gz → rna_seq_analysis\\GSE269147\\GSM8307018_shJmjd1a_RNAseq_006_15_shJmjd1aWT_Day2_Rep3_HISAT2_featureCounts_Counts.txt\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE270461\\GSM8343421_Donor25P0_filtered_feature_bc_matrix.tar.gz\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE270461\\GSM8343421_Donor25P0_raw_feature_bc_matrix.tar.gz\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE270461\\GSM8343422_Donor49P0_filtered_feature_bc_matrix.tar.gz\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE270461\\GSM8343422_Donor49P0_raw_feature_bc_matrix.tar.gz\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE270461\\GSM8343423_Donor25Control_filtered_feature_bc_matrix.tar.gz\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE270461\\GSM8343423_Donor25Control_raw_feature_bc_matrix.tar.gz\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE270461\\GSM8343424_Donor25Stimulated_filtered_feature_bc_matrix.tar.gz\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE270461\\GSM8343424_Donor25Stimulated_raw_feature_bc_matrix.tar.gz\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE270461\\unnamed_file\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE272915\\GSE272915_RAW.tar\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE272915\\GSM8414614_AIM2_1_quant.sf\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE272915\\GSM8414615_AIM2_2_quant.sf\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE272915\\GSM8414616_AIM2_3_quant.sf\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE272915\\GSM8414617_WT1_quant.sf\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE272915\\GSM8414618_WT3_quant.sf\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE272915\\GSM8414619_WT4_quant.sf\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE272915\\GSM8414620_WT5_quant.sf\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE275000\\GSE275000_RAW.tar\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464115_1_S1_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464116_2_S2_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464117_3_S3_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464118_4_S4_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464119_5_S5_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464120_6_S6_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464121_7_S7_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464122_8_S8_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464123_9_S9_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464124_10_S10_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464125_11_S11_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464126_12_S12_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464127_13_S13_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464128_14_S14_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464129_15_S1_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464130_16_S2_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464131_17_S3_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464132_18_S4_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464133_19_S5_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464134_20_S6_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464135_21_S7_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464136_22_S8_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464137_23_S15_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464138_24_S16_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464139_25_S9_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464140_26_S10_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464141_27_S11_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464142_28_S12_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464143_29_S13_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464144_30_S14_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464145_31_S15_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464146_32_S16_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464147_33_S1_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464148_34_S2_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464149_35_S3_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464150_36_S4_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464151_37_S5_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464152_38_S6_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464153_39_S7_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464154_40_S8_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464155_41_S9_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464156_42_S10_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464157_43_S11_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464158_44_S12_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464159_45_S13_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464160_46_S14_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464161_47_S15_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464162_48_S16_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464163_49_S1_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464164_50_S2_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464165_51_S3_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE275000\\GSM8464166_52_S4_R1_trimmed.fq_trimmed_bismark_hisat2.bismark.cov\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE278137\\GSE278137_RAW.tar\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540679_D1_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540680_D2_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540681_D3_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540682_D4_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540683_D5_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540684_D6_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540685_D7_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540686_KO1_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540687_KO2_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540688_KO3_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540689_KO4_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540690_KO5_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540691_KO6_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540692_KO7_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540693_N1_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540694_N2_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540695_N3_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540696_N4_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540697_N5_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540698_N6_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540699_N7_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540700_R1_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540701_R2_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540702_R3_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540703_R4_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540704_R5_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540705_R6_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE278137\\GSM8540706_R7_.genes.results\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE279800\\GSM8580453_sw480_CAF_rep1_S19_L001_R1_001.counts\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE279800\\GSM8580454_sw480_CAF_rep2_S21_L001_R1_001.counts\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE279800\\GSM8580455_sw480_CN_rep1_S18_L001_R1_001.counts\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE279800\\GSM8580456_sw480_CN_rep2_S20_L001_R1_001.counts\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE279800\\GSM8580457_sw620_CAF_rep1_S23_L001_R1_001.counts\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE279800\\GSM8580458_sw620_CAF_rep2_S25_L001_R1_001.counts\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE279800\\GSM8580459_sw620_CN_rep1_S22_L001_R1_001.counts\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE279800\\GSM8580460_sw620_CN_rep2_S24_L001_R1_001.counts\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE279800\\unnamed_file\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE281885\\GSM8631770_JY19.STARReadsPerGene.out.tab.deseq.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE281885\\GSM8631771_JY20.STARReadsPerGene.out.tab.deseq.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE281885\\GSM8631772_JY21.STARReadsPerGene.out.tab.deseq.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE281885\\GSM8631773_JY22.STARReadsPerGene.out.tab.deseq.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE281885\\GSM8631774_JY23.STARReadsPerGene.out.tab.deseq.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE281885\\GSM8631775_JY24.STARReadsPerGene.out.tab.deseq.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE281885\\GSM8631776_JY25.STARReadsPerGene.out.tab.deseq.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE281885\\GSM8631777_JY26.STARReadsPerGene.out.tab.deseq.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE281885\\GSM8631778_JY27.STARReadsPerGene.out.tab.deseq.tab\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE281885\\GSM8631779_JY28.STARReadsPerGene.out.tab.deseq.tab\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE281885\\unnamed_file\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE282987\\GSE282987_readcount.xls\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE283392\\GSE283392_RAW.tar\n",
      "❌ Failed to extract TAR GSE283392_RAW.tar: unexpected end of data\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE283392\\GSM8661316_WJL2015_177.mm10.preA.CD8.Cont.STAT5B.RP.nr.bed.wig\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE283392\\GSM8661317_WJL2015_178.mm10.preA.CD8.IL2.STAT5B.RP.nr.bed.wig\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE283392\\GSM8661318_WJL2015_179.mm10.iTreg.Cont.STAT5B.RP.nr.bed.wig\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE283392\\GSM8661319_WJL2015_180.mm10.iTreg.IL2.STAT5B.RP.nr.bed.wig\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE283392\\GSM8661320_WJL2016_281.mm10.IgG.sorted.bed.wig\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE283392\\GSM8661321_WJL2016_282.mm10.WT.IgE.STAT5B.sorted.bed.wig\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE283392\\GSM8661322_WJL2019_348.mm10.NK.Cont.nr.bed.wig\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE283392\\GSM8661323_WJL2019_349.mm10.NK.IL15.nr.bed.wig\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE283392\\GSM8661324_WJL2019_365.mm10.DN2.Cont.STAT5B.nr.bed.wig\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE283392\\GSM8661325_WJL2019_366.mm10.DN2.Cont.H3K27ac.nr.bed.wig\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE283392\\GSM8661326_WJL2019_367.mm10.DN2.IL7.2h.STAT5B.nr.bed.wig\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE283392\\GSM8661327_WJL2019_368.mm10.DN2.IL7.2h.H3K27ac.nr.bed.wig\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE283392\\GSM8661328_WJL2020_190.mm10.WT.CD3+IL2.24h.H3K27ac.sorted.bam.wig\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE283392\\GSM8661329_WJL2020_194.mm10.6032.CD3+IL2.24h.H3K27ac.sorted.bam.wig\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE283392\\GSM8661330_WJL2020_198.mm10.UP16.CD3+IL2.24h.H3K27ac.sorted.bam.wig\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE286458\\GSE286458_RAW.tar\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE286534\\GSE286534_RAW.tar\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE286534\\GSM8729003_OV1_peak-gene.xls\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE286534\\GSM8729004_OV2_peak-gene.xls\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE286534\\GSM8729005_OV3_peak-gene.xls\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE286534\\GSM8729006_NV1_peak-gene.xls\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE286534\\GSM8729007_NV2_peak-gene.xls\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE286534\\GSM8729008_NV3_peak-gene.xls\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE287383\\GSE287383_DMSO.16k.bal.cool\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE290839\\GSE290839_MergedDataset26Feb2025.loom\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE292962\\GSE292962_APP_qdefault_consensus.bed\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE293371\\GSE293371_OSCC_organoid_maftools.maf\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE294152\\GSE294152_RAW.tar\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE294153\\GSE294153_RAW.tar\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE296557\\GSE296557_RAW.tar\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE296557\\GSM8972609_Ctrl1-Counts.tabular\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE296557\\GSM8972610_Ctrl2-counts.tabular\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE296557\\GSM8972611_Ctrl3-counts.tabular\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE296557\\GSM8972612_VGD1_1-counts.tabular\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE296557\\GSM8972613_VGD1_2-counts.tabular\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE296557\\GSM8972614_VGD1_3-counts.tabular\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE83317\\GSE83317_RAW.tar\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE83317\\GSM2199455_JS.actin.bed\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE83317\\GSM2199456_JS.mycn-low.bed\n",
      "⚠️ Skipping (already exists): rna_seq_analysis\\GSE83317\\GSM2199457_JS.mycn-high.bed\n",
      "📦 Extracting TAR: rna_seq_analysis\\GSE96686\\GSE96686_RAW.tar\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE96686\\GSM2536316_Kdm4a_WT1.wig.gz → rna_seq_analysis\\GSE96686\\GSM2536316_Kdm4a_WT1.wig\n",
      "❌ Failed to decompress GZ GSM2536316_Kdm4a_WT1.wig.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE96686\\GSM2536317_Kdm4a_WT2.wig.gz → rna_seq_analysis\\GSE96686\\GSM2536317_Kdm4a_WT2.wig\n",
      "❌ Failed to decompress GZ GSM2536317_Kdm4a_WT2.wig.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE96686\\GSM2536318_Kdm4a_KO.wig.gz → rna_seq_analysis\\GSE96686\\GSM2536318_Kdm4a_KO.wig\n",
      "❌ Failed to decompress GZ GSM2536318_Kdm4a_KO.wig.gz: [Errno 28] No space left on device\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE96686\\GSM2536319_H3K4me3_WT1.wig.gz → rna_seq_analysis\\GSE96686\\GSM2536319_H3K4me3_WT1.wig\n",
      "❌ Failed to decompress GZ GSM2536319_H3K4me3_WT1.wig.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE96686\\\\GSM2536319_H3K4me3_WT1.wig'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE96686\\GSM2536320_H3K4me3_WT2.wig.gz → rna_seq_analysis\\GSE96686\\GSM2536320_H3K4me3_WT2.wig\n",
      "❌ Failed to decompress GZ GSM2536320_H3K4me3_WT2.wig.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE96686\\\\GSM2536320_H3K4me3_WT2.wig'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE96686\\GSM2536321_H3K4me3_KO1.wig.gz → rna_seq_analysis\\GSE96686\\GSM2536321_H3K4me3_KO1.wig\n",
      "❌ Failed to decompress GZ GSM2536321_H3K4me3_KO1.wig.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE96686\\\\GSM2536321_H3K4me3_KO1.wig'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE96686\\GSM2536322_H3K4me3_KO2.wig.gz → rna_seq_analysis\\GSE96686\\GSM2536322_H3K4me3_KO2.wig\n",
      "❌ Failed to decompress GZ GSM2536322_H3K4me3_KO2.wig.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE96686\\\\GSM2536322_H3K4me3_KO2.wig'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE96686\\GSM2536323_H3K9me3_WT1.wig.gz → rna_seq_analysis\\GSE96686\\GSM2536323_H3K9me3_WT1.wig\n",
      "❌ Failed to decompress GZ GSM2536323_H3K9me3_WT1.wig.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE96686\\\\GSM2536323_H3K9me3_WT1.wig'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE96686\\GSM2536324_H3K9me3_WT2.wig.gz → rna_seq_analysis\\GSE96686\\GSM2536324_H3K9me3_WT2.wig\n",
      "❌ Failed to decompress GZ GSM2536324_H3K9me3_WT2.wig.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE96686\\\\GSM2536324_H3K9me3_WT2.wig'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE96686\\GSM2536325_H3K9me3_KO1.wig.gz → rna_seq_analysis\\GSE96686\\GSM2536325_H3K9me3_KO1.wig\n",
      "❌ Failed to decompress GZ GSM2536325_H3K9me3_KO1.wig.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE96686\\\\GSM2536325_H3K9me3_KO1.wig'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE96686\\GSM2536326_H3K9me3_KO2.wig.gz → rna_seq_analysis\\GSE96686\\GSM2536326_H3K9me3_KO2.wig\n",
      "❌ Failed to decompress GZ GSM2536326_H3K9me3_KO2.wig.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE96686\\\\GSM2536326_H3K9me3_KO2.wig'\n",
      "🗜️ Decompressing GZ: rna_seq_analysis\\GSE96686\\GSM2536329_IgG_WT.wig.gz → rna_seq_analysis\\GSE96686\\GSM2536329_IgG_WT.wig\n",
      "❌ Failed to decompress GZ GSM2536329_IgG_WT.wig.gz: [Errno 28] No space left on device: 'rna_seq_analysis\\\\GSE96686\\\\GSM2536329_IgG_WT.wig'\n",
      "Running DESeq2 for: GSE236921\n",
      "📥 Downloading GSE236921_RAW.tar → rna_seq_analysis/GSE236921/GSE236921_RAW.tar\n",
      "❌ Download failed for GSE236921: Command '['curl', '-L', '-o', 'rna_seq_analysis/GSE236921/GSE236921_RAW.tar', 'https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE236921&format=file']' returned non-zero exit status 23.\n",
      "🔍 Found 70 count files for GSE236921\n",
      "✅ No zero counts — no normalization applied.\n",
      "✅ Final merged count matrix shape: (117135, 70)\n",
      "Error running DESeq2 for GSE236921: \"None of [Index(['group'], dtype='object')] are in the [columns]\"\n",
      "Running DESeq2 for: GSE180233\n",
      "📥 Downloading GSE180233_RAW.tar → rna_seq_analysis/GSE180233/GSE180233_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "🔍 Found 16 count files for GSE180233\n",
      "⚠️ Zero counts detected in merged matrix — applying +1 normalization.\n",
      "✅ Final merged count matrix shape: (57772, 16)\n",
      "Error running DESeq2 for GSE180233: \"None of [Index(['group'], dtype='object')] are in the [columns]\"\n",
      "Running DESeq2 for: GSE173995\n",
      "📥 Downloading GSE173995_RAW.tar → rna_seq_analysis/GSE173995/GSE173995_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "🔍 Found 12 count files for GSE173995\n",
      "❌ Failed to read GSM5284291_K9me3_30uM_IOX1.broadPeak.gz: Unable to allocate 23.0 GiB for an array with shape (3082846534,) and data type int64\n",
      "❌ Failed to read GSM5284292_K9me3_50uM_IOX1.broadPeak.gz: Unable to allocate 2.97 GiB for an array with shape (398752702,) and data type object\n",
      "❌ Failed to read GSM5284293_K9me3_100uM_IOX1.broadPeak.gz: Unable to allocate 5.05 GiB for an array with shape (4, 169305464) and data type float64\n",
      "❌ Failed to read GSM5284294_K9me3_DMSO.broadPeak.gz: Unable to allocate 11.1 GiB for an array with shape (1486400416,) and data type int64\n",
      "❌ Failed to read GSM5284295_K27me3_30uM_IOX1.broadPeak.gz: Unable to allocate 6.04 GiB for an array with shape (810457944,) and data type object\n",
      "❌ Failed to read GSM5284296_K27me3_50uM_IOX1.broadPeak.gz: Unable to allocate 4.54 GiB for an array with shape (4, 152403378) and data type float64\n",
      "❌ Failed to read GSM5284298_K27me3_DMSO.broadPeak.gz: Unable to allocate 2.98 TiB for an array with shape (409032482912,) and data type int64\n",
      "⚠️ Zero counts detected in merged matrix — applying +1 normalization.\n",
      "✅ Final merged count matrix shape: (85518475, 5)\n",
      "Error running DESeq2 for GSE173995: \"None of [Index(['group'], dtype='object')] are in the [columns]\"\n",
      "Running DESeq2 for: GSE236921\n",
      "📥 Downloading GSE236921_RAW.tar → rna_seq_analysis/GSE236921/GSE236921_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "🔍 Found 70 count files for GSE236921\n",
      "✅ No zero counts — no normalization applied.\n",
      "✅ Final merged count matrix shape: (117135, 70)\n",
      "Error running DESeq2 for GSE236921: \"None of [Index(['group'], dtype='object')] are in the [columns]\"\n",
      "Running DESeq2 for: GSE180233\n",
      "📥 Downloading GSE180233_RAW.tar → rna_seq_analysis/GSE180233/GSE180233_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "🔍 Found 16 count files for GSE180233\n",
      "⚠️ Zero counts detected in merged matrix — applying +1 normalization.\n",
      "✅ Final merged count matrix shape: (57772, 16)\n",
      "Error running DESeq2 for GSE180233: \"None of [Index(['group'], dtype='object')] are in the [columns]\"\n",
      "Running DESeq2 for: GSE173995\n",
      "📥 Downloading GSE173995_RAW.tar → rna_seq_analysis/GSE173995/GSE173995_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "🔍 Found 12 count files for GSE173995\n",
      "❌ Failed to read GSM5284291_K9me3_30uM_IOX1.broadPeak.gz: Unable to allocate 23.0 GiB for an array with shape (3082846534,) and data type int64\n",
      "❌ Failed to read GSM5284292_K9me3_50uM_IOX1.broadPeak.gz: Unable to allocate 2.97 GiB for an array with shape (398752702, 1) and data type float64\n",
      "❌ Failed to read GSM5284293_K9me3_100uM_IOX1.broadPeak.gz: Unable to allocate 5.05 GiB for an array with shape (4, 169305464) and data type float64\n",
      "❌ Failed to read GSM5284294_K9me3_DMSO.broadPeak.gz: Unable to allocate 11.1 GiB for an array with shape (1486400416,) and data type int64\n",
      "❌ Failed to read GSM5284295_K27me3_30uM_IOX1.broadPeak.gz: Unable to allocate 6.04 GiB for an array with shape (810457944,) and data type object\n",
      "❌ Failed to read GSM5284296_K27me3_50uM_IOX1.broadPeak.gz: Unable to allocate 4.54 GiB for an array with shape (4, 152403378) and data type float64\n",
      "❌ Failed to read GSM5284298_K27me3_DMSO.broadPeak.gz: Unable to allocate 2.98 TiB for an array with shape (409032482912,) and data type int64\n",
      "⚠️ Zero counts detected in merged matrix — applying +1 normalization.\n",
      "✅ Final merged count matrix shape: (85518475, 5)\n",
      "Error running DESeq2 for GSE173995: \"None of [Index(['group'], dtype='object')] are in the [columns]\"\n",
      "Running DESeq2 for: GSE236921\n",
      "📥 Downloading GSE236921_RAW.tar → rna_seq_analysis/GSE236921/GSE236921_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "🔍 Found 70 count files for GSE236921\n",
      "✅ No zero counts — no normalization applied.\n",
      "✅ Final merged count matrix shape: (117135, 70)\n",
      "Error running DESeq2 for GSE236921: \"None of [Index(['group'], dtype='object')] are in the [columns]\"\n",
      "Running DESeq2 for: GSE180233\n",
      "📥 Downloading GSE180233_RAW.tar → rna_seq_analysis/GSE180233/GSE180233_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "🔍 Found 16 count files for GSE180233\n",
      "⚠️ Zero counts detected in merged matrix — applying +1 normalization.\n",
      "✅ Final merged count matrix shape: (57772, 16)\n",
      "Error running DESeq2 for GSE180233: \"None of [Index(['group'], dtype='object')] are in the [columns]\"\n",
      "Running DESeq2 for: GSE173995\n",
      "📥 Downloading GSE173995_RAW.tar → rna_seq_analysis/GSE173995/GSE173995_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "🔍 Found 12 count files for GSE173995\n",
      "❌ Failed to read GSM5284291_K9me3_30uM_IOX1.broadPeak.gz: Unable to allocate 23.0 GiB for an array with shape (3082846534,) and data type int64\n",
      "❌ Failed to read GSM5284292_K9me3_50uM_IOX1.broadPeak.gz: Unable to allocate 2.97 GiB for an array with shape (398752702, 1) and data type float64\n",
      "❌ Failed to read GSM5284293_K9me3_100uM_IOX1.broadPeak.gz: Unable to allocate 5.05 GiB for an array with shape (4, 169305464) and data type float64\n",
      "❌ Failed to read GSM5284294_K9me3_DMSO.broadPeak.gz: Unable to allocate 11.1 GiB for an array with shape (1486400416,) and data type int64\n",
      "❌ Failed to read GSM5284295_K27me3_30uM_IOX1.broadPeak.gz: Unable to allocate 6.04 GiB for an array with shape (810457944,) and data type object\n",
      "❌ Failed to read GSM5284296_K27me3_50uM_IOX1.broadPeak.gz: Unable to allocate 4.54 GiB for an array with shape (4, 152403378) and data type float64\n",
      "❌ Failed to read GSM5284298_K27me3_DMSO.broadPeak.gz: Unable to allocate 2.98 TiB for an array with shape (409032482912,) and data type int64\n",
      "⚠️ Zero counts detected in merged matrix — applying +1 normalization.\n",
      "✅ Final merged count matrix shape: (85518475, 5)\n",
      "Error running DESeq2 for GSE173995: \"None of [Index(['group'], dtype='object')] are in the [columns]\"\n",
      "Running DESeq2 for: GSE236921\n",
      "📥 Downloading GSE236921_RAW.tar → rna_seq_analysis/GSE236921/GSE236921_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "🔍 Found 70 count files for GSE236921\n",
      "✅ No zero counts — no normalization applied.\n",
      "✅ Final merged count matrix shape: (117135, 70)\n",
      "Error running DESeq2 for GSE236921: \"None of [Index(['group'], dtype='object')] are in the [columns]\"\n",
      "Running DESeq2 for: GSE180233\n",
      "📥 Downloading GSE180233_RAW.tar → rna_seq_analysis/GSE180233/GSE180233_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "🔍 Found 16 count files for GSE180233\n",
      "⚠️ Zero counts detected in merged matrix — applying +1 normalization.\n",
      "✅ Final merged count matrix shape: (57772, 16)\n",
      "Error running DESeq2 for GSE180233: \"None of [Index(['group'], dtype='object')] are in the [columns]\"\n",
      "Running DESeq2 for: GSE173995\n",
      "📥 Downloading GSE173995_RAW.tar → rna_seq_analysis/GSE173995/GSE173995_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "🔍 Found 12 count files for GSE173995\n",
      "❌ Failed to read GSM5284291_K9me3_30uM_IOX1.broadPeak.gz: Unable to allocate 23.0 GiB for an array with shape (3082846534,) and data type int64\n",
      "❌ Failed to read GSM5284292_K9me3_50uM_IOX1.broadPeak.gz: Unable to allocate 2.97 GiB for an array with shape (398752702, 1) and data type float64\n",
      "❌ Failed to read GSM5284293_K9me3_100uM_IOX1.broadPeak.gz: Unable to allocate 5.05 GiB for an array with shape (4, 169305464) and data type float64\n",
      "❌ Failed to read GSM5284294_K9me3_DMSO.broadPeak.gz: Unable to allocate 11.1 GiB for an array with shape (1486400416,) and data type int64\n",
      "❌ Failed to read GSM5284295_K27me3_30uM_IOX1.broadPeak.gz: Unable to allocate 6.04 GiB for an array with shape (810457944,) and data type object\n",
      "❌ Failed to read GSM5284296_K27me3_50uM_IOX1.broadPeak.gz: Unable to allocate 4.54 GiB for an array with shape (4, 152403378) and data type float64\n",
      "❌ Failed to read GSM5284298_K27me3_DMSO.broadPeak.gz: Unable to allocate 2.98 TiB for an array with shape (409032482912,) and data type int64\n",
      "⚠️ Zero counts detected in merged matrix — applying +1 normalization.\n",
      "✅ Final merged count matrix shape: (85518475, 5)\n",
      "Error running DESeq2 for GSE173995: \"None of [Index(['group'], dtype='object')] are in the [columns]\"\n"
     ]
    },
    {
     "ename": "TypeError",
     "evalue": "object AddableValuesDict can't be used in 'await' expression",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[12], line 1\u001b[0m\n\u001b[1;32m----> 1\u001b[0m result \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mawait\u001b[39;00m graph\u001b[38;5;241m.\u001b[39minvoke(initial_state)\n",
      "\u001b[1;31mTypeError\u001b[0m: object AddableValuesDict can't be used in 'await' expression"
     ]
    }
   ],
   "source": [
    "result = await graph.invoke(initial_state)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "7a52d907-f765-422a-ac77-8b9fc611da68",
   "metadata": {},
   "outputs": [],
   "source": [
    "import gzip\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import json\n",
    "import re\n",
    "from collections import defaultdict\n",
    "from pathlib import Path\n",
    "\n",
    "def parse_series_matrix_to_dataframe(series_matrix_path):\n",
    "    data = {}\n",
    "    sample_ids = None\n",
    "    field_counter = defaultdict(int)\n",
    "\n",
    "    open_func = gzip.open if series_matrix_path.endswith(\".gz\") else open\n",
    "    with open_func(series_matrix_path, 'rt', encoding='utf-8') as f:\n",
    "        for line in f:\n",
    "            if line.startswith(\"!Sample_\"):\n",
    "                parts = line.strip().split(\"\\t\")\n",
    "                raw_field = parts[0].lstrip(\"!\")\n",
    "                count = field_counter[raw_field]\n",
    "                field_name = raw_field if count == 0 else f\"{raw_field}_{count}\"\n",
    "                field_counter[raw_field] += 1\n",
    "                values = [p.strip('\"') for p in parts[1:]]\n",
    "                data[field_name] = values\n",
    "                if raw_field == \"Sample_geo_accession\":\n",
    "                    sample_ids = values\n",
    "\n",
    "    if sample_ids is None:\n",
    "        raise ValueError(\"❌ No Sample_geo_accession found.\")\n",
    "\n",
    "    df = pd.DataFrame(data)\n",
    "    df.columns = df.columns.str.replace(r'\\\\', '', regex=True).str.strip()\n",
    "    df.index = sample_ids\n",
    "    df.index.name = \"geo_accession\"\n",
    "    print(f\"✅ Series matrix parsed: {df.shape}\")\n",
    "    return df\n",
    "\n",
    "def remove_uniform_columns(df):\n",
    "    nunique = df.nunique(dropna=False)\n",
    "    non_uniform_cols = nunique[nunique > 1].index\n",
    "    cleaned_df = df[non_uniform_cols].copy()\n",
    "    print(f\"✅ Removed uniform columns: {df.shape[1] - cleaned_df.shape[1]} dropped, {cleaned_df.shape[1]} kept\")\n",
    "    return cleaned_df\n",
    "\n",
    "def send_mini_table_to_llm(client, gse_id, series_df):\n",
    "    mini_df = series_df.sample(n=min(6, len(series_df)), random_state=42)\n",
    "    records = mini_df.reset_index().to_dict(orient=\"records\")\n",
    "\n",
    "    prompt = f\"\"\"\n",
    "You are a biomedical data curator assisting with RNA-seq DESeq2 analysis.\n",
    "\n",
    "Below is a compact metadata table for GEO study {gse_id}.\n",
    "Each row represents a sample (geo_accession). The columns represent metadata fields.\n",
    "\n",
    "Your task:\n",
    "- Identify the columns (metadata fields) that contain key information useful for defining groups for differential expression analysis.\n",
    "- Focus on metadata about: *treatment*, *time*, *gender*, *dose*, *genotype*, *WT/KO*.\n",
    "- \"control\" often labelled as control, mock, media, DMSO, untreated, blank etc. No extra text, no explanation.\n",
    "- Return ONLY a JSON list of column names that are useful. No extra text, no explanation.\n",
    "\n",
    "Here is the table:\n",
    "{json.dumps(records, indent=2)}\n",
    "\"\"\"\n",
    "    print(f\"📝 Sending prompt (first 500 chars): {prompt[:500]}...\")\n",
    "    response = client.chat.completions.create(\n",
    "        model=\"gpt-4o\",\n",
    "        temperature=0.2,\n",
    "        messages=[{\"role\": \"user\", \"content\": prompt}]\n",
    "    )\n",
    "    content = response.choices[0].message.content.strip()\n",
    "    content = re.sub(r\"^```json\", \"\", content, flags=re.IGNORECASE).strip()\n",
    "    content = re.sub(r\"```$\", \"\", content).strip()\n",
    "\n",
    "    try:\n",
    "        col_list = json.loads(content)\n",
    "        print(f\"✅ LLM returned {len(col_list)} useful columns from original metadata\")\n",
    "        return col_list\n",
    "    except json.JSONDecodeError:\n",
    "        print(\"❌ Failed to parse LLM output. Raw content preview:\")\n",
    "        print(content[:500])\n",
    "        raise\n",
    "\n",
    "def derive_condition_with_llm(client, gse_id, metadata_df, useful_cols):\n",
    "    df_for_llm = metadata_df[useful_cols].copy()\n",
    "    df_for_llm[\"geo_accession\"] = metadata_df.index\n",
    "    records = df_for_llm.to_dict(orient=\"records\")\n",
    "\n",
    "    prompt = f\"\"\"\n",
    "You are a biomedical data curator assisting with RNA-seq DESeq2 analysis.\n",
    "\n",
    "Below is metadata for GEO study {gse_id}. Each row represents a sample.\n",
    "\n",
    "Your task:\n",
    "- Determine if each sample represents a \"treated\" or \"control\" condition.\n",
    "- Use the provided metadata fields (treatment, time, genotype, etc).\n",
    "- \"control\" often labelled as control, mock, media, DMSO, untreated, blank etc. No extra text, no explanation.\n",
    "- Return ONLY a JSON mapping geo_accession → \"treated with X\" or \"control\" or \"treated with Y, or other based on key metadata field. No explanation.\n",
    "\n",
    "Here is the metadata:\n",
    "{json.dumps(records, indent=2)}\n",
    "\"\"\"\n",
    "    print(f\"📝 Sending LLM condition prompt (first 500 chars): {prompt[:500]}...\")\n",
    "    response = client.chat.completions.create(\n",
    "        model=\"gpt-4o\",\n",
    "        temperature=0.2,\n",
    "        messages=[{\"role\": \"user\", \"content\": prompt}]\n",
    "    )\n",
    "    content = response.choices[0].message.content.strip()\n",
    "    content = re.sub(r\"^```json\", \"\", content, flags=re.IGNORECASE).strip()\n",
    "    content = re.sub(r\"```$\", \"\", content).strip()\n",
    "\n",
    "    try:\n",
    "        condition_map = json.loads(content)\n",
    "        print(f\"✅ LLM returned conditions for {len(condition_map)} samples\")\n",
    "    except json.JSONDecodeError:\n",
    "        print(\"❌ Failed to parse LLM output. Raw content preview:\")\n",
    "        print(content[:500])\n",
    "        raise\n",
    "\n",
    "    # Apply condition map correctly (do not use dict as indexer)\n",
    "    metadata_df[\"condition\"] = metadata_df.index.to_series().map(lambda x: condition_map.get(x, None))\n",
    "    print(f\"✅ derived condition LLM construct design {metadata_df} from minitable metadata\")\n",
    "    return metadata_df\n",
    "\n",
    "\n",
    "def run_deg_for_single_gse_run(state, client):\n",
    "    gse_id = state.get(\"selected_gse\")\n",
    "    if not gse_id:\n",
    "        print(\"❌ No selected GSE found.\")\n",
    "        return state\n",
    "\n",
    "    deg_paths = {}\n",
    "    series_matrix_path = f\"geofetch_metadata/{gse_id}_series_matrix.txt.gz\"\n",
    "    decompress_gse_in_dir(gse_id)\n",
    "\n",
    "    try:\n",
    "        series_df = parse_series_matrix_to_dataframe(series_matrix_path)\n",
    "        series_df = remove_uniform_columns(series_df)\n",
    "\n",
    "        useful_cols = send_mini_table_to_llm(client, gse_id, series_df)\n",
    "        metadata_df = series_df[useful_cols].copy()\n",
    "        metadata_df.index.name = \"sample_geo_accession\"\n",
    "\n",
    "        metadata_df = derive_condition_with_llm(client, gse_id, metadata_df, useful_cols)\n",
    "\n",
    "        download_dir = download_and_extract_gse(gse_id)\n",
    "        counts_df = find_and_merge_count_files(gse_id)\n",
    "        if counts_df is None:\n",
    "            print(f\"⚠️ No count matrix found for {gse_id}\")\n",
    "            return state\n",
    "\n",
    "        valid_samples = metadata_df.index.intersection(counts_df.columns)\n",
    "        counts_df = counts_df[valid_samples].T\n",
    "        metadata_df = metadata_df.loc[valid_samples]\n",
    "        #metadata_df = metadata_df[metadata_df[\"condition\"].isin([\"treated\", \"control\"])]\n",
    "\n",
    "        if len(metadata_df[\"condition\"].unique()) < 2:\n",
    "            raise ValueError(f\"❌ Need both treated and control groups for DESeq2, found: {metadata_df['condition'].unique()}\")\n",
    "\n",
    "        # Dynamically determine contrast\n",
    "        cond_values = metadata_df[\"condition\"].dropna().unique()\n",
    "        treated_group, control_group = cond_values[0], cond_values[1]\n",
    "        print(f\"✅ Using contrast: {treated_group} vs {control_group}\")\n",
    "\n",
    "        print(f\"✅ Counts shape: {counts_df.shape}\")\n",
    "        print(f\"✅ Metadata shape: {metadata_df.shape}\")\n",
    "        print(f\"✅ Condition counts:\\n{metadata_df['condition'].value_counts()}\")\n",
    "\n",
    "        inference = DefaultInference(n_cpus=4)\n",
    "        dds = DeseqDataSet(\n",
    "            counts=counts_df,\n",
    "            metadata=metadata_df,\n",
    "            design_factors=\"condition\",\n",
    "            refit_cooks=True,\n",
    "            inference=inference\n",
    "        )\n",
    "        dds.fit_size_factors()\n",
    "        dds.fit_genewise_dispersions()\n",
    "        dds.fit_dispersion_trend()\n",
    "        dds.fit_dispersion_prior()\n",
    "        dds.fit_MAP_dispersions()\n",
    "        dds.fit_LFC()\n",
    "        dds.calculate_cooks()\n",
    "        if dds.refit_cooks:\n",
    "            dds.refit()\n",
    "\n",
    "        stat_res = DeseqStats(\n",
    "            dds,\n",
    "            alpha=0.05,\n",
    "            contrast=[\"condition\", treated_group, control_group],\n",
    "            cooks_filter=True,\n",
    "            independent_filter=True\n",
    "        )\n",
    "        stat_res.summary()\n",
    "\n",
    "        res = stat_res.results_df\n",
    "        output_csv = f\"rna_seq_analysis/{gse_id}_deseq2_results.csv\"\n",
    "        res.to_csv(output_csv, index=False)\n",
    "        deg_paths[gse_id] = output_csv\n",
    "        print(f\"✅ DESeq2 results saved: {output_csv}\")\n",
    "\n",
    "        sig_genes = res[(res[\"padj\"] < 0.05) & (abs(res[\"log2FoldChange\"]) > 1)]\n",
    "        print(f\"🧬 {sig_genes.shape[0]} significant genes found\")\n",
    "        # Additional visualization + analysis\n",
    "        # Add log1p layer\n",
    "        dds.layers[\"log1p\"] = np.log1p(dds.layers[\"normed_counts\"])\n",
    "        \n",
    "        # Select significant genes dynamically\n",
    "        sigs = res[(res[\"padj\"] < 0.05) & (res[\"log2FoldChange\"].abs() > 1.5)]\n",
    "        if sigs.empty:\n",
    "            print(\"⚠️ No significant genes found for heatmap/volcano plot.\")\n",
    "        else:\n",
    "            # Build grapher matrix\n",
    "            dds_sigs = dds[:, sigs.index]\n",
    "            grapher = pd.DataFrame(\n",
    "                dds_sigs.layers[\"log1p\"],\n",
    "                index=dds_sigs.obs_names,\n",
    "                columns=dds_sigs.var_names\n",
    "            ).T\n",
    "            grapher.index.name = \"gene\"\n",
    "        \n",
    "            # Dynamically generate column labels\n",
    "            labeled_columns = [\n",
    "                f\"{sample}_{metadata_df.loc[sample, 'condition']}\"\n",
    "                for sample in grapher.columns\n",
    "            ]\n",
    "            grapher.columns = labeled_columns\n",
    "        \n",
    "            # Volcano plot\n",
    "            res[\"symbol\"] = res.index\n",
    "            volcano(\n",
    "                res,\n",
    "                symbol=\"symbol\",\n",
    "                log2fc=\"log2FoldChange\",\n",
    "                pval_thresh=0.05,\n",
    "                to_label=10,\n",
    "                log2fc_thresh=0.75,\n",
    "                colors=[\"dimgrey\", \"lightgrey\", \"green\"]\n",
    "            )\n",
    "            plt.show()\n",
    "        \n",
    "            # Heatmap\n",
    "            sns.clustermap(grapher, z_score=0, cmap=\"RdBu_r\", col_cluster=False)\n",
    "            plt.show()\n",
    "        \n",
    "            # PCA\n",
    "            sc.tl.pca(dds)\n",
    "            sc.pl.pca(dds, color=\"condition\", size=200)\n",
    "\n",
    "    except Exception as e:\n",
    "        print(f\"❌ Error: {e}\")\n",
    "\n",
    "    return {**state, \"deg_results\": deg_paths}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "17f02c7f-cd06-4a9f-a6cc-6c029f272cc1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "🔍 Scanning for compressed files in: rna_seq_analysis\\GSE224084\n",
      "⚠️ Directory does not exist: rna_seq_analysis\\GSE224084\n",
      "❌ Error: [Errno 2] No such file or directory: 'geofetch_metadata/GSE224084_series_matrix.txt.gz'\n"
     ]
    }
   ],
   "source": [
    "## it may not work if counts table author provided already combined: GSE259434\n",
    "state = {\n",
    "    \"selected_gse\": \"GSE224084\"\n",
    "}\n",
    "result = run_deg_for_single_gse_run(state, client)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5a2bdcf2-aa98-4e0d-9d7d-470ed2ed0334",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "47b078b8-7f70-41a8-b6da-dbd4b646beed",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "6a701c9f-ab7f-4e3c-a214-f27a4552cdf8",
   "metadata": {},
   "outputs": [],
   "source": [
    "initial_state = AgentState(\n",
    "    query_gene=\"IL2\",\n",
    "    pathway_info={\"genes\": [], \"drugs\": [], \"pathways\": []},\n",
    "    metadata={},\n",
    "    gse_list=[],\n",
    "    research_plan=\"\",\n",
    "    series_matrix_path=\"\",\n",
    "    selected_gse=\"\",\n",
    "    deg_results={}\n",
    ")\n",
    "result = await graph.invoke(initial_state)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cab80a37-54b4-45bc-b1fe-6bde3dc4a4d9",
   "metadata": {},
   "outputs": [],
   "source": [
    "get_geofetch_projects(gse_list, metadata_folder=\"geofetch_metadata\") "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d5f7916e-9817-4e23-be62-b075f430e147",
   "metadata": {},
   "outputs": [],
   "source": [
    "############ redo fucntion of extract label for metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "id": "3347d10e-c05f-42df-8e3e-6032245b2b92",
   "metadata": {},
   "outputs": [],
   "source": [
    "########## working!!!\n",
    "import gzip\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import json\n",
    "import re\n",
    "from collections import defaultdict\n",
    "\n",
    "def parse_series_matrix_to_dataframe(series_matrix_path):\n",
    "    \"\"\"\n",
    "    Parse GEO series matrix (.gz or plain) into a DataFrame:\n",
    "    Rows = geo_accession\n",
    "    Columns = metadata fields (with unique names)\n",
    "    \"\"\"\n",
    "    data = {}\n",
    "    sample_ids = None\n",
    "    field_counter = defaultdict(int)\n",
    "\n",
    "    open_func = gzip.open if series_matrix_path.endswith(\".gz\") else open\n",
    "    with open_func(series_matrix_path, 'rt', encoding='utf-8') as f:\n",
    "        for line in f:\n",
    "            if line.startswith(\"!Sample_\"):\n",
    "                parts = line.strip().split(\"\\t\")\n",
    "                raw_field = parts[0].lstrip(\"!\")\n",
    "                count = field_counter[raw_field]\n",
    "                field_name = raw_field if count == 0 else f\"{raw_field}_{count}\"\n",
    "                field_counter[raw_field] += 1\n",
    "\n",
    "                values = [p.strip('\"') for p in parts[1:]]\n",
    "                data[field_name] = values\n",
    "                if raw_field == \"Sample_geo_accession\":\n",
    "                    sample_ids = values\n",
    "\n",
    "    if sample_ids is None:\n",
    "        raise ValueError(\"❌ Could not find Sample_geo_accession row!\")\n",
    "\n",
    "    df = pd.DataFrame(data)\n",
    "    df.columns = df.columns.str.replace(r'\\\\', '', regex=True).str.strip()\n",
    "    df.index = sample_ids\n",
    "    df.index.name = \"geo_accession\"\n",
    "    print(f\"✅ Parsed DataFrame shape: {df.shape}\")\n",
    "    return df\n",
    "\n",
    "def send_mini_table_to_llm(client, gse_id, sample_labels, mini_df):\n",
    "    \"\"\"\n",
    "    Send mini table (6 random rows + all columns) to LLM; get important metadata columns.\n",
    "    \"\"\"\n",
    "    records = mini_df.reset_index().to_dict(orient=\"records\")\n",
    "\n",
    "    prompt = f\"\"\"\n",
    "You are a biomedical data curator assisting with RNA-seq DESeq2 analysis.\n",
    "\n",
    "Below is a compact metadata table for GEO study {gse_id}.\n",
    "Each row represents a sample (geo_accession). The columns represent metadata fields.\n",
    "\n",
    "Your task:\n",
    "- Identify the columns (metadata fields) that contain key information useful for defining groups for differential expression analysis.\n",
    "- Focus on metadata about: *treatment*, *time*, *gender*, *dose*, *genotype*, *WT/KO*.\n",
    "- Return ONLY a JSON list of column names that are useful. No extra text, no explanation.\n",
    "\n",
    "Here is the table:\n",
    "{json.dumps(records, indent=2)}\n",
    "\"\"\"\n",
    "\n",
    "    print(f\"📝 Sending prompt (first 500 chars): {prompt[:500]}...\")\n",
    "    response = client.chat.completions.create(\n",
    "        model=\"gpt-4o\",\n",
    "        temperature=0.2,\n",
    "        messages=[{\"role\": \"user\", \"content\": prompt}]\n",
    "    )\n",
    "    content = response.choices[0].message.content.strip()\n",
    "\n",
    "    content = re.sub(r\"^```json\", \"\", content, flags=re.IGNORECASE).strip()\n",
    "    content = re.sub(r\"```$\", \"\", content).strip()\n",
    "\n",
    "    try:\n",
    "        col_list = json.loads(content)\n",
    "        print(f\"✅ LLM returned {len(col_list)} useful columns\")\n",
    "        return col_list\n",
    "    except json.JSONDecodeError:\n",
    "        print(\"❌ Failed to parse LLM output. Raw content preview:\")\n",
    "        print(content[:500])\n",
    "        raise\n",
    "\n",
    "def filter_dataframe_by_columns(df, col_list):\n",
    "    \"\"\"\n",
    "    Filter DataFrame to only include the useful columns.\n",
    "    \"\"\"\n",
    "    valid_cols = [col for col in col_list if col in df.columns]\n",
    "    return df[valid_cols]\n",
    "\n",
    "def run_metadata_selection(series_matrix_path, gse_id, client, sample_labels):\n",
    "    \"\"\"\n",
    "    Parse matrix, sample 6 random rows, send to LLM, get useful columns, filter DataFrame.\n",
    "    \"\"\"\n",
    "    df = parse_series_matrix_to_dataframe(series_matrix_path)\n",
    "\n",
    "    # ✅ Randomly sample 6 sample rows\n",
    "    mini_df = df.sample(n=min(6, len(df)), random_state=42)\n",
    "\n",
    "    print(f\"📊 Mini table shape: {mini_df.shape} (6 rows, {df.shape[1]} columns)\")\n",
    "    print(mini_df.head())\n",
    "\n",
    "    # Send to LLM to get useful column names\n",
    "    useful_cols = send_mini_table_to_llm(client, gse_id, sample_labels, mini_df)\n",
    "\n",
    "    # Filter DataFrame to those columns\n",
    "    df_filtered = filter_dataframe_by_columns(df, useful_cols)\n",
    "\n",
    "    print(f\"🎉 Final filtered DataFrame shape: {df_filtered.shape}\")\n",
    "    return df_filtered\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "id": "5064fcf6-78c0-403d-8475-07852614da5e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "✅ Parsed DataFrame shape: (256, 38)\n",
      "📊 Mini table shape: (6, 38) (6 rows, 38 columns)\n",
      "              Sample_title Sample_geo_accession          Sample_status  \\\n",
      "geo_accession                                                            \n",
      "GSM8978094          334-6W           GSM8978094  Public on May 12 2025   \n",
      "GSM8977872          139-6W           GSM8977872  Public on May 12 2025   \n",
      "GSM8977945          278-6W           GSM8977945  Public on May 12 2025   \n",
      "GSM8978072          296-3W           GSM8978072  Public on May 12 2025   \n",
      "GSM8977983          203-8W           GSM8977983  Public on May 12 2025   \n",
      "\n",
      "              Sample_submission_date Sample_last_update_date Sample_type  \\\n",
      "geo_accession                                                              \n",
      "GSM8978094               May 12 2025             May 12 2025         SRA   \n",
      "GSM8977872               May 12 2025             May 12 2025         SRA   \n",
      "GSM8977945               May 12 2025             May 12 2025         SRA   \n",
      "GSM8978072               May 12 2025             May 12 2025         SRA   \n",
      "GSM8977983               May 12 2025             May 12 2025         SRA   \n",
      "\n",
      "              Sample_channel_count Sample_source_name_ch1 Sample_organism_ch1  \\\n",
      "geo_accession                                                                   \n",
      "GSM8978094                       1                  Blood        Homo sapiens   \n",
      "GSM8977872                       1                  Blood        Homo sapiens   \n",
      "GSM8977945                       1                  Blood        Homo sapiens   \n",
      "GSM8978072                       1                  Blood        Homo sapiens   \n",
      "GSM8977983                       1                  Blood        Homo sapiens   \n",
      "\n",
      "              Sample_characteristics_ch1  ... Sample_contact_zip/postal_code  \\\n",
      "geo_accession                             ...                                  \n",
      "GSM8978094                 tissue: Blood  ...                     75390-9082   \n",
      "GSM8977872                 tissue: Blood  ...                     75390-9082   \n",
      "GSM8977945                 tissue: Blood  ...                     75390-9082   \n",
      "GSM8978072                 tissue: Blood  ...                     75390-9082   \n",
      "GSM8977983                 tissue: Blood  ...                     75390-9082   \n",
      "\n",
      "              Sample_contact_country Sample_data_row_count  \\\n",
      "geo_accession                                                \n",
      "GSM8978094                       USA                     0   \n",
      "GSM8977872                       USA                     0   \n",
      "GSM8977945                       USA                     0   \n",
      "GSM8978072                       USA                     0   \n",
      "GSM8977983                       USA                     0   \n",
      "\n",
      "              Sample_instrument_model Sample_library_selection  \\\n",
      "geo_accession                                                    \n",
      "GSM8978094      Illumina NovaSeq 6000                     cDNA   \n",
      "GSM8977872      Illumina NovaSeq 6000                     cDNA   \n",
      "GSM8977945      Illumina NovaSeq 6000                     cDNA   \n",
      "GSM8978072      Illumina NovaSeq 6000                     cDNA   \n",
      "GSM8977983      Illumina NovaSeq 6000                     cDNA   \n",
      "\n",
      "              Sample_library_source Sample_library_strategy  \\\n",
      "geo_accession                                                 \n",
      "GSM8978094           transcriptomic                 RNA-Seq   \n",
      "GSM8977872           transcriptomic                 RNA-Seq   \n",
      "GSM8977945           transcriptomic                 RNA-Seq   \n",
      "GSM8978072           transcriptomic                 RNA-Seq   \n",
      "GSM8977983           transcriptomic                 RNA-Seq   \n",
      "\n",
      "                                                 Sample_relation  \\\n",
      "geo_accession                                                      \n",
      "GSM8978094     BioSample: https://www.ncbi.nlm.nih.gov/biosam...   \n",
      "GSM8977872     BioSample: https://www.ncbi.nlm.nih.gov/biosam...   \n",
      "GSM8977945     BioSample: https://www.ncbi.nlm.nih.gov/biosam...   \n",
      "GSM8978072     BioSample: https://www.ncbi.nlm.nih.gov/biosam...   \n",
      "GSM8977983     BioSample: https://www.ncbi.nlm.nih.gov/biosam...   \n",
      "\n",
      "                                               Sample_relation_1  \\\n",
      "geo_accession                                                      \n",
      "GSM8978094     SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX...   \n",
      "GSM8977872     SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX...   \n",
      "GSM8977945     SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX...   \n",
      "GSM8978072     SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX...   \n",
      "GSM8977983     SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX...   \n",
      "\n",
      "                                     Sample_supplementary_file_1  \n",
      "geo_accession                                                     \n",
      "GSM8978094     ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM8978...  \n",
      "GSM8977872     ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM8977...  \n",
      "GSM8977945     ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM8977...  \n",
      "GSM8978072     ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM8978...  \n",
      "GSM8977983     ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM8977...  \n",
      "\n",
      "[5 rows x 38 columns]\n",
      "📝 Sending prompt (first 500 chars): \n",
      "You are a biomedical data curator assisting with RNA-seq DESeq2 analysis.\n",
      "\n",
      "Below is a compact metadata table for GEO study GSE296826.\n",
      "Each row represents a sample (geo_accession). The columns represent metadata fields.\n",
      "\n",
      "Your task:\n",
      "- Identify the columns (metadata fields) that contain key information useful for defining groups for differential expression analysis.\n",
      "- Focus on metadata about: *treatment*, *time*, *gender*, *dose*, *genotype*, *WT/KO*.\n",
      "- Return ONLY a JSON list of column names that...\n",
      "✅ LLM returned 2 useful columns\n",
      "🎉 Final filtered DataFrame shape: (256, 2)\n",
      "                  Sample_characteristics_ch1_2 Sample_characteristics_ch1_3\n",
      "geo_accession                                                              \n",
      "GSM8977866                treatment: nivolumab                     time: 6W\n",
      "GSM8977867                treatment: nivolumab                     time: BL\n",
      "GSM8977868                treatment: nivolumab                     time: 3M\n",
      "GSM8977869                treatment: nivolumab                     time: BL\n",
      "GSM8977870     treatment: ipilimumab+nivolumab                     time: 6W\n"
     ]
    }
   ],
   "source": [
    "series_matrix_path = \"geofetch_metadata/GSE296826_series_matrix.txt.gz\"\n",
    "gse_id = \"GSE296826\"\n",
    "#sample_labels = [\"122-6W\", \"122-BL\", \"125-3M\"]  # Example sample labels from your count matrix\n",
    "sample_labels = [\"GSM8977872 \", \"GSM8977945\"] \n",
    "# Make sure your OpenAI client (or relevant client) is initialized, e.g.:\n",
    "# import openai\n",
    "# client = openai.OpenAI(api_key=\"YOUR_KEY\")\n",
    "\n",
    "df_filtered = run_metadata_selection(series_matrix_path, gse_id, client, sample_labels)\n",
    "\n",
    "print(df_filtered.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 239,
   "id": "8ccbb9e6-5ec7-44c9-98c9-df39055ba786",
   "metadata": {},
   "outputs": [],
   "source": [
    "import gzip\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import json\n",
    "import re\n",
    "from collections import defaultdict\n",
    "from pathlib import Path\n",
    "\n",
    "def parse_series_matrix_to_dataframe(series_matrix_path):\n",
    "    data = {}\n",
    "    sample_ids = None\n",
    "    field_counter = defaultdict(int)\n",
    "\n",
    "    open_func = gzip.open if series_matrix_path.endswith(\".gz\") else open\n",
    "    with open_func(series_matrix_path, 'rt', encoding='utf-8') as f:\n",
    "        for line in f:\n",
    "            if line.startswith(\"!Sample_\"):\n",
    "                parts = line.strip().split(\"\\t\")\n",
    "                raw_field = parts[0].lstrip(\"!\")\n",
    "                count = field_counter[raw_field]\n",
    "                field_name = raw_field if count == 0 else f\"{raw_field}_{count}\"\n",
    "                field_counter[raw_field] += 1\n",
    "                values = [p.strip('\"') for p in parts[1:]]\n",
    "                data[field_name] = values\n",
    "                if raw_field == \"Sample_geo_accession\":\n",
    "                    sample_ids = values\n",
    "\n",
    "    if sample_ids is None:\n",
    "        raise ValueError(\"❌ No Sample_geo_accession found.\")\n",
    "\n",
    "    df = pd.DataFrame(data)\n",
    "    df.columns = df.columns.str.replace(r'\\\\', '', regex=True).str.strip()\n",
    "    df.index = sample_ids\n",
    "    df.index.name = \"geo_accession\"\n",
    "    print(f\"✅ Series matrix parsed: {df.shape}\")\n",
    "    return df\n",
    "\n",
    "def send_mini_table_to_llm(client, gse_id, mini_df):\n",
    "    records = mini_df.reset_index().to_dict(orient=\"records\")\n",
    "    prompt = f\"\"\"\n",
    "You are a biomedical data curator assisting with RNA-seq DESeq2 analysis.\n",
    "\n",
    "Below is a metadata table for GEO study {gse_id}. Each row represents a sample.\n",
    "\n",
    "Your task:\n",
    "- Identify the columns that contain key information useful for defining groups for differential expression analysis.\n",
    "- Focus on metadata about: *treatment*, *time*, *gender*, *dose*, *genotype*, *WT/KO*.\n",
    "- Return ONLY a JSON list of useful column names. No explanation.\n",
    "\n",
    "Here is the metadata:\n",
    "{json.dumps(records, indent=2)}\n",
    "\"\"\"\n",
    "    print(f\"📝 Sending LLM prompt (first 500 chars): {prompt[:500]}...\")\n",
    "    response = client.chat.completions.create(\n",
    "        model=\"gpt-4o\",\n",
    "        temperature=0.2,\n",
    "        messages=[{\"role\": \"user\", \"content\": prompt}]\n",
    "    )\n",
    "    content = response.choices[0].message.content.strip()\n",
    "    content = re.sub(r\"^```json\", \"\", content, flags=re.IGNORECASE).strip()\n",
    "    content = re.sub(r\"```$\", \"\", content).strip()\n",
    "    return json.loads(content)\n",
    "\n",
    "def llm_curate_conditions(client, gse_id, metadata_df, batch_size=50):\n",
    "    records = metadata_df.reset_index().to_dict(orient=\"records\")\n",
    "    all_map = {}\n",
    "    for i in range(0, len(records), batch_size):\n",
    "        batch_records = records[i:i+batch_size]\n",
    "        prompt = f\"\"\"\n",
    "You are a biomedical data curator assisting with RNA-seq DESeq2 analysis.\n",
    "\n",
    "Below is a metadata table for GEO study {gse_id}. Each row represents a sample.\n",
    "\n",
    "Your task:\n",
    "- Determine if each sample represents a \"treated\" or \"control\" condition.\n",
    "- Use the metadata fields provided (treatment, time, genotype, etc).\n",
    "- Return ONLY a JSON mapping geo_accession → \"treated\" or \"control\". No explanation.\n",
    "\n",
    "Here is the metadata:\n",
    "{json.dumps(batch_records, indent=2)}\n",
    "\"\"\"\n",
    "        print(f\"📝 Sending LLM prompt batch {i//batch_size+1} (first 500 chars): {prompt[:500]}...\")\n",
    "        response = client.chat.completions.create(\n",
    "            model=\"gpt-4o\",\n",
    "            temperature=0.2,\n",
    "            messages=[{\"role\": \"user\", \"content\": prompt}]\n",
    "        )\n",
    "        content = response.choices[0].message.content.strip()\n",
    "        content = re.sub(r\"^```json\", \"\", content, flags=re.IGNORECASE).strip()\n",
    "        content = re.sub(r\"```$\", \"\", content).strip()\n",
    "        batch_map = json.loads(content)\n",
    "        all_map.update(batch_map)\n",
    "    print(f\"✅ LLM returned conditions for {len(all_map)} samples\")\n",
    "    return all_map\n",
    "\n",
    "def run_deg_for_single_gse_run(state, client):\n",
    "    gse_id = state.get(\"selected_gse\")\n",
    "    if not gse_id:\n",
    "        print(\"❌ No selected GSE found.\")\n",
    "        return state\n",
    "\n",
    "    deg_paths = {}\n",
    "    series_matrix_path = f\"geofetch_metadata/{gse_id}_series_matrix.txt.gz\"\n",
    "    decompress_gse_in_dir(gse_id)\n",
    "\n",
    "    try:\n",
    "        series_df = parse_series_matrix_to_dataframe(series_matrix_path)\n",
    "        mini_df = series_df.sample(n=min(6, len(series_df)), random_state=42)\n",
    "\n",
    "        useful_cols = send_mini_table_to_llm(client, gse_id, mini_df)\n",
    "        metadata_df = series_df[useful_cols].copy()\n",
    "        metadata_df.index.name = \"sample_geo_accession\"\n",
    "\n",
    "        condition_map = llm_curate_conditions(client, gse_id, metadata_df)\n",
    "        metadata_df[\"condition\"] = metadata_df.index.map(condition_map)\n",
    "\n",
    "        download_dir = download_and_extract_gse(gse_id)\n",
    "        counts_df = find_and_merge_count_files(gse_id)\n",
    "        if counts_df is None:\n",
    "            print(f\"⚠️ No count matrix found for {gse_id}\")\n",
    "            return state\n",
    "\n",
    "        valid_samples = metadata_df.index.intersection(counts_df.columns)\n",
    "        counts_df = counts_df[valid_samples].T\n",
    "        metadata_df = metadata_df.loc[valid_samples]\n",
    "        metadata_df = metadata_df[metadata_df[\"condition\"].isin([\"treated\", \"control\"])]\n",
    "\n",
    "        if len(metadata_df[\"condition\"].unique()) < 2:\n",
    "            raise ValueError(\"❌ Need both treated and control groups for DESeq2.\")\n",
    "\n",
    "        print(f\"✅ Counts shape: {counts_df.shape}\")\n",
    "        print(f\"✅ Metadata shape: {metadata_df.shape}\")\n",
    "        print(f\"✅ Condition counts:\\n{metadata_df['condition'].value_counts()}\")\n",
    "\n",
    "        inference = DefaultInference(n_cpus=4)\n",
    "        dds = DeseqDataSet(\n",
    "            counts=counts_df,\n",
    "            metadata=metadata_df,\n",
    "            design_factors=\"condition\",\n",
    "            refit_cooks=True,\n",
    "            inference=inference\n",
    "        )\n",
    "        dds.fit_size_factors()\n",
    "        dds.fit_genewise_dispersions()\n",
    "        dds.fit_dispersion_trend()\n",
    "        dds.fit_dispersion_prior()\n",
    "        dds.fit_MAP_dispersions()\n",
    "        dds.fit_LFC()\n",
    "        dds.calculate_cooks()\n",
    "        if dds.refit_cooks:\n",
    "            dds.refit()\n",
    "\n",
    "        stat_res = DeseqStats(\n",
    "            dds, alpha=0.05, contrast=[\"condition\", \"treated\", \"control\"],\n",
    "            cooks_filter=True, independent_filter=True\n",
    "        )\n",
    "        stat_res.summary()\n",
    "\n",
    "        res = stat_res.results_df\n",
    "        output_csv = f\"rna_seq_analysis/{gse_id}_deseq2_results.csv\"\n",
    "        res.to_csv(output_csv, index=False)\n",
    "        deg_paths[gse_id] = output_csv\n",
    "        print(f\"✅ DESeq2 results saved: {output_csv}\")\n",
    "\n",
    "        sig_genes = res[(res[\"padj\"] < 0.05) & (abs(res[\"log2FoldChange\"]) > 1)]\n",
    "        print(f\"🧬 {sig_genes.shape[0]} significant genes found\")\n",
    "\n",
    "    except Exception as e:\n",
    "        print(f\"❌ Error: {e}\")\n",
    "\n",
    "    return {**state, \"deg_results\": deg_paths}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 240,
   "id": "f3686950-0ebf-4375-9c62-64226c6c1a2a",
   "metadata": {},
   "outputs": [],
   "source": [
    "series_matrix_path = \"geofetch_metadata/GSE296826_series_matrix.txt.gz\"\n",
    "gse_id = \"GSE296826\"\n",
    "sample_labels = [\"GSM8977866\",\"GSM8977867\",\"GSM8977868\",\"GSM8977869\",\"GSM8977870\"]  # Your count matrix labels"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 241,
   "id": "42ee7d4e-5788-472a-981c-137837f37fb4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "🔍 Scanning for compressed files in: rna_seq_analysis\\GSE296826\n",
      "✅ Series matrix parsed: (256, 38)\n",
      "📝 Sending LLM prompt (first 500 chars): \n",
      "You are a biomedical data curator assisting with RNA-seq DESeq2 analysis.\n",
      "\n",
      "Below is a metadata table for GEO study GSE296826. Each row represents a sample.\n",
      "\n",
      "Your task:\n",
      "- Identify the columns that contain key information useful for defining groups for differential expression analysis.\n",
      "- Focus on metadata about: *treatment*, *time*, *gender*, *dose*, *genotype*, *WT/KO*.\n",
      "- Return ONLY a JSON list of useful column names. No explanation.\n",
      "\n",
      "Here is the metadata:\n",
      "[\n",
      "  {\n",
      "    \"geo_accession\": \"GSM8978094...\n",
      "📝 Sending LLM prompt batch 1 (first 500 chars): \n",
      "You are a biomedical data curator assisting with RNA-seq DESeq2 analysis.\n",
      "\n",
      "Below is a metadata table for GEO study GSE296826. Each row represents a sample.\n",
      "\n",
      "Your task:\n",
      "- Determine if each sample represents a \"treated\" or \"control\" condition.\n",
      "- Use the metadata fields provided (treatment, time, genotype, etc).\n",
      "- Return ONLY a JSON mapping geo_accession → \"treated\" or \"control\". No explanation.\n",
      "\n",
      "Here is the metadata:\n",
      "[\n",
      "  {\n",
      "    \"sample_geo_accession\": \"GSM8977866\",\n",
      "    \"Sample_characteristics_ch1_...\n",
      "📝 Sending LLM prompt batch 2 (first 500 chars): \n",
      "You are a biomedical data curator assisting with RNA-seq DESeq2 analysis.\n",
      "\n",
      "Below is a metadata table for GEO study GSE296826. Each row represents a sample.\n",
      "\n",
      "Your task:\n",
      "- Determine if each sample represents a \"treated\" or \"control\" condition.\n",
      "- Use the metadata fields provided (treatment, time, genotype, etc).\n",
      "- Return ONLY a JSON mapping geo_accession → \"treated\" or \"control\". No explanation.\n",
      "\n",
      "Here is the metadata:\n",
      "[\n",
      "  {\n",
      "    \"sample_geo_accession\": \"GSM8977916\",\n",
      "    \"Sample_characteristics_ch1_...\n",
      "📝 Sending LLM prompt batch 3 (first 500 chars): \n",
      "You are a biomedical data curator assisting with RNA-seq DESeq2 analysis.\n",
      "\n",
      "Below is a metadata table for GEO study GSE296826. Each row represents a sample.\n",
      "\n",
      "Your task:\n",
      "- Determine if each sample represents a \"treated\" or \"control\" condition.\n",
      "- Use the metadata fields provided (treatment, time, genotype, etc).\n",
      "- Return ONLY a JSON mapping geo_accession → \"treated\" or \"control\". No explanation.\n",
      "\n",
      "Here is the metadata:\n",
      "[\n",
      "  {\n",
      "    \"sample_geo_accession\": \"GSM8977966\",\n",
      "    \"Sample_characteristics_ch1_...\n",
      "📝 Sending LLM prompt batch 4 (first 500 chars): \n",
      "You are a biomedical data curator assisting with RNA-seq DESeq2 analysis.\n",
      "\n",
      "Below is a metadata table for GEO study GSE296826. Each row represents a sample.\n",
      "\n",
      "Your task:\n",
      "- Determine if each sample represents a \"treated\" or \"control\" condition.\n",
      "- Use the metadata fields provided (treatment, time, genotype, etc).\n",
      "- Return ONLY a JSON mapping geo_accession → \"treated\" or \"control\". No explanation.\n",
      "\n",
      "Here is the metadata:\n",
      "[\n",
      "  {\n",
      "    \"sample_geo_accession\": \"GSM8978016\",\n",
      "    \"Sample_characteristics_ch1_...\n",
      "📝 Sending LLM prompt batch 5 (first 500 chars): \n",
      "You are a biomedical data curator assisting with RNA-seq DESeq2 analysis.\n",
      "\n",
      "Below is a metadata table for GEO study GSE296826. Each row represents a sample.\n",
      "\n",
      "Your task:\n",
      "- Determine if each sample represents a \"treated\" or \"control\" condition.\n",
      "- Use the metadata fields provided (treatment, time, genotype, etc).\n",
      "- Return ONLY a JSON mapping geo_accession → \"treated\" or \"control\". No explanation.\n",
      "\n",
      "Here is the metadata:\n",
      "[\n",
      "  {\n",
      "    \"sample_geo_accession\": \"GSM8978066\",\n",
      "    \"Sample_characteristics_ch1_...\n",
      "📝 Sending LLM prompt batch 6 (first 500 chars): \n",
      "You are a biomedical data curator assisting with RNA-seq DESeq2 analysis.\n",
      "\n",
      "Below is a metadata table for GEO study GSE296826. Each row represents a sample.\n",
      "\n",
      "Your task:\n",
      "- Determine if each sample represents a \"treated\" or \"control\" condition.\n",
      "- Use the metadata fields provided (treatment, time, genotype, etc).\n",
      "- Return ONLY a JSON mapping geo_accession → \"treated\" or \"control\". No explanation.\n",
      "\n",
      "Here is the metadata:\n",
      "[\n",
      "  {\n",
      "    \"sample_geo_accession\": \"GSM8978116\",\n",
      "    \"Sample_characteristics_ch1_...\n",
      "✅ LLM returned conditions for 256 samples\n",
      "📥 Downloading GSE296826_RAW.tar → rna_seq_analysis/GSE296826/GSE296826_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "⚠️ Skipping compressed duplicate: GSM8977866_Receptor_00000000000111_BCR_122-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977867_Receptor_00000000000112_BCR_122-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977868_Receptor_00000000000113_BCR_125-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977869_Receptor_00000000000114_BCR_125-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977870_Receptor_00000000000115_BCR_126-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977871_Receptor_00000000000116_BCR_126-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977872_Receptor_00000000000117_BCR_139-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977873_Receptor_00000000000118_BCR_139-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977874_Receptor_00000000000119_BCR_141-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977875_Receptor_00000000000120_BCR_141-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977876_Receptor_00000000000121_BCR_143-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977877_Receptor_00000000000122_BCR_143-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977878_Receptor_00000000000123_BCR_153-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977879_Receptor_00000000000124_BCR_153-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977880_Receptor_00000000000125_BCR_155-12M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977881_Receptor_00000000000126_BCR_155-6M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977882_Receptor_00000000000127_BCR_155-9M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977883_Receptor_00000000000128_BCR_155-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977884_Receptor_00000000000129_BCR_174-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977885_Receptor_00000000000130_BCR_174-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977886_Receptor_00000000000131_BCR_191-4W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977887_Receptor_00000000000132_BCR_191-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977888_Receptor_00000000000133_BCR_195-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977889_Receptor_00000000000134_BCR_195-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977890_Receptor_00000000000135_BCR_196-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977891_Receptor_00000000000136_BCR_197-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977892_Receptor_00000000000137_BCR_197-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977893_Receptor_00000000000138_BCR_201-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977894_Receptor_00000000000139_BCR_201-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977895_Receptor_00000000000140_BCR_202-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977896_Receptor_00000000000141_BCR_205-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977897_Receptor_00000000000142_BCR_212-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977898_Receptor_00000000000143_BCR_212-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977899_Receptor_00000000000144_BCR_213-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977900_Receptor_00000000000145_BCR_213-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977901_Receptor_00000000000146_BCR_215-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977902_Receptor_00000000000147_BCR_216-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977903_Receptor_00000000000148_BCR_216-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977904_Receptor_00000000000150_BCR_217-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977905_Receptor_00000000000149_BCR_217-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977906_Receptor_00000000000151_BCR_220-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977907_Receptor_00000000000152_BCR_220-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977908_Receptor_00000000000153_BCR_222-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977909_Receptor_00000000000154_BCR_222-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977910_Receptor_00000000000155_BCR_223-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977911_Receptor_00000000000156_BCR_223-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977912_Receptor_00000000000157_BCR_224-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977913_Receptor_00000000000158_BCR_228-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977914_Receptor_00000000000159_BCR_228-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977915_Receptor_00000000000160_BCR_233-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977916_Receptor_00000000000161_BCR_235-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977917_Receptor_00000000000162_BCR_235-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977918_Receptor_00000000000163_BCR_242-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977919_Receptor_00000000000164_BCR_242-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977920_Receptor_00000000000165_BCR_248-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977921_Receptor_00000000000166_BCR_248-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977922_Receptor_00000000000167_BCR_252-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977923_Receptor_00000000000168_BCR_252-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977924_Receptor_00000000000169_BCR_254-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977925_Receptor_00000000000170_BCR_254-6M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977926_Receptor_00000000000171_BCR_254-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977927_Receptor_00000000000172_BCR_254-9M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977928_Receptor_00000000000173_BCR_254-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977929_Receptor_00000000000174_BCR_259-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977930_Receptor_00000000000175_BCR_259-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977931_Receptor_00000000000176_BCR_262-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977932_Receptor_00000000000177_BCR_262-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977933_Receptor_00000000000178_BCR_263-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977934_Receptor_00000000000179_BCR_263-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977935_Receptor_00000000000180_BCR_264-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977936_Receptor_00000000000181_BCR_264-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977937_Receptor_00000000000182_BCR_266-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977938_Receptor_00000000000183_BCR_270-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977939_Receptor_00000000000184_BCR_273-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977940_Receptor_00000000000185_BCR_273-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977941_Receptor_00000000000186_BCR_274-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977942_Receptor_00000000000187_BCR_274-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977943_Receptor_00000000000188_BCR_277-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977944_Receptor_00000000000189_BCR_277-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977945_Receptor_00000000000190_BCR_278-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977946_Receptor_00000000000191_BCR_278-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977947_Receptor_00000000000192_BCR_279-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977948_Receptor_00000000000193_BCR_279-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977949_Receptor_00000000000194_BCR_284-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977950_Receptor_00000000000195_BCR_284-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977951_Receptor_00000000000196_BCR_289-4W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977952_Receptor_00000000000197_BCR_289-6M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977953_Receptor_00000000000198_BCR_289-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977954_Receptor_00000000000199_BCR_290-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977955_Receptor_00000000000200_BCR_290-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977956_Receptor_00000000000201_BCR_295-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977957_Receptor_00000000000202_BCR_302-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977958_Receptor_00000000000203_BCR_302-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977959_Receptor_00000000000204_BCR_304-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977960_Receptor_00000000000205_BCR_304-6M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977961_Receptor_00000000000206_BCR_304-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977962_Receptor_00000000000207_BCR_305-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977963_Receptor_00000000000208_BCR_305-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977964_Receptor_00000000000209_BCR_306-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977965_Receptor_00000000000210_BCR_306-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977966_Receptor_00000000000211_BCR_313-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977967_Receptor_00000000000212_BCR_313-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977968_Receptor_00000000000213_BCR_317-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977969_Receptor_00000000000214_BCR_317-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977970_Receptor_00000000000215_BCR_319-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977971_Receptor_00000000000216_BCR_319-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977972_Receptor_00000000000217_BCR_323-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977973_Receptor_00000000000218_BCR_323-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977974_Receptor_00000000000219_BCR_325-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977975_Receptor_00000000000220_BCR_325-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977976_Receptor_00000000000221_BCR_122-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977977_Receptor_00000000000222_BCR_125-6-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977978_Receptor_00000000000223_BCR_126-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977979_Receptor_00000000000224_BCR_153-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977980_Receptor_00000000000225_BCR_155-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977981_Receptor_00000000000226_BCR_196-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977982_Receptor_00000000000227_BCR_202-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977983_Receptor_00000000000228_BCR_203-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977984_Receptor_00000000000229_BCR_203-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977985_Receptor_00000000000230_BCR_205-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977986_Receptor_00000000000231_BCR_213-8M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977987_Receptor_00000000000232_BCR_215-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977988_Receptor_00000000000233_BCR_216-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977989_Receptor_00000000000234_BCR_223-TOX.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977990_Receptor_00000000000235_BCR_224-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977991_Receptor_00000000000236_BCR_233-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977992_Receptor_00000000000237_BCR_252-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977993_Receptor_00000000000238_BCR_254-27W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977994_Receptor_00000000000239_BCR_266-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977995_Receptor_00000000000240_BCR_270-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977996_Receptor_00000000000241_BCR_279-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977997_Receptor_00000000000242_BCR_290-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977998_Receptor_00000000000243_BCR_295-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977999_Receptor_00000000000244_BCR_304-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978000_Receptor_00000000000245_BCR_323-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978001_Receptor_00000000000246_BCR_140-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978002_Receptor_00000000000247_BCR_140-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978003_Receptor_00000000000248_BCR_146-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978004_Receptor_00000000000249_BCR_146-6M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978005_Receptor_00000000000250_BCR_146-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978006_Receptor_00000000000251_BCR_146-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978007_Receptor_00000000000252_BCR_167-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978008_Receptor_00000000000253_BCR_167-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978009_Receptor_00000000000254_BCR_170-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978010_Receptor_00000000000255_BCR_170-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978011_Receptor_00000000000256_BCR_172-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978012_Receptor_00000000000257_BCR_172-Bl.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978013_Receptor_00000000000258_BCR_173-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978014_Receptor_00000000000259_BCR_173-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978015_Receptor_00000000000260_BCR_181-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978016_Receptor_00000000000261_BCR_181-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978017_Receptor_00000000000262_BCR_185-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978018_Receptor_00000000000263_BCR_185-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978019_Receptor_00000000000264_BCR_186-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978020_Receptor_00000000000265_BCR_186-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978021_Receptor_00000000000266_BCR_186-9M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978022_Receptor_00000000000267_BCR_186-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978023_Receptor_00000000000268_BCR_189-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978024_Receptor_00000000000269_BCR_189-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978025_Receptor_00000000000270_BCR_192-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978026_Receptor_00000000000271_BCR_192-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978027_Receptor_00000000000272_BCR_193-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978028_Receptor_00000000000273_BCR_193-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978029_Receptor_00000000000274_BCR_208-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978030_Receptor_00000000000275_BCR_208-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978031_Receptor_00000000000276_BCR_209-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978032_Receptor_00000000000277_BCR_209-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978033_Receptor_00000000000278_BCR_211-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978034_Receptor_00000000000279_BCR_211-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978035_Receptor_00000000000280_BCR_214-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978036_Receptor_00000000000281_BCR_214-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978037_Receptor_00000000000282_BCR_218-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978038_Receptor_00000000000283_BCR_218-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978039_Receptor_00000000000284_BCR_221-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978040_Receptor_00000000000285_BCR_221-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978041_Receptor_00000000000286_BCR_225-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978042_Receptor_00000000000287_BCR_225-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978043_Receptor_00000000000288_BCR_227-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978044_Receptor_00000000000289_BCR_227-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978045_Receptor_00000000000290_BCR_230-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978046_Receptor_00000000000291_BCR_230-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978047_Receptor_00000000000292_BCR_234-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978048_Receptor_00000000000293_BCR_234-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978049_Receptor_00000000000294_BCR_237-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978050_Receptor_00000000000295_BCR_237-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978051_Receptor_00000000000296_BCR_244-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978052_Receptor_00000000000297_BCR_244-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978053_Receptor_00000000000298_BCR_245-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978054_Receptor_00000000000299_BCR_245-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978055_Receptor_00000000000300_BCR_249-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978056_Receptor_00000000000301_BCR_249-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978057_Receptor_00000000000302_BCR_257-7W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978058_Receptor_00000000000303_BCR_257-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978059_Receptor_00000000000304_BCR_260-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978060_Receptor_00000000000305_BCR_260-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978061_Receptor_00000000000306_BCR_261-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978062_Receptor_00000000000307_BCR_261-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978063_Receptor_00000000000308_BCR_269-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978064_Receptor_00000000000309_BCR_269-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978065_Receptor_00000000000310_BCR_283-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978066_Receptor_00000000000311_BCR_283-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978067_Receptor_00000000000312_BCR_285-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978068_Receptor_00000000000313_BCR_285-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978069_Receptor_00000000000314_BCR_285-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978070_Receptor_00000000000315_BCR_293-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978071_Receptor_00000000000316_BCR_293-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978072_Receptor_00000000000317_BCR_296-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978073_Receptor_00000000000318_BCR_296-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978074_Receptor_00000000000319_BCR_298-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978075_Receptor_00000000000320_BCR_298-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978076_Receptor_00000000000321_BCR_299-10W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978077_Receptor_00000000000322_BCR_299-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978078_Receptor_00000000000323_BCR_299-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978079_Receptor_00000000000324_BCR_299-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978080_Receptor_00000000000325_BCR_299-4W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978081_Receptor_00000000000326_BCR_299-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978082_Receptor_00000000000327_BCR_310-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978083_Receptor_00000000000328_BCR_310-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978084_Receptor_00000000000329_BCR_320-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978085_Receptor_00000000000330_BCR_320-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978086_Receptor_00000000000331_BCR_326-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978087_Receptor_00000000000332_BCR_326-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978088_Receptor_00000000000333_BCR_327-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978089_Receptor_00000000000334_BCR_327-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978090_Receptor_00000000000335_BCR_328-4W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978091_Receptor_00000000000336_BCR_328-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978092_Receptor_00000000000337_BCR_329-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978093_Receptor_00000000000338_BCR_329-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978094_Receptor_00000000000339_BCR_334-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978095_Receptor_00000000000340_BCR_334-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978096_Receptor_00000000000341_BCR_336-4W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978097_Receptor_00000000000342_BCR_336-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978098_Receptor_00000000000343_BCR_338-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978099_Receptor_00000000000344_BCR_338-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978100_Receptor_00000000000345_BCR_346-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978101_Receptor_00000000000346_BCR_346-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978102_Receptor_00000000000347_BCR_350-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978103_Receptor_00000000000348_BCR_350-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978104_Receptor_00000000000349_BCR_351-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978105_Receptor_00000000000350_BCR_351-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978106_Receptor_00000000000351_BCR_352-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978107_Receptor_00000000000352_BCR_352-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978108_Receptor_00000000000353_BCR_358-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978109_Receptor_00000000000354_BCR_358-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978110_Receptor_00000000000355_BCR_360-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978111_Receptor_00000000000356_BCR_360-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978112_Receptor_00000000000357_BCR_386-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978113_Receptor_00000000000358_BCR_386-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978114_Receptor_00000000000359_BCR_60-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978115_Receptor_00000000000360_BCR_60-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978116_Receptor_00000000000361_BCR_74-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978117_Receptor_00000000000362_BCR_74-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978118_Receptor_00000000000363_BCR_90-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978119_Receptor_00000000000364_BCR_90-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978120_Receptor_00000000000365_BCR_92-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978121_Receptor_00000000000366_BCR_92-BL.tsv.gz\n",
      "🔍 Found 256 count files for GSE296826\n",
      "⚠️ Zero counts detected in merged matrix — applying +1 normalization.\n",
      "✅ Final merged count matrix shape: (24901, 256)\n",
      "✅ Counts shape: (256, 24901)\n",
      "✅ Metadata shape: (256, 3)\n",
      "✅ Condition counts:\n",
      "condition\n",
      "treated    143\n",
      "control    113\n",
      "Name: count, dtype: int64\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\difen\\venv1\\Lib\\site-packages\\anndata\\_core\\aligned_df.py:68: ImplicitModificationWarning: Transforming to str index.\n",
      "  warnings.warn(\"Transforming to str index.\", ImplicitModificationWarning)\n",
      "C:\\Users\\difen\\AppData\\Local\\Temp\\ipykernel_38836\\1630641471.py:136: DeprecationWarning: design_factors is deprecated and will soon be removed.Please consider providing a formulaic formula using the design argumentinstead.\n",
      "  dds = DeseqDataSet(\n",
      "Fitting size factors...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "... done in 0.24 seconds.\n",
      "\n",
      "Fitting dispersions...\n",
      "... done in 2.18 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "C:\\Users\\difen\\venv1\\Lib\\site-packages\\pydeseq2\\dds.py:805: UserWarning: The dispersion trend curve fitting did not converge. Switching to a mean-based dispersion trend.\n",
      "  self._fit_parametric_dispersion_trend(vst)\n",
      "... done in 0.31 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 9.06 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 3.27 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.38 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 3.68 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: condition treated vs control\n",
      "         baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "Gene                                                                     \n",
      "0      103.969669        0.079685  0.345011  0.230965  0.817342  0.999662\n",
      "1       67.352032       -0.195212  0.316313 -0.617148  0.537137  0.999662\n",
      "2       53.945877       -0.111967  0.295605 -0.378773  0.704856  0.999662\n",
      "3       47.332183       -0.487050  0.287178 -1.695986  0.089888  0.999662\n",
      "4       42.383649        0.009939  0.281012  0.035368  0.971787  0.999662\n",
      "...           ...             ...       ...       ...       ...       ...\n",
      "25406    1.000099        0.051398  0.184873  0.278019  0.780998  0.999662\n",
      "25407    1.000099        0.051398  0.184873  0.278019  0.780998  0.999662\n",
      "25408    1.000099        0.051398  0.184873  0.278019  0.780998  0.999662\n",
      "25409    1.000099        0.051398  0.184873  0.278019  0.780998  0.999662\n",
      "25410    1.000099        0.051398  0.184873  0.278019  0.780998  0.999662\n",
      "\n",
      "[24901 rows x 6 columns]\n",
      "✅ DESeq2 results saved: rna_seq_analysis/GSE296826_deseq2_results.csv\n",
      "🧬 0 significant genes found\n"
     ]
    }
   ],
   "source": [
    "state = {\n",
    "    \"selected_gse\": \"GSE296826\"\n",
    "}\n",
    "result = run_deg_for_single_gse_run(state, client)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4e4478f9-2166-49fd-acef-8618ec995ca2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "161353fd-e233-4309-b868-a96e4f28cc83",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "id": "c9a7b7cd-2c7d-42cd-af04-b1abe364f099",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📂 Parsing series matrix: geofetch_metadata/GSE296826_series_matrix.txt.gz\n",
      "✅ Parsed 256 samples\n",
      "🧠 Building prompt for batch 1 with 100 samples...\n",
      "📝 Prompt preview (first 500 chars): \n",
      "You are a biomedical data curator.\n",
      "\n",
      "Below is a list of samples from GEO study GSE296826. Each sample has a unique 'row_id'.\n",
      "Each contains metadata fields like geo_accession, title, description, characteristics.\n",
      "\n",
      "Your task:\n",
      "- For each count matrix label, identify the matching sample (if any) and return its row_id.\n",
      "- Also return:\n",
      "  - geo_accession\n",
      "  - condition: treated or control\n",
      "  - time: if available\n",
      "\n",
      "Return a JSON object where each key is the count matrix label, and the value is:\n",
      "{\"row_id\": ....\n",
      "💬 Raw LLM output length: 12705\n",
      "❌ JSON decode error. Raw content preview:\n",
      "```json\n",
      "{\n",
      "  \"GSM8977866\": {\n",
      "    \"row_id\": \"row_0\",\n",
      "    \"geo_accession\": \"GSM8977866\",\n",
      "    \"condition\": \"treated\",\n",
      "    \"time\": \"6W\"\n",
      "  },\n",
      "  \"GSM8977867\": {\n",
      "    \"row_id\": \"row_1\",\n",
      "    \"geo_accession\": \"GSM8977867\",\n",
      "    \"condition\": \"control\",\n",
      "    \"time\": \"BL\"\n",
      "  },\n",
      "  \"GSM8977868\": {\n",
      "    \"row_id\": \"row_2\",\n",
      "    \"geo_accession\": \"GSM8977868\",\n",
      "    \"condition\": \"treated\",\n",
      "    \"time\": \"3M\"\n",
      "  },\n",
      "  \"GSM8977869\": {\n",
      "    \"row_id\": \"row_3\",\n",
      "    \"geo_accession\": \"GSM8977869\",\n",
      "    \"condition\": \"control\",\n",
      "    \"tim\n"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "Could not parse LLM JSON output.",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mJSONDecodeError\u001b[0m                           Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[177], line 64\u001b[0m, in \u001b[0;36mextract_labels_using_llm\u001b[1;34m(series_matrix_path, gse_id, client, sample_labels_from_counts, batch_size)\u001b[0m\n\u001b[0;32m     63\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m---> 64\u001b[0m     batch_metadata \u001b[38;5;241m=\u001b[39m \u001b[43mjson\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mloads\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcontent\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m     65\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m json\u001b[38;5;241m.\u001b[39mJSONDecodeError \u001b[38;5;28;01mas\u001b[39;00m e:\n",
      "File \u001b[1;32m~\\.pyenv\\pyenv-win\\versions\\3.11.9\\Lib\\json\\__init__.py:346\u001b[0m, in \u001b[0;36mloads\u001b[1;34m(s, cls, object_hook, parse_float, parse_int, parse_constant, object_pairs_hook, **kw)\u001b[0m\n\u001b[0;32m    343\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m (\u001b[38;5;28mcls\u001b[39m \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;129;01mand\u001b[39;00m object_hook \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;129;01mand\u001b[39;00m\n\u001b[0;32m    344\u001b[0m         parse_int \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;129;01mand\u001b[39;00m parse_float \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;129;01mand\u001b[39;00m\n\u001b[0;32m    345\u001b[0m         parse_constant \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;129;01mand\u001b[39;00m object_pairs_hook \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m kw):\n\u001b[1;32m--> 346\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43m_default_decoder\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdecode\u001b[49m\u001b[43m(\u001b[49m\u001b[43ms\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    347\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mcls\u001b[39m \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n",
      "File \u001b[1;32m~\\.pyenv\\pyenv-win\\versions\\3.11.9\\Lib\\json\\decoder.py:337\u001b[0m, in \u001b[0;36mJSONDecoder.decode\u001b[1;34m(self, s, _w)\u001b[0m\n\u001b[0;32m    333\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124;03m\"\"\"Return the Python representation of ``s`` (a ``str`` instance\u001b[39;00m\n\u001b[0;32m    334\u001b[0m \u001b[38;5;124;03mcontaining a JSON document).\u001b[39;00m\n\u001b[0;32m    335\u001b[0m \n\u001b[0;32m    336\u001b[0m \u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m--> 337\u001b[0m obj, end \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mraw_decode\u001b[49m\u001b[43m(\u001b[49m\u001b[43ms\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43midx\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m_w\u001b[49m\u001b[43m(\u001b[49m\u001b[43ms\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m0\u001b[39;49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mend\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    338\u001b[0m end \u001b[38;5;241m=\u001b[39m _w(s, end)\u001b[38;5;241m.\u001b[39mend()\n",
      "File \u001b[1;32m~\\.pyenv\\pyenv-win\\versions\\3.11.9\\Lib\\json\\decoder.py:355\u001b[0m, in \u001b[0;36mJSONDecoder.raw_decode\u001b[1;34m(self, s, idx)\u001b[0m\n\u001b[0;32m    354\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mStopIteration\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m err:\n\u001b[1;32m--> 355\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m JSONDecodeError(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mExpecting value\u001b[39m\u001b[38;5;124m\"\u001b[39m, s, err\u001b[38;5;241m.\u001b[39mvalue) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m    356\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m obj, end\n",
      "\u001b[1;31mJSONDecodeError\u001b[0m: Expecting value: line 1 column 1 (char 0)",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[178], line 1\u001b[0m\n\u001b[1;32m----> 1\u001b[0m all_metadata, sample_table \u001b[38;5;241m=\u001b[39m \u001b[43mextract_labels_using_llm\u001b[49m\u001b[43m(\u001b[49m\n\u001b[0;32m      2\u001b[0m \u001b[43m    \u001b[49m\u001b[43mseries_matrix_path\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mgeofetch_metadata/GSE296826_series_matrix.txt.gz\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[0;32m      3\u001b[0m \u001b[43m    \u001b[49m\u001b[43mgse_id\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mGSE296826\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[0;32m      4\u001b[0m \u001b[43m    \u001b[49m\u001b[43mclient\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mclient\u001b[49m\u001b[43m,\u001b[49m\u001b[43m  \u001b[49m\u001b[38;5;66;43;03m# your OpenAI or similar client\u001b[39;49;00m\n\u001b[0;32m      5\u001b[0m \u001b[43m    \u001b[49m\u001b[43msample_labels_from_counts\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcdf\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcolumns\u001b[49m\u001b[43m,\u001b[49m\u001b[43m  \u001b[49m\u001b[38;5;66;43;03m# or list of your sample labels\u001b[39;49;00m\n\u001b[0;32m      6\u001b[0m \u001b[43m    \u001b[49m\u001b[43mbatch_size\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m100\u001b[39;49m\n\u001b[0;32m      7\u001b[0m \u001b[43m)\u001b[49m\n",
      "Cell \u001b[1;32mIn[177], line 68\u001b[0m, in \u001b[0;36mextract_labels_using_llm\u001b[1;34m(series_matrix_path, gse_id, client, sample_labels_from_counts, batch_size)\u001b[0m\n\u001b[0;32m     66\u001b[0m     \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m❌ JSON decode error. Raw content preview:\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m     67\u001b[0m     \u001b[38;5;28mprint\u001b[39m(content[:\u001b[38;5;241m500\u001b[39m])\n\u001b[1;32m---> 68\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mCould not parse LLM JSON output.\u001b[39m\u001b[38;5;124m\"\u001b[39m) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01me\u001b[39;00m\n\u001b[0;32m     70\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m✅ Parsed JSON batch metadata for \u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[38;5;28mlen\u001b[39m(batch_metadata)\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m sample labels\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m     71\u001b[0m all_metadata\u001b[38;5;241m.\u001b[39mupdate(batch_metadata)\n",
      "\u001b[1;31mValueError\u001b[0m: Could not parse LLM JSON output."
     ]
    }
   ],
   "source": [
    "all_metadata, sample_table = extract_labels_using_llm(\n",
    "    series_matrix_path=\"geofetch_metadata/GSE296826_series_matrix.txt.gz\",\n",
    "    gse_id=\"GSE296826\",\n",
    "    client=client,  # your OpenAI or similar client\n",
    "    sample_labels_from_counts=cdf.columns,  # or list of your sample labels\n",
    "    batch_size=100\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "14e937c7-ba8c-4ffb-8318-1321cd15afbe",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "id": "0e628242-3243-4e2a-88f1-2a40b244491b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GSM8977866</th>\n",
       "      <th>GSM8977867</th>\n",
       "      <th>GSM8977868</th>\n",
       "      <th>GSM8977869</th>\n",
       "      <th>GSM8977870</th>\n",
       "      <th>GSM8977871</th>\n",
       "      <th>GSM8977872</th>\n",
       "      <th>GSM8977873</th>\n",
       "      <th>GSM8977874</th>\n",
       "      <th>GSM8977875</th>\n",
       "      <th>...</th>\n",
       "      <th>GSM8978112</th>\n",
       "      <th>GSM8978113</th>\n",
       "      <th>GSM8978114</th>\n",
       "      <th>GSM8978115</th>\n",
       "      <th>GSM8978116</th>\n",
       "      <th>GSM8978117</th>\n",
       "      <th>GSM8978118</th>\n",
       "      <th>GSM8978119</th>\n",
       "      <th>GSM8978120</th>\n",
       "      <th>GSM8978121</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>376</td>\n",
       "      <td>61</td>\n",
       "      <td>111</td>\n",
       "      <td>208</td>\n",
       "      <td>267</td>\n",
       "      <td>121</td>\n",
       "      <td>1</td>\n",
       "      <td>105</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>85</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>286</td>\n",
       "      <td>229</td>\n",
       "      <td>123</td>\n",
       "      <td>271</td>\n",
       "      <td>610</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>208</td>\n",
       "      <td>49</td>\n",
       "      <td>107</td>\n",
       "      <td>173</td>\n",
       "      <td>246</td>\n",
       "      <td>68</td>\n",
       "      <td>1</td>\n",
       "      <td>86</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>83</td>\n",
       "      <td>145</td>\n",
       "      <td>1</td>\n",
       "      <td>262</td>\n",
       "      <td>180</td>\n",
       "      <td>99</td>\n",
       "      <td>214</td>\n",
       "      <td>383</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>181</td>\n",
       "      <td>46</td>\n",
       "      <td>97</td>\n",
       "      <td>121</td>\n",
       "      <td>195</td>\n",
       "      <td>41</td>\n",
       "      <td>1</td>\n",
       "      <td>77</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>53</td>\n",
       "      <td>127</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>134</td>\n",
       "      <td>94</td>\n",
       "      <td>136</td>\n",
       "      <td>221</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>60</td>\n",
       "      <td>120</td>\n",
       "      <td>46</td>\n",
       "      <td>1</td>\n",
       "      <td>96</td>\n",
       "      <td>180</td>\n",
       "      <td>38</td>\n",
       "      <td>1</td>\n",
       "      <td>34</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>108</td>\n",
       "      <td>1</td>\n",
       "      <td>101</td>\n",
       "      <td>128</td>\n",
       "      <td>83</td>\n",
       "      <td>126</td>\n",
       "      <td>128</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>58</td>\n",
       "      <td>82</td>\n",
       "      <td>40</td>\n",
       "      <td>62</td>\n",
       "      <td>86</td>\n",
       "      <td>113</td>\n",
       "      <td>28</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>27</td>\n",
       "      <td>47</td>\n",
       "      <td>103</td>\n",
       "      <td>1</td>\n",
       "      <td>96</td>\n",
       "      <td>120</td>\n",
       "      <td>62</td>\n",
       "      <td>109</td>\n",
       "      <td>125</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25406</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25407</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25408</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25409</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25410</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>24901 rows × 256 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       GSM8977866  GSM8977867  GSM8977868  GSM8977869  GSM8977870  GSM8977871  \\\n",
       "Gene                                                                            \n",
       "0               1           1         376          61         111         208   \n",
       "1               1           1         208          49         107         173   \n",
       "2               1           1         181          46          97         121   \n",
       "3               1          60         120          46           1          96   \n",
       "4               1          58          82          40          62          86   \n",
       "...           ...         ...         ...         ...         ...         ...   \n",
       "25406           1           1           1           1           1           1   \n",
       "25407           1           1           1           1           1           1   \n",
       "25408           1           1           1           1           1           1   \n",
       "25409           1           1           1           1           1           1   \n",
       "25410           1           1           1           1           1           1   \n",
       "\n",
       "       GSM8977872  GSM8977873  GSM8977874  GSM8977875  ...  GSM8978112  \\\n",
       "Gene                                                   ...               \n",
       "0             267         121           1         105  ...           1   \n",
       "1             246          68           1          86  ...           1   \n",
       "2             195          41           1          77  ...           1   \n",
       "3             180          38           1          34  ...           1   \n",
       "4             113          28           1           1  ...          27   \n",
       "...           ...         ...         ...         ...  ...         ...   \n",
       "25406           1           1           1           1  ...           1   \n",
       "25407           1           1           1           1  ...           1   \n",
       "25408           1           1           1           1  ...           1   \n",
       "25409           1           1           1           1  ...           1   \n",
       "25410           1           1           1           1  ...           1   \n",
       "\n",
       "       GSM8978113  GSM8978114  GSM8978115  GSM8978116  GSM8978117  GSM8978118  \\\n",
       "Gene                                                                            \n",
       "0              85           1           1         286         229         123   \n",
       "1              83         145           1         262         180          99   \n",
       "2              53         127           1           1         134          94   \n",
       "3               1         108           1         101         128          83   \n",
       "4              47         103           1          96         120          62   \n",
       "...           ...         ...         ...         ...         ...         ...   \n",
       "25406           1           1           1           1           1           1   \n",
       "25407           1           1           1           1           1           1   \n",
       "25408           1           1           1           1           1           1   \n",
       "25409           1           1           1           1           1           1   \n",
       "25410           1           1           1           1           1           1   \n",
       "\n",
       "       GSM8978119  GSM8978120  GSM8978121  \n",
       "Gene                                       \n",
       "0             271         610          58  \n",
       "1             214         383          47  \n",
       "2             136         221          39  \n",
       "3             126         128          38  \n",
       "4             109         125          35  \n",
       "...           ...         ...         ...  \n",
       "25406           1           1           1  \n",
       "25407           1           1           1  \n",
       "25408           1           1           1  \n",
       "25409           1           1           1  \n",
       "25410           1           1           1  \n",
       "\n",
       "[24901 rows x 256 columns]"
      ]
     },
     "execution_count": 176,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9b118ca3-24c1-4c25-b0f8-87037ad4b9f4",
   "metadata": {},
   "outputs": [],
   "source": [
    "#selected_gse = \"GSE259434\"\n",
    "selected_gse = \"GSE296826\" \n",
    "initial_state: Dict = {\n",
    "    \"query_gene\": \"\",\n",
    "    \"pathway_info\": {\"genes\": [], \"drugs\": [], \"pathways\": []},\n",
    "    \"metadata\": {},\n",
    "    \"gse_list\": [],\n",
    "    \"research_plan\": \"\",\n",
    "    \"selected_gse\": selected_gse,\n",
    "    \"deg_results\": {}\n",
    "}\n",
    "\n",
    "# Inject the path dynamically\n",
    "initial_state[\"series_matrix_path\"] = f\"geofetch_metadata/{selected_gse}_series_matrix.txt.gz\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ed9a6eef-6e34-4451-9ded-dfba4f0b3bf9",
   "metadata": {},
   "outputs": [],
   "source": [
    "initial_state"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "8fa581c3-9b79-4976-9f8c-b45684927bd2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "🔍 Scanning for compressed files in: rna_seq_analysis\\GSE296826\n",
      "Running DESeq2 for: GSE296826\n",
      "📥 Downloading GSE296826_RAW.tar → rna_seq_analysis/GSE296826/GSE296826_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "⚠️ Skipping compressed duplicate: GSM8977866_Receptor_00000000000111_BCR_122-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977867_Receptor_00000000000112_BCR_122-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977868_Receptor_00000000000113_BCR_125-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977869_Receptor_00000000000114_BCR_125-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977870_Receptor_00000000000115_BCR_126-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977871_Receptor_00000000000116_BCR_126-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977872_Receptor_00000000000117_BCR_139-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977873_Receptor_00000000000118_BCR_139-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977874_Receptor_00000000000119_BCR_141-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977875_Receptor_00000000000120_BCR_141-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977876_Receptor_00000000000121_BCR_143-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977877_Receptor_00000000000122_BCR_143-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977878_Receptor_00000000000123_BCR_153-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977879_Receptor_00000000000124_BCR_153-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977880_Receptor_00000000000125_BCR_155-12M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977881_Receptor_00000000000126_BCR_155-6M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977882_Receptor_00000000000127_BCR_155-9M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977883_Receptor_00000000000128_BCR_155-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977884_Receptor_00000000000129_BCR_174-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977885_Receptor_00000000000130_BCR_174-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977886_Receptor_00000000000131_BCR_191-4W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977887_Receptor_00000000000132_BCR_191-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977888_Receptor_00000000000133_BCR_195-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977889_Receptor_00000000000134_BCR_195-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977890_Receptor_00000000000135_BCR_196-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977891_Receptor_00000000000136_BCR_197-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977892_Receptor_00000000000137_BCR_197-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977893_Receptor_00000000000138_BCR_201-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977894_Receptor_00000000000139_BCR_201-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977895_Receptor_00000000000140_BCR_202-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977896_Receptor_00000000000141_BCR_205-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977897_Receptor_00000000000142_BCR_212-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977898_Receptor_00000000000143_BCR_212-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977899_Receptor_00000000000144_BCR_213-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977900_Receptor_00000000000145_BCR_213-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977901_Receptor_00000000000146_BCR_215-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977902_Receptor_00000000000147_BCR_216-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977903_Receptor_00000000000148_BCR_216-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977904_Receptor_00000000000150_BCR_217-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977905_Receptor_00000000000149_BCR_217-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977906_Receptor_00000000000151_BCR_220-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977907_Receptor_00000000000152_BCR_220-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977908_Receptor_00000000000153_BCR_222-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977909_Receptor_00000000000154_BCR_222-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977910_Receptor_00000000000155_BCR_223-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977911_Receptor_00000000000156_BCR_223-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977912_Receptor_00000000000157_BCR_224-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977913_Receptor_00000000000158_BCR_228-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977914_Receptor_00000000000159_BCR_228-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977915_Receptor_00000000000160_BCR_233-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977916_Receptor_00000000000161_BCR_235-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977917_Receptor_00000000000162_BCR_235-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977918_Receptor_00000000000163_BCR_242-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977919_Receptor_00000000000164_BCR_242-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977920_Receptor_00000000000165_BCR_248-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977921_Receptor_00000000000166_BCR_248-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977922_Receptor_00000000000167_BCR_252-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977923_Receptor_00000000000168_BCR_252-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977924_Receptor_00000000000169_BCR_254-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977925_Receptor_00000000000170_BCR_254-6M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977926_Receptor_00000000000171_BCR_254-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977927_Receptor_00000000000172_BCR_254-9M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977928_Receptor_00000000000173_BCR_254-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977929_Receptor_00000000000174_BCR_259-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977930_Receptor_00000000000175_BCR_259-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977931_Receptor_00000000000176_BCR_262-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977932_Receptor_00000000000177_BCR_262-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977933_Receptor_00000000000178_BCR_263-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977934_Receptor_00000000000179_BCR_263-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977935_Receptor_00000000000180_BCR_264-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977936_Receptor_00000000000181_BCR_264-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977937_Receptor_00000000000182_BCR_266-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977938_Receptor_00000000000183_BCR_270-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977939_Receptor_00000000000184_BCR_273-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977940_Receptor_00000000000185_BCR_273-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977941_Receptor_00000000000186_BCR_274-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977942_Receptor_00000000000187_BCR_274-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977943_Receptor_00000000000188_BCR_277-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977944_Receptor_00000000000189_BCR_277-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977945_Receptor_00000000000190_BCR_278-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977946_Receptor_00000000000191_BCR_278-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977947_Receptor_00000000000192_BCR_279-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977948_Receptor_00000000000193_BCR_279-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977949_Receptor_00000000000194_BCR_284-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977950_Receptor_00000000000195_BCR_284-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977951_Receptor_00000000000196_BCR_289-4W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977952_Receptor_00000000000197_BCR_289-6M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977953_Receptor_00000000000198_BCR_289-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977954_Receptor_00000000000199_BCR_290-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977955_Receptor_00000000000200_BCR_290-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977956_Receptor_00000000000201_BCR_295-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977957_Receptor_00000000000202_BCR_302-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977958_Receptor_00000000000203_BCR_302-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977959_Receptor_00000000000204_BCR_304-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977960_Receptor_00000000000205_BCR_304-6M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977961_Receptor_00000000000206_BCR_304-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977962_Receptor_00000000000207_BCR_305-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977963_Receptor_00000000000208_BCR_305-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977964_Receptor_00000000000209_BCR_306-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977965_Receptor_00000000000210_BCR_306-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977966_Receptor_00000000000211_BCR_313-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977967_Receptor_00000000000212_BCR_313-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977968_Receptor_00000000000213_BCR_317-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977969_Receptor_00000000000214_BCR_317-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977970_Receptor_00000000000215_BCR_319-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977971_Receptor_00000000000216_BCR_319-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977972_Receptor_00000000000217_BCR_323-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977973_Receptor_00000000000218_BCR_323-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977974_Receptor_00000000000219_BCR_325-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977975_Receptor_00000000000220_BCR_325-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977976_Receptor_00000000000221_BCR_122-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977977_Receptor_00000000000222_BCR_125-6-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977978_Receptor_00000000000223_BCR_126-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977979_Receptor_00000000000224_BCR_153-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977980_Receptor_00000000000225_BCR_155-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977981_Receptor_00000000000226_BCR_196-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977982_Receptor_00000000000227_BCR_202-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977983_Receptor_00000000000228_BCR_203-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977984_Receptor_00000000000229_BCR_203-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977985_Receptor_00000000000230_BCR_205-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977986_Receptor_00000000000231_BCR_213-8M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977987_Receptor_00000000000232_BCR_215-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977988_Receptor_00000000000233_BCR_216-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977989_Receptor_00000000000234_BCR_223-TOX.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977990_Receptor_00000000000235_BCR_224-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977991_Receptor_00000000000236_BCR_233-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977992_Receptor_00000000000237_BCR_252-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977993_Receptor_00000000000238_BCR_254-27W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977994_Receptor_00000000000239_BCR_266-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977995_Receptor_00000000000240_BCR_270-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977996_Receptor_00000000000241_BCR_279-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977997_Receptor_00000000000242_BCR_290-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977998_Receptor_00000000000243_BCR_295-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977999_Receptor_00000000000244_BCR_304-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978000_Receptor_00000000000245_BCR_323-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978001_Receptor_00000000000246_BCR_140-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978002_Receptor_00000000000247_BCR_140-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978003_Receptor_00000000000248_BCR_146-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978004_Receptor_00000000000249_BCR_146-6M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978005_Receptor_00000000000250_BCR_146-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978006_Receptor_00000000000251_BCR_146-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978007_Receptor_00000000000252_BCR_167-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978008_Receptor_00000000000253_BCR_167-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978009_Receptor_00000000000254_BCR_170-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978010_Receptor_00000000000255_BCR_170-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978011_Receptor_00000000000256_BCR_172-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978012_Receptor_00000000000257_BCR_172-Bl.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978013_Receptor_00000000000258_BCR_173-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978014_Receptor_00000000000259_BCR_173-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978015_Receptor_00000000000260_BCR_181-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978016_Receptor_00000000000261_BCR_181-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978017_Receptor_00000000000262_BCR_185-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978018_Receptor_00000000000263_BCR_185-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978019_Receptor_00000000000264_BCR_186-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978020_Receptor_00000000000265_BCR_186-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978021_Receptor_00000000000266_BCR_186-9M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978022_Receptor_00000000000267_BCR_186-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978023_Receptor_00000000000268_BCR_189-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978024_Receptor_00000000000269_BCR_189-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978025_Receptor_00000000000270_BCR_192-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978026_Receptor_00000000000271_BCR_192-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978027_Receptor_00000000000272_BCR_193-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978028_Receptor_00000000000273_BCR_193-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978029_Receptor_00000000000274_BCR_208-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978030_Receptor_00000000000275_BCR_208-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978031_Receptor_00000000000276_BCR_209-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978032_Receptor_00000000000277_BCR_209-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978033_Receptor_00000000000278_BCR_211-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978034_Receptor_00000000000279_BCR_211-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978035_Receptor_00000000000280_BCR_214-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978036_Receptor_00000000000281_BCR_214-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978037_Receptor_00000000000282_BCR_218-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978038_Receptor_00000000000283_BCR_218-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978039_Receptor_00000000000284_BCR_221-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978040_Receptor_00000000000285_BCR_221-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978041_Receptor_00000000000286_BCR_225-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978042_Receptor_00000000000287_BCR_225-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978043_Receptor_00000000000288_BCR_227-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978044_Receptor_00000000000289_BCR_227-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978045_Receptor_00000000000290_BCR_230-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978046_Receptor_00000000000291_BCR_230-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978047_Receptor_00000000000292_BCR_234-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978048_Receptor_00000000000293_BCR_234-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978049_Receptor_00000000000294_BCR_237-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978050_Receptor_00000000000295_BCR_237-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978051_Receptor_00000000000296_BCR_244-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978052_Receptor_00000000000297_BCR_244-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978053_Receptor_00000000000298_BCR_245-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978054_Receptor_00000000000299_BCR_245-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978055_Receptor_00000000000300_BCR_249-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978056_Receptor_00000000000301_BCR_249-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978057_Receptor_00000000000302_BCR_257-7W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978058_Receptor_00000000000303_BCR_257-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978059_Receptor_00000000000304_BCR_260-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978060_Receptor_00000000000305_BCR_260-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978061_Receptor_00000000000306_BCR_261-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978062_Receptor_00000000000307_BCR_261-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978063_Receptor_00000000000308_BCR_269-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978064_Receptor_00000000000309_BCR_269-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978065_Receptor_00000000000310_BCR_283-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978066_Receptor_00000000000311_BCR_283-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978067_Receptor_00000000000312_BCR_285-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978068_Receptor_00000000000313_BCR_285-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978069_Receptor_00000000000314_BCR_285-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978070_Receptor_00000000000315_BCR_293-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978071_Receptor_00000000000316_BCR_293-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978072_Receptor_00000000000317_BCR_296-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978073_Receptor_00000000000318_BCR_296-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978074_Receptor_00000000000319_BCR_298-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978075_Receptor_00000000000320_BCR_298-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978076_Receptor_00000000000321_BCR_299-10W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978077_Receptor_00000000000322_BCR_299-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978078_Receptor_00000000000323_BCR_299-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978079_Receptor_00000000000324_BCR_299-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978080_Receptor_00000000000325_BCR_299-4W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978081_Receptor_00000000000326_BCR_299-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978082_Receptor_00000000000327_BCR_310-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978083_Receptor_00000000000328_BCR_310-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978084_Receptor_00000000000329_BCR_320-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978085_Receptor_00000000000330_BCR_320-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978086_Receptor_00000000000331_BCR_326-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978087_Receptor_00000000000332_BCR_326-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978088_Receptor_00000000000333_BCR_327-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978089_Receptor_00000000000334_BCR_327-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978090_Receptor_00000000000335_BCR_328-4W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978091_Receptor_00000000000336_BCR_328-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978092_Receptor_00000000000337_BCR_329-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978093_Receptor_00000000000338_BCR_329-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978094_Receptor_00000000000339_BCR_334-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978095_Receptor_00000000000340_BCR_334-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978096_Receptor_00000000000341_BCR_336-4W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978097_Receptor_00000000000342_BCR_336-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978098_Receptor_00000000000343_BCR_338-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978099_Receptor_00000000000344_BCR_338-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978100_Receptor_00000000000345_BCR_346-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978101_Receptor_00000000000346_BCR_346-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978102_Receptor_00000000000347_BCR_350-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978103_Receptor_00000000000348_BCR_350-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978104_Receptor_00000000000349_BCR_351-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978105_Receptor_00000000000350_BCR_351-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978106_Receptor_00000000000351_BCR_352-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978107_Receptor_00000000000352_BCR_352-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978108_Receptor_00000000000353_BCR_358-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978109_Receptor_00000000000354_BCR_358-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978110_Receptor_00000000000355_BCR_360-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978111_Receptor_00000000000356_BCR_360-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978112_Receptor_00000000000357_BCR_386-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978113_Receptor_00000000000358_BCR_386-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978114_Receptor_00000000000359_BCR_60-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978115_Receptor_00000000000360_BCR_60-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978116_Receptor_00000000000361_BCR_74-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978117_Receptor_00000000000362_BCR_74-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978118_Receptor_00000000000363_BCR_90-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978119_Receptor_00000000000364_BCR_90-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978120_Receptor_00000000000365_BCR_92-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978121_Receptor_00000000000366_BCR_92-BL.tsv.gz\n",
      "🔍 Found 256 count files for GSE296826\n",
      "⚠️ Zero counts detected in merged matrix — applying +1 normalization.\n",
      "✅ Final merged count matrix shape: (24901, 256)\n",
      "       GSM8977866  GSM8977867  GSM8977868  GSM8977869  GSM8977870  GSM8977871  \\\n",
      "Gene                                                                            \n",
      "0               1           1         376          61         111         208   \n",
      "1               1           1         208          49         107         173   \n",
      "2               1           1         181          46          97         121   \n",
      "3               1          60         120          46           1          96   \n",
      "4               1          58          82          40          62          86   \n",
      "...           ...         ...         ...         ...         ...         ...   \n",
      "25406           1           1           1           1           1           1   \n",
      "25407           1           1           1           1           1           1   \n",
      "25408           1           1           1           1           1           1   \n",
      "25409           1           1           1           1           1           1   \n",
      "25410           1           1           1           1           1           1   \n",
      "\n",
      "       GSM8977872  GSM8977873  GSM8977874  GSM8977875  ...  GSM8978112  \\\n",
      "Gene                                                   ...               \n",
      "0             267         121           1         105  ...           1   \n",
      "1             246          68           1          86  ...           1   \n",
      "2             195          41           1          77  ...           1   \n",
      "3             180          38           1          34  ...           1   \n",
      "4             113          28           1           1  ...          27   \n",
      "...           ...         ...         ...         ...  ...         ...   \n",
      "25406           1           1           1           1  ...           1   \n",
      "25407           1           1           1           1  ...           1   \n",
      "25408           1           1           1           1  ...           1   \n",
      "25409           1           1           1           1  ...           1   \n",
      "25410           1           1           1           1  ...           1   \n",
      "\n",
      "       GSM8978113  GSM8978114  GSM8978115  GSM8978116  GSM8978117  GSM8978118  \\\n",
      "Gene                                                                            \n",
      "0              85           1           1         286         229         123   \n",
      "1              83         145           1         262         180          99   \n",
      "2              53         127           1           1         134          94   \n",
      "3               1         108           1         101         128          83   \n",
      "4              47         103           1          96         120          62   \n",
      "...           ...         ...         ...         ...         ...         ...   \n",
      "25406           1           1           1           1           1           1   \n",
      "25407           1           1           1           1           1           1   \n",
      "25408           1           1           1           1           1           1   \n",
      "25409           1           1           1           1           1           1   \n",
      "25410           1           1           1           1           1           1   \n",
      "\n",
      "       GSM8978119  GSM8978120  GSM8978121  \n",
      "Gene                                       \n",
      "0             271         610          58  \n",
      "1             214         383          47  \n",
      "2             136         221          39  \n",
      "3             126         128          38  \n",
      "4             109         125          35  \n",
      "...           ...         ...         ...  \n",
      "25406           1           1           1  \n",
      "25407           1           1           1  \n",
      "25408           1           1           1  \n",
      "25409           1           1           1  \n",
      "25410           1           1           1  \n",
      "\n",
      "[24901 rows x 256 columns]\n",
      "📂 Parsing series matrix: geofetch_metadata/GSE296826_series_matrix.txt.gz\n",
      "✅ Parsed 256 samples\n",
      "🔍 Example parsed sample metadata row:\n",
      "{'title': '122-6W', 'geo_accession': 'GSM8977866', 'status': 'Public on May 12 2025', 'submission_date': 'May 12 2025', 'last_update_date': 'May 12 2025', 'type': 'SRA', 'channel_count': '1', 'source_name_ch1': 'Blood', 'organism_ch1': 'Homo sapiens', 'characteristics_ch1': 'tissue: Blood', 'molecule_ch1': 'polyA RNA', 'extract_protocol_ch1': 'Total RNA was isolated via poly‑A mRNA enrichment using oligo‑dT magnetic beads as part of the TruSeq Stranded mRNA kit workflow (oligo‑dT capture, wash, and elution)', 'taxid_ch1': '9606', 'description': 'Library name: Sample_00000000000013', 'data_processing': 'Raw paired‑end FASTQ files were processed with MiXCR (vX.X) to assemble BCR heavy‑chain sequences, call clonotypes, and quantify clone counts and frequencies \\u200b', 'platform_id': 'GPL24676', 'contact_name': 'David,,Gerber', 'contact_email': 'David.Gerber@UTSouthwestern.edu', 'contact_institute': 'UT Southwestern Medical Center', 'contact_address': '5323 Harry Hines Blvd.', 'contact_city': 'Dallas', 'contact_state': 'Texas', 'contact_zip/postal_code': '75390-9082', 'contact_country': 'USA', 'data_row_count': '0', 'instrument_model': 'Illumina NovaSeq 6000', 'library_selection': 'cDNA', 'library_source': 'transcriptomic', 'library_strategy': 'RNA-Seq', 'relation': 'BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN48436107', 'supplementary_file_1': 'ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM8977nnn/GSM8977866/suppl/GSM8977866_Receptor_00000000000111_BCR_122-6W.tsv.gz'}\n",
      "🧠 Building prompt for batch 1 with 100 samples...\n",
      "📝 Prompt preview (first 500 chars): \n",
      "You are a biomedical data curator.\n",
      "\n",
      "Below is a list of samples from GEO study GSE296826.\n",
      "Each sample contains metadata fields like geo_accession, title, description, and characteristics.\n",
      "\n",
      "Your task:\n",
      "- Identify which metadata field contains the sample titles that match these count matrix sample labels.\n",
      "- The count matrix sample labels are:\n",
      "[\n",
      "  \"GSM8977866\",\n",
      "  \"GSM8977867\",\n",
      "  \"GSM8977868\",\n",
      "  \"GSM8977869\",\n",
      "  \"GSM8977870\",\n",
      "  \"GSM8977871\",\n",
      "  \"GSM8977872\",\n",
      "  \"GSM8977873\",\n",
      "  \"GSM8977874\",\n",
      "  \"GSM897787...\n",
      "💬 Raw LLM output (first 500 chars): ```json\n",
      "{\n",
      "  \"GSM8977866\": {\n",
      "    \"geo_accession\": \"GSM8977866\",\n",
      "    \"condition\": \"treated\",\n",
      "    \"time\": \"6W\"\n",
      "  },\n",
      "  \"GSM8977867\": {\n",
      "    \"geo_accession\": \"GSM8977867\",\n",
      "    \"condition\": \"control\",\n",
      "    \"time\": null\n",
      "  },\n",
      "  \"GSM8977868\": {\n",
      "    \"geo_accession\": \"GSM8977868\",\n",
      "    \"condition\": \"treated\",\n",
      "    \"time\": \"3M\"\n",
      "  },\n",
      "  \"GSM8977869\": {\n",
      "    \"geo_accession\": \"GSM8977869\",\n",
      "    \"condition\": \"control\",\n",
      "    \"time\": null\n",
      "  },\n",
      "  \"GSM8977870\": {\n",
      "    \"geo_accession\": \"GSM8977870\",\n",
      "    \"condition\": \"treated...\n",
      "✅ Parsed JSON batch metadata for 256 samples\n",
      "🧠 Building prompt for batch 2 with 100 samples...\n",
      "📝 Prompt preview (first 500 chars): \n",
      "You are a biomedical data curator.\n",
      "\n",
      "Below is a list of samples from GEO study GSE296826.\n",
      "Each sample contains metadata fields like geo_accession, title, description, and characteristics.\n",
      "\n",
      "Your task:\n",
      "- Identify which metadata field contains the sample titles that match these count matrix sample labels.\n",
      "- The count matrix sample labels are:\n",
      "[\n",
      "  \"GSM8977866\",\n",
      "  \"GSM8977867\",\n",
      "  \"GSM8977868\",\n",
      "  \"GSM8977869\",\n",
      "  \"GSM8977870\",\n",
      "  \"GSM8977871\",\n",
      "  \"GSM8977872\",\n",
      "  \"GSM8977873\",\n",
      "  \"GSM8977874\",\n",
      "  \"GSM897787...\n",
      "💬 Raw LLM output (first 500 chars): ```json\n",
      "[\n",
      "    {\n",
      "        \"geo_accession\": \"GSM8977866\",\n",
      "        \"condition\": null,\n",
      "        \"time\": null\n",
      "    },\n",
      "    {\n",
      "        \"geo_accession\": \"GSM8977867\",\n",
      "        \"condition\": null,\n",
      "        \"time\": null\n",
      "    },\n",
      "    {\n",
      "        \"geo_accession\": \"GSM8977868\",\n",
      "        \"condition\": null,\n",
      "        \"time\": null\n",
      "    },\n",
      "    {\n",
      "        \"geo_accession\": \"GSM8977869\",\n",
      "        \"condition\": null,\n",
      "        \"time\": null\n",
      "    },\n",
      "    {\n",
      "        \"geo_accession\": \"GSM8977870\",\n",
      "        \"condition\": null,\n",
      "        \"time\": nu...\n",
      "❌ Failed to extract JSON: Extra data: line 5 column 6 (char 94)\n",
      "✅ Parsed JSON batch metadata for 0 samples\n",
      "🧠 Building prompt for batch 3 with 56 samples...\n",
      "📝 Prompt preview (first 500 chars): \n",
      "You are a biomedical data curator.\n",
      "\n",
      "Below is a list of samples from GEO study GSE296826.\n",
      "Each sample contains metadata fields like geo_accession, title, description, and characteristics.\n",
      "\n",
      "Your task:\n",
      "- Identify which metadata field contains the sample titles that match these count matrix sample labels.\n",
      "- The count matrix sample labels are:\n",
      "[\n",
      "  \"GSM8977866\",\n",
      "  \"GSM8977867\",\n",
      "  \"GSM8977868\",\n",
      "  \"GSM8977869\",\n",
      "  \"GSM8977870\",\n",
      "  \"GSM8977871\",\n",
      "  \"GSM8977872\",\n",
      "  \"GSM8977873\",\n",
      "  \"GSM8977874\",\n",
      "  \"GSM897787...\n",
      "💬 Raw LLM output (first 500 chars): ```json\n",
      "{\n",
      "  \"GSM8977866\": {\n",
      "    \"geo_accession\": null,\n",
      "    \"condition\": null,\n",
      "    \"time\": null\n",
      "  },\n",
      "  \"GSM8977867\": {\n",
      "    \"geo_accession\": null,\n",
      "    \"condition\": null,\n",
      "    \"time\": null\n",
      "  },\n",
      "  \"GSM8977868\": {\n",
      "    \"geo_accession\": null,\n",
      "    \"condition\": null,\n",
      "    \"time\": null\n",
      "  },\n",
      "  \"GSM8977869\": {\n",
      "    \"geo_accession\": null,\n",
      "    \"condition\": null,\n",
      "    \"time\": null\n",
      "  },\n",
      "  \"GSM8977870\": {\n",
      "    \"geo_accession\": null,\n",
      "    \"condition\": null,\n",
      "    \"time\": null\n",
      "  },\n",
      "  \"GSM8977871\": {\n",
      "    \"geo_accession\": n...\n",
      "✅ Parsed JSON batch metadata for 256 samples\n",
      "🎉 Final metadata contains 256 sample labels\n",
      "🔑 Metadata key summary: {'geo_accession': 256, 'condition': 256, 'time': 256}\n",
      "Error running DESeq2 for GSE296826: too many values to unpack (expected 2)\n"
     ]
    }
   ],
   "source": [
    "# Run the DEG analysis directly\n",
    "result_state = run_deg_for_single_gse_debug(initial_state)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "15cbe780-bd4e-4e94-9c28-dc5a0d6e1fca",
   "metadata": {},
   "outputs": [],
   "source": [
    "### test this function ###########"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "09fca73c-da5c-4ef7-8408-87cdd4f837d1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "✅ Mock count matrix created:\n",
      "       122-6W  122-BL  125-3M  125-BL\n",
      "Gene1      42      55      53      54\n",
      "Gene2      51      64      41      45\n",
      "Gene3      44      36      45      43\n",
      "Gene4      60      49      49      46\n",
      "Gene5      48      54      47      55\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "# Mock count matrix with 4 samples and 100 genes\n",
    "genes = [f\"Gene{i}\" for i in range(1, 101)]\n",
    "samples = [\"122-6W\", \"122-BL\", \"125-3M\", \"125-BL\"]\n",
    "\n",
    "data = np.random.poisson(lam=50, size=(100, 4))\n",
    "counts_df = pd.DataFrame(data, index=genes, columns=samples)\n",
    "print(\"✅ Mock count matrix created:\")\n",
    "print(counts_df.head())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "2c448ec4-417c-4eec-8a1e-f01d97a16945",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "✅ Mock metadata DataFrame created:\n",
      "                     condition      time\n",
      "sample_geo_accession                    \n",
      "122-6W                 treated   6 weeks\n",
      "122-BL                 control  baseline\n",
      "125-3M                 treated  3 months\n",
      "125-BL                 control  baseline\n"
     ]
    }
   ],
   "source": [
    "# Mock metadata dict, simulating what LLM would return\n",
    "mock_metadata = {\n",
    "    \"122-6W\": {\"condition\": \"treated\", \"time\": \"6 weeks\"},\n",
    "    \"122-BL\": {\"condition\": \"control\", \"time\": \"baseline\"},\n",
    "    \"125-3M\": {\"condition\": \"treated\", \"time\": \"3 months\"},\n",
    "    \"125-BL\": {\"condition\": \"control\", \"time\": \"baseline\"}\n",
    "}\n",
    "\n",
    "# Convert to DataFrame\n",
    "metadata_df = pd.DataFrame.from_dict(mock_metadata, orient=\"index\")\n",
    "metadata_df.index.name = \"sample_geo_accession\"\n",
    "print(\"✅ Mock metadata DataFrame created:\")\n",
    "print(metadata_df)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "e20b6059-97de-4074-8ecb-9a280d59aea3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "✅ Aligned 4 samples\n",
      "Counts columns: ['122-6W', '122-BL', '125-3M', '125-BL']\n",
      "Metadata index: ['122-6W', '122-BL', '125-3M', '125-BL']\n"
     ]
    }
   ],
   "source": [
    "# Align count matrix columns to metadata index\n",
    "valid_samples = metadata_df.index.intersection(counts_df.columns)\n",
    "counts_df = counts_df[valid_samples]\n",
    "metadata_df = metadata_df.loc[valid_samples]\n",
    "\n",
    "print(f\"✅ Aligned {len(valid_samples)} samples\")\n",
    "print(\"Counts columns:\", counts_df.columns.tolist())\n",
    "print(\"Metadata index:\", metadata_df.index.tolist())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "8eefe332-955e-4e54-8094-58adca5d04e0",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\difen\\AppData\\Local\\Temp\\ipykernel_38836\\2860708067.py:7: DeprecationWarning: design_factors is deprecated and will soon be removed.Please consider providing a formulaic formula using the design argumentinstead.\n",
      "  dds = DeseqDataSet(\n",
      "Fitting size factors...\n",
      "... done in 0.00 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 0.05 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "C:\\Users\\difen\\venv1\\Lib\\site-packages\\pydeseq2\\dds.py:805: UserWarning: The dispersion trend curve fitting did not converge. Switching to a mean-based dispersion trend.\n",
      "  self._fit_parametric_dispersion_trend(vst)\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "C:\\Users\\difen\\AppData\\Local\\Temp\\ipykernel_38836\\2860708067.py:17: UserWarning: As the residual degrees of freedom is less than 3, the distribution of log dispersions is especially asymmetric and likely to be poorly estimated by the MAD.\n",
      "  dds.fit_dispersion_prior()\n",
      "Fitting MAP dispersions...\n",
      "... done in 0.05 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 0.04 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: condition treated vs control\n",
      "          baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "Gene1    51.013182       -0.210079  0.324775 -0.646844  0.517733  0.986019\n",
      "Gene2    50.084666       -0.257575  0.330620 -0.779067  0.435940  0.986019\n",
      "Gene3    41.979408        0.157063  0.336942  0.466144  0.641113  0.986019\n",
      "Gene4    50.861889        0.183145  0.321944  0.568871  0.569444  0.986019\n",
      "Gene5    50.962442       -0.212529  0.320128 -0.663888  0.506762  0.986019\n",
      "...            ...             ...       ...       ...       ...       ...\n",
      "Gene96   51.880639       -0.095749  0.321562 -0.297761  0.765885  0.986019\n",
      "Gene97   48.938245        0.403851  0.328076  1.230969  0.218334  0.986019\n",
      "Gene98   52.302913        0.287630  0.324709  0.885809  0.375721  0.986019\n",
      "Gene99   51.423540        0.013483  0.319110  0.042251  0.966299  0.986019\n",
      "Gene100  49.824090        0.070216  0.329188  0.213300  0.831093  0.986019\n",
      "\n",
      "[100 rows x 6 columns]\n",
      "✅ DESeq2 result preview:\n",
      "        baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "Gene1  51.013182       -0.210079  0.324775 -0.646844  0.517733  0.986019\n",
      "Gene2  50.084666       -0.257575  0.330620 -0.779067  0.435940  0.986019\n",
      "Gene3  41.979408        0.157063  0.336942  0.466144  0.641113  0.986019\n",
      "Gene4  50.861889        0.183145  0.321944  0.568871  0.569444  0.986019\n",
      "Gene5  50.962442       -0.212529  0.320128 -0.663888  0.506762  0.986019\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "... done in 0.17 seconds.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from pydeseq2.dds import DeseqDataSet\n",
    "from pydeseq2.ds import DeseqStats\n",
    "#from pydeseq2.inference import DefaultInference\n",
    "\n",
    "tcounts = counts_df.T\n",
    "inference = DefaultInference(n_cpus=2)\n",
    "dds = DeseqDataSet(\n",
    "    counts=tcounts,\n",
    "    metadata=metadata_df,\n",
    "    design_factors=\"condition\",\n",
    "    refit_cooks=True,\n",
    "    inference=inference,\n",
    ")\n",
    "dds.fit_size_factors()\n",
    "dds.fit_genewise_dispersions()\n",
    "dds.fit_dispersion_trend()\n",
    "dds.fit_dispersion_prior()\n",
    "dds.fit_MAP_dispersions()\n",
    "dds.fit_LFC()\n",
    "dds.calculate_cooks()\n",
    "\n",
    "if dds.refit_cooks:\n",
    "    dds.refit()\n",
    "\n",
    "stat_res = DeseqStats(\n",
    "    dds, alpha=0.05, contrast=[\"condition\", \"treated\", \"control\"],\n",
    "    cooks_filter=True, independent_filter=True\n",
    ")\n",
    "stat_res.summary()\n",
    "\n",
    "res = stat_res.results_df\n",
    "print(\"✅ DESeq2 result preview:\")\n",
    "print(res.head())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5d1cb1e8-1a5e-435d-9e99-9e31064ede7f",
   "metadata": {},
   "outputs": [],
   "source": [
    "####"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "b95271e7-dc3d-4ca5-bcfd-627a18b8ccca",
   "metadata": {},
   "outputs": [],
   "source": [
    "gse_id=\"GSE296826\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "a06b5ea1-cd3d-4ea1-a7a7-bb2e5cc0cd3f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "WindowsPath('rna_seq_analysis')"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Path(\"rna_seq_analysis\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "29ac10c0-9ef1-4246-90e4-274989cb841e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "WindowsPath('rna_seq_analysis/GSE296826')"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Path(\"rna_seq_analysis\")/gse_id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "f54d66df-9b67-4366-a6fb-03ad0cf8df28",
   "metadata": {},
   "outputs": [],
   "source": [
    "if counts_df is None:\n",
    "    print(f\"⚠️ No count matrix found for {gse_id}. Trying to load DE results directly...\")\n",
    "    gse_dir = Path(\"rna_seq_analysis\")/gse_id\n",
    "    for file in gse_dir.glob(\"*\"):\n",
    "        if file.suffix in [\".txt\", \".tsv\", \".csv\", \".gz\"]:\n",
    "            de_df = load_de_result_file(file)\n",
    "            if de_df is not None:\n",
    "                print(\"✅ DE result loaded. Skipping DESeq2 analysis.\")\n",
    "                output_csv = f\"rna_seq_analysis/{gse_id}_de_results_direct.csv\"\n",
    "                de_df.to_csv(output_csv)\n",
    "                deg_paths[gse_id] = output_csv\n",
    "                filtered_df = de_df[\n",
    "                    (de_df.get(\"padj\", 1) < 0.05) & \n",
    "                    (abs(de_df.get(\"log2FoldChange\", 0)) > 1)\n",
    "                ]\n",
    "                print(f\"🧬 DE results from file — {filtered_df.shape[0]} significant genes found\")\n",
    "                #return {**state, \"deg_results\": deg_paths}\n",
    "    print(f\"❌ No DE result files found in {gse_dir}\")\n",
    "    #return state\n",
    "#    return {**state, \"deg_results\": deg_paths}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "id": "6699b517-bc0a-45e1-8d55-e14283901b2e",
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_json_block(text):\n",
    "    try:\n",
    "        start = text.index(\"{\")\n",
    "        end = text.rindex(\"}\") + 1\n",
    "        return json.loads(text[start:end])\n",
    "    except Exception as e:\n",
    "        print(f\"\\u274c Failed to extract JSON: {e}\")\n",
    "        return {}\n",
    "def extract_labels_using_llm(series_matrix_path, gse_id, client, sample_labels_from_counts, batch_size=100):\n",
    "    \"\"\"\n",
    "    Extract sample metadata by matching count matrix sample labels to series matrix metadata using LLM.\n",
    "    Matches are based on sample titles (e.g. 122-6W, 125-BL).\n",
    "    \"\"\"\n",
    "    from collections import defaultdict\n",
    "\n",
    "    print(f\"📂 Parsing series matrix: {series_matrix_path}\")\n",
    "    sample_table = parse_series_matrix_to_sample_table(series_matrix_path)\n",
    "    print(f\"✅ Parsed {len(sample_table)} samples\")\n",
    "\n",
    "    # Debug: show an example parsed row\n",
    "    print(\"🔍 Example parsed sample metadata row:\")\n",
    "    print(sample_table[0])\n",
    "\n",
    "    all_metadata = {}\n",
    "    for batch_num, batch_table in enumerate(split_sample_table(sample_table, batch_size=batch_size), 1):\n",
    "        print(f\"🧠 Building prompt for batch {batch_num} with {len(batch_table)} samples...\")\n",
    "\n",
    "        # Construct a focused prompt\n",
    "        prompt = f\"\"\"\n",
    "You are a biomedical data curator.\n",
    "\n",
    "Below is a list of samples from GEO study {gse_id}.\n",
    "Each sample contains metadata fields like geo_accession, title, description, and characteristics.\n",
    "\n",
    "Your task:\n",
    "- Identify which metadata field contains the sample titles that match these count matrix sample labels.\n",
    "- The count matrix sample labels are:\n",
    "{json.dumps(list(sample_labels_from_counts), indent=2)}\n",
    "\n",
    "- Map each count matrix sample label to:\n",
    "  - geo_accession (from the matching sample)\n",
    "  - condition: treated or control (based on title, characteristics, or description)\n",
    "  - time: if available (e.g. 6W, 3M)\n",
    "\n",
    "If no match, include the sample label with all fields null.\n",
    "\n",
    "Return **only** the final JSON dictionary, no explanations.\n",
    "\"\"\"\n",
    "\n",
    "        prompt += \"\\nHere is the metadata table:\\n\"\n",
    "        prompt += json.dumps(batch_table, indent=2)\n",
    "\n",
    "        print(f\"📝 Prompt preview (first 500 chars): {prompt[:500]}...\")\n",
    "\n",
    "        response = client.chat.completions.create(\n",
    "            model=\"gpt-4o\",\n",
    "            temperature=0.2,\n",
    "            messages=[{\"role\": \"user\", \"content\": prompt}]\n",
    "        )\n",
    "\n",
    "        content = response.choices[0].message.content.strip()\n",
    "        print(f\"💬 Raw LLM output (first 500 chars): {content[:500]}...\")\n",
    "\n",
    "        batch_metadata = extract_json_block(content)\n",
    "        print(f\"✅ Parsed JSON batch metadata for {len(batch_metadata)} samples\")\n",
    "\n",
    "        all_metadata.update(batch_metadata)\n",
    "\n",
    "    unmatched_labels = set(sample_labels_from_counts) - set(all_metadata.keys())\n",
    "    if unmatched_labels:\n",
    "        print(f\"⚠️ Unmatched sample labels: {list(unmatched_labels)[:5]} (total {len(unmatched_labels)})\")\n",
    "        # Ensure we return them with null fields\n",
    "        for label in unmatched_labels:\n",
    "            all_metadata[label] = {\"geo_accession\": None, \"condition\": None, \"time\": None}\n",
    "\n",
    "    # Normalize all metadata so each sample has the same keys\n",
    "    all_keys = set()\n",
    "    for meta in all_metadata.values():\n",
    "        all_keys.update(meta.keys())\n",
    "    for label, meta in all_metadata.items():\n",
    "        for key in all_keys:\n",
    "            if key not in meta:\n",
    "                meta[key] = None\n",
    "\n",
    "    # Summary\n",
    "    print(f\"🎉 Final metadata contains {len(all_metadata)} sample labels\")\n",
    "    key_counter = defaultdict(int)\n",
    "    for meta in all_metadata.values():\n",
    "        for k in meta.keys():\n",
    "            key_counter[k] += 1\n",
    "    print(f\"🔑 Metadata key summary: {dict(key_counter)}\")\n",
    "\n",
    "    return all_metadata\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "id": "6479dbaf-3c1a-4eb6-b838-bdbe17a15230",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<function __main__.extract_labels_using_llm(series_matrix_path, gse_id, client, sample_labels_from_counts, batch_size=100)>"
      ]
     },
     "execution_count": 154,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "extract_labels_using_llm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2d40bb94-f992-4b59-855e-4f598f76b548",
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_json_block(text):\n",
    "    try:\n",
    "        start = text.index(\"{\")\n",
    "        end = text.rindex(\"}\") + 1\n",
    "        return json.loads(text[start:end])\n",
    "    except Exception as e:\n",
    "        print(f\"\\u274c Failed to extract JSON: {e}\")\n",
    "        return {}\n",
    "def extract_labels_using_llm(series_matrix_path, gse_id, client, sample_labels_from_counts, batch_size=100):\n",
    "    \"\"\"\n",
    "    Extract sample metadata by matching count matrix sample labels to series matrix metadata using LLM.\n",
    "    Matches are based on sample titles (e.g. 122-6W, 125-BL).\n",
    "    \"\"\"\n",
    "    from collections import defaultdict\n",
    "\n",
    "    print(f\"📂 Parsing series matrix: {series_matrix_path}\")\n",
    "    sample_table = parse_series_matrix_to_sample_table(series_matrix_path)\n",
    "    print(f\"✅ Parsed {len(sample_table)} samples\")\n",
    "\n",
    "    # Debug: show an example parsed row\n",
    "    print(\"🔍 Example parsed sample metadata row:\")\n",
    "    print(sample_table[0])\n",
    "\n",
    "    all_metadata = {}\n",
    "    for batch_num, batch_table in enumerate(split_sample_table(sample_table, batch_size=batch_size), 1):\n",
    "        print(f\"🧠 Building prompt for batch {batch_num} with {len(batch_table)} samples...\")\n",
    "\n",
    "        # Construct a focused prompt\n",
    "        prompt = f\"\"\"\n",
    "You are a biomedical data curator.\n",
    "\n",
    "Below is a list of samples from GEO study {gse_id}.\n",
    "Each sample contains metadata fields like geo_accession, title, description, and characteristics.\n",
    "\n",
    "Your task:\n",
    "- Identify which metadata field contains the sample titles that match these count matrix sample labels.\n",
    "- The count matrix sample labels are:\n",
    "{json.dumps(list(sample_labels_from_counts), indent=2)}\n",
    "\n",
    "- Map each count matrix sample label to:\n",
    "  - geo_accession (from the matching sample)\n",
    "  - condition: treated or control (based on title, characteristics, or description)\n",
    "  - time: if available (e.g. 6W, 3M)\n",
    "\n",
    "If no match, include the sample label with all fields null.\n",
    "\n",
    "Return **only** the final JSON dictionary, no explanations.\n",
    "\"\"\"\n",
    "\n",
    "        prompt += \"\\nHere is the metadata table:\\n\"\n",
    "        prompt += json.dumps(batch_table, indent=2)\n",
    "\n",
    "        print(f\"📝 Prompt preview (first 500 chars): {prompt[:500]}...\")\n",
    "\n",
    "        response = client.chat.completions.create(\n",
    "            model=\"gpt-4o\",\n",
    "            temperature=0.2,\n",
    "            messages=[{\"role\": \"user\", \"content\": prompt}]\n",
    "        )\n",
    "\n",
    "        content = response.choices[0].message.content.strip()\n",
    "        print(f\"💬 Raw LLM output (first 500 chars): {content[:500]}...\")\n",
    "\n",
    "        batch_metadata = extract_json_block(content)\n",
    "        print(f\"✅ Parsed JSON batch metadata for {len(batch_metadata)} samples\")\n",
    "\n",
    "        all_metadata.update(batch_metadata)\n",
    "\n",
    "    unmatched_labels = set(sample_labels_from_counts) - set(all_metadata.keys())\n",
    "    if unmatched_labels:\n",
    "        print(f\"⚠️ Unmatched sample labels: {list(unmatched_labels)[:5]} (total {len(unmatched_labels)})\")\n",
    "        # Ensure we return them with null fields\n",
    "        for label in unmatched_labels:\n",
    "            all_metadata[label] = {\"geo_accession\": None, \"condition\": None, \"time\": None}\n",
    "\n",
    "    # Normalize all metadata so each sample has the same keys\n",
    "    all_keys = set()\n",
    "    for meta in all_metadata.values():\n",
    "        all_keys.update(meta.keys())\n",
    "    for label, meta in all_metadata.items():\n",
    "        for key in all_keys:\n",
    "            if key not in meta:\n",
    "                meta[key] = None\n",
    "\n",
    "    # Summary\n",
    "    print(f\"🎉 Final metadata contains {len(all_metadata)} sample labels\")\n",
    "    key_counter = defaultdict(int)\n",
    "    for meta in all_metadata.values():\n",
    "        for k in meta.keys():\n",
    "            key_counter[k] += 1\n",
    "    print(f\"🔑 Metadata key summary: {dict(key_counter)}\")\n",
    "\n",
    "    return all_metadata\n",
    "\n",
    "def run_deg_for_single_gse_run(state):\n",
    "    gse_id = state.get(\"selected_gse\")\n",
    "    if not gse_id:\n",
    "        print(\"No selected GSE found.\")\n",
    "        return state\n",
    "\n",
    "    deg_paths = {}\n",
    "    decompress_gse_in_dir(gse_id)\n",
    "    series_matrix_path = f\"geofetch_metadata/{gse_id}_series_matrix.txt.gz\"\n",
    "\n",
    "    try:\n",
    "        print(f\"Running DESeq2 for: {gse_id}\")\n",
    "        download_dir = download_and_extract_gse(gse_id)\n",
    "        counts_df = find_and_merge_count_files(gse_id)\n",
    "\n",
    "        if counts_df is None:\n",
    "            print(f\"⚠️ No count matrix found for {gse_id}. Trying to load DE results directly...\")\n",
    "            gse_dir = Path(\"rna_seq_analysis\")/gse_id\n",
    "            for file in gse_dir.glob(\"*\"):\n",
    "                if file.suffix in [\".txt\", \".tsv\", \".csv\", \".gz\"]:\n",
    "                    de_df = load_de_result_file(file)\n",
    "                    if de_df is not None:\n",
    "                        print(\"✅ DE result loaded. Skipping DESeq2 analysis.\")\n",
    "                        output_csv = f\"rna_seq_analysis/{gse_id}_de_results_direct.csv\"\n",
    "                        de_df.to_csv(output_csv)\n",
    "                        deg_paths[gse_id] = output_csv\n",
    "                        filtered_df = de_df[\n",
    "                            (de_df.get(\"padj\", 1) < 0.05) & \n",
    "                            (abs(de_df.get(\"log2FoldChange\", 0)) > 1)\n",
    "                        ]\n",
    "                        print(f\"🧬 DE results from file — {filtered_df.shape[0]} significant genes found\")\n",
    "                        return {**state, \"deg_results\": deg_paths}\n",
    "            print(f\"❌ No DE result files found in {gse_dir}\")\n",
    "            return state\n",
    "            #print(\"✅ Count matrix loaded:\")\n",
    "            #print(counts_df.shape)\n",
    "            #print(\"Columns (first 5):\", list(counts_df.columns)[:5])\n",
    "\n",
    "              \n",
    "\n",
    "        print(counts_df)\n",
    "        # Extract labels from count matrix columns\n",
    "        sample_labels_from_counts = counts_df.columns\n",
    "        \n",
    "        # LLM label mapping\n",
    "        group_label_mapping = extract_labels_using_llm(series_matrix_path, gse_id, client, sample_labels_from_counts)\n",
    "        \n",
    "        print(f\"✅ Group label mapping keys: {list(group_label_mapping.keys())[:5]}\")\n",
    "        print(f\"✅ Group label mapping size: {len(group_label_mapping)}\")\n",
    "        \n",
    "        # Create metadata DataFrame\n",
    "\n",
    "# Build metadata DataFrame and rename to unique names\n",
    "        metadata_df = pd.DataFrame.from_dict(group_label_mapping, orient=\"index\")\n",
    "        metadata_df.index = [name_map.get(idx, idx) for idx in metadata_df.index]\n",
    "        metadata_df.index.name = \"sample_geo_accession\"\n",
    "\n",
    "        print(\"✅ Metadata DataFrame:\")\n",
    "        print(metadata_df.head())\n",
    "        \n",
    "        # Align count matrix columns to metadata\n",
    "        print(\"Before alignment:\")\n",
    "        print(\"Counts columns:\", list(counts_df.columns)[:5])\n",
    "        print(\"Metadata index:\", list(metadata_df.index)[:5])\n",
    "\n",
    "        # Align counts + metadata\n",
    "        valid_samples = metadata_df.index.intersection(counts_df.columns)\n",
    "        print(f\"✅ Valid samples after intersection: {list(valid_samples)}\")\n",
    "               \n",
    "               \n",
    "        counts_df = counts_df[valid_samples]\n",
    "        metadata_df = metadata_df.loc[valid_samples]\n",
    "\n",
    "        if counts_df.shape[1] == 0:\n",
    "            raise ValueError(\"❌ No valid samples left after alignment — cannot run DESeq2.\")\n",
    "\n",
    "        \n",
    "        inference = DefaultInference(n_cpus=4)\n",
    "       \n",
    "        tcounts = counts_df.T\n",
    "        print(\"✅ Final counts shape:\", tcounts.shape)\n",
    "        print(\"✅ Final metadata shape:\", metadata_df.shape)\n",
    "\n",
    "        ### convert counts_df same as tcounts #######\n",
    "        counts_df=counts_df.T ### both matrix should use sample id as rows, genes in cols\n",
    "        # Filter metadata to only those with non-null condition\n",
    "        metadata_valid = metadata_df[metadata_df[\"condition\"].notnull()]\n",
    "        # Get the intersection of sample names that are in both counts_df and valid metadata\n",
    "        valid_samples = metadata_valid.index.intersection(counts_df.index)\n",
    "        # Subset counts and metadata\n",
    "        counts_df = counts_df.loc[valid_samples]\n",
    "        metadata_df = metadata_valid.loc[valid_samples]\n",
    "        if counts_df.shape[1] == 0:\n",
    "            raise ValueError(\"❌ No valid samples left after alignment — cannot run DESeq2.\")\n",
    "        # Sanity check\n",
    "        print(f\"✅ Number of valid samples: {len(valid_samples)}\")\n",
    "        print(\"Counts shape:\", counts_df.shape)\n",
    "        print(\"Metadata shape:\", metadata_df.shape)\n",
    "        dds = DeseqDataSet(\n",
    "            counts=counts_df,\n",
    "            metadata=metadata_df,\n",
    "            design_factors=\"condition\",\n",
    "            refit_cooks=True,\n",
    "            inference=DefaultInference(n_cpus=4)\n",
    "        )\n",
    "        dds.fit_size_factors()\n",
    "        dds.fit_genewise_dispersions()\n",
    "        dds.fit_dispersion_trend()\n",
    "        dds.fit_dispersion_prior()\n",
    "        dds.fit_MAP_dispersions()\n",
    "        dds.fit_LFC()\n",
    "        dds.calculate_cooks()\n",
    "        \n",
    "        if dds.refit_cooks:\n",
    "            dds.refit()\n",
    "        \n",
    "        stat_res = DeseqStats(\n",
    "            dds, alpha=0.05, contrast=[\"condition\", \"treated\", \"control\"],\n",
    "            cooks_filter=True, independent_filter=True\n",
    "        )\n",
    "        stat_res.summary()\n",
    "        \n",
    "        res = stat_res.results_df\n",
    "        print(\"✅ DESeq2 result preview:\")\n",
    "        print(res.head())\n",
    "\n",
    "       \n",
    "\n",
    "        output_csv = f\"rna_seq_analysis/{gse_id}_deseq2_results.csv\"\n",
    "        res.to_csv(output_csv, index=False)\n",
    "        deg_paths[gse_id] = output_csv\n",
    "        print(f\"✅ DESeq2 results saved: {output_csv}\")\n",
    "\n",
    "        filtered_df = res[\n",
    "            (res[\"padj\"] < 0.05) & (res[\"log2FoldChange\"].abs() > 1)\n",
    "        ]\n",
    "        print(f\"🧬 {filtered_df.shape[0]} significant genes found\")\n",
    "\n",
    "        # Additional visualization + analysis\n",
    "        dds.layers[\"log1p\"] = np.log1p(dds.layers[\"normed_counts\"])\n",
    "        sigs = res[(res.padj < 0.05) & (abs(res.log2FoldChange) > 1.5)]\n",
    "        dds_sigs = dds[:, sigs.index]\n",
    "\n",
    "        grapher = pd.DataFrame(\n",
    "            dds_sigs.layers['log1p'],\n",
    "            index=dds_sigs.obs_names,\n",
    "            columns=dds_sigs.var_names\n",
    "        ).T\n",
    "        grapher.index.name = \"gene\"\n",
    "        grapher_cp = grapher.copy()\n",
    "\n",
    "        sample_names = list(dds_sigs.obs_names)\n",
    "        sample_labels = [group_label_mapping[sample][\"treatment\"] for sample in sample_names]\n",
    "        labeled_columns = [f\"{sample}_{label}\" for sample, label in zip(sample_labels, sample_names)]\n",
    "        grapher.columns = labeled_columns\n",
    "\n",
    "        sc.tl.pca(dds)\n",
    "        stat_res.lfc_shrink(coeff=\"condition[T.treated]\")\n",
    "        sc.pl.pca(dds, color=\"condition\", size=200)\n",
    "\n",
    "        res[\"symbol\"] = res.index\n",
    "        volcano(\n",
    "            res, symbol='symbol', log2fc='log2FoldChange',\n",
    "            pval_thresh=0.05, to_label=10, log2fc_thresh=0.75,\n",
    "            colors=['dimgrey', 'lightgrey', 'green']\n",
    "        )\n",
    "        plt.show()\n",
    "\n",
    "        annotate_clustermap_with_metadata(grapher_cp, metadata_df)\n",
    "\n",
    "    except Exception as e:\n",
    "        print(f\"Error running DESeq2 for {gse_id}: {e}\")\n",
    "\n",
    "    return {**state, \"deg_results\": deg_paths}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5626f7fb-69d6-45ea-96ff-86a41d5500ac",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "initial_state: Dict = {\n",
    "    \"query_gene\": \"\",\n",
    "    \"pathway_info\": {\"genes\": [], \"drugs\": [], \"pathways\": []},\n",
    "    \"metadata\": {},\n",
    "    \"gse_list\": [],\n",
    "    \"research_plan\": \"\",\n",
    "  #  \"series_matrix_path\":'geofetch_metadata/GSE263141_series_matrix.txt.gz',\n",
    "    \"series_matrix_path\":'geofetch_metadata/GSE296826_series_matrix.txt.gz',\n",
    "    \"selected_gse\": 'GSE296826',#\"GSE263141\", #\"GSE236204\",\n",
    "  #  \"selected_gses\": [\"GSE242272\"],\n",
    "    \"deg_results\": {}\n",
    "}\n",
    "# Run the DEG analysis directly\n",
    "result_state = run_deg_for_single_gse_run(initial_state)\n",
    "\n",
    "# Output the DEG results\n",
    "print(\"\\n✅ DEG Results by GSE ID:\")\n",
    "for gse_id, path in result_state.get(\"deg_results\", {}).items():\n",
    "    print(f\"- {gse_id}: {path}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cbd173e0-6e59-4568-8623-9104d855be54",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "id": "d2ae62c3-c590-4d9b-8729-7c57ecd8e640",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'GSE296826'"
      ]
     },
     "execution_count": 170,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gse_id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "id": "ed31b6a0-e64d-4648-be21-605727ab991d",
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_json_block(content):\n",
    "    content = content.strip()\n",
    "    if not content:\n",
    "        raise ValueError(\"❌ LLM returned empty content — cannot parse JSON.\")\n",
    "\n",
    "    # Try direct load first\n",
    "    try:\n",
    "        return json.loads(content)\n",
    "    except json.JSONDecodeError:\n",
    "        pass  # We'll try to clean it up\n",
    "\n",
    "    # Use regex to extract JSON array\n",
    "    json_match = re.search(r\"\\[\\s*{.*}\\s*\\]\", content, re.DOTALL)\n",
    "    if not json_match:\n",
    "        raise ValueError(\"❌ Could not find JSON array in LLM output.\")\n",
    "\n",
    "    json_str = json_match.group(0)\n",
    "    parsed_list = json.loads(json_str)\n",
    "\n",
    "    # Convert to dict keyed by geo_accession\n",
    "    return {\n",
    "        entry[\"geo_accession\"]: {k: v for k, v in entry.items() if k != \"geo_accession\"}\n",
    "        for entry in parsed_list\n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c3a284a9-ac05-4ddc-8d3d-ded7eb6d0580",
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_json_block(content):\n",
    "    content = content.strip()\n",
    "    if not content:\n",
    "        raise ValueError(\"❌ LLM returned empty content — cannot parse JSON.\")\n",
    "\n",
    "    # Remove common markdown wrappers\n",
    "    content = re.sub(r\"^```json\", \"\", content, flags=re.IGNORECASE).strip()\n",
    "    content = re.sub(r\"```$\", \"\", content).strip()\n",
    "\n",
    "    # Try direct parse\n",
    "    try:\n",
    "        return json.loads(content)\n",
    "    except json.JSONDecodeError:\n",
    "        pass  # We'll try regex next\n",
    "\n",
    "    # Try to find JSON array\n",
    "    array_match = re.search(r\"(\\[\\s*{.*}\\s*\\])\", content, re.DOTALL)\n",
    "    if array_match:\n",
    "        json_str = array_match.group(1)\n",
    "        parsed_list = json.loads(json_str)\n",
    "        return {\n",
    "            entry[\"geo_accession\"]: {k: v for k, v in entry.items() if k != \"geo_accession\"}\n",
    "            for entry in parsed_list\n",
    "        }\n",
    "\n",
    "    # Try to find JSON dict\n",
    "    dict_match = re.search(r\"(\\{\\s*\\\".*?\\}\\s*)\", content, re.DOTALL)\n",
    "    if dict_match:\n",
    "        json_str = dict_match.group(1)\n",
    "        return json.loads(json_str)\n",
    "\n",
    "    # Fail if no match\n",
    "    print(f\"⚠️ Raw LLM output for inspection:\\n{content[:500]}\")\n",
    "    raise ValueError(\"❌ Could not find JSON block in LLM output.\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "id": "dba97805-145f-4154-8aae-6467718f61c7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📥 Downloading GSE296826_RAW.tar → rna_seq_analysis/GSE296826/GSE296826_RAW.tar\n",
      "🗂️ Extracting TAR...\n",
      "⚠️ Skipping compressed duplicate: GSM8977866_Receptor_00000000000111_BCR_122-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977867_Receptor_00000000000112_BCR_122-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977868_Receptor_00000000000113_BCR_125-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977869_Receptor_00000000000114_BCR_125-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977870_Receptor_00000000000115_BCR_126-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977871_Receptor_00000000000116_BCR_126-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977872_Receptor_00000000000117_BCR_139-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977873_Receptor_00000000000118_BCR_139-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977874_Receptor_00000000000119_BCR_141-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977875_Receptor_00000000000120_BCR_141-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977876_Receptor_00000000000121_BCR_143-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977877_Receptor_00000000000122_BCR_143-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977878_Receptor_00000000000123_BCR_153-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977879_Receptor_00000000000124_BCR_153-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977880_Receptor_00000000000125_BCR_155-12M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977881_Receptor_00000000000126_BCR_155-6M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977882_Receptor_00000000000127_BCR_155-9M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977883_Receptor_00000000000128_BCR_155-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977884_Receptor_00000000000129_BCR_174-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977885_Receptor_00000000000130_BCR_174-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977886_Receptor_00000000000131_BCR_191-4W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977887_Receptor_00000000000132_BCR_191-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977888_Receptor_00000000000133_BCR_195-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977889_Receptor_00000000000134_BCR_195-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977890_Receptor_00000000000135_BCR_196-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977891_Receptor_00000000000136_BCR_197-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977892_Receptor_00000000000137_BCR_197-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977893_Receptor_00000000000138_BCR_201-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977894_Receptor_00000000000139_BCR_201-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977895_Receptor_00000000000140_BCR_202-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977896_Receptor_00000000000141_BCR_205-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977897_Receptor_00000000000142_BCR_212-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977898_Receptor_00000000000143_BCR_212-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977899_Receptor_00000000000144_BCR_213-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977900_Receptor_00000000000145_BCR_213-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977901_Receptor_00000000000146_BCR_215-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977902_Receptor_00000000000147_BCR_216-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977903_Receptor_00000000000148_BCR_216-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977904_Receptor_00000000000150_BCR_217-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977905_Receptor_00000000000149_BCR_217-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977906_Receptor_00000000000151_BCR_220-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977907_Receptor_00000000000152_BCR_220-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977908_Receptor_00000000000153_BCR_222-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977909_Receptor_00000000000154_BCR_222-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977910_Receptor_00000000000155_BCR_223-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977911_Receptor_00000000000156_BCR_223-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977912_Receptor_00000000000157_BCR_224-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977913_Receptor_00000000000158_BCR_228-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977914_Receptor_00000000000159_BCR_228-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977915_Receptor_00000000000160_BCR_233-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977916_Receptor_00000000000161_BCR_235-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977917_Receptor_00000000000162_BCR_235-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977918_Receptor_00000000000163_BCR_242-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977919_Receptor_00000000000164_BCR_242-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977920_Receptor_00000000000165_BCR_248-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977921_Receptor_00000000000166_BCR_248-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977922_Receptor_00000000000167_BCR_252-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977923_Receptor_00000000000168_BCR_252-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977924_Receptor_00000000000169_BCR_254-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977925_Receptor_00000000000170_BCR_254-6M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977926_Receptor_00000000000171_BCR_254-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977927_Receptor_00000000000172_BCR_254-9M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977928_Receptor_00000000000173_BCR_254-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977929_Receptor_00000000000174_BCR_259-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977930_Receptor_00000000000175_BCR_259-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977931_Receptor_00000000000176_BCR_262-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977932_Receptor_00000000000177_BCR_262-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977933_Receptor_00000000000178_BCR_263-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977934_Receptor_00000000000179_BCR_263-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977935_Receptor_00000000000180_BCR_264-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977936_Receptor_00000000000181_BCR_264-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977937_Receptor_00000000000182_BCR_266-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977938_Receptor_00000000000183_BCR_270-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977939_Receptor_00000000000184_BCR_273-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977940_Receptor_00000000000185_BCR_273-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977941_Receptor_00000000000186_BCR_274-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977942_Receptor_00000000000187_BCR_274-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977943_Receptor_00000000000188_BCR_277-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977944_Receptor_00000000000189_BCR_277-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977945_Receptor_00000000000190_BCR_278-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977946_Receptor_00000000000191_BCR_278-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977947_Receptor_00000000000192_BCR_279-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977948_Receptor_00000000000193_BCR_279-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977949_Receptor_00000000000194_BCR_284-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977950_Receptor_00000000000195_BCR_284-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977951_Receptor_00000000000196_BCR_289-4W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977952_Receptor_00000000000197_BCR_289-6M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977953_Receptor_00000000000198_BCR_289-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977954_Receptor_00000000000199_BCR_290-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977955_Receptor_00000000000200_BCR_290-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977956_Receptor_00000000000201_BCR_295-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977957_Receptor_00000000000202_BCR_302-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977958_Receptor_00000000000203_BCR_302-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977959_Receptor_00000000000204_BCR_304-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977960_Receptor_00000000000205_BCR_304-6M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977961_Receptor_00000000000206_BCR_304-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977962_Receptor_00000000000207_BCR_305-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977963_Receptor_00000000000208_BCR_305-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977964_Receptor_00000000000209_BCR_306-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977965_Receptor_00000000000210_BCR_306-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977966_Receptor_00000000000211_BCR_313-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977967_Receptor_00000000000212_BCR_313-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977968_Receptor_00000000000213_BCR_317-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977969_Receptor_00000000000214_BCR_317-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977970_Receptor_00000000000215_BCR_319-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977971_Receptor_00000000000216_BCR_319-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977972_Receptor_00000000000217_BCR_323-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977973_Receptor_00000000000218_BCR_323-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977974_Receptor_00000000000219_BCR_325-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977975_Receptor_00000000000220_BCR_325-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977976_Receptor_00000000000221_BCR_122-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977977_Receptor_00000000000222_BCR_125-6-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977978_Receptor_00000000000223_BCR_126-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977979_Receptor_00000000000224_BCR_153-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977980_Receptor_00000000000225_BCR_155-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977981_Receptor_00000000000226_BCR_196-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977982_Receptor_00000000000227_BCR_202-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977983_Receptor_00000000000228_BCR_203-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977984_Receptor_00000000000229_BCR_203-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977985_Receptor_00000000000230_BCR_205-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977986_Receptor_00000000000231_BCR_213-8M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977987_Receptor_00000000000232_BCR_215-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977988_Receptor_00000000000233_BCR_216-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977989_Receptor_00000000000234_BCR_223-TOX.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977990_Receptor_00000000000235_BCR_224-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977991_Receptor_00000000000236_BCR_233-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977992_Receptor_00000000000237_BCR_252-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977993_Receptor_00000000000238_BCR_254-27W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977994_Receptor_00000000000239_BCR_266-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977995_Receptor_00000000000240_BCR_270-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977996_Receptor_00000000000241_BCR_279-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977997_Receptor_00000000000242_BCR_290-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977998_Receptor_00000000000243_BCR_295-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8977999_Receptor_00000000000244_BCR_304-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978000_Receptor_00000000000245_BCR_323-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978001_Receptor_00000000000246_BCR_140-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978002_Receptor_00000000000247_BCR_140-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978003_Receptor_00000000000248_BCR_146-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978004_Receptor_00000000000249_BCR_146-6M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978005_Receptor_00000000000250_BCR_146-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978006_Receptor_00000000000251_BCR_146-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978007_Receptor_00000000000252_BCR_167-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978008_Receptor_00000000000253_BCR_167-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978009_Receptor_00000000000254_BCR_170-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978010_Receptor_00000000000255_BCR_170-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978011_Receptor_00000000000256_BCR_172-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978012_Receptor_00000000000257_BCR_172-Bl.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978013_Receptor_00000000000258_BCR_173-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978014_Receptor_00000000000259_BCR_173-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978015_Receptor_00000000000260_BCR_181-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978016_Receptor_00000000000261_BCR_181-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978017_Receptor_00000000000262_BCR_185-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978018_Receptor_00000000000263_BCR_185-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978019_Receptor_00000000000264_BCR_186-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978020_Receptor_00000000000265_BCR_186-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978021_Receptor_00000000000266_BCR_186-9M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978022_Receptor_00000000000267_BCR_186-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978023_Receptor_00000000000268_BCR_189-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978024_Receptor_00000000000269_BCR_189-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978025_Receptor_00000000000270_BCR_192-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978026_Receptor_00000000000271_BCR_192-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978027_Receptor_00000000000272_BCR_193-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978028_Receptor_00000000000273_BCR_193-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978029_Receptor_00000000000274_BCR_208-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978030_Receptor_00000000000275_BCR_208-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978031_Receptor_00000000000276_BCR_209-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978032_Receptor_00000000000277_BCR_209-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978033_Receptor_00000000000278_BCR_211-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978034_Receptor_00000000000279_BCR_211-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978035_Receptor_00000000000280_BCR_214-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978036_Receptor_00000000000281_BCR_214-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978037_Receptor_00000000000282_BCR_218-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978038_Receptor_00000000000283_BCR_218-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978039_Receptor_00000000000284_BCR_221-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978040_Receptor_00000000000285_BCR_221-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978041_Receptor_00000000000286_BCR_225-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978042_Receptor_00000000000287_BCR_225-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978043_Receptor_00000000000288_BCR_227-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978044_Receptor_00000000000289_BCR_227-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978045_Receptor_00000000000290_BCR_230-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978046_Receptor_00000000000291_BCR_230-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978047_Receptor_00000000000292_BCR_234-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978048_Receptor_00000000000293_BCR_234-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978049_Receptor_00000000000294_BCR_237-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978050_Receptor_00000000000295_BCR_237-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978051_Receptor_00000000000296_BCR_244-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978052_Receptor_00000000000297_BCR_244-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978053_Receptor_00000000000298_BCR_245-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978054_Receptor_00000000000299_BCR_245-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978055_Receptor_00000000000300_BCR_249-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978056_Receptor_00000000000301_BCR_249-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978057_Receptor_00000000000302_BCR_257-7W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978058_Receptor_00000000000303_BCR_257-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978059_Receptor_00000000000304_BCR_260-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978060_Receptor_00000000000305_BCR_260-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978061_Receptor_00000000000306_BCR_261-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978062_Receptor_00000000000307_BCR_261-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978063_Receptor_00000000000308_BCR_269-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978064_Receptor_00000000000309_BCR_269-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978065_Receptor_00000000000310_BCR_283-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978066_Receptor_00000000000311_BCR_283-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978067_Receptor_00000000000312_BCR_285-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978068_Receptor_00000000000313_BCR_285-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978069_Receptor_00000000000314_BCR_285-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978070_Receptor_00000000000315_BCR_293-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978071_Receptor_00000000000316_BCR_293-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978072_Receptor_00000000000317_BCR_296-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978073_Receptor_00000000000318_BCR_296-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978074_Receptor_00000000000319_BCR_298-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978075_Receptor_00000000000320_BCR_298-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978076_Receptor_00000000000321_BCR_299-10W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978077_Receptor_00000000000322_BCR_299-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978078_Receptor_00000000000323_BCR_299-3M.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978079_Receptor_00000000000324_BCR_299-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978080_Receptor_00000000000325_BCR_299-4W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978081_Receptor_00000000000326_BCR_299-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978082_Receptor_00000000000327_BCR_310-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978083_Receptor_00000000000328_BCR_310-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978084_Receptor_00000000000329_BCR_320-8W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978085_Receptor_00000000000330_BCR_320-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978086_Receptor_00000000000331_BCR_326-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978087_Receptor_00000000000332_BCR_326-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978088_Receptor_00000000000333_BCR_327-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978089_Receptor_00000000000334_BCR_327-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978090_Receptor_00000000000335_BCR_328-4W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978091_Receptor_00000000000336_BCR_328-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978092_Receptor_00000000000337_BCR_329-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978093_Receptor_00000000000338_BCR_329-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978094_Receptor_00000000000339_BCR_334-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978095_Receptor_00000000000340_BCR_334-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978096_Receptor_00000000000341_BCR_336-4W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978097_Receptor_00000000000342_BCR_336-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978098_Receptor_00000000000343_BCR_338-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978099_Receptor_00000000000344_BCR_338-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978100_Receptor_00000000000345_BCR_346-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978101_Receptor_00000000000346_BCR_346-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978102_Receptor_00000000000347_BCR_350-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978103_Receptor_00000000000348_BCR_350-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978104_Receptor_00000000000349_BCR_351-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978105_Receptor_00000000000350_BCR_351-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978106_Receptor_00000000000351_BCR_352-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978107_Receptor_00000000000352_BCR_352-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978108_Receptor_00000000000353_BCR_358-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978109_Receptor_00000000000354_BCR_358-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978110_Receptor_00000000000355_BCR_360-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978111_Receptor_00000000000356_BCR_360-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978112_Receptor_00000000000357_BCR_386-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978113_Receptor_00000000000358_BCR_386-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978114_Receptor_00000000000359_BCR_60-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978115_Receptor_00000000000360_BCR_60-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978116_Receptor_00000000000361_BCR_74-2W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978117_Receptor_00000000000362_BCR_74-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978118_Receptor_00000000000363_BCR_90-3W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978119_Receptor_00000000000364_BCR_90-BL.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978120_Receptor_00000000000365_BCR_92-6W.tsv.gz\n",
      "⚠️ Skipping compressed duplicate: GSM8978121_Receptor_00000000000366_BCR_92-BL.tsv.gz\n",
      "🔍 Found 256 count files for GSE296826\n",
      "⚠️ Zero counts detected in merged matrix — applying +1 normalization.\n",
      "✅ Final merged count matrix shape: (24901, 256)\n",
      "✅ Count matrix loaded:\n",
      "(24901, 256)\n",
      "Columns (first 5): ['GSM8977866', 'GSM8977867', 'GSM8977868', 'GSM8977869', 'GSM8977870']\n",
      "📂 Parsing series matrix: geofetch_metadata/GSE296826_series_matrix.txt.gz\n",
      "✅ Parsed 256 samples\n",
      "🔍 Example parsed sample metadata row:\n",
      "{'title': '122-6W', 'geo_accession': 'GSM8977866', 'status': 'Public on May 12 2025', 'submission_date': 'May 12 2025', 'last_update_date': 'May 12 2025', 'type': 'SRA', 'channel_count': '1', 'source_name_ch1': 'Blood', 'organism_ch1': 'Homo sapiens', 'characteristics_ch1': 'tissue: Blood', 'molecule_ch1': 'polyA RNA', 'extract_protocol_ch1': 'Total RNA was isolated via poly‑A mRNA enrichment using oligo‑dT magnetic beads as part of the TruSeq Stranded mRNA kit workflow (oligo‑dT capture, wash, and elution)', 'taxid_ch1': '9606', 'description': 'Library name: Sample_00000000000013', 'data_processing': 'Raw paired‑end FASTQ files were processed with MiXCR (vX.X) to assemble BCR heavy‑chain sequences, call clonotypes, and quantify clone counts and frequencies \\u200b', 'platform_id': 'GPL24676', 'contact_name': 'David,,Gerber', 'contact_email': 'David.Gerber@UTSouthwestern.edu', 'contact_institute': 'UT Southwestern Medical Center', 'contact_address': '5323 Harry Hines Blvd.', 'contact_city': 'Dallas', 'contact_state': 'Texas', 'contact_zip/postal_code': '75390-9082', 'contact_country': 'USA', 'data_row_count': '0', 'instrument_model': 'Illumina NovaSeq 6000', 'library_selection': 'cDNA', 'library_source': 'transcriptomic', 'library_strategy': 'RNA-Seq', 'relation': 'BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN48436107', 'supplementary_file_1': 'ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM8977nnn/GSM8977866/suppl/GSM8977866_Receptor_00000000000111_BCR_122-6W.tsv.gz'}\n",
      "🧠 Building prompt for batch 1 with 100 samples...\n",
      "📝 Prompt preview (first 500 chars): \n",
      "You are a biomedical data curator.\n",
      "\n",
      "Below is a list of samples from GEO study GSE296826.\n",
      "Each sample contains metadata fields like geo_accession, title, description, and characteristics.\n",
      "\n",
      "Your task:\n",
      "- Identify which metadata field contains the sample titles that match these count matrix sample labels.\n",
      "- The count matrix sample labels are:\n",
      "[\n",
      "  \"GSM8977866\",\n",
      "  \"GSM8977867\",\n",
      "  \"GSM8977868\",\n",
      "  \"GSM8977869\",\n",
      "  \"GSM8977870\",\n",
      "  \"GSM8977871\",\n",
      "  \"GSM8977872\",\n",
      "  \"GSM8977873\",\n",
      "  \"GSM8977874\",\n",
      "  \"GSM897787...\n",
      "💬 Raw LLM output (first 500 chars): ```json\n",
      "{\n",
      "  \"GSM8977866\": {\n",
      "    \"geo_accession\": \"GSM8977866\",\n",
      "    \"condition\": \"treated\",\n",
      "    \"time\": \"6W\"\n",
      "  },\n",
      "  \"GSM8977867\": {\n",
      "    \"geo_accession\": \"GSM8977867\",\n",
      "    \"condition\": \"control\",\n",
      "    \"time\": null\n",
      "  },\n",
      "  \"GSM8977868\": {\n",
      "    \"geo_accession\": \"GSM8977868\",\n",
      "    \"condition\": \"treated\",\n",
      "    \"time\": \"3M\"\n",
      "  },\n",
      "  \"GSM8977869\": {\n",
      "    \"geo_accession\": \"GSM8977869\",\n",
      "    \"condition\": \"control\",\n",
      "    \"time\": null\n",
      "  },\n",
      "  \"GSM8977870\": {\n",
      "    \"geo_accession\": \"GSM8977870\",\n",
      "    \"condition\": \"treated...\n"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "❌ Could not find JSON array in LLM output.",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[175], line 34\u001b[0m\n\u001b[0;32m     31\u001b[0m sample_labels_from_counts \u001b[38;5;241m=\u001b[39m counts_df\u001b[38;5;241m.\u001b[39mcolumns\n\u001b[0;32m     33\u001b[0m \u001b[38;5;66;03m# LLM label mapping\u001b[39;00m\n\u001b[1;32m---> 34\u001b[0m group_label_mapping \u001b[38;5;241m=\u001b[39m \u001b[43mextract_labels_using_llm\u001b[49m\u001b[43m(\u001b[49m\u001b[43mseries_matrix_path\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mgse_id\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mclient\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43msample_labels_from_counts\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m     36\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m✅ Group label mapping keys: \u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[38;5;28mlist\u001b[39m(group_label_mapping\u001b[38;5;241m.\u001b[39mkeys())[:\u001b[38;5;241m5\u001b[39m]\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m     37\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m✅ Group label mapping size: \u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[38;5;28mlen\u001b[39m(group_label_mapping)\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m)\n",
      "Cell \u001b[1;32mIn[153], line 78\u001b[0m, in \u001b[0;36mextract_labels_using_llm\u001b[1;34m(series_matrix_path, gse_id, client, sample_labels_from_counts, batch_size)\u001b[0m\n\u001b[0;32m     75\u001b[0m content \u001b[38;5;241m=\u001b[39m response\u001b[38;5;241m.\u001b[39mchoices[\u001b[38;5;241m0\u001b[39m]\u001b[38;5;241m.\u001b[39mmessage\u001b[38;5;241m.\u001b[39mcontent\u001b[38;5;241m.\u001b[39mstrip()\n\u001b[0;32m     76\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m💬 Raw LLM output (first 500 chars): \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mcontent[:\u001b[38;5;241m500\u001b[39m]\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m...\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m---> 78\u001b[0m batch_metadata \u001b[38;5;241m=\u001b[39m \u001b[43mextract_json_block\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcontent\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m     79\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m✅ Parsed JSON batch metadata for \u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[38;5;28mlen\u001b[39m(batch_metadata)\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m samples\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m     81\u001b[0m all_metadata\u001b[38;5;241m.\u001b[39mupdate(batch_metadata)\n",
      "Cell \u001b[1;32mIn[174], line 15\u001b[0m, in \u001b[0;36mextract_json_block\u001b[1;34m(content)\u001b[0m\n\u001b[0;32m     13\u001b[0m json_match \u001b[38;5;241m=\u001b[39m re\u001b[38;5;241m.\u001b[39msearch(\u001b[38;5;124mr\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\\\u001b[39m\u001b[38;5;124m[\u001b[39m\u001b[38;5;124m\\\u001b[39m\u001b[38;5;124ms*\u001b[39m\u001b[38;5;124m{\u001b[39m\u001b[38;5;124m.*}\u001b[39m\u001b[38;5;124m\\\u001b[39m\u001b[38;5;124ms*\u001b[39m\u001b[38;5;124m\\\u001b[39m\u001b[38;5;124m]\u001b[39m\u001b[38;5;124m\"\u001b[39m, content, re\u001b[38;5;241m.\u001b[39mDOTALL)\n\u001b[0;32m     14\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m json_match:\n\u001b[1;32m---> 15\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m❌ Could not find JSON array in LLM output.\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m     17\u001b[0m json_str \u001b[38;5;241m=\u001b[39m json_match\u001b[38;5;241m.\u001b[39mgroup(\u001b[38;5;241m0\u001b[39m)\n\u001b[0;32m     18\u001b[0m parsed_list \u001b[38;5;241m=\u001b[39m json\u001b[38;5;241m.\u001b[39mloads(json_str)\n",
      "\u001b[1;31mValueError\u001b[0m: ❌ Could not find JSON array in LLM output."
     ]
    }
   ],
   "source": [
    "download_dir = download_and_extract_gse(gse_id)\n",
    "\n",
    "# Try to find and merge count files\n",
    "counts_df = find_and_merge_count_files(gse_id)\n",
    "series_matrix_path=f\"geofetch_metadata/{gse_id}_series_matrix.txt.gz\"\n",
    "\n",
    "if counts_df is None:\n",
    "    print(f\"⚠️ No count matrix found for {gse_id}. Trying to load DE results directly...\")\n",
    "    gse_dir = Path(\"rna_seq_analysis\")/gse_id\n",
    "    for file in gse_dir.glob(\"*\"):\n",
    "        if file.suffix in [\".txt\", \".tsv\", \".csv\", \".gz\"]:\n",
    "            de_df = load_de_result_file(file)\n",
    "            if de_df is not None:\n",
    "                print(\"✅ DE result loaded. Skipping DESeq2 analysis.\")\n",
    "                output_csv = f\"rna_seq_analysis/{gse_id}_de_results_direct.csv\"\n",
    "                de_df.to_csv(output_csv)\n",
    "                deg_paths[gse_id] = output_csv\n",
    "                filtered_df = de_df[\n",
    "                    (de_df.get(\"padj\", 1) < 0.05) & \n",
    "                    (abs(de_df.get(\"log2FoldChange\", 0)) > 1)\n",
    "                ]\n",
    "                print(f\"🧬 DE results from file — {filtered_df.shape[0]} significant genes found\")\n",
    "                #return {**state, \"deg_results\": deg_paths}\n",
    "    print(f\"❌ No DE result files found in {gse_dir}\")\n",
    "    #return state\n",
    "print(\"✅ Count matrix loaded:\")\n",
    "print(counts_df.shape)\n",
    "print(\"Columns (first 5):\", list(counts_df.columns)[:5])\n",
    "\n",
    "# Extract labels from count matrix columns\n",
    "sample_labels_from_counts = counts_df.columns\n",
    "\n",
    "# LLM label mapping\n",
    "group_label_mapping = extract_labels_using_llm(series_matrix_path, gse_id, client, sample_labels_from_counts)\n",
    "\n",
    "print(f\"✅ Group label mapping keys: {list(group_label_mapping.keys())[:5]}\")\n",
    "print(f\"✅ Group label mapping size: {len(group_label_mapping)}\")\n",
    "\n",
    "# Create metadata DataFrame\n",
    "metadata_df = pd.DataFrame.from_dict(group_label_mapping, orient=\"index\")\n",
    "metadata_df.index.name = \"sample_geo_accession\"\n",
    "print(\"✅ Metadata DataFrame:\")\n",
    "print(metadata_df.head())\n",
    "\n",
    "# Align count matrix columns to metadata\n",
    "print(\"Before alignment:\")\n",
    "print(\"Counts columns:\", list(counts_df.columns)[:5])\n",
    "print(\"Metadata index:\", list(metadata_df.index)[:5])\n",
    "### match counts and metadata sample ID\n",
    "valid_samples = metadata_df.index.intersection(counts_df.columns)\n",
    "print(f\"✅ Valid samples after intersection: {list(valid_samples)}\")\n",
    "\n",
    "if len(valid_samples) == 0:\n",
    "    raise ValueError(\"❌ No valid samples left after alignment — cannot run DESeq2.\")\n",
    "\n",
    "counts_df = counts_df[valid_samples]\n",
    "metadata_df = metadata_df.loc[valid_samples]\n",
    "### debug to see if both metdata is complete or match with counts data \n",
    "tcounts = counts_df.T\n",
    "print(\"✅ Final counts shape:\", tcounts.shape)\n",
    "print(\"✅ Final metadata shape:\", metadata_df.shape)\n",
    "\n",
    "### convert counts_df same as tcounts #######\n",
    "counts_df=counts_df.T ### both matrix should use sample id as rows, genes in cols\n",
    "# Filter metadata to only those with non-null condition\n",
    "metadata_valid = metadata_df[metadata_df[\"condition\"].notnull()]\n",
    "# Get the intersection of sample names that are in both counts_df and valid metadata\n",
    "valid_samples = metadata_valid.index.intersection(counts_df.index)\n",
    "# Subset counts and metadata\n",
    "counts_df = counts_df.loc[valid_samples]\n",
    "metadata_df = metadata_valid.loc[valid_samples]\n",
    "\n",
    "# Sanity check\n",
    "print(f\"✅ Number of valid samples: {len(valid_samples)}\")\n",
    "print(\"Counts shape:\", counts_df.shape)\n",
    "print(\"Metadata shape:\", metadata_df.shape)\n",
    "dds = DeseqDataSet(\n",
    "    counts=counts_df,\n",
    "    metadata=metadata_df,\n",
    "    design_factors=\"condition\",\n",
    "    refit_cooks=True,\n",
    "    inference=DefaultInference(n_cpus=4)\n",
    ")\n",
    "dds.fit_size_factors()\n",
    "dds.fit_genewise_dispersions()\n",
    "dds.fit_dispersion_trend()\n",
    "dds.fit_dispersion_prior()\n",
    "dds.fit_MAP_dispersions()\n",
    "dds.fit_LFC()\n",
    "dds.calculate_cooks()\n",
    "\n",
    "if dds.refit_cooks:\n",
    "    dds.refit()\n",
    "\n",
    "stat_res = DeseqStats(\n",
    "    dds, alpha=0.05, contrast=[\"condition\", \"treated\", \"control\"],\n",
    "    cooks_filter=True, independent_filter=True\n",
    ")\n",
    "stat_res.summary()\n",
    "\n",
    "res = stat_res.results_df\n",
    "print(\"✅ DESeq2 result preview:\")\n",
    "print(res.head())\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "id": "4ca0ab51-8ac7-472d-91ea-8183b0b5e88a",
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_json_block(content):\n",
    "    content = content.strip()\n",
    "    if not content:\n",
    "        raise ValueError(\"❌ LLM returned empty content — cannot parse JSON.\")\n",
    "\n",
    "    # Try direct load first\n",
    "    try:\n",
    "        return json.loads(content)\n",
    "    except json.JSONDecodeError:\n",
    "        pass  # We'll try to clean it up\n",
    "\n",
    "    # Use regex to extract JSON array\n",
    "    json_match = re.search(r\"\\[\\s*{.*}\\s*\\]\", content, re.DOTALL)\n",
    "    if not json_match:\n",
    "        raise ValueError(\"❌ Could not find JSON array in LLM output.\")\n",
    "\n",
    "    json_str = json_match.group(0)\n",
    "    parsed_list = json.loads(json_str)\n",
    "\n",
    "    # Convert to dict keyed by geo_accession\n",
    "    return {\n",
    "        entry[\"geo_accession\"]: {k: v for k, v in entry.items() if k != \"geo_accession\"}\n",
    "        for entry in parsed_list\n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "id": "646abb29-6d60-4958-88fa-200939762b28",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'content' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[173], line 1\u001b[0m\n\u001b[1;32m----> 1\u001b[0m batch_metadata \u001b[38;5;241m=\u001b[39m extract_json_block(\u001b[43mcontent\u001b[49m)\n\u001b[0;32m      2\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m✅ Parsed JSON batch metadata for \u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[38;5;28mlen\u001b[39m(batch_metadata)\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m samples\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "\u001b[1;31mNameError\u001b[0m: name 'content' is not defined"
     ]
    }
   ],
   "source": [
    "batch_metadata = extract_json_block(content)\n",
    "print(f\"✅ Parsed JSON batch metadata for {len(batch_metadata)} samples\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "id": "09e59da5-6b37-42d8-8a37-e3abb751c550",
   "metadata": {},
   "outputs": [],
   "source": [
    "#cdf=counts_df\n",
    "#mtf=metadata_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "id": "1b807fe5-799e-4b6c-b270-f31d22336728",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "✅ Number of valid samples: 100\n",
      "Counts shape: (100, 24901)\n",
      "Metadata shape: (100, 3)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\difen\\venv1\\Lib\\site-packages\\anndata\\_core\\aligned_df.py:68: ImplicitModificationWarning: Transforming to str index.\n",
      "  warnings.warn(\"Transforming to str index.\", ImplicitModificationWarning)\n",
      "C:\\Users\\difen\\AppData\\Local\\Temp\\ipykernel_38836\\2318307309.py:16: DeprecationWarning: design_factors is deprecated and will soon be removed.Please consider providing a formulaic formula using the design argumentinstead.\n",
      "  dds = DeseqDataSet(\n"
     ]
    }
   ],
   "source": [
    "counts_df=counts_df.T ### both matrix should use sample id as rows, genes in cols\n",
    "# Filter metadata to only those with non-null condition\n",
    "metadata_valid = metadata_df[metadata_df[\"condition\"].notnull()]\n",
    "# Get the intersection of sample names that are in both counts_df and valid metadata\n",
    "valid_samples = metadata_valid.index.intersection(counts_df.index)\n",
    "# Subset counts and metadata\n",
    "counts_df = counts_df.loc[valid_samples]\n",
    "metadata_df = metadata_valid.loc[valid_samples]\n",
    "\n",
    "# Sanity check\n",
    "print(f\"✅ Number of valid samples: {len(valid_samples)}\")\n",
    "print(\"Counts shape:\", counts_df.shape)\n",
    "print(\"Metadata shape:\", metadata_df.shape)\n",
    "dds = DeseqDataSet(\n",
    "    counts=counts_df,\n",
    "    metadata=metadata_df,\n",
    "    design_factors=\"condition\",\n",
    "    refit_cooks=True,\n",
    "    inference=DefaultInference(n_cpus=4)\n",
    ")\n",
    "dds.fit_size_factors()\n",
    "dds.fit_genewise_dispersions()\n",
    "dds.fit_dispersion_trend()\n",
    "dds.fit_dispersion_prior()\n",
    "dds.fit_MAP_dispersions()\n",
    "dds.fit_LFC()\n",
    "dds.calculate_cooks()\n",
    "\n",
    "if dds.refit_cooks:\n",
    "    dds.refit()\n",
    "\n",
    "stat_res = DeseqStats(\n",
    "    dds, alpha=0.05, contrast=[\"condition\", \"treated\", \"control\"],\n",
    "    cooks_filter=True, independent_filter=True\n",
    ")\n",
    "stat_res.summary()\n",
    "\n",
    "res = stat_res.results_df\n",
    "print(\"✅ DESeq2 result preview:\")\n",
    "print(res.head())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "74c2dd59-1fa4-4d41-a732-96e9af884082",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d5b51694-bae7-463a-9c31-62cc34a2fa28",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "4515410f-1477-4ee5-b7f6-831f7794d3a7",
   "metadata": {},
   "outputs": [],
   "source": [
    "import GEOparse\n",
    "def fetch_geo_metadata(gse_id):\n",
    "    \"\"\"\n",
    "    Fetch metadata for a GEO Series using GEOparse.\n",
    "\n",
    "    Returns a dictionary with key study fields.\n",
    "    \"\"\"\n",
    "    gse = GEOparse.get_GEO(geo=gse_id, destdir=\".\")\n",
    "    metadata = {\n",
    "        \"Title\": gse.metadata.get(\"title\", [\"\"])[0],\n",
    "        \"Summary\": gse.metadata.get(\"summary\", [\"\"])[0],\n",
    "        \"Overall Design\": gse.metadata.get(\"overall_design\", [\"\"])[0],\n",
    "        \"Type\": gse.metadata.get(\"type\", [\"\"])[0],\n",
    "        \"Platform\": \", \".join(gse.metadata.get(\"platform_id\", []))\n",
    "    }\n",
    "    return metadata\n",
    "\n",
    "def construct_llm_prompt(gse_id, metadata):\n",
    "    prompt = f\"\"\"\n",
    "    You are a biomedical data expert.\n",
    "    \n",
    "    Based on the following metadata for GEO Series **{gse_id}**, identify the **primary data type** used in the study.\n",
    "    \n",
    "    Choose from:\n",
    "    - RNA-seq\n",
    "    - Microarray\n",
    "    - ChIP-seq\n",
    "    - DNA methylation sequencing\n",
    "    - RT-PCR\n",
    "    - Single-cell RNA-seq\n",
    "    - Spatial transcriptomics (Visium)\n",
    "    - Other (specify)\n",
    "    \n",
    "    Respond in this JSON format:\n",
    "    ```json\n",
    "    {{\n",
    "      \"data_type\": \"...\",\n",
    "      \"explanation\": \"...\"\n",
    "    }}\n",
    "    Metadata:\n",
    "    Title: {metadata['Title']}\n",
    "    Summary: {metadata['Summary']}\n",
    "    Overall Design: {metadata['Overall Design']}\n",
    "    Type: {metadata['Type']}\n",
    "    Platform: {metadata['Platform']}\n",
    "\n",
    "    \"\"\"\n",
    "    return prompt\n",
    "\n",
    "def identify_geo_data_type_via_llm(gse_id, client):\n",
    "    \"\"\"\n",
    "    Args:\n",
    "        gse_id (str): The GEO accession (e.g., \"GSE292329\")\n",
    "        client: Your OpenAI-compatible client\n",
    "    \n",
    "    Returns:\n",
    "    dict with \"data_type\" and \"explanation\"\n",
    "    \"\"\"\n",
    "    metadata = fetch_geo_metadata(gse_id)\n",
    "    prompt = construct_llm_prompt(gse_id, metadata)\n",
    "    \n",
    "    response = client.chat.completions.create(\n",
    "        model=\"gpt-4o\",\n",
    "        messages=[{\"role\": \"user\", \"content\": prompt}],\n",
    "        temperature=0.2\n",
    "    )\n",
    "    return response.choices[0].message.content.strip()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "60f28671-ecb5-4d42-9e09-b64c1352cd0a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "🔍 Scanning for compressed files in: rna_seq_analysis\\GSE292329\n",
      "Running DESeq2 for: GSE292329\n",
      "📥 Downloading GSE292329_FPKM.txt.gz → rna_seq_analysis/GSE292329/GSE292329_FPKM.txt.gz\n",
      "🗂️ Extracting GZ...\n",
      "⚠️ Skipping compressed duplicate: GSE292329_FPKM.txt.gz\n",
      "🔍 Found 1 count files for GSE292329\n",
      "📄 Only one uncompressed count file found: GSE292329_FPKM.txt — attempting to load without merging.\n",
      "⚠️ Zero counts detected — applying +1 normalization.\n",
      "✅ Single count file processed. Shape: (56181, 48)\n",
      "                     K401   K402   K403   K404   K405   K406   M401   M402  \\\n",
      "gene_id                                                                      \n",
      "ENSMUSG00000024653  23844  15269  14003  25157  17148  30449  14814  17045   \n",
      "ENSMUSG00000076258  12570   8903   8039   2922   1263   7481   3193   4599   \n",
      "ENSMUSG00000022097   7202   7878   7523  12282   6747  12200   9419   4442   \n",
      "ENSMUSG00000064351   5445   5709   4933   5987   5747   5222   4334   4408   \n",
      "ENSMUSG00000064370   3171   3103   2700   3340   3481   3020   2238   1912   \n",
      "...                   ...    ...    ...    ...    ...    ...    ...    ...   \n",
      "novel.552               1      1      1      1      1      1      1      1   \n",
      "novel.589               1      1      1      1      1      1      1      1   \n",
      "novel.639               1      1      1      1      1      1      1      1   \n",
      "novel.663               1      1      1      1      1      1      1      1   \n",
      "novel.681               1      1      1      1      1      1      1      1   \n",
      "\n",
      "                    M403   M404  ...  K1003  K1004  K1005  K1006  M1001  \\\n",
      "gene_id                          ...                                      \n",
      "ENSMUSG00000024653  6403  16965  ...  25254  23527  25610  21750    753   \n",
      "ENSMUSG00000076258  8288   2606  ...  18065   4070   4370   3441   4701   \n",
      "ENSMUSG00000022097  7131   7148  ...   9749   8867  10050   8637    143   \n",
      "ENSMUSG00000064351  3642   4211  ...   4852   4318   4148   3992   3706   \n",
      "ENSMUSG00000064370  1720   2062  ...   2699   2416   2535   2422   1316   \n",
      "...                  ...    ...  ...    ...    ...    ...    ...    ...   \n",
      "novel.552              1      1  ...      1      1      1      1      1   \n",
      "novel.589              1      1  ...      1      1      1      1      1   \n",
      "novel.639              1      1  ...      1      1      1      1      1   \n",
      "novel.663              1      1  ...      1      1      1      1      1   \n",
      "novel.681              1      1  ...      1      1      1      1      1   \n",
      "\n",
      "                    M1002  M1003  M1004  M1005  M1006  \n",
      "gene_id                                                \n",
      "ENSMUSG00000024653   1343   4453   5260   5207   6155  \n",
      "ENSMUSG00000076258   7000   3652   6251   7886   8779  \n",
      "ENSMUSG00000022097   2140   1574    887   2100   3442  \n",
      "ENSMUSG00000064351   3419   4014   4523   4136   3465  \n",
      "ENSMUSG00000064370   1430   1578   1733   1606   1481  \n",
      "...                   ...    ...    ...    ...    ...  \n",
      "novel.552               1      1      1      1      1  \n",
      "novel.589               1      1      1      1      1  \n",
      "novel.639               1      1      1      1      1  \n",
      "novel.663               1      1      1      1      1  \n",
      "novel.681               1      1      1      1      1  \n",
      "\n",
      "[56181 rows x 48 columns]\n",
      "Error running DESeq2 for GSE292329: name 'series_matrix_path' is not defined\n",
      "\n",
      "✅ DEG Results by GSE ID:\n"
     ]
    }
   ],
   "source": [
    "initial_state: Dict = {\n",
    "    \"query_gene\": \"\",\n",
    "    \"pathway_info\": {\"genes\": [], \"drugs\": [], \"pathways\": []},\n",
    "    \"metadata\": {},\n",
    "    \"gse_list\": [],\n",
    "    \"research_plan\": \"\",\n",
    "    \"selected_gse\": \"GSE292329\" ,#\"GSE94840\", #\"GSE236204\",\n",
    "  #  \"selected_gses\": [\"GSE242272\"],\n",
    "    \"deg_results\": {}\n",
    "}\n",
    "# Run the DEG analysis directly\n",
    "result_state = run_deg_for_single_gse_debug(initial_state)\n",
    "\n",
    "# Output the DEG results\n",
    "print(\"\\n✅ DEG Results by GSE ID:\")\n",
    "for gse_id, path in result_state.get(\"deg_results\", {}).items():\n",
    "    print(f\"- {gse_id}: {path}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "37c6e1f8-333d-428a-b242-9ecee71cbe51",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "🔍 Scanning for compressed files in: rna_seq_analysis\\GSE259434\n",
      "Running DESeq2 for: GSE259434\n",
      "📥 Downloading GSE259434_log2cpm.csv.gz → rna_seq_analysis/GSE259434/GSE259434_log2cpm.csv.gz\n",
      "🗂️ Extracting GZ...\n",
      "⚠️ Skipping compressed duplicate: GSE259434_log2cpm.csv.gz\n",
      "🔍 Found 1 count files for GSE259434\n",
      "📄 Only one uncompressed count file found: GSE259434_log2cpm.csv — attempting to load without merging.\n",
      "⚠️ Zero counts detected — applying +1 normalization.\n",
      "✅ Single count file processed. Shape: (18563, 15)\n",
      "                       M1-IL2  M1-IL7  M1-IL15  M1-R  M3-IL2  M3-IL7  M3-R  \\\n",
      "Unnamed: 0                                                                   \n",
      "ENSMUSG00000000001.4        9       8        9     9       9       9     9   \n",
      "ENSMUSG00000000028.15       4       3        2     2       4       3     2   \n",
      "ENSMUSG00000000037.16       0       1        0    -2       0      -1     0   \n",
      "ENSMUSG00000000056.7        6       5        6     6       6       5     6   \n",
      "ENSMUSG00000000058.6        1       1        1     1       1       1     1   \n",
      "...                       ...     ...      ...   ...     ...     ...   ...   \n",
      "ENSMUSG00000118318.1        1       2        2     2       2       1     2   \n",
      "ENSMUSG00000118332.1        6       5        6     6       6       5     6   \n",
      "ENSMUSG00000118346.1        5       5        4     5       4       4     5   \n",
      "ENSMUSG00000118353.1        0      -1       -1    -2      -2      -1     0   \n",
      "ENSMUSG00000118382.1        0      -1        1     0      -1       0    -1   \n",
      "\n",
      "                       M4-IL2  M4-IL7  M4-IL15  M4-R  M5-IL2  M5-IL7  M5-IL15  \\\n",
      "Unnamed: 0                                                                      \n",
      "ENSMUSG00000000001.4        9       9        9     9       9       9        9   \n",
      "ENSMUSG00000000028.15       5       3        4     2       4       3        3   \n",
      "ENSMUSG00000000037.16       0      -2        0     1      -1      -2        0   \n",
      "ENSMUSG00000000056.7        5       6        6     6       6       5        5   \n",
      "ENSMUSG00000000058.6        1       1        1     1       1       1        1   \n",
      "...                       ...     ...      ...   ...     ...     ...      ...   \n",
      "ENSMUSG00000118318.1        2       1        1     2       2       1        1   \n",
      "ENSMUSG00000118332.1        5       5        5     5       6       5        6   \n",
      "ENSMUSG00000118346.1        4       5        4     4       5       5        5   \n",
      "ENSMUSG00000118353.1       -1      -2       -1    -1       0      -1       -1   \n",
      "ENSMUSG00000118382.1        0      -2        0     1       0       1        0   \n",
      "\n",
      "                       M5-R  \n",
      "Unnamed: 0                   \n",
      "ENSMUSG00000000001.4      8  \n",
      "ENSMUSG00000000028.15     2  \n",
      "ENSMUSG00000000037.16     0  \n",
      "ENSMUSG00000000056.7      6  \n",
      "ENSMUSG00000000058.6      1  \n",
      "...                     ...  \n",
      "ENSMUSG00000118318.1      2  \n",
      "ENSMUSG00000118332.1      6  \n",
      "ENSMUSG00000118346.1      5  \n",
      "ENSMUSG00000118353.1     -2  \n",
      "ENSMUSG00000118382.1      1  \n",
      "\n",
      "[18563 rows x 15 columns]\n",
      "Error running DESeq2 for GSE259434: name 'series_matrix_path' is not defined\n",
      "\n",
      "✅ DEG Results by GSE ID:\n"
     ]
    }
   ],
   "source": [
    "initial_state: Dict = {\n",
    "    \"query_gene\": \"\",\n",
    "    \"pathway_info\": {\"genes\": [], \"drugs\": [], \"pathways\": []},\n",
    "    \"metadata\": {},\n",
    "    \"gse_list\": [],\n",
    "    \"research_plan\": \"\",\n",
    "    \"series_matrix_path\":'geofetch_metadata/GSE259434_series_matrix.txt.gz',\n",
    "    \"selected_gse\": \"GSE259434\", #\"GSE236204\",\n",
    "  #  \"selected_gses\": [\"GSE242272\"],\n",
    "    \"deg_results\": {}\n",
    "}\n",
    "# Run the DEG analysis directly\n",
    "result_state = run_deg_for_single_gse_debug(initial_state)\n",
    "\n",
    "# Output the DEG results\n",
    "print(\"\\n✅ DEG Results by GSE ID:\")\n",
    "for gse_id, path in result_state.get(\"deg_results\", {}).items():\n",
    "    print(f\"- {gse_id}: {path}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "970eedc9-153b-4627-9d26-40bc3e4a4b0a",
   "metadata": {},
   "outputs": [
    {
     "ename": "MissingSchema",
     "evalue": "Invalid URL 'None': No scheme supplied. Perhaps you meant https://None?",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mMissingSchema\u001b[0m                             Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[13], line 2\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[38;5;28mid\u001b[39m\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mGSE292393\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m----> 2\u001b[0m \u001b[43mdownload_series_matrix\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mid\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43moutput_dir\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mgeofetch_metadata\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "Cell \u001b[1;32mIn[9], line 602\u001b[0m, in \u001b[0;36mdownload_series_matrix\u001b[1;34m(gse_id, output_dir)\u001b[0m\n\u001b[0;32m    600\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m output_path\u001b[38;5;241m.\u001b[39mexists():\n\u001b[0;32m    601\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mstr\u001b[39m(output_path)\n\u001b[1;32m--> 602\u001b[0m response \u001b[38;5;241m=\u001b[39m \u001b[43mrequests\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mstream\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m)\u001b[49m\n\u001b[0;32m    603\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m response\u001b[38;5;241m.\u001b[39mstatus_code \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m200\u001b[39m:\n\u001b[0;32m    604\u001b[0m     \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mopen\u001b[39m(output_path, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mwb\u001b[39m\u001b[38;5;124m\"\u001b[39m) \u001b[38;5;28;01mas\u001b[39;00m f:\n",
      "File \u001b[1;32m~\\venv1\\Lib\\site-packages\\requests\\api.py:73\u001b[0m, in \u001b[0;36mget\u001b[1;34m(url, params, **kwargs)\u001b[0m\n\u001b[0;32m     62\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mget\u001b[39m(url, params\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs):\n\u001b[0;32m     63\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124mr\u001b[39m\u001b[38;5;124;03m\"\"\"Sends a GET request.\u001b[39;00m\n\u001b[0;32m     64\u001b[0m \n\u001b[0;32m     65\u001b[0m \u001b[38;5;124;03m    :param url: URL for the new :class:`Request` object.\u001b[39;00m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m     70\u001b[0m \u001b[38;5;124;03m    :rtype: requests.Response\u001b[39;00m\n\u001b[0;32m     71\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[1;32m---> 73\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mrequest\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mget\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mparams\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mparams\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32m~\\venv1\\Lib\\site-packages\\requests\\api.py:59\u001b[0m, in \u001b[0;36mrequest\u001b[1;34m(method, url, **kwargs)\u001b[0m\n\u001b[0;32m     55\u001b[0m \u001b[38;5;66;03m# By using the 'with' statement we are sure the session is closed, thus we\u001b[39;00m\n\u001b[0;32m     56\u001b[0m \u001b[38;5;66;03m# avoid leaving sockets open which can trigger a ResourceWarning in some\u001b[39;00m\n\u001b[0;32m     57\u001b[0m \u001b[38;5;66;03m# cases, and look like a memory leak in others.\u001b[39;00m\n\u001b[0;32m     58\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m sessions\u001b[38;5;241m.\u001b[39mSession() \u001b[38;5;28;01mas\u001b[39;00m session:\n\u001b[1;32m---> 59\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43msession\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[43m(\u001b[49m\u001b[43mmethod\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43murl\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32m~\\venv1\\Lib\\site-packages\\requests\\sessions.py:575\u001b[0m, in \u001b[0;36mSession.request\u001b[1;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[0;32m    562\u001b[0m \u001b[38;5;66;03m# Create the Request.\u001b[39;00m\n\u001b[0;32m    563\u001b[0m req \u001b[38;5;241m=\u001b[39m Request(\n\u001b[0;32m    564\u001b[0m     method\u001b[38;5;241m=\u001b[39mmethod\u001b[38;5;241m.\u001b[39mupper(),\n\u001b[0;32m    565\u001b[0m     url\u001b[38;5;241m=\u001b[39murl,\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    573\u001b[0m     hooks\u001b[38;5;241m=\u001b[39mhooks,\n\u001b[0;32m    574\u001b[0m )\n\u001b[1;32m--> 575\u001b[0m prep \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mprepare_request\u001b[49m\u001b[43m(\u001b[49m\u001b[43mreq\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    577\u001b[0m proxies \u001b[38;5;241m=\u001b[39m proxies \u001b[38;5;129;01mor\u001b[39;00m {}\n\u001b[0;32m    579\u001b[0m settings \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mmerge_environment_settings(\n\u001b[0;32m    580\u001b[0m     prep\u001b[38;5;241m.\u001b[39murl, proxies, stream, verify, cert\n\u001b[0;32m    581\u001b[0m )\n",
      "File \u001b[1;32m~\\venv1\\Lib\\site-packages\\requests\\sessions.py:484\u001b[0m, in \u001b[0;36mSession.prepare_request\u001b[1;34m(self, request)\u001b[0m\n\u001b[0;32m    481\u001b[0m     auth \u001b[38;5;241m=\u001b[39m get_netrc_auth(request\u001b[38;5;241m.\u001b[39murl)\n\u001b[0;32m    483\u001b[0m p \u001b[38;5;241m=\u001b[39m PreparedRequest()\n\u001b[1;32m--> 484\u001b[0m \u001b[43mp\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mprepare\u001b[49m\u001b[43m(\u001b[49m\n\u001b[0;32m    485\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmethod\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmethod\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mupper\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    486\u001b[0m \u001b[43m    \u001b[49m\u001b[43murl\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    487\u001b[0m \u001b[43m    \u001b[49m\u001b[43mfiles\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mfiles\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    488\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdata\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdata\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    489\u001b[0m \u001b[43m    \u001b[49m\u001b[43mjson\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mjson\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    490\u001b[0m \u001b[43m    \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmerge_setting\u001b[49m\u001b[43m(\u001b[49m\n\u001b[0;32m    491\u001b[0m \u001b[43m        \u001b[49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mheaders\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mheaders\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdict_class\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mCaseInsensitiveDict\u001b[49m\n\u001b[0;32m    492\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    493\u001b[0m \u001b[43m    \u001b[49m\u001b[43mparams\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmerge_setting\u001b[49m\u001b[43m(\u001b[49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mparams\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mparams\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    494\u001b[0m \u001b[43m    \u001b[49m\u001b[43mauth\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmerge_setting\u001b[49m\u001b[43m(\u001b[49m\u001b[43mauth\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mauth\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    495\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcookies\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmerged_cookies\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    496\u001b[0m \u001b[43m    \u001b[49m\u001b[43mhooks\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmerge_hooks\u001b[49m\u001b[43m(\u001b[49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mhooks\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mhooks\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    497\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    498\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m p\n",
      "File \u001b[1;32m~\\venv1\\Lib\\site-packages\\requests\\models.py:367\u001b[0m, in \u001b[0;36mPreparedRequest.prepare\u001b[1;34m(self, method, url, headers, files, data, params, auth, cookies, hooks, json)\u001b[0m\n\u001b[0;32m    364\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124;03m\"\"\"Prepares the entire request with the given parameters.\"\"\"\u001b[39;00m\n\u001b[0;32m    366\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mprepare_method(method)\n\u001b[1;32m--> 367\u001b[0m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mprepare_url\u001b[49m\u001b[43m(\u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mparams\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    368\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mprepare_headers(headers)\n\u001b[0;32m    369\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mprepare_cookies(cookies)\n",
      "File \u001b[1;32m~\\venv1\\Lib\\site-packages\\requests\\models.py:438\u001b[0m, in \u001b[0;36mPreparedRequest.prepare_url\u001b[1;34m(self, url, params)\u001b[0m\n\u001b[0;32m    435\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m InvalidURL(\u001b[38;5;241m*\u001b[39me\u001b[38;5;241m.\u001b[39margs)\n\u001b[0;32m    437\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m scheme:\n\u001b[1;32m--> 438\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m MissingSchema(\n\u001b[0;32m    439\u001b[0m         \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mInvalid URL \u001b[39m\u001b[38;5;132;01m{\u001b[39;00murl\u001b[38;5;132;01m!r}\u001b[39;00m\u001b[38;5;124m: No scheme supplied. \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m    440\u001b[0m         \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mPerhaps you meant https://\u001b[39m\u001b[38;5;132;01m{\u001b[39;00murl\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m?\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m    441\u001b[0m     )\n\u001b[0;32m    443\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m host:\n\u001b[0;32m    444\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m InvalidURL(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mInvalid URL \u001b[39m\u001b[38;5;132;01m{\u001b[39;00murl\u001b[38;5;132;01m!r}\u001b[39;00m\u001b[38;5;124m: No host supplied\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "\u001b[1;31mMissingSchema\u001b[0m: Invalid URL 'None': No scheme supplied. Perhaps you meant https://None?"
     ]
    }
   ],
   "source": [
    "id=\"GSE292393\"\n",
    "download_series_matrix(id, output_dir=\"geofetch_metadata\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "2ca9180a-e724-44e3-a1a5-fe30be4662b8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<function __main__.download_series_matrix(gse_id, output_dir='geofetch_metadata')>"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "download_series_matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "fc10eeda-7776-4e69-be1a-988df5b7f9e5",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "01-Jun-2025 22:29:20 DEBUG utils - Directory . already exists. Skipping.\n",
      "01-Jun-2025 22:29:20 INFO GEOparse - Downloading ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE292nnn/GSE292329/soft/GSE292329_family.soft.gz to .\\GSE292329_family.soft.gz\n",
      "100%|█████████████████████████████████████████████████████████████████████████████| 4.16k/4.16k [00:00<00:00, 12.2kB/s]\n",
      "01-Jun-2025 22:29:21 DEBUG downloader - Size validation passed\n",
      "01-Jun-2025 22:29:21 DEBUG downloader - Moving C:\\Users\\difen\\AppData\\Local\\Temp\\tmp3vp36xw4 to C:\\Users\\difen\\POPPER\\GSE292329_family.soft.gz\n",
      "01-Jun-2025 22:29:21 DEBUG downloader - Successfully downloaded ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE292nnn/GSE292329/soft/GSE292329_family.soft.gz\n",
      "01-Jun-2025 22:29:21 INFO GEOparse - Parsing .\\GSE292329_family.soft.gz: \n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - DATABASE: GeoMiame\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SERIES: GSE292329\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - PLATFORM: GPL34290\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856560\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856561\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856562\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856563\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856564\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856565\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856566\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856567\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856568\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856569\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856570\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856571\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856572\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856573\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856574\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856575\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856576\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856577\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856578\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856579\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856580\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856581\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856582\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856583\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856584\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856585\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856586\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856587\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856588\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856589\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856590\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856591\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856592\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856593\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856594\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856595\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856596\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856597\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856598\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856599\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856600\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856601\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856602\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856603\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856604\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856605\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856606\n",
      "01-Jun-2025 22:29:21 DEBUG GEOparse - SAMPLE: GSM8856607\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "```json\n",
      "{\n",
      "  \"data_type\": \"RNA-seq\",\n",
      "  \"explanation\": \"The metadata specifies that the study involves 'RNA-seq profiling of lung tissues from Balb/c mice on days 4, 6, 8, and 10 post-infection with influenza A virus.' This indicates that the primary data type used in the study is RNA-seq, which is a method for expression profiling by high throughput sequencing.\"\n",
      "}\n",
      "```\n"
     ]
    }
   ],
   "source": [
    "result = identify_geo_data_type_via_llm(\"GSE292329\", client)\n",
    "print(result)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "id": "0608b985-87a4-4556-b0fd-c057975d2a9b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "To accurately identify the primary data type used in the study for GEO Series GSE292329, I would need to review the specific metadata details provided for this series. The metadata typically includes information such as the platform used, the experimental design, the type of biological material analyzed, and the technology applied. \n",
      "\n",
      "However, since the metadata details are not provided here, I can guide you on how to interpret typical metadata entries to determine the data type:\n",
      "\n",
      "1. **Platform**: Look for the platform or instrument used. For example, Illumina platforms are commonly used for RNA-seq and DNA methylation sequencing, while Affymetrix or Agilent platforms are often associated with microarrays.\n",
      "\n",
      "2. **Experiment Type**: Check if the metadata mentions terms like \"transcriptome sequencing\" (RNA-seq), \"gene expression profiling\" (Microarray), \"chromatin immunoprecipitation\" (ChIP-seq), \"methylation profiling\" (DNA methylation sequencing), \"quantitative PCR\" (RT-PCR), \"single-cell analysis\" (Single-cell RNA-seq), or \"spatial gene expression\" (Spatial transcriptomics).\n",
      "\n",
      "3. **Sample Description**: The description of the biological samples and the preparation methods can also provide clues. For instance, single-cell RNA-seq often involves dissociation of tissues into single cells.\n",
      "\n",
      "4. **Data Files**: The type of data files associated with the series (e.g., FASTQ, CEL, BED, etc.) can also indicate the data type.\n",
      "\n",
      "If you can provide specific metadata details, I can offer a more precise classification.\n"
     ]
    }
   ],
   "source": [
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "\n",
    "def fetch_geo_description_for_llm(gse_id):\n",
    "    \"\"\"\n",
    "    Fetch description text from GEO study page to be used for LLM judgment.\n",
    "\n",
    "    Args:\n",
    "        gse_id (str): GEO Series ID, e.g., \"GSE292329\"\n",
    "\n",
    "    Returns:\n",
    "        str: Combined descriptive metadata from the GEO study page.\n",
    "    \"\"\"\n",
    "    url = f\"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc={gse_id}\"\n",
    "    response = requests.get(url)\n",
    "    soup = BeautifulSoup(response.text, \"html.parser\")\n",
    "\n",
    "    description_parts = []\n",
    "\n",
    "    for header in soup.find_all(\"td\", class_=\"label\"):\n",
    "        if header and header.find_next_sibling(\"td\", class_=\"content\"):\n",
    "            label = header.text.strip()\n",
    "            content = header.find_next_sibling(\"td\", class_=\"content\").text.strip()\n",
    "            description_parts.append(f\"{label}: {content}\")\n",
    "\n",
    "    return \"\\n\".join(description_parts)\n",
    "def construct_llm_prompt(gse_id, metadata_text):\n",
    "    \"\"\"\n",
    "    Construct a prompt for the LLM to determine the data type of a GEO study.\n",
    "\n",
    "    Args:\n",
    "        gse_id (str): GEO Series ID.\n",
    "        metadata_text (str): Descriptive metadata of the GEO study.\n",
    "\n",
    "    Returns:\n",
    "        str: Prompt to be sent to the LLM.\n",
    "    \"\"\"\n",
    "    prompt = f\"\"\"\n",
    "You are a biomedical data expert.\n",
    "\n",
    "Given the following metadata for GEO Series {gse_id}, identify the primary data type used in the study. Possible data types include:\n",
    "\n",
    "- RNA-seq\n",
    "- Microarray\n",
    "- ChIP-seq\n",
    "- DNA methylation sequencing\n",
    "- RT-PCR\n",
    "- Single-cell RNA-seq\n",
    "- Spatial transcriptomics\n",
    "- Other (please specify)\n",
    "\n",
    "Provide a brief explanation for your classification.\n",
    "\n",
    "Metadata:\n",
    "{metadata_text}\n",
    "\"\"\"\n",
    "    return prompt\n",
    "\n",
    "\n",
    "def get_data_type_from_llm(prompt):\n",
    "    \"\"\"\n",
    "    Send the prompt to the LLM and retrieve the data type classification.\n",
    "\n",
    "    Args:\n",
    "        prompt (str): The prompt constructed for the LLM.\n",
    "\n",
    "    Returns:\n",
    "        str: The LLM's response containing the data type classification.\n",
    "    \"\"\"\n",
    "    response = client.chat.completions.create(\n",
    "        model=\"gpt-4o\",\n",
    "        messages=[{\"role\": \"user\", \"content\": prompt}],\n",
    "        temperature=0.2\n",
    "    )\n",
    "    return response.choices[0].message.content.strip()\n",
    "\n",
    "\n",
    "gse_id = \"GSE292329\"\n",
    "metadata_text = fetch_geo_description_for_llm(gse_id)\n",
    "prompt = construct_llm_prompt(gse_id, metadata_text)\n",
    "data_type_classification = get_data_type_from_llm(prompt)\n",
    "print(data_type_classification)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "37931ce7-d09d-4f4a-a70a-dc76eebb81e0",
   "metadata": {},
   "outputs": [],
   "source": [
    "def annotate_clustermap_with_metadata(grapher, metadata_df):\n",
    "    print(\"🔍 Starting clustermap annotation...\")\n",
    "\n",
    "    # 1. Build treatment label to GSM mapping\n",
    "    label_map = {}\n",
    "    for gsm, row in metadata_df.iterrows():\n",
    "        label = str(row['treatment']).strip()\n",
    "        if label not in label_map:\n",
    "            label_map[label] = []\n",
    "        label_map[label].append(gsm)\n",
    "    print(f\"✅ Built label_map with labels: {list(label_map.keys())}\")\n",
    "\n",
    "    # 2. Check if column names in grapher need remapping\n",
    "    if not all(label in metadata_df.index for label in grapher.columns):\n",
    "        print(\"⚠️ Column names in grapher are not pure GSM IDs. Attempting to remap based on treatment labels...\")\n",
    "        new_cols = []\n",
    "        for label in grapher.columns:\n",
    "            # Assume format: \"Vehicle (DMSO)_GSM7757588\"\n",
    "            try:\n",
    "                treatment_label = \"_\".join(label.split('_')[:-1])  # everything except last piece\n",
    "            except Exception as e:\n",
    "                raise ValueError(f\"❌ Failed to parse treatment label from: {label} — {e}\")\n",
    "\n",
    "            matches = label_map.get(treatment_label)\n",
    "            if not matches:\n",
    "                raise ValueError(f\"❌ No GSM match found for treatment label: {treatment_label} (from full label: {label})\")\n",
    "            new_cols.append(matches[0])  # Use only the first match\n",
    "            print(f\"🔄 Remapped {label} -> {matches[0]}\")\n",
    "        grapher.columns = new_cols\n",
    "    else:\n",
    "        print(\"✅ All grapher columns are valid GSM IDs\")\n",
    "\n",
    "    # 3. Extract annotations\n",
    "    annotation_cols = [\"condition\", \"treatment\", \"time\"]\n",
    "    missing_cols = [col for col in annotation_cols if col not in metadata_df.columns]\n",
    "    if missing_cols:\n",
    "        raise KeyError(f\"❌ Metadata is missing required columns: {missing_cols}\")\n",
    "\n",
    "    try:\n",
    "        col_annotation_df = metadata_df.loc[grapher.columns, annotation_cols].copy()\n",
    "    except KeyError as e:\n",
    "        raise KeyError(f\"❌ Some GSM IDs in grapher not found in metadata: {e}\")\n",
    "\n",
    "    col_annotation_df = col_annotation_df.applymap(lambda x: str(x).strip() if pd.notna(x) else \"NA\")\n",
    "    print(\"✅ Extracted column annotations.\")\n",
    "\n",
    "    # 4. Create annotation color maps\n",
    "    col_colors = pd.DataFrame(index=col_annotation_df.index)\n",
    "    palette_cache = {}\n",
    "\n",
    "    palette_map = {\n",
    "        \"condition\": \"Set1\",\n",
    "        \"treatment\": \"Dark2\",\n",
    "        \"time\": \"Paired\"\n",
    "    }\n",
    "\n",
    "    for col in annotation_cols:\n",
    "        unique_vals = sorted(col_annotation_df[col].unique())\n",
    "        real_vals = [v for v in unique_vals if v != \"NA\"]\n",
    "        print(f\"🎨 Mapping colors for {col}: {real_vals}\")\n",
    "        palette_name = palette_map.get(col, \"Set2\")\n",
    "        palette = sns.color_palette(palette_name, len(real_vals))\n",
    "        lut = dict(zip(real_vals, palette))\n",
    "        lut[\"NA\"] = \"#ffffff\"\n",
    "        col_colors[col] = col_annotation_df[col].map(lut)\n",
    "        palette_cache[col] = lut\n",
    "\n",
    "    print(\"✅ Created color palettes for annotations.\")\n",
    "\n",
    "    # 5. Plot the clustermap\n",
    "    print(\"📊 Generating clustermap...\")\n",
    "    g = sns.clustermap(\n",
    "        grapher,\n",
    "        z_score=0,\n",
    "        cmap=\"RdYlBu_r\",\n",
    "        col_cluster=False,\n",
    "        col_colors=col_colors\n",
    "    )\n",
    "\n",
    "    # 6. Add legend\n",
    "    legend_handles = []\n",
    "    for col, lut in palette_cache.items():\n",
    "        for label, color in lut.items():\n",
    "            legend_handles.append(Patch(facecolor=color, label=f\"{col}: {label}\"))\n",
    "\n",
    "    plt.gcf().legend(handles=legend_handles, bbox_to_anchor=(1.02, 1), loc=\"upper left\", title=\"Annotations\")\n",
    "    plt.show()\n",
    "    print(\"✅ Clustermap rendered successfully.\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "aa56f363-df59-4f2f-b6c2-bde027a520aa",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "🔍 Starting clustermap annotation...\n",
      "✅ Built label_map with labels: ['Vehicle (DMSO)', 'PGE2']\n",
      "✅ All grapher columns are valid GSM IDs\n",
      "✅ Extracted column annotations.\n",
      "🎨 Mapping colors for condition: ['control', 'treated']\n",
      "🎨 Mapping colors for treatment: ['PGE2', 'Vehicle (DMSO)']\n",
      "🎨 Mapping colors for time: ['24h', '48h', '60h']\n",
      "✅ Created color palettes for annotations.\n",
      "📊 Generating clustermap...\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAABPMAAAPmCAYAAACB3NoIAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjkuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8hTgPZAAAACXBIWXMAAA9hAAAPYQGoP6dpAAEAAElEQVR4nOzdB3RUVdeH8Z2EFHog9N57lSZNwAYIKkgTUUAQEKVJE6SjSBEUFMVOE0Wk6QsCKgIqIL0pvYnSeyeEJN/ax2/GSUhC+pxJnt9ad2Vy5869ZyaZkPzZ+xyv8PDwcAEAAAAAAABgPW93DwAAAAAAAABA7BDmAQAAAAAAAB6CMA8AAAAAAADwEIR5AAAAAAAAgIcgzAMAAAAAAAA8BGEeAAAAAAAA4CEI8wAAAAAAAAAPQZgHAAAAAAAAeIg07h4AAAAAAABIeqGhoRISEuLuYQCIgq+vr/j4+EhsEOYBAAAAAJCChYeHy6lTp+TSpUvuHgqAGAQGBkquXLnEy8srpsMI8wAAAAAASMkcQV6OHDkkXbp09wwKACR/4H7jxg05c+aM+Tx37twxHk+YBwAAAABACm6tdQR5QUFB7h4OgGikTZvWfNRAT9+vMbXcsgAGAAAAAAAplGOOPK3IA2A3x/v0XnNbEuYBAAAAAJDC0VoLpJz3KWEeAAAAAAAA4CEI8wAAAAAAAFKBjh07SrNmzdw9DCQQYR4AAAAAAEAU1q9fbxYiaNKkiVuuv3r1atN6qYuYxMXRo0fN47Zv3x5h/5QpU2TGjBmJPEokN8I8AAAAAACAKHz22WfSs2dP+eWXX+TEiRPi6TJnziyBgYHuHgYSiDAPAAAAAAAgkmvXrsnXX38t3bt3N5V5rhVtjoq5lStXStWqVc0qpLVq1ZJ9+/Y5jxk5cqRUqlRJZs+eLYUKFTJB2tNPPy1Xr151HhMcHCy9evWSHDlySEBAgNSpU0c2bdrkrK5r0KCBuZ0lSxZzPW2TVcuXLzfHajAXFBQkTZs2lUOHDjnPW7hwYfOxcuXK5nH169ePss02puvH9nnu2LHDjDNjxoySKVMmqVKlimzevDmRvxpwRZgHAAAAAAAQybx586RUqVJSsmRJefbZZ+Xzzz+X8PDwCMcMGTJEJk2aZMKrNGnSSKdOnSLcrwHb4sWLZcmSJWZbs2aNjBs3znn/wIEDZcGCBTJz5kzZunWrFCtWTBo2bCgXLlyQ/Pnzm/uUhmcnT540bbLq+vXr0rdvX3NdDdq8vb2lefPmEhYWZu7fuHGj+fjTTz+Zxy1cuDDK5xjT9WP7PNu1ayf58uUzIeCWLVtk0KBB4uvrm8BXHzEhzAMAAAAAAIiixVZDPNWoUSO5fPmyCeNcjRkzRurVqydlypQxIda6devk1q1bzvs1XNOKvnLlykndunXlueeeM+GbI5CbNm2avPXWW9K4cWNzjk8++UTSpk1rrq1z9WXNmtUcq5VzuXLlMtV9qkWLFvLUU0+Z8E2r/zRo3LVrl+zevdvcnz17dvNRq/b0cY7zuLrX9WP7PI8dOyYPP/ywCT6LFy8urVq1kooVKybq1wIREeYBAAAAAAC40Eo4rW5r27at+Vyr0dq0aXNXyFWhQgXn7dy5c5uPZ86cce7T9lptP3U9xnG/Vu2FhIRI7dq1nfdrRVv16tVlz549MY7vwIEDZmxFihQxra16HUewFltxuX5Mz1MrBF944QUT6GnVoWu7L5JGmoQ8+EuvkpISPBP+X683AAAAAABI3TS0u3PnjuTJk8e5T1ts/f39ZerUqc59ru2kOreccrS6Rr7fcYzr/fH1+OOPS8GCBU0lnY5Rz6nVf7dv35akENPz1LkBn3nmGVm6dKksW7ZMRowYIXPnzjVtv7CwMi9NmpSxAQAAAAAAKA3xZs2aZeaI2759u3PThR40OPvqq68S5TpFixYVPz8/Wbt2rXOfVsrp3HPazqr0fhUaGuo85vz586ZycOjQofLQQw9J6dKl5eLFixHOHdXj4nP92CpRooS88sor8sMPP5j23+nTp8fp8YibBEVZ3jTpAgAAAACAFEQXqtBwrHPnzs456hx0rjqt2tN55hIqffr0ZqXcAQMGmDntChQoIBMmTJAbN26YayutvtNKOB3TY489Zuaz05VtdS68jz/+2LS8amutzmPnSufY02N11VtdnEJXqo38XGJz/Xu5efOmeXzLli3NCrr//POPCQP1dULS8U5omJcSNgAAAAAAAKVhnc7/Fjn8UhpS6YquO3fuTJRr6Rxzek5dGOO+++6TgwcPyooVK0xgp/LmzSujRo0yYV3OnDmlR48eZuVabWPVlWO1tVYr4iKHizrH37vvvisfffSRqSZ88skn43X9e9FFOrRSsH379qY6r3Xr1mYxDR0zko5XeOR1lePguwwpY868J64xZx4AAAAAIOXRFUePHDliqqa0OguA579fabMFAAAAAAAAPESCwjwWjwAAAAAAAACSD3PmxfEVGDt2rFSrVk0yZsxoJpRs1qyZWUUGAAAAAAAASGpU5sXRmjVr5OWXXzaBni5X/dprr8mjjz4qu3fvNivBAAAAAAAAAEklQXGcLo+c2uiyzq5mzJhhKvR0FZkHHnjAbeMCAAAAAABAyscCGAl0+fJl8zFr1qzuHgoAAAAAAABSONpsRSQ4ONhsrvz9/c0Wk7CwMOnTp4/Url1bypUrl8SjBAAAAAAAQGrHAhje/y5qkTlz5gib7rsXnTvvjz/+kLlz5ybkZQQAAAAAAACSPszTyryUsA0ePNi0y7puui8mPXr0kCVLlsiqVaskX758CXkZAQAAAACABQoVKiSTJ0+OsFbA4sWLY3xMx44dpVmzZskwOs929OhR83pu377d3UPxeMyZF8uWWofw8HDp2bOnLFq0SFavXi2FCxdO8vEBAAAAAIDkd/LkScmSJYszjNIMYNu2bVKpUiXnMVOmTDFZQUqkQeWlS5fuGWjCk+bM85FUR1trv/zyS/n2228lY8aMcurUKbNfW3PTpk3r7uEBAAAAABArx/PmT7Zr5T3+t3iiXLly3fMYzQNSu5CQEPH19XX3MFIN5syL4yswbdo004Zbv359yZ07t3P7+uuvE/JSAgAAAAAAlwUnJ0yYIMWKFTOddAUKFJAxY8Y479+1a5c8+OCDpqgmKChIunbtKteuXbur9XXixInmb3Y9RotzNHRyOHPmjDz++OPmHFpxN2fOnLvG4dpm6+jMq1y5stmvuYDrtRx0gc1evXpJjhw5JCAgQOrUqSObNm1y3q9dfvr4lStXStWqVSVdunRSq1Yt2bdvX5xfp//9739SrVo1c51s2bJJ8+bNnfddvHhR2rdvbyoL9RqNGzeWAwcOOO+fMWOGBAYGyooVK6R06dKSIUMGadSokalGVCNHjpSZM2eaYiYdr246dke7rOYg9erVM9fW106/ZqNHjzZTkenXTKsXly9fHufnhHsjzIvjK6Cls1Ft+uYFAAAAAAAJp/PYjxs3ToYNGya7d+82HXI5c+Y0912/fl0aNmxoQioNyb755hv56aefzNz2rnSO+0OHDpmPGkppeKWbg/4d//fff5v758+fLx988IEJ+KKzceNG81GvpYHXwoULozxu4MCBsmDBAnPNrVu3mkBSx3vhwoUIxw0ZMkQmTZokmzdvljRp0kinTp2c9zkCMw3PorN06VIT3j322GOm9VfDwerVq0d4fnru7777TtavX2+yCz3WNdC8ceOGCTxnz54tv/zyixw7dkz69+9v7tOPrVu3dgZ8umno6DBo0CDp3bu37Nmzxzw/bTfW56Pn27lzp9n3xBNPRAgQYUObbRqvRBoGAAAAAACAyNWrV00wNHXqVOnQoYPZV7RoUVPhpjTYu3XrlsyaNUvSp09v9umxWmU3fvx4Z+inYZ/u9/HxkVKlSkmTJk1M4NWlSxfZv3+/LFu2zAR0WtmmPvvsM1OhFp3s2bObj1rlF137rQaN2tGnoaFWwqlPPvlEfvzxR3P+AQMGOI/VSkOtbHMEYzo+fV5a6aYtqyVLljQVddHRxz/99NMyatQo576KFSuajxqgaYi3du1aZwCn1XP58+c3lYatWrUy+zTY+/DDD83rqzQQ1eo6pZV6WrWolYZRPd8+ffrIU0895fxcQ7xXX33VjEnp10KDUl1Q5P3334/2eSCZwzwvH8I8AAAAAACQeLTSSwOkhx56KNr7NbRyBHmqdu3aps1TW1UdYV7ZsmVNkOeg7bbanus4h1bDValSxXm/Bn7adpoQWgmoAZmOx0GDOa2Y02u6qlChQoSxKa0M1JbivHnzyt69e2O8lq4Kq8FkVBzPr0aNGs59GkJqQOg6Dg0LHUGeYxwxVSe60hZhhytXrsiJEyciPG+ln+/YsSNW50NyrWabJoUsZwsAAAAAAKyQWItLRl6QQdtWNfCzhev4dGwqLuNLjNcpqtcotivzuoapSF7eCa3MSwkbAAAAAACwQ/HixU1QpS2xUdFWWK320pZWB20n9fb2NpVnsaFVeHfu3JEtW7Y492lV36VLl6J9jJ+fn/kYGhoa7TFa5abH6XgctFJP5/YrU6aMJCat7IvpNdLnt2HDBue+8+fPm+cYl3Hoc4np+TpkypRJ8uTJE+F5K/08sZ83EhrmeXuliA0AAAAAANhB54zTudd0IQmdF09bV3///Xcz55xq166dOUbn0/vjjz/MvGw9e/aU5557ztliey8a+unCDt26dTOBl4Z6L7zwQozVbro6rd6vK7SePn1aLl++HGW1Wvfu3c3ceHqcLt6hrbC60ETnzp1j/RocP37cBI6ORTeiMmLECPnqq6/MR22d1RZinafOEYg++eST5tq//fabCT+fffZZ076r+2OrUKFCZjELDQHPnTsXYfGMyPQ56/V1lVs9XucB1FZgXSQDFoV5Pr7eKWIDAAAAAAD20FVs+/XrJ8OHDzdVZm3atHHO5abzvK1YscKsDquLV7Rs2dLMr6eLXcTF9OnTTTWZLkKhCzl07drVBHbR0Tno3n33Xfnoo4/M46ILxXQV3hYtWphw8b777pODBw+a8eqCHLGloZkGYhoCRqd+/fpmJV9d6KJSpUry4IMPRgj/9PnpnIBNmzaVmjVrmvbZ77///q7W2phoGKjBp86PpwuARK68c9WrVy/p27ev+bqVL1/ehJk6Ng0Wkbi8wmPbDB2FY49UlpSgwI/b3D0EAAAAAAASna6OeuTIESlcuLCpZgPg+e/XhC2A4fvfqjAAAAAAAAAAklaCwjwWjwAAAAAAAACST8Iq89Iw3xwAAAAAAACQXKjMAwAAAAAAAFJFZZ43YR4AAAAAAACQXBK4AAZttgAAAAAAAIBntNlSmQcAAAAAAAB4SmWeT+KNBAAAAAAAAECMqMwDAAAAAAAAPESCJr3z8fVOERsAAAAAAEChQoVk8uTJzs+9vLxk8eLFMT6mY8eO0qxZs2QYXcoxY8YMCQwMdPcwPFaCkiwvH68UsQEAAAAAAER28uRJady4sbl99OhRE+5t3749wjFTpkwx4VRyGzlypFSqVCnZrkcAZw/abAEAAAAASIXuH7Ei2a71+6iG4oly5cp1z2MyZ84sNgsJCRFfX193DwPWVOb5eqeIDQAAAAAA2CMsLEwmTJggxYoVE39/fylQoICMGTPGef+uXbvkwQcflLRp00pQUJB07dpVrl27dlfr68SJEyV37tzmmJdfftkEWw5nzpyRxx9/3JyjcOHCMmfOnLvG4dpmq8eoypUrm/3169ePcC2H4OBg6dWrl+TIkUMCAgKkTp06smnTJuf9q1evNo9fuXKlVK1aVdKlSye1atWSffv2xalKbtSoUbJjxw5zLt0c1YF6e9q0afLEE09I+vTpna/bt99+K/fdd58ZU5EiRczj79y54zzn22+/LeXLlzePyZ8/v7z00kvO11TH/Pzzz8vly5ed19PKQMfz7d+/v+TNm9c8tkaNGub4yOPVr6E+1+bNm8v58+dj/Vxxt4QlWT7eKWMDAAAAAADWGDx4sIwbN06GDRsmu3fvli+//FJy5sxp7rt+/bo0bNhQsmTJYkKyb775Rn766Sfp0aNHhHOsWrVKDh06ZD7OnDnTBEqu7bAawv3999/m/vnz58sHH3xgAr7obNy40XzUa2n77cKFC6M8buDAgbJgwQJzza1bt5pAUsd74cKFCMcNGTJEJk2aJJs3b5Y0adJIp06dnPc5Wnojh2IObdq0kX79+knZsmXNWHTTfQ4atGlopqGnnvfXX3+V9u3bS+/evc3r+dFHH5nXwjUg9fb2lnfffVf+/PNPM/aff/7ZPBelYaPOJZgpUybn9TTAU/q6r1+/XubOnSs7d+6UVq1aSaNGjeTAgQPm/g0bNkjnzp3Ncdqi3KBBA3njjTeifZ2R1G22VLUBAAAAAIBEdPXqVTMP3dSpU6VDhw5mX9GiRU2Fm9Jg79atWzJr1ixTCab0WK2yGz9+vDP007BP9/v4+EipUqWkSZMmphquS5cusn//flm2bJkJ6KpVq2aO/+yzz6R06dLRjit79uzmo1b5Rdd+q0GjVsVpUOaYa++TTz6RH3/80Zx/wIABzmM1SKtXr565PWjQIDM+fV5aOadtsSVLljSVbFHRasIMGTKYEDCqsTzzzDOmks5BAz29huP11Mq8119/3YR1I0aMMPv69OkTYSEQDdxefPFFE3L6+fmZdmINGF2vd+zYMZk+fbr5mCdPHrNPQ77ly5eb/W+++ab5Wmq45wgGS5QoIevWrTPHwA1hnjBnHgAAAAAASER79uwxrZsPPfRQtPdXrFjRGeSp2rVrm9ZcbVV1hHlataZBnoO222qlmuMcGoRVqVLFeb8Gfgld4EErAbWVV8fjoMFc9erVzTVdVahQIcLYlFYGajuqtqzu3bs33uPQ9l1X2o67du3aCJV4oaGhJjy8ceOGCQ214nDs2LHmuleuXDEtuK73R0VfTz2PBnSu9OunoafS561Vgq5q1qxJmJcAVOYBAAAAAABraNVZYoi86INWlWngZwvX8enYVGKNzzXoVDr3nc6R99RTT911rFYCaltv06ZNpXv37ibwy5o1q/z222+mPfb27dvRhnl6Xg1Mt2zZEiE4VVo5CBsr85hvDgAAAAAAJKLixYubQE9bYl944YW77tdWWG1j1ZZWR2ilVWc655u2psaGVuFp5ZmGUI42W63qu3TpUrSP0VZTpZVo0dF2YD1Ox1OwYEGzTyv1dG4/1zbWxKDXiWksrnThC31+On9fVPR10CBR5/DT11HNmzfvntfTxUB0n1YU1q1bN8pz69dL581z9fvvv8dq3EiK1Wy9vVLEBgAAAAAA7KCVYq+++qqZY03nxdPWVQ1/dM451a5dO3OMzv/2xx9/mAUsevbsKc8995yzxfZeNPTTedy6detmgiYNszQ4jKkqUFen1fu1PfT06dNmZdfINFzU6jadG0+P08UmdI4+bVXVKrfYOn78uAkcHYtuREXntTty5IhZVOLcuXOmtTU6w4cPN6+lVufpAhfa+qoLVgwdOtTcryGfho7vvfeeHD58WGbPni0ffvjhXdfTSjwNWfV6+py0vVa/Hrq4hi4IouPRMWu77tKlS83jdGVffS10ZWFdFEPnMaTFNmESVlrn550yNgAAAAAAYA1dxVZXa9UQSiu7dKVWx0qz2vK5YsUKszqsVtW1bNnSzK+nIVFc6AINumiDLkKh7addu3Y1gV10dI49Xe1VV4LVxz355JNRHqer8LZo0cKEi1oRd/DgQTNeXZAjtjRY00o6Dcyio9fQQFJXh9XFOb766qtoj9XVdJcsWSI//PCDec3uv/9+eeedd5zVgzoH4dtvv20WEClXrpzMmTPHBHKudEVbXRBDvxZ6vQkTJjhfRw3z9OulIWmzZs1MJaLO/af0WroIiC6EodfRMThCRMSPV3h4eHg8Hyu332khKYHfKwvcPQQAAAAAABKdLmCg1VKFCxc21WwAPP/9mrA583wjTm4IAAAAAAAAwNY583y8UsQWF7/88os8/vjjpqRWV5tZvHhxQl5CAAAAAAAAINYSNmGcLh6RErY40NVytMf7/fffT9BLBwAAAAAAAMRVgtpsvXxT3+IRjRs3NhsAAAAAAACQ3BI2Z55PygjzdPnmyEs4+/v7mw0AAAAAAABIGXPm+XqniE2XW86cOXOELfISzAAAAAAAAICHV+bFbb45Ww0ePFj69u0bYR9VeQAAAAAAALANc+bRUgsAAAAAAIBUUZnnnTLCPAAAAAAAAMATeCc4zEsJWxxcu3ZNtm/fbjZ15MgRc/vYsWMJeikBAAAAAIB7FSpUSCZPnuz83MvLSxYvXhzjYzp27CjNmjVLhtEBiRHmpfFJGVscbN68WSpXrmw2pXPt6e3hw4cn6KUEAAAAAAB2OXnypDRu3NjcPnr0qAn3HMU9DlOmTJEZM2Yk+9j0mjqeRo0aRdh/6dIls3/16tV3PaZbt27i4+Mj33zzTTKOFImNNts4ql+/voSHh7t7GAAAAAAAIInlypXrnsdkzpxZ3CVNmjTy008/yapVq6RBgwYxHnvjxg2ZO3euDBw4UD7//HNp1apVso0TVlXmpUkZGwAAAAAAsEZYWJhMmDBBihUrZhasLFCggIwZM8Z5/65du+TBBx+UtGnTSlBQkHTt2tVMixW59XXixImSO3duc8zLL78sISEhzmPOnDkjjz/+uDlH4cKFZc6cOXeNw7XNVo9R2p2n+7XYx/VaDsHBwdKrVy/JkSOHBAQESJ06dWTTpk3O+7ViTh+/cuVKqVq1qqRLl05q1aol+/bti/PrlD59eunUqZMMGjTonsdqNV6ZMmXMsb/88ov8/fffcb4e7MCceamwuhAAAAAAAJsNHjxYxo0bJ8OGDZPdu3fLl19+KTlz5jT3Xb9+XRo2bChZsmQxIZmGVFqd1qNHjwjn0Gq1Q4cOmY8zZ840bamu7bAawmmgpffPnz9fPvjgAxPwRWfjxo3mo15L228XLlwY5XFa+bZgwQJzza1bt5pAUsd74cKFCMcNGTJEJk2aZKbz0go7DeUcHC29UbXKRjZy5EgTbupziMlnn30mzz77rKkk1NZhd7QGI3FQmUdlHgAAAAAA1rh69aqZh04r8zp06CBFixY11W0vvPCCuV+DvVu3bsmsWbOkXLlypkJv6tSpMnv2bDl9+rTzPBr26f5SpUpJ06ZNpUmTJqYaTu3fv1+WLVsmn3zyidx///1SpUoVE3bdvHkz2nFlz57dfNQqP22/zZo1613HaNA4bdo0eeutt0xgppVweg2t/tPzu9JKw3r16jmr5datW2eel/L19ZWSJUuaqr17yZMnj/Tu3duEg3fu3InymAMHDsjvv/8ubdq0MZ9rqDd9+nSmEfNQCazM80oZGwAAAAAAsMKePXtMq+pDDz0U7f0VK1Y0LaYOtWvXNq25rq2qZcuWNYs9OGi7raPyTs+h1XAa4jlo6BcYGJigsWsloLby6ngcNJirXr26uaarChUqRBibcowvb968snfvXvO42Hj11Vfl7NmzZi68qOh+rQ7Mli2b+fyxxx6Ty5cvy88//xyPZwl3YwEMAAAAAABgDa1iSwwaornStlUN/GzhOj4dm4rv+DSE1NbkUaNGmSpEV6Ghoabl99SpUybAdN2vIV90oSnslaA0zssnTYrYAAAAAACAHYoXL24CPUdLbGSlS5eWHTt2mJZWh7Vr14q3t7dpTY0NrcLTltQtW7Y492lV36VLl6J9jJ+fnzMEi462BOtxOh4HrdTTuf20nTYp9ezZ07wG2qLs6vvvvzety9u2bZPt27c7t6+++srM+xfTc4adWACD6kIAAAAAAKyhK8Bq26guJKHz4mnrqs735phzrl27duYYnU/vjz/+MAtYaJD13HPPORfJuBcN/Ro1aiTdunWTDRs2mFBP5+SLqSpQV6fV+5cvX27m5tM21ci09bd79+4yYMAAc5wu3tGlSxe5ceOGdO7cOdavwfHjx03g6Fh0Izb0NdHKvHfffTfCfn3ddL5AbU3WOQYdW+vWrU1FX1Sr+CJFL4DhkzI2AAAAAABgDV3Ftl+/fjJ8+HBTiacLNzjmk9NFIVasWGFWh61WrZq0bNnStIrqYhdxoQtA6OIRugjFU089JV27djWBXXS0RVWDso8++sg87sknn4zyOF2Ft0WLFiZcvO++++TgwYNmvLogR2xpNZ9WCmoIGBcacBYpUsT5uYaOS5cuNeOJTKv4mjdvftfCHLCfV3gCli4J3zFaUgKvisPdPQQAAAAAABKdro565MgRKVy4sKncAuD579eETRjnMnEiAAAAAAAAgKTFarYAAAAAAACAhyDMAwAAAAAAADxEAttsWTwCAAAAAAAASC5U5gEAAAAAAACpIszzYQEMAAAAAAAAILkkLI3zojIPAAAAAAAASC602QIAAAAAAACpI8xjAQwAAAAAAAAgudBmCwAAAAAAAHgI7wQvgJESNgAAAAAAYI369etLnz59ku16HTt2lGbNmoktZsyYIYGBgfF+7by8vMwWEBAgZcqUkQ8++CDCMbdv35a33npL7rvvPkmfPr1kzpxZKlasKEOHDpUTJ05EeF0c53LdGjVqZO6/cOGC9OzZU0qWLClp06aVAgUKSK9eveTy5csJfAUQEyrzAAAAAABIhfJNH5Rs1/rn+XGJfs7w8HAJDQ2VNGko0omsS5cuMnr0aLlx44bMmjVLXn75ZcmSJYu0bdtWgoOD5dFHH5WdO3fKqFGjpHbt2pI9e3Y5cuSIfPXVV/Lee+/J2LFjnefS4G769OkRzu/v728+avCn28SJE01o+Ndff8mLL75o9s2fPz/Zn3dq4Z3gOfNSwgYAAAAAAKyg1WBr1qyRKVOmOCvBjh49KqtXrza3ly1bJlWqVDGB0m+//SZhYWEmfCpcuLCpDtMKM9cgSQO/zp07O+/XKjI9t8PIkSNl5syZ8u233zqvp9fSa+rtefPmSd26dc1jq1WrJvv375dNmzZJ1apVJUOGDNK4cWM5e/ZshOfw6aefSunSpU1lXKlSpSJUxjnOu3DhQmnQoIGkS5fOjHn9+vXmfr32888/b6rbHOPRMcaFnjNXrlxSpEgR89jixYvLd999Z+575513zOv2888/myo6fS21oq5evXry4YcfyptvvhnhXPo667lcNw0GVbly5WTBggXy+OOPS9GiReXBBx+UMWPGyP/+9z+5c+dOnMaM2GM1WwAAAAAAYA0N2jQw06BIq8uUVo5pCKYGDRpkKsE0qNJQSYO8L774wgRRGlr98ssv8uyzz5rHaEClYV++fPnkm2++kaCgIFm3bp107dpVcufOLa1bt5b+/fvLnj175MqVK84KtKxZszrbTUeMGCGTJ082gVenTp3kmWeekYwZM5pxamim5xg+fLhMmzbNHD9nzhzz+dSpU6Vy5cqybds2Uymn7awdOnRwPs8hQ4aY56Fj1ttaNXfw4EGpVauWuZ6eY9++feZYDQ2VBnPagut4LWJLg0htrVVafffII4+YsUVFw8OE0BAyU6ZMVEwmoQS9sl602QIAAAAAgESk87f5+fk5q8si04BPwyilLaNaSfbTTz9JzZo1zT4N+bTy7KOPPjJhnq+vr2knddAKPa2C04o7DeI0KNOwS88V1fU07GvYsKG53bt3bxO6rVy50rSnKq3604DNQcO/SZMmyVNPPeW83u7du814XMM8PW+TJk3MbR1f2bJlTZinlXz6GmioFnk82bJlMxVwsaVViRreaUutBphKg1KdV89V8+bN5ccffzS3K1SoYAJPhyVLljjDRIfXXnvNbJGdO3dOXn/9dee1YGVlHikrAAAAAABIPtre6qDhl84L5wj3HLQKzbXy7P3335fPP/9cjh07Jjdv3jT3V6pUKVbX03DLIWfOnOZj+fLlI+w7c+aMuX39+nU5dOiQCfi0Gs9BW041oIvuvFolqPQ8GuZFp0ePHma7F23r1VZffZ4+Pj7yyiuvSPfu3WM8Xsf+7rvvmspGV9oK7Kg6dNDKxci0slHDSZ07L65twUjWBTASVnoJAAAAAAAQF9qu6nDt2jXzcenSpZI3b94oF2mYO3euqYLTajmt3tMWWV3JdcOGDbG6nlb2RW5BjbxPW3ldx/PJJ59IjRo1IpxHQ7V7nddxnoRq166dad3VikMNCr1dpknTtl5H+27kMDGqkE5f72LFisV4vatXr5qFMvS1XbRoUYTnBtvCPB8q8wAAAAAAQOLSNlttEb0XrQLT0E4r7rSlNipr164189C99NJLzn1aPRef692LVunlyZNHDh8+bAK1+EroeLQKMLoATtuEhw4daubyi27evLjQijxtQ9avgy6yoYt+wOrKPObMAwAAAAAAiatQoUKmck4XetD52qKqGFNaCaZVd9pGqlVtderUMQswaICnizDoHHVaiTZr1ixZsWKFmb9u9uzZZjVave16Pb1fK9Z0kYzILbFxofPf6Sqxeg6tVtO5+DZv3iwXL16Uvn37xvr5a5Wfzs2nK93q/IG66aIaWvmm++NLXyutZHzooYfM/H66Uq8uJKJz6elKwZErCHX8p06dirBPF7fQ+fs0yHv00UdNq7MuQqKf66Z0AZLI50Li8E5wZV5K2OJB++31zaWJs5bObty4MUEvJQAAAAAA+JcGdBoEaeWdhkJaeRcdXXBh2LBhZlXb0qVLmwBNwypHWNetWzezGEWbNm3M3+/nz5+PUKWndH67kiVLmvn49HoaBsbXCy+8YOar05VxdW49rRjUBTJcw8N70UrCF1980YxZxzNhwgTnAhORqwrjSnMMDQNfffVVM0YNQPV169Onj1nUY/HixRGOX758uWnDdd30MWrr1q0mdN21a5epBHQ95u+//07QOBE9r/Dw8HCJrzs/SIqQ5tE4Hf71119L+/btzbLX+oNAl4zWJa41wc+RI0eSDRMAAAAAgLi4deuWHDlyxARJtD8CKeP9mrDKPJ2gMSVscfT222+b1P755583/0ugoZ6Wu+rKOAAAAAAAAICdc+Z5p4wFMLT/WzdXOnGjY+UbV7qs85YtW2Tw4MHOfboqzMMPPyzr169PlvECAAAAAAAgdUpQZV64l3eK2LSvXiemdN10X1S0P11XlNEValzp55EnhAQAAAAAAAASU4JK68LCfSUl0Cq7yCvKRFWVBwAAAAAAAHhsmBcenrAp92wRXUttVHTpZV1R5/Tp0xH26+e5cuVKohECAAAAAAAACW2zFe8UscWFn5+fVKlSxSzj7BAWFmY+r1mzZkJeTgAAAAAAACDpKvNCU0ibbVyfhbbkdujQQapWrSrVq1eXyZMny/Xr183qtgAAAAAAAEBSoc02Htq0aSNnz56V4cOHm0UvKlWqJMuXL79rUQwAAAAAAAAgMXmFh4eHx/fB1+78KSlBhjRl3T0EAAAAAAAS3a1bt+TIkSNSuHBhCQgIcPdwACTC+5XKPAAAAAAAAMBDJCjMC0shc+YBAAAAAJDa7O/ok2zXKjEjNNmuhYTz8vKSRYsWSbNmzaK8f/Xq1dKgQQO5ePGiBAYG3vN89evXN1OU6ZoDCaGLj/bo0UP++OMP8fFJvu/fhDp37pyUKVNGtm7dKvny5Uvw+VjNNmEvAQAAAAAASGQa/vTp0yfZrtexY8dogyt3mDFjRqxCssjKly8vL774YpT3zZ49W/z9/U2wlFC1atWSkydPSubMmSU5DRw4UIYOHeoM8vR10uBRN92XJUsWqVGjhowePVouX75819dYj4vq9Xn55ZfNfXqMg66V0L17dylQoIB53XLlyiUNGzaUtWvXRnjsunXr5LHHHjPX1tZY/Rq8/fbbEhr6X4CdLVs2ad++vYwYMSJRXocEJVlh4d4pYgMAAAAAAJ5FlwC4c+eOu4dhlc6dO8vcuXPl5s2bd903ffp0eeKJJ0ywlFB+fn4m3NIALLn89ttvcujQIWnRokWE/ZkyZTLB4j///GOCta5du8qsWbNMJeCJEyciHJs/f/67Xh+dp+7LL780oZ0rvc62bdtk5syZsn//fvnuu+9MyHz+/HnnMVq9WK9ePVNtt2rVKtm7d6/07t1b3njjDXn66afN96jD888/L3PmzJELFy64O8zzTREbAAAAAACwg1ZHrVmzRqZMmeKsujp69Khp7dTby5YtkypVqphqKQ14wsLCZOzYsWbRgLRp00rFihVl/vz5zvNphZSGXI77S5Ysac7tMHLkSBPYfPvtt87r6bX0mnp73rx5UrduXfPYatWqmWBn06ZNUrVqVcmQIYM0btzYVHG5+vTTT6V06dKmUqtUqVLywQcfOO9znHfhwoWmVTVdunRmzOvXrzf367U1+NHKMsd4dIyx8eyzz5qgasGCBRH266IKel59HZQ+1/vuu8+Mr0iRIjJq1Ki7glGt4GvevLkZX/HixU2Y5eD4Wly6dMm5TyvWNOzS47VKTavYtA03KsHBwdK/f3/JmzevpE+f3lTT6TljoiHcI488ctfCEDoODRZz585tXnN9jhrqXbt2zVTyudLnrIGevvYOeluDvMqVKzv36fP69ddfZfz48eZrVLBgQalevboMHjzYBKLq+vXr0qVLF/P5xx9/bMLDQoUKyQsvvGC+n/R7UL93HMqWLSt58uQxAWBC0WZLmy0AAAAAANbQoK1mzZomKNGKK900gHEYNGiQjBs3Tvbs2SMVKlQwQZ5WYn344Yfy559/yiuvvGJCLQ0ElYZ9Wjn1zTffyO7du2X48OHy2muvOYMWDZVat24tjRo1cl5P20gdtDVSWzt1vrM0adLIM888Y0IiHacGPgcPHjTndNDqK/18zJgxZoxvvvmmDBs2zAQ8roYMGWKuvX37dilRooS0bdvWBGp6bZ1bzlFxppsepzTU08AoOlp19+STT8rnn38eYb+2o+pr8Oijj5oxa8unVpDp6/HRRx+Z+3W8rjTg09dl586dpo20Xbt20VaV6XN46KGHzLxwGkpqyPr4449HaDV1pfPe6XEa0On5W7VqZV7/AwcORPvcdNwaoMZGjhw5zHg1gIw8hk6dOpkqRQd9rTQ8daUhrW6LFy82wWNUfvjhB1Ol5/jauNLnrl/Tr776KsJ+DQT1ebh1AYxQqtoAAAAAAEAi0nnYtI1TK7y04ioynQ9NK7SUBi0alv30008mAFRaaaZhkoZU2gLp6+trgikHrdDTIEnDPA2rNLTRqjs9V1TX07BGq8yUBmAauulCDLVr1zb7tBJMwzDX8G/SpEny1FNPOa/nCM06dOgQ4bxNmjQxt3V8WrmlwaBW8ulr4Kg4ixzWFS1aNMbXT8ej1YJajafX1lZPDRL12t7e3uZaGog6xqKv1+uvv24CStc53bRCUp+r0tf43XfflY0bN5rQLbIJEyaYoM21AlGfT1SOHTtmwjT9qJVqjtdi+fLlZr9eKyp//fWX8/jY0Nfx6tWrJnDTcM9Bg16tsNPzOSoKNVR0rQzU0Fa/phooa0isFX36vaStsxogK63QVFoNGN31Hcc46Pi1ddetYV44880BAAAAAIBk5FqdpeHXjRs3nOGew+3btyO0Tb7//vumAksDJG1D1fu1LTI2HOGNypkzp/moixy47jtz5oyz9VLnddNATYMgB624i7xYhOt5tUVU6Xk0BIqOVrTpFhN9LbQKT4MxDT41eNTn7ag+27FjhwmwXCvxtHpN547T11JD1Mjj01ZYrRR0PM+oKvO0ui42du3aZa6nlWuuNEwNCgqK9nH6dYvcYhsTx3x1kef1y549uwlRNazTY/R2VPMI6px5ep9W0v3++++mvVtDS22hdl0ow3VevHvR0FhfY7eGeWHhCXo4AAAAAABAnGiw5KDzoqmlS5ea+ddc6Zx6SquutPJLq+W0ei9jxozy1ltvyYYNG2J1Pa3sc3AEQ5H3aSuv63g++eQTMw+cK8cKrDGd13GehNDqOw2btBpP23I11NN537QCzzFGrc5zVA66cg3LXMcX+XlGFVLFll5fX4stW7bc9ZpolWR0NHCLbg6+qGiLswaQUQWE2mrrCEU16I2Ovh4ajuqmrdI6H55WL+rr6wgj9Tqubdmu19e2Y1fapqxhonvDPIn4ogMAAAAAACSUttlGN9+aKw1LNLTTyjNtg4yKVqFp2PLSSy8592n1XHyudy9apaetlIcPHzZztsVXQsejVXi6oqou7qALLmg1mYO2jO7bt0+KFSsmiUWr+LQC0LWdOTpaManPTav8dGGR2NLHabtybOi5dYXaZs2amXAzMm0V1upMDSgdLdSxod9vOo+e0vkHs2bNakLiyGGeztWn8/9p+7KrP/74wywSklC02QIAAAAAAKvoIg9aOacrv2q1loYmUdEqO62600UvtGqsTp06ZhVYDfC0KkvnhdOVWHWBjBUrVpg55GbPnm1Wo9XbrtfT+zXk0kquyC2xcaGBVq9evcw5NDTS9tHNmzebqrK+ffvG+vlrBZsGZLrSrba+6jZ16lQTzun+mOhze/DBB6Vr164m7HStwtPFOZo2bWpWcG3ZsqUJu7T1VoMmDQDjQ+eg09ZjDUxffPFFE0auWrXKtN5GbmHVijYNOnURDg3CNKTT1YD1OWko6JhHMDIN3SIvIuJocz116pT5qKvQ6nyIOu+evv66UEpUtCJQK+cctyPTefZ07FrBp2PS7zP9GmqbrS4w4qgQ1XkQdR49fZ210k+/5/R5DBgwwLy2Oiejg7bXajVidHMCJuMCGBH7jgEAAAAAgGcoMSPhlWhJRQM6DeK0EkrnStPFHKKj1U/auqir2mpFXGBgoKk+0xVrVbdu3cyiA23atDGVWLqog4ZOOgeag85vpwsg6Hx8GqJpEBXTqrEx0VZMDd60lVdDHQ19NOjq06dPrM+hlV4aiumYNVjS1k5tmT137txdVYXR0Xn7NFjS5+raPquh2JIlS8x8euPHjzfttDpPn447vjSg09Vd9TXXFVu17VbbjB0LaESmrb8aHPbr10+OHz9uAr/777/fhIzR0QBQF+nYt2+flCxZ0rn/ypUrZs5B/dpqmKb36feOLlain0cnpvs0QNbxv/POO+b1DgkJMSsq6/eJ4/tKaWCn3ys6/6BWGeq8gxoe60rF+vV2na/v22+/NQFqXKoRo+MVHpeZ+iI5dOWWpARFM8V+AkUAAAAAADyFhguOVU3jsngAYCMNR69cuWIq4jyNhpVasfnMM88k+P2aoMq8O2HxzgEBAAAAAACAWNOKtw8++MC0VEc1F56ttKJSW52jq1RM1sq8PRcTvpyuDUpn+XfZZQAAAAAAUhIq8wDPkSyVeSFU5gEAAAAAAADJJkFhXlj8i/oAAAAAAAAAJG+Yl5BHAwAAAAAAAIgLFsAAAAAAAAAAPARttgAAAAAAAICHoDIPAAAAAAAA8BBU5gEAAAAAAAAewjshD9bCvJSwAQAAAAAAe9SvX1/69OmTbNfr2LGjNGvWTGwxY8YMCQwMjPdr5+XlJXPnzo2wf/LkyVKoUKG7jr9586ZkzZpVsmXLJsHBwfEeMzwkzLsTFpYiNgAAAAAA4FnCw8Plzp077h6GlQICAmTo0KESEhJyz2MXLFggZcuWlVKlSsnixYuTZXxIGCrzqMwDAAAAAMAaWiW3Zs0amTJliqkw0+3o0aOyevVqc3vZsmVSpUoV8ff3l99++03CwsJk7NixUrhwYUmbNq1UrFhR5s+f7zxfaGiodO7c2Xl/yZIlzbkdRo4cKTNnzpRvv/3WeT29ll5Tb8+bN0/q1q1rHlutWjXZv3+/bNq0SapWrSoZMmSQxo0by9mzZyM8h08//VRKly5tQjUNyT744APnfY7zLly4UBo0aCDp0qUzY16/fr25X6/9/PPPy+XLl53j0THGRdu2beXSpUvyySef3PPYzz77TJ599lmz6W2k8DnzQkjCAAAAAABAItKgTQOzcuXKyejRo82+7NmzmxBMDRo0SCZOnChFihSRLFmymCDviy++kA8//FCKFy8uv/zyiwmm9DH16tUzYV++fPnkm2++kaCgIFm3bp107dpVcufOLa1bt5b+/fvLnj175MqVKzJ9+nRzDW07PXHihLk9YsQI06JaoEAB6dSpkzzzzDOSMWNGM04N4vQcw4cPl2nTppnj58yZYz6fOnWqVK5cWbZt2yZdunSR9OnTS4cOHZzPc8iQIeZ56Jj1tgZwBw8elFq1apnr6Tn27dtnjtXQUGmopy24jtciOpkyZTLn1NdPr6nXjsqhQ4dMiKjBolY6vvLKK/LXX39JwYIFE+EriaTCAhgAAAAAAMAamTNnFj8/PxOU5cqV6677NaB65JFHzG2d4+3NN9+Un376SWrWrGn2acinFXsfffSRCfN8fX1l1KhRzsdrhZ4GWFpxp0GcBmVadafniup6GvY1bNjQ3O7du7cJ3VauXCm1a9c2+7TqTwM2Bw3/Jk2aJE899ZTzert37zbjcQ3z9LxNmjQxt3V82uqqYZ5W8ulroBV5kcej89oVLVo0Vq/jSy+9ZALHt99+W4YNGxblMZ9//rmpLNRQVOnz1EAzrpWA8KAw7w6VeQAAAAAAIBlpe6uDhl83btxwhnsOt2/fNlVxDu+//74Jro4dO2YWfND7K1WqFKvrVahQwXk7Z86c5mP58uUj7Dtz5oy5ff36dVPtpgGfVuM56Nx+GtBFd16tElR6Hg3zotOjRw+zxYa2IWvw2bNnT+nevftd92v7sbYXu7Yca0WjhoxaFejtnaCZ2ZCEqMxLQmPGjJGlS5fK9u3bzf8qaL86AAAAAACIP9eW0WvXrpmP+rd33rx57wqzlK7qqgGVVstp9Z62yL711luyYcOGWF1PK/sctFouqn3ayus6Hp2rrkaNGhHO4+Pjc8/zOs6TWDSc01beN954466VbFesWCHHjx+XNm3a3BXyaeVh5IAUKSXMYyHYGGnS36pVK/PDgkkkAQAAAACIHS2I0VDpXsqUKWNCO62405baqKxdu9bMQ6dtpw5aPRef692LVunlyZNHDh8+LO3atYv3eRJrPFpdp3MKastv5Oo8zSmefvppM7de5MIkvY8wL6W22YaS5sXE0ZPv2jsPAAAAAABiplVkWjmnCz3onHa6IEVUtMpOq+504QataqtTp45ZBVYDPF0EQueo0wUmZs2aZSrRdP662bNnm9Vo9bbr9fR+XXBCF8mI3BIb1yygV69e5hyNGjUyc/Ft3rxZLl68KH379o3189cqP62Q05Vudf5A3XRRjUWLFpn9saXz8mmVoM7Z52gT1tV3//e//8l3331nFhpx1b59e2nevLlcuHAh2tcd7pWgBujQsPAUsQEAAAAAAHtoQKdtqVp5p6vSauVddF5//XWzwINWoJUuXdoEaNp26wjrunXrZirTtJ1UQ63z589HqNJTOr9dyZIlzXx8ej0NA+PrhRdekE8//dQsJKFz62nFoBb5uIaH96KVhC+++KIZs45nwoQJZv+5c+fuqiqMjfHjx8utW7ecn2u4qe3KDz300F3H6j5dEERXCIadvMJ17eF4mrHjuKQEbUtlM0m5Ky3TdfTXJ5S+afv06cOceQAAAACAZKUBzpEjR0yQFBAQ4O7hAEiE92uCKvPCwsJTxKbpvZa/um66LyqDBg0yE1PGtO3duzchLysAAAAAAACQBHPm3UkZc+YNHjz4rr716Kry+vXrJx07dozxfEWKFEnU8QEAAAAAAAAJX802/h26VolLS632qusGAAAAAAAAeFSYx+IRMdMJOnX1F/2oS0pv377d7C9WrJhZjQcAAAAAAACIC9psk9Dw4cNl5syZzs8rV65sPq5atUrq16/vxpEBAAAAAADAE9Fmm4R0FVvdAAAAAAAAgMRAZR4AAAAAAACQKubMozIPAAAAAAAA8JA2WwrzAAAAAAAAgGTjndA225SwAQAAAAAAu61evVq8vLzk0qVLklIUKlRIJk+e7O5hwMOwAAYAAAAAAKnQV9v+SbZrta2cL07H169fXypVqhQh6KpVq5acPHlSMmfOLDYEi++8845s3LhRrly5IsWLF5cBAwZIu3btojx+7ty50rZtW3nyySdl8eLFyT5epCwJCvNCqGoDAAAAAADJwM/PT3LlyiU2WLdunVSoUEFeffVVyZkzpyxZskTat29vgsamTZtGOPbo0aPSv39/qVu3rtvGi5QlQW22YWHhKWIDAAAAAAB26Nixo6xZs0amTJli2mp100AscpvtjBkzJDAw0ARpJUuWlHTp0knLli3lxo0bMnPmTNPCmiVLFunVq5eEhoY6zx8cHGzCtbx580r69OmlRo0a5txx8dprr8nrr79uqgWLFi0qvXv3lkaNGsnChQsjHKfX1Wq9UaNGSZEiRaI8l463U6dOkjFjRilQoIB8/PHH8XrdkHokbM680LAUsQEAAAAAADtoiFezZk3p0qWLaavVLX/+/NEGYe+++65pY12+fLkJ5Zo3by7ff/+92WbPni0fffSRzJ8/3/mYHj16yPr1681jdu7cKa1atTJB3IEDB5zHaGioYWFcXL58WbJmzRph3+jRoyVHjhzSuXPnaB83adIkqVq1qmzbtk1eeukl6d69u+zbty9O10bqksDVbKlqAwAAAAAAiUdbVbWlVivt7tVWGxISItOmTTPVcUor8zTAO336tGTIkEHKlCkjDRo0kFWrVkmbNm3k2LFjMn36dPMxT5485jFapadBoO5/8803zT6t9IvL3Hzz5s2TTZs2meDQ4bfffpPPPvtMtm/fHuNjH3vsMRPiKW3b1bn4dLw6BiAqhHkAAAAAAMAjaeDnCPKUzl+n7bUa5LnuO3PmjLm9a9cu0/paokSJCOfR1tugoCDn53v37o31GDR4e/755+WTTz6RsmXLmn1Xr16V5557zuzLli1bjI/XufdcKwI1wHSMF0j0MC805L+ecwAAAAAAgOTk6+sb4XMNw6LaFxb27xRb165dEx8fH9myZYv56Mo1AIwtndvv8ccfN9V0ugCGw6FDh8w8f3qfg2MMadKkMW20jhAypvECUaEyDwAAAAAAWEXbbF0XrUgslStXNufVyreEri6r8/PpyrXjx4+Xrl27RrivVKlSpgrQ1dChQ03Fns4JGN0cgECSh3l37pAUAwAAAACAxKWtshs2bDDVbVoxF3lhifjS9lpdXVar6HThCQ33zp49KytXrjTtrk2aNHGGcWPHjjWLaUTXWqtBnq5i26JFCzl16pQzhNSxBgQESLly5SI8RlfeVZH3A8m6mm1YaHiK2AAAAAAAgD10UQptg9UFLLJnz24WrEgsutCFhnn9+vUzi0w0a9bMLF5RoEAB5zHaBqur00Zn5syZZiVdDfxy587t3J566qlEGycQHa/w8PB4p1mdP1wvKcFnL9Z09xAAAAAAAEh0t27dkiNHjkjhwoVNtRgAz3+/JnDOPNpsAQAAAAAAgOTCAhgAAAAAAABAqlgAI4TKPAAAAAAAACC5UJkHAAAAAAAApIYwL/QOlXkAAAAAAACAZ1TmhVKZBwAAAAAAAHjGnHl3QhNvJAAAAAAAAABixJx5AAAAAAAAgIcgzAMAAAAAAABSxQIYISyAAQAAAAAAkt7q1aulQYMGcvHiRQkMDJSUoH79+lKpUiWZPHmyu4cCD0JlHgAAAAAAqVChwUuT7VpHxzZJcMhVq1YtOXnypGTOnFlscvDgQalcubL4+PjIpUuXItyn4582bZocO3ZMsmXLJi1btpSxY8dKQECA28aL1B7mhbAABgAAAAAASHp+fn6SK1cusUlISIi0bdtW6tatK+vWrYtw35dffimDBg2Szz//3ASR+/fvl44dO4qXl5e8/fbbbhszPJ93Qh4cHhaeIjYAAAAAAGAHDbzWrFkjU6ZMMcGXbkePHjVttnrbUf02Y8YM0267ZMkSKVmypKRLl85Uvt24cUNmzpwphQoVkixZskivXr0kNPS/YqTg4GDp37+/5M2bV9KnTy81atQw546PoUOHSqlSpaR169Z33afhXu3ateWZZ54xY3n00UdN8Ldx48YIx4WFhcnAgQMla9asJqwcOXJkvMaC1CNhYV5oeIrYkoL+oOncubMULlxY0qZNK0WLFpURI0bI7du3k+R6AAAAAACkBBri1axZU7p06WLaanXLnz9/lMdqcPfuu+/K3LlzZfny5SaUa968uXz//fdmmz17tnz00Ucyf/5852N69Ogh69evN4/ZuXOntGrVSho1aiQHDhxwHqOhoYaFMfn555/lm2++kffffz/K+7Uab8uWLc7w7vDhw2ZMjz32WITjNHjUUHHDhg0yYcIEGT16tPz4449xes2QuiSozTacNtto7d2716Tr+kOjWLFi8scff5gfRNevX5eJEye6e3gAAAAAAFhJ58TTllqttLtXW622ueqcdFpAo7QyTwO806dPS4YMGaRMmTJm0YxVq1ZJmzZtzNx106dPNx/z5MljHqNVehoE6v4333zT7NNKv5jm5jt//rypIPziiy8kU6ZMUR6jFXnnzp2TOnXqSHh4uNy5c0defPFFee211yIcV6FCBVP8o4oXLy5Tp06VlStXyiOPPBLHVw6pRcLCPFpUo6Wpvm4ORYoUkX379pkfMoR5AAAAAAAknAZ+jiBP5cyZ07S0apDnuu/MmTPm9q5du0zLbYkSJSKcR1tvg4KCIhToxESLdTSse+CBB6I9RqsENRz84IMPTCuvLpTRu3dvef3112XYsGERwjxXuXPndo4XSIIFMMIS8vBU5/Lly6YHHgAAAAAAJJyvr2+Ez7U9Nqp92jmnrl27Zlad1fZX/ejKNQC8F22x/e6775zFOlp5p9dIkyaNfPzxx9KpUycT2D333HPywgsvmGPKly9vuvW6du0qQ4YMEW/vf2c+i2m8QKKHeZJCvrk0gdfNlb+/v9kSiybw7733HlV5AAAAAADcg7bZui5akVgqV65szquVb7oCbXzpnHuu4/v2229l/PjxZtELXVjDMZ+fI7BzcASIGv4BblkAQyvzUsI2duxY0wvvuum+qOiy0o7VdKLbIpfjHj9+3LTc6qSaWooLAAAAAACip62yuiCELi6p884lVqWatte2a9dO2rdvLwsXLpQjR46YBSo0A1i6dKnzOF2hdtGiRdGep3Tp0lKuXDnnpgGeBnd6W1fQVY8//riZaksX2tDr6KIWWq2n+yNXBQLJN2deEq0Em9wGDx4sffv2jbAvuqq8fv36mUkuY6Lz4zmcOHHCTLapq9hoqS0AAAAAAIiZLkrRoUMHs4DFzZs3TRiWWHShizfeeMP8fa/FN9myZZP7779fmjZt6jxG57zXqbISYujQoabgRz/qdbJnz26CvDFjxiTCs0Bq5hWegNrO2q3mSEqw9pt2SXJefbNqkFelShWzwg3Je+I7njfq5clhvxYvfOruISCeFnz675wf8Dy87zwT7znPxXvOc/0+qqG7hwAkmlu3bpkgrHDhwhIQEODu4QBIhPdrwirzQhK/fz2l0CCvfv36UrBgQTNP3tmzZ5333WtpbQAAAAAAACAqtNkmEe2F10UvdMuXL1+E+5joEgAAAAAAAMke5uniEYiazqt3r7n1AAAAAAAAgGQL8ySRVpMBAAAAAAAAcG9U5gEAAAAAAAAegjnzAAAAAAAAAA/hneA225SwAQAAAICHKFSokEyePNn5uZeXlyxevDjGx+h83s2aNUuG0QEAkhpttgAAAADgwU6ePClZsmQxt48ePSqFCxeWbdu2SaVKlZzHTJkyRcLD6awCAEntYd7WDS8l3kgAAAAAAHGWK1euex6TOXPmZBkLAMD2NlsAAAAASMXCwsJkwoQJUqxYMfH395cCBQrImDFjzH27du2SBx98UNKmTStBQUHStWtXuXbt2l2trxMnTpTcuXObY15++WUJCQlxHnPmzBl5/PHHzTm04m7OnDl3jcG1zVaPUZUrVzb769evH+FaDsHBwdKrVy/JkSOHBAQESJ06dWTTpk3O+1evXm0ev3LlSqlataqkS5dOatWqJfv27UuS1xGIDcf35aVLlySl0Pdonz593D0MpKbKPAAAAABIzQYPHiyffPKJvPPOOyYQ05bXvXv3yvXr16Vhw4ZSs2ZNE5JpKPfCCy9Ijx49ZMaMGc7Hr1q1ygR5+vHgwYPSpk0b0x7bpUsXZwh34sQJc7+vr68J4PRc0dm4caNUr15dfvrpJylbtqz4+flFedzAgQNlwYIFMnPmTClYsKAJJHW8OoasWbM6jxsyZIhMmjRJsmfPLi+++KJ06tRJ1q5dm6ivIdznf0ffSrZrPV5oQJxDLn0vuM4PqYGyvsdsqTTV1nV9f3z88cfy119/SbZs2eSll14y7xvXALJv377y559/Sv78+WXo0KHmfQ0kBGEeAAAAALjQqjXdXGnVnW6url69auaimzp1qnTo0MHsK1q0qAn1NOC7deuWzJo1S9KnT2/u0+O0ym78+PGSM2dOs0/nutP9Pj4+UqpUKWnSpImphtMwb//+/bJs2TIT0FWrVs0c/9lnn0np0qWjHbuGbkqr/KJrv9Wgcdq0aSZUbNy4sdmn4/3xxx/N+QcM+C900SrDevXqmduDBg0y49PnpdV8QHLTcDo2beXJpXfv3vLDDz+Y6try5cvLhQsXzOZw5MgR857RIFyravW9raG+BvgangPxRZstAAAAALgYO3asqfxx3XRfZHv27DGh30MPPRTlfRUrVnQGeap27dqmLde1VVWr5zTIc9A/8h2Vd3qONGnSSJUqVZz3a+AXGBiYoOd36NAh08qr43HQqj+t6NNruqpQoUKEsamYKgOBxKCVa2vWrDFhubbV6qaLu0Rus9VAWt8PS5YskZIlS5p28JYtW8qNGzdM1amu/KyBuVa0hoaGOs+v79v+/ftL3rx5zXu0Ro0a5txxoe8VDcW//fZbeeKJJ0yLu75XH3nkEecxH374odmv1Xsawmtlro5PK3ld6c8FrZbVqlgNK0eOHJng1xApG2EeAAAAAERqnb18+XKETfdFpvPYJZSGaK40qNA/7G3hOj4dm7JpfEiZNMTTFnWtUNW2Wt20RTUqGty9++67MnfuXFm+fLkJ5Zo3by7ff/+92WbPni0fffSRzJ8/3/kYDdXWr19vHrNz505p1aqVNGrUSA4cOBDh+921JT6y//3vf1KkSBETJGpgp8GhVt25VubpNR5++OEIj9OKPN3vSoNHDRU3bNhgWt5Hjx5tKmWB6BDmAQAAAIALbafNlClThC1yi60qXry4CfS0dS4yrcLZsWOHaWl10LnmvL29TQVRbGgV3p07d2TLli3OfVrVF9Pk/4458lyrkCLTVmA9znXuO63U07n9ypQpE6uxAUlJq2H1e1Qr7bRSTTfXClZX+r2rFXK66MsDDzxgKt9+++030zKu389NmzaVBg0amHkn1bFjx2T69OnyzTffSN26dc37Qav0tD1e9zvo+zSmufkOHz5s5snT82g7vQZ/+l7V6zucOnXK2VLvoJ9fuXJFbt68GaECdsSIEeZnSvv27c2iM1H9XAEcmDMPAAAAAOJB54179dVXTXucBg/atnr27Fkz0X27du3MH+c6l562zOn+nj17ynPPPXfXH/fR0TBBq4W6detmwgptudVVL2OqCNTVafV+rVDKly+fGWPkQEIrgLp3727mxtO2Pl2BV6uBtMKpc+fOCX5dgOSkgZ8Gcg76/tIquQwZMkTY52gP11WmNewuUaJEhPNo663ONemgC9nERCtU9TEa5DnOpQGittpq6B7b0D5yO3vkdnsgKoR5AAAAABBPw4YNMyHb8OHDzaqz+ke4TnavAcOKFSvMBPm6eIV+3qJFC3n77bfjdH6tFNLWPV2EQgOJN954w1wzOjoWbTnUNj0dk1YeRTUX2Lhx40wYoeGiLuShlUA6Xp1fDPAkUbWqx9S+fu3aNVPlp1V0kav9XAPAe9H3ur7fXENBx+I0Wv2nYZ5WFJ4+fTrC4/RzrfZ1DeVtb7eHfQjzAAAAACCetG12yJAhZotMV7f8+eefo31sVPNxTZ48OcLnGgbonFyuNIBzFR4eHuFzDf90i+laWrGnoZ9uUalfv/5d561UqdJd+4CkotWuMbWLx5e24+p5tfJNw+740kpcbYPXBWUclYG6ArUqWLCg+ajz/um8fa50LjzdDyQEc+YBAAAAAACraKusLgihq9ieO3cu0SrVtJJO2+B1brqFCxfKkSNHZOPGjWbF6qVLl0aYs3LRokXRnkcXtrjvvvukU6dOsm3bNlPppy3xupqto1pPq3R1bj1txde23Q8++EDmzZsnr7zySqI8F6RehHkAAAAAAMAquiiFtsHqIhbZs2c3rauJRdvXNczr16+faYdt1qyZWQBG54900HnvdCXrmKpydUXbbNmymYU3mjRpYtpsdYVcB13lVgNCrcarWLGiTJo0ST799FOzoi2QEF7h1EnDgx3PG/Xy5LBfixc+dfcQEE8LPo3YtgPPwfvOM/Ge81y85zzX76P4Qxspx61bt0z1mQZL2l4NwPPfr1TmAQAAAAAAAB6CMA8AAAAAAADwEIR5AAAAAAAAgIcgzAMAAAAAAAA8RBp3DwBIiOuPnHD3EBBP/xRa5e4hIJ5433ku3neeifec5+I958lYAAMAYC8q8wAAAAAAAAAPQZgHAAAAAAAAeAjCPAAAAAAAAMBDEOYBAAAAAAAAHoIwDwAAAAAAWG/16tXi5eUlly5dcvdQALcizAMAAAAAAFapX7++9OnTJ8K+WrVqycmTJyVz5sxiS7BYtmxZCQ0NjXBfYGCgzJgx467HjB07Vnx8fOStt95KxpEiJSLMAwAAAAAA1vPz85NcuXKZEM0Whw8fllmzZsXq2M8//1wGDhxoPgIJQZgHAAAAAIlYPZRUOnbsKM2aNRNbaOWRViABSfG9vmbNGpkyZYoJ7nQ7evToXW22ju/BJUuWSMmSJSVdunTSsmVLuXHjhsycOVMKFSokWbJkkV69ekWongsODpb+/ftL3rx5JX369FKjRg1z7vjo2bOnjBgxwpwzJvp8bt68KaNHj5YrV67IunXr4nU9QBHmAQAAAEASCQ8Plzt37rh7GIBH0RCvZs2a0qVLF9NWq1v+/PmjPFaDu3fffVfmzp0ry5cvN6Fc8+bN5fvvvzfb7Nmz5aOPPpL58+c7H9OjRw9Zv369eczOnTulVatW0qhRIzlw4IDzGA0No2qVjUzDfH2Pv/feezEe99lnn0nbtm3F19fXfNTPgfgizAMAAACARKoe0j/+9eOyZcukSpUq4u/vL7/99puEhYWZ+bIKFy4sadOmlYoVK0YIF7RqqHPnzs77tcpIz+swcuRIU2n07bffOq+loYVWK+ntefPmSd26dc1jq1WrJvv375dNmzZJ1apVJUOGDNK4cWM5e/ZshPF/+umnUrp0aQkICJBSpUrJBx984LzPcd6FCxdKgwYNTMWTjlkDEKXXfv755+Xy5cvO8egYgcSgc+JpS61+32lbrW4611xUQkJCZNq0aVK5cmV54IEHTGWevuc0LCtTpow0bdrUfA+vWrXKHH/s2DGZPn26fPPNN+Y9U7RoUVOlV6dOHbPfQd+DsZmbT8eolXn6/tb3Q1S0Ek/f788++6z5XD/qe/batWvxfIWQ2qVx9wAAAAAAwBNp2KahWbly5UzrnPrzzz/Nx0GDBsnEiROlSJEips1P/9D/4osv5MMPP5TixYvLL7/8Yv6gz549u9SrV8+Effny5TMBQ1BQkGnB69q1q+TOnVtat25twoY9e/aYUMAROGTNmlVOnDhhbmuYMHnyZClQoIB06tRJnnnmGcmYMaMZo4YNeo7hw4eb0EPNmTPHfD516lQTgmzbts1UQWnLYYcOHZzPcciQIeZ56Jj1tlYUHTx40CxEoNfTc+zbt88cq6EhkNz0+1sDOYecOXOa9lrX70fdd+bMGXN7165dJjwvUaJEhPNom6y+9xz27t0b6zFoED9p0iQZP368vPnmm3fd/9VXX5kxaiCuKlWqJAULFpSvv/7aPBaIK8I8AAAAAIj0R33k+a+0wk63mKqHXAMADfceeeQR5/n0D/yffvrJtA4qDfm0ekjb/zTM09a7UaNGOc+tFXpaBafVOxrEaTChVXd6Lse1XGnY17BhQ3O7d+/eJnRbuXKl1K5d2+zTwMC1ZVDDPw0fnnrqKef1du/ebcbjGubpeZs0aWJu6/h05U4N87SST5+/VuRFNR4gueh7x5V+T0a1TwNzpdVwWuW3ZcuWu6r94htIp0mTRsaMGWOqdbWFNzKtEtSgX49z0PHoQhiEeYgPwjwAAAAAcKFVdK7BmiP8iksbqba3Omj4pfN6OcI9h9u3b5uqOIf333/f/HGvbYA6Ub7erxU8sVGhQoUIVUiqfPnyUVYmXb9+XQ4dOmRCBK3Gc9B5vyK3FbqeV6sElZ5HwzwgKWlQ7rpoRWLR95yeV7+Ptc02sei8e2+99dZdPzu0EnDz5s2mNV2raR0uXLhgFtDR/wDg/YS4IswDAAAAABeDBw+Wvn37RtgXuSrvXrRd1cExL9bSpUvN6plRnVcn4tcqOK2W0+o9bZHVYGDDhg2xup5rJZJWIUW1z7UySX3yySdmFU9XkSuVojqv4zxAUtJWWf3+1/kbtWLONQhLCG2vbdeunbRv39683zTc0/kktZJVw2tHJaoGbBrs62IasTVu3DhnhaxrVV716tXNfH6R6fyWer++14G4IMwDAAAAgHu01Cakekgn4dfzacWdttRGZe3atWYeupdeesm5T6vn4nqt2NAqvTx58sjhw4dNqGFb5RSgNNzWlm99/2il6pEjRxLt3Drv5BtvvCH9+vWT48ePS7Zs2eT+++83i2U46FyQ0S1oEZ0HH3zQbD/88IP5XKtrda7MV199NcrjW7RoYQJFbcOP3BoMxIQwDwAAAAASqXooqqo1rbLTYOKVV14x9+uqmRoSaICXKVMmE1joAhOzZs2SFStWmPnrZs+ebVaj1duu19L7NWTQifpjs9JmdLQVsFevXuYcjRo1MnPxaSvgxYsX76pKjOm5a5WfVjTpxP46d6BuQGJV0DlWT3b9ngsPD3d+rnPU6eZK2+Ejt8S7zhepHHNURm6JdeV6nahoi2xUx+h71NW5c+eiPcfAgQPNBsSVd5wfAQAAAAAwNKTT1lStHtKVabX6Liqvv/66DBs2zLTtlS5d2gRo2nbrCOu6detmFqNo06aNaX09f/58hCo9pfPblSxZ0szHp9fSMDC+XnjhBfn0009NhZLOracVgxp4uIaH96KVhC+++KIZs45nwoQJ8R4PACD2vMLvFTcDFtvfMeKcHvAcD9Yb4O4hIJ5+XsOcHp6K951n4j3nuXjPea5/nh/n7iEAiebWrVumRVWD2oCAAHcPB0AivF+pzAMAAAAAAAA8BGEeAAAAAAAA4CEI8wAAAAAAAAAPQZgHAAAAAEAKx3T5QMp5nxLmAQAAAACQQvn6+pqPN27ccPdQANyD433qeN9GJ829TgQAAAAAADyTj4+PBAYGypkzZ8zn6dKlEy8vL3cPC0CkijwN8vR9qu9Xfd/GhDAPAAAAAIAULFeuXOajI9ADYCcN8hzv15h4hdM4Dw/21bZ/3D0EINUZPG+Hu4cApCrvddvt7iEAqc7jhQa4ewhAkggNDZWQkBB3DwNAFLS19l4VeQ5U5gEAAAAAkApoUBDbsACAvVgAAwAAAAAAAPAQhHkAAAAAAACAhyDMAwAAAAAAADwEYR4AAAAAAADgIQjzAAAAAAAAAA9BmAcAAAAAiWT16tXi5eUlly5dcvdQAAApFGEeAAAAAMRT/fr1pU+fPs7Pa9WqJSdPnpTMmTO7dVwAgJQrjbsHAAAAAAAphZ+fn+TKlcvdwwAApGBU5gEAAABAPHTs2FHWrFkjU6ZMMa21us2YMSNCm61+HhgYKEuWLJGSJUtKunTppGXLlnLjxg2ZOXOmFCpUSLJkySK9evWS0NBQ57mDg4Olf//+kjdvXkmfPr3UqFHDtPACAEBlHgAAAAC40CBNN1f+/v5mc6Uh3v79+6VcuXIyevRos+/PP/+863wa3L377rsyd+5cuXr1qjz11FPSvHlzE/J9//33cvjwYWnRooXUrl1b2rRpYx7To0cP2b17t3lMnjx5ZNGiRdKoUSPZtWuXFC9ePEmfPwDAblTmAQAAAICLsWPHmjnvXDfdF5nu17ZarbbT1lrdfHx87jouJCREpk2bJpUrV5YHHnjAVOb99ttv8tlnn0mZMmWkadOm0qBBA1m1apU5/tixYzJ9+nT55ptvpG7dulK0aFFTpVenTh2zHwCQulGZBwAAAAAuBg8eLH379o2wL3JVXlxo2KeBnEPOnDlNe22GDBki7Dtz5oy5rdV32nJbokSJCOfRasGgoKB4jwMAkDIQ5gEAAADAPVpqE8LX1zfC5zqnXlT7wsLCzO1r166ZCr8tW7bcVennGgACAFInwjwAAAAAiCdts3VduCIxaDuunlMr9bTNFgAAV8yZBwAAAADxpO2yGzZskKNHj8q5c+ec1XUJoe217dq1k/bt28vChQvlyJEjsnHjRjNv39KlSxNl3AAAz0WYBwAAAADxpAtTaCusLmSRPXt2s3hFYtCFLjTM69evn5QsWVKaNWsmmzZtkgIFCiTK+QEAnssrPDw83N2DAOLrq23/uHsIQKozeN4Odw8BSFXe67bb3UMAUp3HCw1w9xAAAIgWlXkAAAAAAACAhyDMAwAAAAAAADwEYR4AAAAAAADgIQjzAAAAAAAAAA9BmAcAAAAAAAB4CMI8AAAAAAAAwEN4hYeHh7t7EAAAAAAAAADujco8AAAAAAAAwEMQ5gEAAABAApw9e1a6d+8uBQoUEH9/f8mVK5c0bNhQ1q5dG6vH169fX/r06XPX/l69ekmVKlXMOStVqnTX/fv27ZMGDRpIzpw5JSAgQIoUKSJDhw6VkJCQRHleAAA7pXH3AAAAAADAk7Vo0UJu374tM2fONIHa6dOnZeXKlXL+/PkEn7tTp06yYcMG2blz5133+fr6Svv27eW+++6TwMBA2bFjh3Tp0kXCwsLkzTffTPC1AQB2Ys48AAAAAIinS5cuSZYsWWT16tVSr169KMO4M2fOyJIlS5z7tHIub968MnbsWPn1119NCOjqyJEjUqhQIefnI0eOlMWLF8v27dvvOZ6+ffvKpk2bzHkBACkTlXnwaL8cvyS2+fKH/e4egkd49+oEsU2aUrnENnf2nhLbHH/+C3cPwSOs3H9WbPNCpb1im7C1K9w9BOtdrNFbbLPoD3ePwDM883UXsY1f2Wxim/Bbd8Q2YR1nSnBwcIR92uqqW2QZMmQwm4Zt999//13HvPDCC/LAAw/IyZMnJXfu3GafBns3btyQNm3aSMuWLWX//v1Srlw5GT16tLk/e/bs8Rr3wYMHZfny5fLUU0/F6/EAAM9AmJeKDB48WG7duiUpSfP+I9w9BAAAAKQwWjE3atSoCPtGjBhhKuQiS5MmjcyYMcO0t3744Yem5VUr9J5++mmpUKGC1KpVS0qWLCmzZ8+WgQMHmsdMnz5dWrVqZUJA5efnJ+nSpTNz7cWHXmPr1q0mgOzataszFAQApEyEeamIBnnvvPOOpCQ2VuYBAADA8/8TXNtVXUVVlec6Z16TJk1Ma+vvv/8uy5YtkwkTJsinn34qHTt2NNV5H3/8sQnzdD49vf/nn39OtPF+/fXXcvXqVTNn3oABA2TixInO4BAAkPIQ5sGjBaX1Fds0rFvY3UPwCEda/ym2uXXLvr6xjNn8xDY7Hrvq7iF4hJqFs4ptwndvENt4V3tAbBN+7m+xyS9/B4htgjK7ewSewevN8WIbn2u7xTpn/xHbeEXTUhsTXU32kUceMduwYcNMgKfVfBrm6SIVgwYNkvXr18u6deukcOHCUrdu3UQbb/78+c3HMmXKSGhoqKnO69evn/j4+CTaNQAA9iDMS0WttDrxLgAAAICkp8GazqOngoKCpFmzZqa9VgO9559/PsKx2marIVxi0JVsdYEN/UiYBwApE2FeKmqlfeWVV1Lc3HovDHvD3UMAAABAKnb+/Hkz/52uWqtz5GXMmFE2b95s2myffPJJ53Faqde0aVMT2nXo0CHCOXTl2g0bNsjRo0fNPHpZs2YVb29vs6DFtWvX5NSpU3Lz5k3narYaFGoAOGfOHPH19ZXy5cubSkK9rv4urwtr6H4AQMpEmAePnlvvzwvX3T0EAAAApGIavtWoUcP8/nzo0CFTFadtr7ogxmuvveY87uGHHzar2ZYtW1by5MkT4Rz9+/c3AZ+GdBraHTlyxAR8GgCuWbPGeVzlypXNR8f9uvjG+PHjzWq44eHhUrBgQenRo0es/xMfAOCZCPNSEZ3HIzb/sHtSO+7FW3fENpn97XtbBaTxFtsU7fHvL6M2ub3/otjm9rmbYpvcGeI2h1ByWPT7X2KbNBVyi23K3AoW24TvtG8ev/Dz58Um/wQ+JrY5df6G2KZwnkxiG/+fPhLbhAfYNwejhIWJbbyKx/5YrYjT1W91i8n169fl4sWL0rlz57vuK1GihGm/jWz16tUxnlMr8HQDAKQu9qUOSDL3+gXDgf/JAwAAABKHzl137tw5mTRpkgQGBsoTTzzh7iEBADwcYR7iXcFngxYDRrp7CAAAAEC0jh07ZlavzZcvn8yYMcO0xgIAkBD8S4J4V/DZ4LcTl909BAAAACBaOredzmcHAEBiIcyDR1vw2xGxzdgW9v2y5ud9TWxz5vvDYpsrf10R2xRoFodJe5LJ2n32zauZI0tasU2ZbBnENoezDRHbXLwVIrYpVvms2OSxW/bNv7j+mH1zjD5c1L6VO8O+uiS2uXPSvsXD/JvVcPcQAADwKPbNig8AAAAAAAAgSlTmudHgwYPl1q1biXpO15Vok+L81qnZwd0jAAAAAAAASDaEeW6kQds777yTqOd0XbgiKc5vm1fmbXf3EAAAAAAAAJINYR48WrECgWKbi8GZxDa5vdeJbTIVyyK2yTm2hdjm9nfrxTb5cmQU2xQNSie2+W7HcbFNn4Lfi3UKVHD3CKx3PSSb2GbPkQtim2U/nhfbfHTotNjGJ0uA2CZ4gX3/1gWUdfcIAACIHnPmAQAAAEAC/P3339KpUyfJkyeP+Pn5ScGCBaV3795y/nzihcxjxoyRWrVqSbp06SQw0L7/0AYAJB8q8xBvNszJV6xNb7deHwAAAKnb4cOHpWbNmlKiRAn56quvpHDhwvLnn3/KgAEDZNmyZfL7779L1qxZE3yd27dvS6tWrcy1Pvvss0QZOwDAMxHmId5smJPv/d+PuvX6AAAASN1efvllU433ww8/SNq0ac2+AgUKSOXKlaVo0aIyZMgQmTZtmsyePVumTJki+/btk/Tp08uDDz4okydPlhw5cpjHXLx4UXr06GHOc+3aNcmXL5+89tpr8vzzz5v7R40aZT7OmDEjynGEhoZK165d5eeff5ZTp06ZMbz00kumQhAAkLIQ5sGjtar47y9MNglIc05sc9Mrv9gmoFklsc6162Ib3yqFxDa5M/mLbdbuPyu22bnrlNjGq1phsU34yX1inbR2zQu550p5sU2OLPb9+3s5m31zZ97a7N4OhqjcOWXfv3UBgfbN4+cVHCzBwcER9vn7+5vN1YULF2TFihWmBdYR5DnkypVL2rVrJ19//bV88MEHEhISIq+//rqULFlSzpw5I3379pWOHTvK99//O5/psGHDZPfu3aaaL1u2bHLw4EG5efNmrMccFhZmAsBvvvlGgoKCZN26dSbcy507t7Ru3TpBrwcAwC6EeYh3K63+EpKQxyeGwW8OSrJzAwAAIHUaO3assxLOYcSIETJy5MgI+w4cOCDh4eFSunTpKM+j+7Xi7uzZs2ZOPYciRYrIu+++K9WqVTNVeBkyZJBjx46Zar6qVauaYwoVitt/6Pn6+kYYs7b7rl+/XubNm0eYBwApDGEe4t1K+8orryTo8YnhzE37VokDAACAZ9P/kNbKOVeRq/JcaaB3L1u2bDFh4I4dO0zAp5V0SkO8MmXKSPfu3aVFixaydetWefTRR6VZs2ZmwYu4eP/99+Xzzz8359SqPp1nr1IlC7shAAAJQpiXwgQEBDhDtthUznm6hbvsa1+pXjiP2Mbfx0tsU/rSZbFNeGBmsc6NG2Kb45fte9+dOGdf21jGzPa1jYWf/1usc+e2WMeyNtsK2UPENlnT/jvHl03Sp/UV2/j95ie28fa9I7YJuxMqtomqpTYqxYoVEy8vL9mzZ480b978rvt1f5YsWcwceQ0bNjTbnDlzJHv27CZw0881cFONGzeWv/76y7Td/vjjj/LQQw+Z+fgmTpwYqzHPnTtX+vfvL5MmTTKLZGTMmFHeeust2bBhQzxeAQCAzQjzUmBLQFwq5wAAAADEj85N98gjj5g58fR3b9d583QRCg3u2rdvL3v37pXz58/LuHHjJH/+f+cy3rx5813n05CvQ4cOZqtbt65ZETe2Yd7atWtNJZ8ueuFw6NChRHmeAAC7EOYhUaoAoxNVdWBizqNXsm2fRDkPAAAAEB9Tp041IZpW2b3xxhtmrro///zTBHF58+Y1i2PoSrO64u17770nL774ovzxxx9mMQxXw4cPlypVqkjZsmXN4htLliyJMBefVvLpghv6Uc+3fft2Z3WgzrlXvHhxmTVrllmQQ8egq+du2rTJ3AYApCyEeUiUKsDoRBX2JeY8eh9u/CtRzgMAAADEh4ZoWmWnC2ToQhMauOlKtjrnne7LmjWrOW7GjBny2muvmYUv7rvvPlNx98QTTzjPo2Gf/qf30aNHTYWfVuZp66xr2Ddz5kzn57pYhlq1apXUr19funXrJtu2bZM2bdqY1t+2bduaKj1dHRcAkLIQ5sGjVSqQRWyTL4N9c3dlT7tXbBN+K1hs4+XSGmONIG+xzYUr9s2Zd+fOv5OI2+TMiatinVD75sqykZePXfOcXQzOJLbZ/s95sY23t33zw4bcsG++Qx9/+379981o33yHcVWwYEET1sVEwzXdols4Y+jQoWaLjp4/pmvoHH/Tp083W1z/Ax4A4Fns+9ccKb4VNzUszAEAAAAAAJAUCPOQpKL6n8CoWm/jO49em1dHxXtsAAAAAAAAnoYwD1aI7zx6v5+6kiTjAQAAAAAAsBFhXipfbfZekqIlNjFbb09ft2/etezpMottsvr7i228M2UU24SfPCXWuW3ffEsZC9k1n5it8hf9d8JzqwSkF+t42TcvpPja9TMzg7d9PwdsnJ/u1Hn75qy1kbePfV+70BvM5wkAQFwQ5qVgiTHZbULDwIS03gIAAAAAACAiwjx4dBVh/R6vJcl4AAAAAAAAbESYB4+uIvz20NlEHwsAAAAAAICtCPPg0Z7MvVZsE37xpFjn+GWxzaUvtrl7CB4h+Ip980Je7X9bbFO8QKDYZveBc2Kbi5kfFNt4e4WJbW7esev7adUR++aCO3HWvjFlTOcrtsnSvba7h+ARvNKmlZQiPDxcunXrJvPnz5eLFy/Ktm3bpE+fPlKpUiWZPHmyu4cHAEghLJx1GgAAAAA8Q8eOHaVZs2bm9vLly2XGjBmyZMkSOXnypJQrV04WLlwor7/+uvP4QoUK3RXs3bp1y5ynfPnykiZNGuf5AACICpV5iJfBgwebXzrc7Z0x9dw9BAAAAMA4dOiQ5M6dW2rVquXclzXrvVc3Dw0NlbRp00qvXr1kwYIFSTxKAICnI8xzY4h15swZ8VT6GrzzzjvuHoasO2lf+2i13PvENr5B9n2v+WdbLbbxLZFFbJP29A2xTb7s6cU2v2w7IbYpbGHrb9aQrWKdsFCxTWDGbGITf9+yYpuwsHCxzanz9v28vHPkkFgn1L6vXXiIfe32/pXj/hitrJs5c6a57eXlJQULFpSjR49K/fr1nW22evuvv/4yC785Fn/T1tz06dPLtGnTzOdr166VS5cuJe4TAgCkKIR5bgyx4rN6KwAAAICkFRwcbDZX/v7+ZovOlClTpGjRovLxxx/Lpk2bxMfH565jtOW2YsWK0rVrV+nSpUuSjB0AkPIR5sGjKx5bDRyZJOMBAABA6jV27FgZNWpUhH0jRoyQkSOj/90zc+bMkjFjRhPi5cqVK8pjtOVW79fjojsGAIB7IcyDR1c82thmCwAAAM//j+a+fftG2BdTVR4AAMmJMA8xCggIiLIdOLbz/cW24i6+8wdmCfAV23iJffO+2ChNgUxiG68A+34k+uRMJ7Y5feGm2KZQHvu+n+6EWvizIDRErONr3x/H528XE5v4pfES29j4/X0r+I7YJuSgffOe2fhvnZevt9gm7T1aagEAcCf7/jWHdS0GCZnvL7YVd8wfCAAAgNTAz8/PrF4LAEB8EebB6grAe+k6fEySjAcAAABICoUKFZJffvlFnn76aVP9ly3bvyto7969W27fvi0XLlyQq1evyvbt281+XQkXAABXhHmwugLwXvZcvJHoYwEAAACSyujRo6Vbt25m5VtdMTc8PNzsf+yxx+Svv/5yHle5cmXz0XE/AAAOhHnwaHkz/CO28bl4QGwTftu+Oc5CT14T23hn9BPb3PnritjmXKG4rzyd1LJmsm9eo2Onrrp7CB7Bxp9PQZkPik1u3ykrtgkLsy9cOPW3fYti+VfPLbbxTmfhr//e9s2ZFxczZsxw3u7Tp4/ZXK1evTrC5/fff7/s2LHjrvMcPXo0CUcJAEhJLPzXHCmpLTa+C1sAAAAAAADgboR5SNK22MiBX2xXt42tURO6J9q5AAAAAAAAbEeYh2QV29VtY+tKyP5EOxcAAAAAAIDtCPPg0S4F5xfbZJQNYp2wULGNVzpfsU3YZfvmggu7dltsU7FokNhmx6HzYpu/D10Q6zwk9rls33QMYZnziU0uXrHvZ9Pff10U2zR+pLjYxvtwgFjHz75/fyUszN0jAADAoxDmIVnn1mMOPQAAAAAAgPgjzEOyzq0X1aIZCZlH75XX34z32AAAAAAAADwNYR48eh69Y9duJvp4AAAAAAAAbEWYB7e23Sa09fbMjTtimwJBucQ24ZdOiW3unLwmtvHPmUOsExoutjl54Ya7h+ARvLy9xDbhd+ybg1G8vMU2oeH+YpP82dOLbc6VyC62sXFuwdAz9v1b55Mrk1jHwp+XAADYjDAPbm27dbTZRtV+GxvtBo9OhFEBAAAA8dOxY0eZOXOmuZ0mTRrJly+ftGrVSkaPHm3+I9ud9u3bJy+++KLs3r1bLl++LHny5JFnnnlGRowYIb6+Fi6GAgCIFcI8WBnwxdbmM1cTdSwAAABAXDVq1EimT58uISEhsmXLFunQoYN4eXnJ+PHj3TouDezat28v9913nwQGBsqOHTukS5cuEhYWJm++ydzTAOCpCPPg0apmWiPWCbPvbXU5Y22xzY5FY8Q6i46LbW7etK/NVjqIddrWKCC2aV4ln9hmr1QT2/xx0742xMLn0opNquW6IrYpFWTfe+78zRCxzZ3qg8Q250Pyim3SeAWLbbLF8Xh/f3/JlevfqVby588vDz/8sPz4448mzDt//rz06NFDfvnlF7l48aIULVpUXnvtNWnbtq3z8fXr15fy5cuLj4+PqfLz8/OTN954w1TR6WPnz58vOXPmlPfee08aN25sHrN69Wpp0KCBLFmyxHS67N+/XypVqiSffvqplCtXzhxTpEgRszkULFjQPO7XX39NlNcJAOAe9k1UAwAAAABuFBwcLFeuXImw6b7Y+OOPP2TdunUmkHMs9lalShVZunSpua9r167y3HPPycaNGyM8TkO8bNmymf09e/aU7t27m3bdWrVqydatW+XRRx81j7txI+LctQMGDJBJkybJpk2bJHv27PL444+bCsGoHDx4UJYvXy716tWL92sDAHA/+0qIEIH+L5v+AoCovTP2IXcPAQAAAClwGphRo0ZF2KfzzI0cOTLK47U6LkOGDHLnzh0T+nl7e8vUqVPNfXnz5pX+/fs7j9WgbsWKFTJv3jypXr26c3/FihVl6NChzr8Bxo0bZ8I9bYtVw4cPl2nTpsnOnTvl/vvvjzCuRx55xBkI6px9ixYtktatWzuPcQSCOjYNE3U+PwCA5yLMs5wGee+88467h2GvW0vcPQIAAACkMBqm9e3b965W2uhou6sGbdevXze/u+tCGC1atDD3hYaGmvnpNLw7fvy43L5924Rq6dKli3COChUqOG9ru21QUJBpvXXQNlt15syZCI+rWbOm83bWrFmlZMmSsmfPngjHfP3113L16lUzZ55W8k2cOFEGDhwYx1cFAGALwjx4tPe3/zsfCGLWvEKg2CZrFrFO2bcfFtt83/JHsU1wSKjYprDvT2KbEz51xDa5T34mtin51xGxjVeawmKTkxk6i22+//OU2MbGf+sC/rLvZ9PF3PZ9P329w765M1++3z/G8C6y9OnTS7Fixcztzz//3FTZffbZZ9K5c2d56623ZMqUKTJ58mQTzumxffr0MaGeq8iry+oCGq779HOli1fElc7jp8qUKWPCRa3O69evnwkNAQCehznzAAAAACCRaIutLnChLbM3b96UtWvXypNPPinPPvusCfl0QQpdrCKx/P77787busCGnrt06dLRHq9hoM6pF59QEABgByrz3CggIEBeeeWVGI+JXEbPHHoRFWvT291DAAAAACLQhSu0nfX999+X4sWLm9VodVGMLFmyyNtvvy2nT582VXKJQee/05ZcbcMdMmSImWevWbNm5r45c+aY6j6tCNRKw82bN5u/J9q0aXNXJSAAwHMQ5rl5Yt17iRz2MYdeRO//ftTdQwAAAAAi0DnzevToIRMmTJBt27bJ4cOHpWHDhmaePG1x1bDt8uXLiXItXSijd+/ecuDAAalUqZL873//c66kq+MYP368qdYLDw+XggULmnHdq6AAAGA3wjx4tJdKrBXbeGXKLrYJPx5xEmQbhOQMENtcW/Cn2KbGY0Fim5qV7Ztfp9f83GKb2vcFi21a37oq1rlxU6yTxq5fj95feVxskyNLWrHNN1sviG1qFu8qtinifUVsY+N8h3ExY8aMKPcPGjTIbGrx4sUxnmP16tV37Tt69O7/tNZALrI6derIH3/8EeV5tQJPNwBAymLXb6u4Zytu5LZbAAAAAAAApB6EeR7WipuQkviUON/e28OqunsIAAAAAAAAyYYwLxVJifPthV+Y4+4hAAAAAMmufv36UbbdAgBSPsI8eLR5px8W27QJjHrOEncK//sfsc2V0/ZViQZfOS22yVExh9hm0JcnxTavNC8vttl+0r55qcJPnhLbeDd4VGyzL/wJsUn7umIdG7+/bVQ9+0axztVzYhvb3nNGOncPAACA6BHmefgcenHBfHsAAAAAAACejTDPw+fQi4uYQkBPnU/v/q7/rhAGAAAAAACQGhDmwaPn0/t6j31tkQAAAAAAAEmFMC8VialF11NbcB8pclNscyGkqNgma3n7Vv1Nn3WF2ObvQ7fFNsWGFRfbPFSykNhm5+mrYhtvby+xjVfePGKbU53t+4+k7HNqiE1WHrVv8i4bv7+tZOH8dBcC7Pr+VjuP2vczvGRgWncPAXHg5eUlixYtkmbNmsnRo0elcOHCsm3bNqlUqZK7hwYASYIwLxWJqUVX22zjOxefOw0b19vdQwAAAEAq1rFjR5k5c+Zd+w8cOGBCpZEjR8oXX3whp06dkjx58pjjhw4dagIopffPnTtX/v77b/Hz85MqVarImDFjpEaN/4LXCxcuSM+ePeV///ufeHt7S4sWLWTKlCmSIUMG5zE7d+6Ul19+WTZt2iTZs2c3xw8cONB5/59//inDhw+XLVu2yF9//WW6cvr06XPXuI8fPy6vvvqqLFu2TG7cuCHFihWT6dOnS9WqVZ2r6K5Zs8Z5fI4cOeSBBx6QiRMnSsGCBZ37jx07Jt27d5dVq1aZcXbo0MH8PZImzX9/gs6ZM0cmTJhgXqvMmTNL48aN5a233pKgoCDnazNq1ChzW5+3vn56zLhx4yRr1qzO85w8eVKyZMkSr68fAHgiwjwkeC4+d7oQfNTdQwAAAEAq16hRIxN4udJAbfz48TJt2jQT9pUtW1Y2b94szz//vAmuevXqZY4rUaKETJ06VYoUKSI3b940Idujjz4qBw8eNOdQ7dq1M4HVjz/+KCEhIeYcXbt2lS+//NLcf+XKFfOYhx9+WD788EPZtWuXdOrUSQIDA81xSoM5vUarVq2i/U/8ixcvSu3ataVBgwYmzNPra9AWOSjr0qWLjB49WsLDw00wqKHgs88+K7/++qu5PzQ0VJo0aSK5cuWSdevWmbG3b99efH195c033zTHrF271uzT5/v444+bEPHFF1805164cKHzWvq6/fTTT+ace/bsMc/r8uXL8vXXXzuP0esAQGpCmAeP9uNh+1ogKuXOJLb58WR9sU2l754Q2zxw+WOxzbwbT4ttKuXMKLYpmfE3sc3X+yuIbY72WSC2KTSjk9hmnnX/toSLbRoXuyy2yeT9l9gmZPpisc2OYSPENg83zinW+fy/yrPY8Pf3jzJQ0iDrySefNMGWKlSokHz11VeyceNG5zHPPPNMhMe8/fbb8tlnn5lKu4ceesgEWMuXLzcVd47quPfee08ee+wxUw2n1Wpa4Xb79m35/PPPTXWfBmDbt28353KEedWqVTObGjQo6kXkNHzMnz9/hGBSqwsjS5cunfP55s6dW3r06CHdunVz3v/DDz/I7t27TQiXM2dO0+76+uuvm4o/rbbTMa5fv968Ho5QU6+j59AxuNJKPse18ubNa8LIyMGpa5ttZBoCakCoXwsdV4ECBaJ87gDgSbzdPQAAAAAAsElwcLCpdnPddF9c1apVS1auXCn79+83n+/YsUN+++030yoaFQ3kPv74Y1O5V7FiRbNPQy+tsHMEeUor8LTtdMOGDc5jtNVVQzKHhg0byr59+0y1XWx999135joamGn7bOXKleWTTz6J8THaAjxv3rwIbcE6nvLly5sgz3U8+jpqu6+qWbOmaS3+/vvvTYXf6dOnZf78+SakjI7Oh7dixYoIzzMm+jXT56LBplYNEuQBSCmozPNAOr+drj4Lkfu7Rv2/igAAAEBCpqBxzNXmMGLECFNVFpUlS5ZEmL9Ow7pvvvnGVMBpgFWqVCnx8fExVWI6H562zUZ+/NNPP21aYbXSTdtps2XLZu7TufY0WItcraZzxul9jmMiV9A5gjS9L7bzyR0+fNi0Bfft21dee+01Uw2olXManumcdw4ffPCBfPrppyaE0zFrq7CGbA56TdcgL/J4lLbzakVhmzZtzN82d+7cMe2277//foTHacuwvrb62jn+BtKKw3u5du2aqYjUQE/n7dOAFABSCsI8D6T/iOncEhD5es9pdw8BAAAAKfA/zzXQitxKGx2dY05DMIf06dObj1qxpoGVzm3naH3V+eW0NdY1HNPH633nzp0zlXCtW7c2VXeRQ7ykFhYWZirzHPPaaWXeH3/8Yebhcx2vhpFDhgwxt7WiTo/XOft0cY2MGWM3HYe24fbu3dssyqFVezqv3oABA8y8edpm7FCyZElTMah/A+lCIvo66eIe99K2bVvJly+f/Pzzz5I2rW3TJwBAwhDmeWDl3ZkzZ5J1PDarWzD6X6rcJcj/39YBm3z9+3//U2wLG792F0fMF9vkfOO/X5xtcfJa3NuMklqhjPatYLfvr9i3NSWX1l/btwL47SD75hZs9a1dVd9vpHvV3UO4S92C9k02v/VC8oYesVHmh+Fim4oN/11QwSYh126LbTS4iym8i0zDO131NTINp7Q6T6vulLae6oIRWvnnGo45Hq/b/fffL8WLFzeBlv5doPPFRf79X6vYtL3VMZecftRQzZXj87gsDqFVgWXKlImwr3Tp0rJgQcQ5V7XKzfF89aOOVR+ri1K88MIL5pqu8wJGNR59DbQ6T18jVaFCBfM61K1bV9544w1zPqVVgY5r6Sq2Wm2nVZM6B19MtF1Xwz9t+X3wwQdj/RoAgCcgzPPAyrvoVp8CAAAAYA9tQdW57Vxpu61WwMVE73fM0adzy126dMlUvVWpUsXs02ozPcYxT50eo5VyutKtrhirtFVXq9pi22KrNFzTefZc6Xx/BQsWjPFx+pyUrsbrGI+2E2sI6agu1PFkypTJGRbqa6PtwlGdR9t3ozN06FATznXv3t1UOEZH7y9Xrpw88cQTsnTpUqlXr949nj0AeA7CPA8UEBBAoPf/Boyxb0U2AAAAQOkccBpq6cIL2ma7bds2M99bp07/ruR9/fp1c78GTlqJpm22Omfc8ePHzcINjsq4Ro0amRVZtd1VAztdPVar/Rxhlq6Iq9VqnTt3NivGamvslClTIhQI6OIa2trquK3X0JZVnY/OUfmmf2Pooh3aNqutvlpdpwty6OZKgzjH3HdacadVcvo3irbaKv2ood1zzz0nEyZMMMdqCPfyyy87Kx71tdHnpO3JjjZbbUGuXr16jCGdBoVaxadjnDp1aoyvv7bj6lx7TZs2lWXLlkmdOnXi8VUEAPsQ5nkgLUnHv07cuOTuIQAAAABReu+992TYsGHy0ksvmSo1Dam6detm5olzVKLt3btXZs6caYK8oKAgqVatmll5VcM/B513TwO8hx56yFT6tWjRQt59990Iba8//PCDCcu0ek8Xz9BrdO3a1XnMiRMnzBx4DhMnTjSbVqytXr3a7NNrL1q0yLT3jh492iyqMXny5LsW7NB5/Ryr3Grln4ZruiqtVgI6npcu6qHVcRq+afusthXrOR06duwoV69eNYFcv379zIq9WnE3fvz4e76uGjrq4zW4zJ8/f4zHakCoVYzadrt8+XITVgKAp/MKj6mGGVH+w5HUi08kxzVSit9OXHb3EDxCbb9v3T0Ej/B3wL//A26Tnw6cF9tMf+tXsc2cj5uJbTL7HRfbBLw3UGzjW7mA2MYrp11zr10p3lpss+ucffOu7T99VWzz6BD7fjblfPLued3c7fyPR8U2ueb/7u4hAAAQrYgTOAAAAAAAAACwFm22KWiV29SoxYCR7h4CAAAAAABAsiHMS0Gr3KZGtNkCAAAAAIDUhDAPHm3b0QtimwdL2TXXkpq6t67YpkehFWKbrIGHxTadSh0S21zvcb/Y5tud/66oZ5MHS+UW25S2cH66n1ovE9vU6VBQbJL5jXJimywBD4ltrt8MEdvkft6+r13w7yfENrlmv+buIQAA4FGYMw8AAAAAAADwEFTmpQIpea69Iq16uXsIAAAAAAAAyYYwLxVIyXPtHbh8U2zzzuI/xDZp0/uJdfLZ1w6V4dY+sU34zStim9L5MottJry3TmxztIZ9X7u30tn3Hzu1n7Ov9TegQWGxSWhANrFNWa9VYpuy1ez7tTb0uFgnTeFAsU3oovliG59nnnD3EAAAiJZ9v/WkIPGtiDtz5kyinvde5wMAAACQPGbMmCF9+vSRS5cumc9Hjhwpixcvlu3bt5vPO3bsaO7TfQAARIUwz8KKuFdeeSVRzxv5fCmp7falkW+6ewgAAABIxTR8mzlzprnt6+srBQoUkPbt28trr70madLE/c+tKVOmSHh4eKKN7/z581KxYkU5fvy4XLx4UQID7avOBADEDWFeKpSS2m5tbLMFAABA6tKoUSOZPn26BAcHy/fffy8vv/yyCfb0P9HjKnPmxJ3OonPnzlKhQgUT5gEAUgbCvFQgICAgQnVeSmq73Xz8stimQH775hMrmCuTWMfXX2xzLk0NsU1g0N9im7N7b4ttFo61a44z1aSzffNn3sx8UGxz6Yh9P8fTNvQRm/jcOie22XnjIbFNnvT/tizaJPPlf6u1bOIVYNf3t/LKlFY8nb+/v+TKlcvc7t69uyxatEi+++47E+ZpW+3w4cPl3Llz0rBhQ6lTp06M54rcZlu/fn0pX768+Pj4mApAPz8/eeONN+SZZ56RHj16yPz58yVnzpzy3nvvSePGjSOca9q0aeZcev1ly5Yl4SsAAEhO3sl6NbjF2LFjTSWeY8uRI4e7hwQAAABYSyvsrly5EmHTfbGVNm1auX37tmzYsMFUxmnopnPiNWjQwARxcaUhXrZs2WTjxo3Ss2dPExi2atVKatWqJVu3bpVHH31UnnvuOblx44bzMbt375bRo0fLrFmzxNubP/sAICWhMi8Vilyp58mqd3nV3UMAAABACvzP8FGjRkXYN2LECLNYRUx0rruVK1fKihUrTOim899pC+7AgQPN/SVKlJB169bJ8uXL4zQenfNu6NCh5rZW+40bN86Ee126dDH7tPJOq/B27twp999/vwke27ZtK2+99ZaZw+/w4cNxfAUAADYjzEulv5ykFF/tPuXuIQAAACCF0cCsb9++d7XSRmfJkiWSIUMGCQkJkbCwMNMCq8GfttQ2b948wrE1a9aMc5inc945aLttUFCQab110DZb1+l0dPylS5eWZ599Nk7XAQB4BsI8IJG1q5FXbJPJ96RY54J97R5B138W61wNFdsUzlpGbPPHuXRim/IP27dAj187+1YAzxyWXmxzLiSb2CQ4LIPY5tS12LcbJpcbIRnFNl5jt7l7CB4hg30/BqRsU/8Yw7vItH1WK+N0Prs8efLEaxXbmOhiGq68vLwi7NPPlQaJ6ueff5Zdu3aZ+fSUY3VcreYbMmTIXVWHAADPQpgHAAAAAAmQPn16KVas2F37tTpO581z9fvvvyf5eBYsWCA3b/73n0qbNm2STp06ya+//ipFixZN8usDAJIWYR5iTcv1b926JTZhzjwAAADYqlevXlK7dm2ZOHGiPPnkk2Yuvbi22MZH5MBOV9J1hIuBgYFJfn0AQNIizEOsaZCnq+HahDnzAAAAYCtdjOKTTz4xi2foIhUPP/ywWcji9ddfd/fQAAAejDAPHs37/+cHsUl4uH1zwd24k1Vsk+n6ZbGNV57iYpvwY3vFNtsvXxTbHDlxRWxz7tQ1sY3PuqViG/tmOxRJX7iI2GR/+o5im0z+9v0KeTv037nCbFKsXCaxjbevfb+npM9u40+C2JsxY0aM92t7q26u+vXr57ytC2W4rpIb+XyrV6++65xHjx69a59jXryo1K9fP8b7AQCexb7fxGBtm6xjdSwAAAAAAAC4B2GehQICAuSVV15J9DAtoW2yUY3J3XPr3d91ULJcBwAAAAAAIEWFeTYujpAUkqM6bezYsQkK01LT3Hpf7zmdLNcBAAAAAABIUWGejYsjJAVPCdRSi31/2Td3V/YMfmKbwAD7xpSjSE2xze3Q9GKdogXFNiVO2jcHVPDtULFNGh/75qXyyp9PrHPJwvkzM9g1z2iJzBvFNudvFRPbXA0JEtvcPH9TbHPlfIjYJn0G++Y9zejuAQAAEAPabJHg9t/ImFsPAAAAAAAgaRDmIcHtv5FFDvySsgU7S8NuSXJeAAAAAAAAGxHmIcklZQt2j9mbxTZ+PrnFNnfCwsU23id2iG3SZswmtgkPuyO2SedbWGxzJzRMbHP1ioXzyF65KrYJv2rfmCT4utjkyu28YpuLwXa1IqvT14PFNnm8vcQ2OUvYN1VCGn8fdw8BAACPQpiHJG/Hpe0WAAAAqUF4eLh069ZN5s+fLxcvXpRt27ZJnz59pFKlSjJ58mR3Dw8AkEIQ5iHJ23GTtO32vnaJcx4AAAAgHjp27CiXLl2SxYsXy/Lly2XGjBmyevVqKVKkiGTLlk0WLlwovr6+5tiQkBAZOnSofP/993L48GHJnDmzPPzwwzJu3DjJkyeP85xjxoyRpUuXyvbt28XPz8+cPzrnz5+XihUryvHjx02AGBgYmCzPGwDgPoR58Oi2WxvbbAEAAJA6HTp0SHLnzi21atVy7sua9b+29Bs3bsjWrVtl2LBhJoDT8K13797yxBNPyObN//1ee/v2bWnVqpXUrFlTPvvssxiv2blzZ6lQoYIJ8wAAqQNhXgoRm2o3d7W7JmXbbdas6cQ2fj7eYps7YfbNJ+Zl4/x0ly1sCb9zW2xz8Lpd84mpoMC0YptzlyycMy8wi9jGu0BxsY7PvxU0tvDysu9neBoL54JD7ARfsW9uwauX7fu3Lkc8K/Rmzpxpbnt5eUnBggXl6NGjUr9+fWebrVbi/fjjjxEeN3XqVKlevbocO3ZMChQoYPaNGjXKfNQqv5hMmzbNVO0NHz5cli1bFo9RAwA8EWFeKqp2i9zuakvbLQAAAGCT4OBgs7ny9/c3W3SmTJkiRYsWlY8//lg2bdokPj6xW9jj8uXLJvyLa3vs7t27ZfTo0bJhwwbTsgsASD0I8+D2Sr2EyPhwl0Q5DwAAAOD6n9GO6jiHESNGyMiRI6N9jFbdZcyY0YR4uXLlivV/yL/66qvStm1byZQp9isNa9Coj3nrrbdMNR9hHgCkLoR5cHulXkIMX7o70c4FAAAAOKaw6du3b4R9MVXlxYcuhtG6dWuzAq62y8Z1fKVLl5Znn302UccEAPAMhHnwaKOanBXbnLieT+xj39xGx6Wh2MYvu31zwaVPc05s0yA4vdhm6k/HxDaNq9r3syD0p/ViG++iucU2obvs+n76rFxFsU1Q5gtimz1H7BtTryD75vMMLBMktjn56z9im3u11CZWkPfXX3/Jzz//HKeqPKWP2bVrl8yfP998roGg0tVzhwwZcldVIQAgZSHMAwAAAIBk4gjyDhw4IKtWrZKgoLgHrAsWLJCbN286P9c5+jp16iS//vqrmbcPAJCyEeYhVivh2urtd5q5ewgAAABArIO8li1bytatW2XJkiUSGhoqp06dMvdlzZpV/Pz8zG1d2fbChQvmox6zfft2s79YsWKSIUOGuwK7c+f+reTX1tu4LqQBAPA8hHmI1Uq4tgqXNe4eAgAAABArx48fl++++87crlSpUoT7tEqvfv365vbw4cNl5syZzvsqV6581zEAgNSLMA8ebfHBMmKbctl9xTaZ/a6IbbIH/yy2ueVv3/dTwIlfxDa7AzqKbQrnidtcQ8lhxdbjYptaoWFim5BfD4htrv5h13ysme9Punm74svb2765WLNmsu91ylonr1jntn0/B/I+WUw82YwZM5y3+/TpYzZXq1evdt4uVKiQc367e53T9bz3ogFfbM4LAEgZCPM8SEBAgLzyyity5syZBD0+svieDwAAAAAAAMmLMM+DjB071nyMKpCLy+Mji8/5bJln74GXX3P3EAAAAAAAAJINYR48ep69RQftaoUCAAAAAABISoR5iLb9Nia2tObO+3a32ObgfXnENlWKZRPbNLjz78ptNknr8+8KcjYJD7kttimY45rY5lbezGKboMAAsY23r30rHHoF2PerSGDOdGKTItkziG3uhNk379rFK+7vGIgs7IJ9Ywq/dUdsEx5i3/eTfTMwAgDwH/t+g0ayi6799l5ttvFt901U+Zq5ewQAAAAAAADJhjAPyRYAJoW2k9a4ewgAAAAAAADJhjAPHi08LFxss2XzcbFNv/r2tbSu+qej2GbXjotim47VK4ptbtzOKrapE7hMbBOWO0hsE77JvhZprwz2NbN55bCrHTks3L5/6x7M/6fYpnjW6mIbr1W+YhufbGnFNodm2vf9VNrdAwAAIAbeMd0JAAAAAEh+HTt2lGbNEm9KmZEjR0qlSpXi9JgbN25IixYtJFOmTOLl5SWXLl1KtPEAAOKPyrxUQOe309VnU6Q8T7p7BAAAAICsX79e6tSpI40aNZKlS5eK7T755BOZNWuW/PHHH+bzKlWqyJtvvinVq/9X5Tpz5kz59ddfZd26dZItWzbJnDl+C06tXr1a3nnnHdm4caNcuXJFihcvLgMGDJB27dol2vMBgNSEMC8V0CBP//FMiZ5+a7W7hwAAAADIZ599Jj179jQfT5w4IXny5BGbacDWtm1bqVWrlgQEBMj48ePl0UcflT///FPy5s1rjjl06JCULl1aypUrl6BraRhYoUIFefXVVyVnzpyyZMkSad++vQkHmzZtmkjPCABSD8I8eLS3XqomtjlyKURs884v58Q21Yp5iW3uhIaJbXy8g8U2f12+Kbb5S+qLdeybglHqZLJvrqwbjXqJbTLc2C02eTDDVrGOfdP4SVDAQbHNscUHxDbeaeybZefUafH4OfOuXbsmX3/9tWzevFlOnTolM2bMkNdee83cFxoaKl27dpWff/7Z3FegQAF56aWXpHfv3s7H6zFaqfb555+Lj4+PdO7cWcIjzZe5fPlyeeONN0wlnR5Ts2ZNmTJlihQtWtR5zD///GPOs2LFCgkODjZB3Pvvvy81atS4a8xz5syJ8Pmnn34qCxYskJUrV5qgrX79+rJmzb+LzWmLbb169UwAWKhQIXnhhRdk//79snDhQgkKCpL33nvPjEf36+OLFClinkvVqlXN4x2vhYM+9x9++ME8njAPAOLOvn/NAQAAAMCNNAjTdlDXTfdFZ968eVKqVCkpWbKkPPvssybIcoRxYWFhki9fPvnmm29k9+7dMnz4cBNu6WMcJk2aZAJAfdxvv/0mFy5ckEWLFkW4xvXr16Vv374mMNTAzNvbW5o3b27O7wgUNXA7fvy4fPfdd7Jjxw4ZOHCg8/7YzI8XEhIiWbP+u9CVBm1dunQxId3JkyfN5w7a9VO7dm3Ztm2bNGnSRJ577jkTAOpz37p1qwkY9fPIgaSry5cvO68FAIgbKvNSmZQ2f17fN95w9xAAAACQwowdO1ZGjRoVYd+IESPMIhJR0dZaDbKUzpmnQZVWtWl1m6+vb4RzFS5c2Myvp2Fe69atzb7Jkyeb39Ofeuop8/mHH35oqutc6UIUrjT4y549uwkItQ32yy+/lLNnz8qmTZucIVmxYsVi/Zy1BVZbgx9++GHzuZ4jXbp04ufnJ7ly5Ypw7GOPPSbdunUztzWcnDZtmlSrVk1atWrlPJeGgKdPn77rsUqfu47zo48+ivX4AAD/IcxLZVLa/Hl/X7/u7iEAAAAghdFgTavgXPn7+0d57L59+8zCDo5KujRp0kibNm1MwKdhntJWVw3fjh07Jjdv3pTbt287V5bV4E8r31xbYfUc2qLqWtl24MABE5xt2LBBzp0756y403NqmLd9+3apXLlyvKrdxo0bJ3PnzjVttDp/3r3o/HcOOgeeKl++/F37zpw5c1eYt2rVKnn++efNAhxly5aN81gBAIR58HDHr4aKbfx87Oter10yh7uH4BFsfJ3+sG+6Q/HzcfcIEF+3G78otgnv0VNsE1Ll3z9CbZGm+WNimw0hrCYfG4c32/d7Studz4htwjp9KbbR4C668C4yDe3u3LkTYcELDeH08VOnTpVly5ZJ//79TSutVqtlzJhR3nrrLRPKxcXjjz8uBQsWNCGYXkvDPA3xNBhUadPGb17UiRMnmjDvp59+ihDSxUSrDR10Pr3o9kVu8dVqRX0eWlygbbgAgPghzAMAAACAeNAQb9asWSao05VgXTVr1ky++uorszqsrhiri1446CqxDrqia+7cuU2498ADDzjPu2XLFrnvvvvM5+fPnzcVgBrk1a1b1+zTufVcaRCni1jofHuxrc6bMGGCjBkzxrT0OharSCpa9aeLXeiqubogCAAg/gjz4NHz6bV5NeJcJgAAAEByWbJkiVy8eNGsPquhXOQ57rRqTxeH0MBPAzOdL2/27Nlmvji97bq6q1bHFS9e3Cyk8fbbb8ulS5ec92fJksWsGvvxxx+b4E9bawcNGhThem3btpU333zThIg6558epwtUaBWfVgRGpqGatu3qXHu6Qq2utKsyZMhgtsSkrbUa5Onz1NfFcS2dj49FMAAg7gjz4NHz6f1+6oq7hwAAAIBUSsM6XTAicpCnNLTSyjetltOFLXQePW0/1dBNq/S0/dahX79+Zt68Dh06mFVqO3XqZFaq1fn0lO7TOe169eplWmt11dx3333XOSefIxj74YcfzLl0gQqt7itTpoyZry8qumiFtui2bNky1gt9xNfMmTPNarkaMurmoKvvasUeACBuCPPg0faeviq2OXrCvoCxbJEgsU3+zPeeXDm5Zfa370fihmMXxTYdi8Ztjp/kcDbN3RUH7vbnufjNXZSU/DbNE9v4v/jfH6LWyGrZz8zQELFNiSwXxDbrT2QU29R/4u6Ax90OWDg/XWhIxHnVPMn//ve/aO+rXr26cwGL6dOnm82Va6ilC17oira6RUdDQ1251pXrAhlK59SbP39+lI/XgM41pDt69KjcS1TjiepxkcehlX6u+2bMmGE2AEDisO8vVyR6m6yuIuWwZ88eeeWVV6J8vOtxAAAAAAAAsA9hXipok3UN73RFqehaaaML+WyeU69ix/7uHgIAAAAAAECyIcyDR8+p99Mx+9p8HioWKLbJ5HtSbLPplH0/fnb+/d9E07Y4c/Gm2GZjrkfcPQSPUDSLt9jG67x9rb/hl+2bmkBO/jsxuy28H3pCbHPwQpB9Lds+9rUjp/H3EdvY2NIaZt+QAACwmn1/TcNtAgICYqzOow0XAAAAAADAvQjzEOUkvFGJKehzVwtuk1eGJfs1AQAAAAAA3IUwDx7dgmtjmy0AAAAAAEBSIcxDorThuqsFNyTsvyXvbfHHWbsWCVFVc6UX29j4tWte3ldss/mUfV+78zdvi22q5rJvTJtP+Ylt8pevLbYJ/36e2MarZDGxSfhl+6a5sPHngI1unrdv3lPftPb9+n/0kH3zHZZ19wAAAIiBff+awyPbcO+1Ei4AAAAAAAASjjAPybJ4RlJ5tPfQZL8mAAAA4OrUqVPmP76XLl0q//zzj2TOnFmKFSsmzz77rHTo0EHSpUuX4Gt07NhRLl26JIsXL46w/4knnpDt27ebTpksWbLIww8/LOPHj5c8efIk+JoAADsR5iFZFs9IKsuOnnfLdQEAAAB1+PBhqV27tgQGBsqbb74p5cuXF39/f9m1a5d8/PHHkjdvXhO4JZUGDRrIa6+9Jrlz55bjx49L//79pWXLlrJu3bokuyYAwL0I8+DR3v1oo9imcKlsYpt3954T23wxspy7h+ARymXPJLbJHrBfbBMcmlFsUz33HbFNyEcfi21CL9g3z2joysNik6tvfiW2qRt4SmyT4fofYps77e37ty78hn3z0xX09RZP9tJLL0maNGlk8+bNkj79f3PdFilSRJ588kkJD/93nmCtqnv11VdNZd3ly5dN5d64ceOkadOmMmPGDOnTp498/fXX5uPff/8tderUkenTp5uQbuTIkTJz5kxzHi8vL/Nx1apVUr9+/QjdMQULFpRBgwZJs2bNJCQkRHx97ZsPGACQcIR5qawFNjg42N3DAQAAAKymvzNH/r1Zq+10c3X+/Hn54YcfTEWea5DnSsO3sLAwady4sVy9elW++OILKVq0qOzevVt8fHycx924cUMmTpwos2fPFm9vb9Oiq1V2c+bMMR/37NkjV65cMQGfypo1613XunDhgjm+Vq1aBHkAkIIR5qWQ+el0jowcOXLcswX20UcfleQ2ePBguXUriSovApL/+QAAACBl09+fR40aFWHfiBEjTIWcq4MHD5rKu5IlS0bYny1bNufvvy+//LI89NBDsnHjRhPIlShRwlm550or6T788EMT9KkePXrI6NGjze0MGTJI2rRpTcCYK1euu8arFX9Tp041geD9998vS5YsSZTXAQBgJ8K8FDI/nc2ryeovMu+8806SnLvx4GVJcl4AAACkXvqf0X379o2wL3JVXkw0uNNqvHbt2pkATheoyJcvnzPIi4oukuEI8pS21+p/2MfGgAEDpHPnzvLXX3+ZELJ9+/Ym0HO05AIAUhbCPCR5tV1sfwmJjzvB9s1J9c+Ri2KbPAUDxTZpvOxr+b4Sklts4+99TWzj631TbHPjThaxjY+XffNS+eS1bw5GG/lVtutnQXqv38U2d7ztmx/2ctoqYhvfP/9th7SJl/9/bZ2IXtooWmqjovPeaWC2b9++CPsdVXdaTef6MSaR22L1vI759u5FKwF107CwdOnSkj9/fvn999+lZs2asXo8AMCzEOYhyavtbK4aBAAAAOIrKChIHnnkEdPi2rNnz2jnzatQoYL8888/sn///hir82Li5+cnoaGh9zxOKwIVc2UDQMpFmAe3zPGXaLweTJrzAgAAALHwwQcfSO3ataVq1apmTj0N7nQBi02bNsnevXulSpUqUq9ePXnggQekRYsW8vbbb5uKPr1Pq+8aNWoUq+sUKlRIVqxYYaoANUTMnDmzbN261VxHV77NkiWLHDp0SIYNG2badanKA4CUizAPbpnjL7E80vd/SXZuAAAA4F40ONu2bZtZ0VanotEKPG3RLVOmjFmF9qWXXjLHLViwwHzetm1buX79ugn0xo0bF+vrdOnSRVavXm1Cw2vXrsmqVatMqLdw4UKzOIeeU+fZ03Bw6NChcZrjDwDgWQjz4NGaPlVWbHPmon3zif3zl33z+PX6+JTYpluL7GKbgDRRt+u4Uya/e8/7k9wC/f4W21wIjrhKoQ1Cz9g3B6OV83fdtmu+w/Dz9n1/p8n6bxufTXx97ft5KaH2vU4S5i228c7gJ55OQ7T33nvPbNHJmjWrfP7551He17FjR7O5atasWYQ587Jnzy4//PDDXY/9+eefEzR2AIDnIcxDvBa5SMpFLQAAAAAAABA1wjzEa5GLqObAi8sKt4mlUIueyXo9AAAAAAAAdyLMg1tWuE0sTYYsF9uUqpzb3UPwCD1bVxTb+PnY13qUK519bZEBYafFNmduVxbb3AwNFNtkLWRfK7kEWDink2VttuJvYftoQEaxjU+YfSt3Xj9v39Qb3mnsa20PP3ldbGPhTyYAAJzs+8sVAAAAAAAAQJSozIPbJEpbbrqGiTUcAAAAAAAA6xHmIV4CAgLumjcvrotiJEZbro1ttgAAAAAAAEmFMA/xMnbs2AQvipEYK+IWLWPf/E8B/va9rdJYOD/OrTthYps7YeFim+t3MottQvyzim28Q++Ibfx9roptvAoXFOvcsm+eMwmz7OfT9Ytim/Bs9n0vXQnJK7a5dcq+ueC8vO2bZSfctveciGQRz+Dl5SWLFi2SZs2aRXn/0aNHpXDhwrJt2zapVKlSso8PAJA07EsdkKpXxAUAAAA8SceOHWXmzJnOz7NmzSrVqlWTCRMmSIUKFdw6tvz588vJkyclW7ZsSfr7/osvvihbtmyRPXv2SNOmTWXx4sVJdj0AAGEeLGvVjbOqzybWcAAAAIB4adSokUyfPt3cPnXqlAwdOtSEWseOHXPruHx8fCRXrlxJeo3Q0FBJmzat9OrVSxYsWJCk1wIA/IswD1a16sZVrzlbEmUsAAAAQHz5+/s7QzP9OGjQIKlbt66cPXtWsmfPLq+++qpph/3nn3/M/e3atZPhw4eLr6+veczIkSNNNZsGYnr7woUL0r59e3nvvfdk0qRJ8vbbb0tYWJj07t1bhgwZEuHaWnnXuHFjWb16teTOndtUBLZs2TLWbbYzZsyQPn36mI8DBgyQv//+W+rVqyeffvqpqexTO3bsMMds3rzZtPYWL15cPvroI6lataqkT59epk2bZo5bu3atXLp0KUlfawAAYR4sWxQjrs6csG9OqsDAtGKb4iWCxDaXg+2b48zbS6xz8pp984llKZB0rTrxdfm2fd/jNgrfd0BsE349gauapwLetWqIdc4cEttkz2Df3IJXCto372maDP+GRzYJCw6VlOTatWvyxRdfSLFixSQo6N9/nzJmzGjCsjx58siuXbukS5cuZt/AgQOdjzt06JAsW7ZMli9fbm5rIHf48GEpUaKErFmzRtatWyedOnWShx9+WGrU+O/nwrBhw2TcuHEyZcoUmT17tjz99NPmGqVLl471mG/cuCFjxoyRWbNmiZ+fn7z00kvmPBrOKQ0fK1eubEI7rfbbvn27M4gEACQ/wrxURv9n0N2LYgAAAAA2Cw4ONlvk6jvdorJkyRLJkCGDuX39+nVTIaf7vP9/wRFtu3UoVKiQ9O/fX+bOnRshzNPKu88//9yEfGXKlJEGDRrIvn375PvvvzfnKVmypIwfP15WrVoVIcxr1arV/7F3H9BRVV/Dxnd6AgmhE3roTaqINKWIgGCh2AALVUWKiPgioBS7KCJF8a8iQQW7oIIgiIKKIIJioUvvPYQSUudb5/AlJiEJCSRz9yTPb60xM3fu3HsyxMlkzy7Sv39/e/2ZZ56RpUuX2oy+N954I8vfb1xcnEyfPj35uKYHoAkGrlmzRpo0aWLLhU3WXs2aNe39JjMPAOAcgnn5TMmSJR07d2aTbS9bmdty9ngAAADI98yH1BMmTEi1bdy4cbYENj0m8JZUanry5EkbSDOlryYYVrFiRfn4449l6tSpNuPOZO7Fx8dLoUKFUh3DBPlMIC9JqVKlbBZcUkAwaVvaypdmzZpddNtkzqWnTp06snv3bnvdlAGbTEDD19fXDu1IYoJ2hQsXtgMtTDBv+PDhNmBoMv9MZqAJIFapUuUSzyIAILcQzMvFoQ25XWKa20G2nF5/ZpNtL9fdLy/P0eMBAAAA5v2xCWCllFFWnmH6xpmy2iSm31xoaKi8/fbb0rlzZ1umaoKDHTp0sNtNVp7phZdS2rJV05suvW0mg+9ymSw/k4VnmKEVWWWCmD179pSFCxfaAKAJbJrvoWvXrpe9FgDA5SOYl4tDGzylxDSjIFtOrD9l8DM3gpsVq+nrk3Xs2Fmnl+AR6oQXFW2ORunrTxfg5yPaFPbR1yurUMh+0SYmIXXGhQq++n7te5XOvfYPly0ySjTxKuRcVn2GfPS9NiUG6XtPkJjgEm1iTur7XedXQN9rU2YltVlhgm4moy46Otr2ujPZeSkHVyRlx+WE1atX22EZKW+b/nbpMetIj8kUNMMtTBaeYcp7zSCLlH33TO8+czHv7Xv06GGn9xLMAwBn6PvNiTwb/PSU4CYAAACQHaa/3qFDh5LLbE3/OVNOe8stt0hUVJTtOWcy2Uwpq8luM5Ntc8qnn35qp8q2bNlS5syZY0t7Z86cma1jmAzAIUOG2FJgU3I7ePBgadq0qQ3umYCk6ZdnBnKYybhmIu9vv/0m3bt3T378xo0bJTY21k7hPX36dHKZb0YTdAEAV4ZgHlSWKGeVf6u+OXo8AAAAILvMBFoz9MIwfe9MzzkTZGvdurXdZt4DmwCZCfqZslszgTaj/nvZZcp3TaDQTKA1a/jwww/tAI3sKFCggIwcOdKW0u7fv9/200sKCJq+fcePH7fZf4cPH5bixYtLt27dUvUU7NSpU6psw6TMQJdLX3YqAOQFBPOgskQ5q0bO/zvHjwkAAABkVUREhL1kZuLEifaS0rBhw5Kvm8Be2uBeesdcvjx1v+ikYJkJ5KXHDNXIakDNBOjMJS1/f38bIMzMrl27snQOAEDOIJgHj/br11tEm8IVQ0Wb27tdJdrcUuF30WZn9PWiTcXg30Sb47E1RZvdUUVEmzU7j4s2fdbr6y3o11TfNESvsheya7RwnTwg6hQsLNokBJYRbQLKBYs2rtOxoo1P8QJOLwEAAI9CMA9XPPXWWS2cXgAAAAAAAIDbEMzDFU+9dVLrfp85vQQAAADAY/Xu3dteAACeg2AePJorUV9T3W5d6og2zSroK0E8I2VFm0pBP4o2kXFVRZu56+JFmzrl40SbIiGBoo3/PR1EG9eO7aKNV9W6ooorUdTx9hFtzsUXFW2OffffQAAtvP28RZvgUgVFmxJOLwAAgEzo+20OAAAAAAAAIF1k5sHD++g1d/DcAAAAAAAA7kUwDx7dR69Vn08dOzcAAAAAAIC7EcxDhgIDA+XRRx9Nvn3kyJGLMvHMNifVur6iaLPiF339cTS6t/hPos25kvqmIxc+v060aVa1o2hzLi5BtNl58JSoUypEtPFupO//O9fZSNHEK6iQqOOj7y1kgM9p0cY3QF9vwYQ4fT0YY8/q63sKAIBm9MxDhl544QWbdZd0KVmyZHImXsptAAAAQH5lJsF6eXnJQw89dNF9gwYNsvdldVrswYMHpWfPnlK9enXx9vaWYcOG5cKKAQCeTt/HqvDo7D23q9fDuXMDAAAAIlK+fHn56KOP7IfdQUFBdpv5EHzu3LlSoUKFLB8nJiZGSpQoIU8++aSjrWwAALoRzMMVZ+856aFZaxw9PwAAANCoUSPZvn27fPHFF9KrVy+7zVw3gbxKlSrZ20ePHpW6devK0KFDZfTo0XbbL7/8Iq1bt5ZFixbJDTfcIOHh4TJlyhR737vvvuvgdwQA0IxgHjxavMK+LzVrFBdt/t2rq/+TEVPnWtHmxcX6ehvd2/Jm0aZusQ2izZn4UqJNrWL6+tPJlu2ijauQvufJq2I90cR18oBo4xVQULSJTdS3Jt8CfqJNzKGzoo2PX97o/NO3b1+ZNWtWcjDPBOP69Okjy5cvt7dNxp3Z1qVLF2nfvr3UqFFD7r33Xhk8eLAN5AEAkFUE8wAAAAAgTbmruaQUEBBgLxm555577LC43bsvDENbuXKlLb1NCuYZnTp1kgEDBtiAX+PGjaVgwYKOV7oAADwPwTxkqz/enj17km+nnWzriFp3OXt+AAAA5DkmwDZhwoRU28aNGyfjx4/P8DEm865z584SEREhLpfLXi9e/OKKjVdeeUWuuuoq+fTTT2XdunWZBggBAEgPwTxk601NymEXSZNtndT/rdWOnh8AAAB5j/nQevjw4am2ZSXoZkptTdms8frrr6e7j+mtd+DAAUlMTJRdu3bZPnoAAGQHwTx4tDq1Sog25Urq6/+078hp0eb4+XKiTdv6+nowViuwQrTZHd1GtPnn6BnR5lxsgmhze6K+n3E5o69/l7h0PU9eRfW9XiYqfAt5+lyYaBN9LFq0OXlC18+3UURiRZtLldRmpGPHjhIbGyteXl7SoUOHi+4395ly3Lvuusv2zOvfv7/8/fffUrJkyRxaOQAgP9D3Tgw5UgILAAAAwL18fHxk06ZNydfTGjNmjJw6dUqmTp0qwcHB8s0339hsvgULFiTvs379evv1zJkzdgKuue3v7y+1a9d243cCANCMYF4eLYF1N6f651XodqGMAQAAANCgUKFC6W43gzBee+01+eGHH5L3ef/996V+/foyY8YMGThwoN3WsGHD5MeYnnpz586VihUr2pJcAAAMgnnIdgZgUtDwyJEjjvfPm/zTdrefEwAAAEhiBl5kZv78+cnX4+LiUt0XHh5uM/VSMsMzAADIDME8ZDsDMImTmYBJDh0/J9okJup7AxZeJv1PiJ10Lt5PtCkdLOr8cqyZaHMuTl+Ps+pFC4g2G4/p6+Mnsan/iNTAdTJS1PlnlWjiVaepaOPy0Td9s2jATtHmoJ+3aFOirL9o46PweQIAQDOCefDoMlz/Vn1z/JgAAAAAAABaEcyDW+RWGe7I+X/n+DEBAAAAAAC0IpiXj3veJUnZ+87T7Nl+QrRJrFREtPH3u3iamtOuKhEi2uyI1Fe2XaFQoGhTLEhfidbxaH3lozWL6avbdh2OF3UC9ZVrireukj/XYX39Yb299f1eCQotKdqElFH4OqCwHUhCXKLTSwAAwKMQzMvHPe+utPdd2mEYmZXSenLAEAAAAAAAQAuCecixYRiZldJmFDC84l565bpc/mMBAAAAAAA8DME8eHQvvR6TVuToegAAAAAAADQjmAfHevXlRPltQry+HishBfT1EwsK0Pe/eomgw6JNoktfv6VC/vp6nEXFeok2/x47K9pUKBIk2lQvWFDUKaSvf6br4CHRxEvhc+RVtJxokxhUTLSJPauvn6e3r66ekEa8wr6nl8vlcsmDDz4on332mZw8eVL++OMPGTZsmDRo0EBee+01p5cHAMgj9P2Fj3zTq+9K+vUBAAAAGvTu3VsiIyNl/vz5snjxYomIiJDly5dL5cqVpXjx4vLFF1+In5+f3TcuLk6efPJJ+eabb2THjh0SGhoq7dq1kxdffFHKlCmTfMwTJ07IkCFD5OuvvxZvb2/p3r27TJkyRYKDg1MFDidNmiRvvfWW7N69257r4YcfljFjxjjyPAAA3IdgHlRm7GVZ2K05uRwAAADgsm3fvl1Kly4tzZs3T95WtGjR5Ovnzp2T33//XZ566impX7++zd575JFH5NZbb5W1a9cm79erVy85ePCgLF261AYA+/TpIw888IDMnTs3eR/zuCVLlsgrr7widevWtQFAcwEA5H0E85BjQbk9e/bkWMZeVt350g9X9HgAAAAgpzL0Zs+eba97eXlJxYoVZdeuXdK6devkMluTiWcCdClNnz5dmjRpYt9LV6hQQTZt2mQz/H777Tdp3Lix3WfatGnSqVMnG7gzGXxmnxkzZsg///wjNWrUsPtUqlTJge8aAOAEgnnIESYo50TJ7LE9p0SbMu2qijalixYQbYp6/ybaFC68V7RJdF0oy9EkwKe8aBNaIEC0+XtvpGjTMjZK1Nm7X7TxqlNTNIkv30y0OXq+umhz8qS+18uiCvvTJcQkiDaBhQMlLzBlsFWqVLFlryYQ5+Pjk6XHnTp1ygb/ChcubG+vWrXKXk8K5BmmFNeU2/7666/StWtXW35ryngXLFggHTt2tCW3Zp+JEyemygQEAORNBPMAAAAAIIWYmBh7SSkgIMBeMmKy7kJCQmwQLywsLEvnOX/+vIwcOVJ69OghhQoVstsOHTokJUumHsrl6+trg3TmPsP02zN98j799FN57733JCEhwX6wfvvtt8v3339/Gd8xAMCTEMxDrhg1apR9c5L7WrnhHAAAAMhvVScTJkxItW3cuHEyfvz4HDuH6YV355132qw6UzKbHYmJiTbYaAJ51atfyJSdOXOmXH311bJly5bk0lsAQN5EMA+5wgTyJk+enOvnaTtofq6fAwAAAPnvg+nhw4en2pZZVt7lBvJMdp3JpEvKyjNMVt+RI0dS7R8fH2+HWyRl/JkhGyZbLymQZ9SqVct+Nb33COYBQN5GMA8e7arm+np3bd+nr4+fxjV1DtfX42zn6WtEm/Px+nob1Smgb/BM9aI3iDZXlQgWbby2FhR1atYWdRLiRZO4xCDRxtcrdfmhBiWC9K3p1IEzoo0r0SXaxJ6JFW1KXKKkNicCedu2bZMffvhBihUrlur+Zs2aSWRkpKxbt85m2hkm4Gey8a699lp7u0WLFjbAZ6bnmj59xtatW+1XM3gDAJC3EcxDjk60TRqCkfbTRAAAACC/M4E809fu999/t8MrTK+7pD54pieev7+/zbAzQy0GDBggb775pn3M4MGD5e6777aTbA0z7KJRo0bSt29fOyXXBPoGDRokN954Y6psPQBA3kQwDznaWyRJZpNtc7SfXuN7cuY4AAAAQC7bv3+/fPXVV/Z6gwYNUt1nsvRat25tr8+ZM8cG8G644QY7xbZ79+4yderU5H3NNjPRdsiQIXL99ddLwYIF5aabbpJJkya5+TsCADiBYB48up/ezWO/FW2atgwXbY5HRos2UfFlnV7CRYoEXPhkXBPvwETR5rirvmgTn6jveYrRVyEtruhzoo3X4QOijq+ut0cFSunLdo/2Liza+HjFiTbBZfSV25+PdMeAsuzxDfARTxYREZF8fdiwYfaS0vLly5Ovh4eH24EXl2Ky9ObOnZvpPiZL7/PPP7+sNQMAPJuud6vIkyW3aVGCCwAAAAAAcHkI5iHXS27TyqwEN9vluCEds7s0AAAAAAAAj0UwDx5djquxzBYAAAAAACC3EMyDqhLc7Jbjent7iTaxcfoaZZUrpa9nT6LLW7RJcAWIPjGijZ+3vh6MZ+P09cyLTdC3JvHW9/+dnNf3My6Flb0WxJwVbYL8IkWbkzEVRZvEaH19/DT2p3MlXLqHHAAA+A/BPKgqwb2cclwAAAAAAID8gmAePDuDL/Qmdy0HAAAAAADAcQTz4NEZfLeOX+KWtQAAAAAAAGhAMA8ebcSAJqLNL1su3evP3a6pWly0ORcv6vx7UtT5Zctpp5fgEZrX0NerUqPEHQdEm4TD+vrBuc7q6nN2YsQA0aZ01ELRpkDIftFmZ4y+16Zyd9YUbeJ2R0le1rp1a2nQoIG89tprGe6za9cuqVSpkvzxxx92XwAAMqOwEzYAAAAAeIbevXtLly5dUm377LPPbMuYSZMmZekY5cuXl4MHD8pVV11lb//555/So0cPuz0oKEhq1aolU6ZMyZX1AwA8D5l5SGXUqFFy/vx58RRdR4xzegkAAABAsnfeeUcGDRokb775pvTp00e+/vrrTPePjY0Vf39/CQsLS962bt06KVmypHzwwQc2oPfLL7/IAw88ID4+PjJ48GA3fBcAAM0I5iEVE8ibPHmyeIof90c6vQQAAADAmjhxoowbN04++ugj6dq1a7r7hIeHS79+/WTbtm0yf/586datm4wfPz5VmW3fvn1TPaZy5cqyatUq+eKLLwjmAQAI5sGz/bHzhGgT5K/vfyuNz1PDkjGizR87A0SbQdfp67f07q/6nqfrAr4SbaZubinatPDT113Dr1ox0eb8L/tEk9IF/hRtzvg1cnoJHsEvyE+0Of2rvt6Zhbvp6+N3OUaOHClvvPGGLFiwQG644YZM933llVdk7NixNvCXVadOnZKiRYvmwEoBAJ5OX9QBbmf6eTz66KP2+pEj+oY3AAAAAO4UExNjLykFBATYS3oWLVokX375pSxbtkzatm17yeObfR577LFUAzAyY8psP/74Y1m4UN/wGQCA+xHMg7zwwgvJ15OCep7SOy+8+xCnlwAAAIA8+P54woQJqbaZLDpTDpueevXqybFjx+w+TZo0keDg4EyP37hx4yyv5Z9//pHbbrvNHrt9+/ZZfhwAIO8imAeP7p035ecdTi8BAAAAeYz5YHv48OGptmWUlWeULVvWTrBt06aNdOzY0WbqhYSEZLh/wYIFs7SOjRs32pJdM/ziySefzMZ3AADIywjmwaNdV6OkaLM3Klq02bDzpGgzbYWPaNOxvr6fp9hEff923Rrq69dzwreNaBNeOlC08Q4KF21cp06JNt4h/qLJiZjKos3uqCKiTYkCXqLNiV1nRJu46DjRJnDFbtGmwK0Zl9RmpGLFirJixYrkgN7ixYszDehdyoYNG2w57v333y/PPffcZR8HAJD3EMxDuv3zknrnaS+3vXf0M04vAQAAALDKly8vy5cvtwG9Dh062IDe5TCltSaQZ45hMgQPHTpkt/v4+EiJEiVyeNUAAE9DMA/p9s9L6p2nvdz296P6PvEGAABA/lWuXLlUAb20gzSywpTsHj16VD744AN7SZn9d6lhGQCAvI9gHjzahz/p65lXq5K+EsQHWmatL4s7/bRPVxmbUSFEX7nf9sjCos3+0+dEm5rFiok2W/fsF20S9mwXbWI3HBNtgobcIpoE7PpQtClcvaVos+5IXdGmXLVCok2BEgVEm8SYBKeXcEUiIiLS7aG3devWdPdPLxgXHh4uLpcr+bYZtJHRsA0AALydXgA8o9wWAAAAAAAAziMzD1kqt02Phn56vtf1cfT8AAAAAAAA7kQwD5dNQz+9x7/4y9HzAwAAAAAAuBPBPGSp3DY9GkpwYxX2WImOiRdtftitrz9OvVLBos2OU/r+7fx9vESbJqUDRBsfb319/MLLhIo2rq36fsa9g/1Em7h5P4gmMQP1TW4/H6Ovn2eIv49oE1hE3+ulK/G/vmxaJMYlOr0EAAA8CsE8ZKncNj2ZleACAAAAAAAg5xHMQ65k7blNk/ucPT8AAAAAAIAbEcxDrmTtucsjH/7h9BIAAAAAAADchmAePFqJYvp6wZ2MihFt2tcqJdpU8/tWtDkR1EC0OX5e379dkO8x0eaNFfr+v3u4lb5eWT4Vi4o2K57S96FMm8OzRJO42CDRJlJhz7wdkfp6Z/psOC7aVB11rWgTu+mE00tQZ/z48TJ//nxZv36900sBACjk7fQCAAAAAMBT9e7dW7y8vC66dOzYMcfOsWHDBunevbuEh4fbY7/22ms5dmwAgOchMy+PGjVqlJw/f17yumI3PeT0EgAAAJDPmcDdrFmps3oDAnIuQ/vcuXNSuXJlueOOO5zvWQ0AcBzBvDzKBPImT54sed2zS7Y4vQQAAADkcyZwFxYWlu59JpPuzTfflK+//lq+//57qVixorz77rtSokQJ6d+/v/z2229Sv359ef/996VKlSrpHuOaa66xF+OJJ57I1e8FAKAfwbw8Ol32yJEjkh+0rK2vn5hG+0/r6ycWWqyZaLPxqL4eZ7WLRYk2hf33ijbRMfp6wRX56wPRxtWhh2jT9gN9HT+8Tu0WTf46U1v00fd7paCfj2hTsn5J0WblI6tEm8BAUeeaMTl7vGeeeUZeffVVexk5cqT07NnTZtqZapoKFSpI3759ZfDgwbJo0aKcPTEAIE8imJdHp8uSfg8AAABcnpiYGHtJm32XUensggULJDg4ONW20aNH24vRp08fufPOO+11E8xr1qyZPPXUU9KhQwe77ZFHHrH7AACQFQTz8kGWXl5222PjnF4CAAAA8uCH5BMmTEi1bdy4cXbKbHratGkjM2bMSLWtaNH/ssbr1auXfL1UqQuVJXXr1k21zbTJiYqKkkKFCuXY9wEAyJsI5uWDLL28bPm+SKeXAAAAgDzGlL8OHz48ywMtChYsKFWrVs3wfj8/v1Q99DLalpiYeEXrBgDkDwTz4NG+//OAaPP7r/r6iWn05YQQ0aZ12BnR5rSrvGjjEn19qZ6++ZRocya+p2hzop2+nnmlWun7Gf9z4HeiSes/9fXzPB1QS7Q5dC5ctCn85oUSS01arF4q2rhadBVtfLwyLqkFAMBpBPMc/sTPpNPj8oW0G+D0EgAAAJDPmf56hw4dSrXN19dXihcvniPHj42NlY0bNyZf379/v6xfv9726cssIxAAkDcRzHOQCeRNnjzZ6WV4tLELL7ypAQAAAJyyePFiKV26dKptNWrUkM2bN+fI8Q8cOCANGzZMvv3KK6/YS6tWrWT58uU5cg4AgOcgmAeP9tiN/qLOjVVEm883BIo2XssmijbvFvs/0abX90NFmzND9P3bafwZ16h1nL5eTAEP3CbaVN/1vmjy7rYmTi/BQxwXbaqW3iPaeFWtIdqc6jdKtCny7oos7xsREWEvGXG5XKluh4eHX7StdevWqbaZQRsph22k9xgAQP7l7fQCAAAAAAAAAGQNmXm5KDAwUB599NEM7z9y5Eimj6en3qWNf2mQ00sAAAAAAABwG4J5ueiFF17I9P7MAn0GPfUu7VTsv04vAQAAAAAAwG0I5sGjxSSEiDbjP9HXH6dpw7KiTcwP+gKxfZ/8XbSJW6OvL+S2yFKiTd+aP4g2R+Ra0Sb0Zn09Pc9P+0K0KTzwetFk7V8HnV6CR9D4u0689XW0iY34WrQp/OL9Ti8BAACPQjDPg8twAQAAAAAAkL8QzPPgMtwk+bm33qjnn3B6CQAAAAAAAG5DMC8PyM+99Y5EH3Z6CQAAAAAAAG5DMM+Dy3CT5Ody3K0nA0WbapWLijYffvinaHPfmP6iTeJSfX2Eji/dJdo07rVUtDnoai3aFA/U1xfSp2Zx0Sb6u52ijStGV7Z7zxurO70Ej/DctF9Em3uarRJtvEeMEnU2fSfqlMyZw0RERMiwYcMkMjIyZw4IAADBPM8uw81uOS4AAACAnNW7d2+ZPXu2ve7n5ycVKlSQ++67T0aPHp3tapuHHnpI1q1bJ5s2bZKbb75Z5s+fn0urBgB4MoJ5+SiDLy/q/vh4p5cAAACAfK5jx44ya9YsiYmJkW+++UYGDRpkA3ulS5fO8jESEhIkKChIhg4dKp9//nmurhcA4NkI5uWjDL686OcDp5xeAgAAAPK5gIAACQsLs9cHDhwo8+bNk6+++koefPDBVPv9+eeftux27dq14uXlJdWqVZP//e9/0rhxYylYsKDMmDHD7rdy5UpKcwEAGSKYB4/WNGyTaNOy2AnRpnvDNqLNhthE0ebLAleJNjVeKCzadNv7rmhTukycaJP44wrRpueWXqJNszsriDZnz8eLJnd2byHaFCofItp8FlZQtPG69SnRxvfENlGnUKjkNSbD7vjx4xdt79WrlzRs2NAG7Xx8fGT9+vU2gw8AgOwgmAcAAAAAKZhyWXNJm31nLplxuVyybNky+fbbb2XIkCEX3b9nzx55/PHHpWbNmva2ycwDACC7COY5YNSoUbbBLa7cy5PucnoJAAAAyINtbCZMmJBq27hx42T8+PT7NS9YsECCg4MlLi5OEhMTpWfPnnbfTz/9NNV+w4cPl/79+8v7778v7dq1kzvuuEOqVKmSq98LACDvIZjnABPImzx5stPLyBNcRyNEG5foU/7cW6LN1tAHRJuGVYuJNlv36esLmbBrg2jjffSYqFOggGjz+lB9fzAeiy4u2kz8aL1oUqSqvnL70A6VRJujn28Vbc7HVBZtPthYXrT565/Dos0boxrZwFtKmWXltWnTxpbO+vv7S5kyZcTXN/0/s0yAzwT6Fi5cKIsWLbIBwo8++ki6du2a498DACDvIpgHAAAAANksqU3JDK+oWrVqlvatXr26vTz66KPSo0cPOwWXYB4AIDsI5uUzea3E99XR9Z1eAgAAAHBJ0dHRtl/e7bffLpUqVZJ9+/bJb7/9Jt27d0/eZ+PGjRIbGysnTpyQ06dP2wEZRoMGDRxcOQBAG4J5+UxeK/HVWGYLAAAApGWm15oJt/fdd58cPnxYihcvLt26dUvVm69Tp06ye/fu5Ntm8m3SYA0AAJIQzINHc61aKdpsavmaaFO7QOrmyxpUKfSnaBOfWFe0+e3MUdHGp46+vmsaRX+4WrQpGjJTtDlU8XHRpmJ4EdEkaEewaHPq252iTdjYm0SbZ3446/QSPIK2/+eyKyIi4w+Xe/fubS+G6af34YcfZnqsXbt25fj6AAB5D8G8fFY2e+TIkVxdDwAAAAAAAHIPwbx8VjZrGu3mpR56k9o4vQIAAAAAAAD3IZiXz3l6D73ErwY4vQQAAAAAAAC3IZiXzwQGBqbKzvP4sltvb9EmOj5RtDlf9ELzZFX0PU1S2/WFaPPeeX3T6068+p1oU3REe9EmqFdz0eZ8ndtEm/cW7BBt7mgeLpr4/ltI1Dl4RrSJW7lRtOl0Z1nR5tNf6MkGAICnI5iXz7zwwguZlt0CAAAAAABAL4J5+VzaTD1PM+kGp1cAAAAAAADgPgTz8rm0mXqeJnHBg04vAQAAAAAAwG0I5sGjebVsLdo02visaOPlo/B/9bCSok1suRaizYs1Foo20SULiDau3XtFncKhok27O/8Sbb76UN/r+Pe7okWTRpWLijZFb2gk6hQpLtpo7E9XrmSwaLNy5W7J67y8vGTevHnSpUsX1gEAuGL6pgcAAAAAgAc5evSoDBw4UCpUqCABAQESFhYmHTp0kJUrV2b7WD///LO0aNFCihUrJkFBQVKzZk2ZPHlyqn3Gjx9vA3MpL2a/7Lr11lvtmk3rndKlS8u9994rBw4cyPZxAADupTBdB7ll1KhRcv78eclLXh3XxOklAAAAIJ/r3r27xMbGyuzZs6Vy5cpy+PBhWbZsmRw/fjzbxypYsKAMHjxY6tWrZ6+b4N6DDz5orz/wwAPJ+9WpU0e+++6/CfO+vtn/065NmzYyevRoG8jbv3+/jBgxQm6//Xb55Zdfsn0sAID7EMzLR0wgL+2nep7OFfmh00sAAABAPhYZGSk//fSTLF++XFq1amW3VaxYUZo0yfhD53379snjjz8u3377rcTExEitWrXk9ddfl2uvvVYaNmxoL0nCw8Pliy++sOdIGcwzwTuTAZiRbdu2Sb9+/WTNmjU2wDhlypSL9kk5CM+s+YknnrAluHFxceLn53dZzwcAIPcRzINnO31C1Dl0RLRxFQgSbWKuvlO0Wbm/vGjTuIq+56lQq+2iTfQP+vpS+ZYLEW0+n/OEaPPHYX0dP2LiYkQVb33PkQQGiDp79fVdKx56IbCjyaxHF4k2w2d2FW1MgM1cUjLls+aSVnBwsL3Mnz9fmjZtmu4+KZ05c8YG/cqWLStfffWVDcj9/vvvkpiYmO7+f/zxh82Ue/bZZy8K1pUpU8aWyDZr1swOtjMls4Y5Vrdu3aRUqVLy66+/yqlTp2TYsGGZruvEiRMyZ84cad68OYE8AFCOYF4+KqU9cuRIniu5fXV4VaeXAAAAgDzGBMYmTJiQatu4ceNsr7q0TIZcRESEDBgwQN58801p1KiRDdbdfffdtlQ2rblz59oee7/99psULXphwE3Vqhe/py1XrpzdLz4+3p63f//+yfeZDD5zzho1asjBgwftWq+77jr5559/JCQkxJbfbt682Wb+mYCf8fzzz8tNN9100XlGjhwp06dPl3Pnztlg5IIFCy7zWQMAuAvBvHxUSpsyjT6vlNy69r7u9BIAAACQx5gPvocPH55qW2YZd6ZnXufOnW0p7OrVq2XRokUyceJEeeedd6R3796p9l2/fr0to00K5GXEHMtk8ZnjmfJXE/Dr0aOHvS9lUM4EDE1wz5TJfvLJJ7a0dtOmTVK+fPnkQJ5hsvfSY8p9zWN2795tg4L33XefDeiZoRoAAJ0I5sGzKSw98r6tp6gTGy3axLn0lf4WL+Av2hw//9+bcC2CtmW/mXduKzC0k2hzbuo3os3Tn+x0egke4abrKokmxyPWizZF2keKNrGb9bXe6DX+MdHm9NjWok1oAX0lnRmV1GbGlLveeOON9vLUU0/ZTDqTzZc2mGcm1GZFpUoXXovq1q1rB2qY7LykYF5ahQsXlurVq8u///4r2VW8eHF7MY83vftMENAEEDMK/gEAnKcvEgIAAAAAHq527dpy9uzZi7abTDqTnWd61GWV6YGXtodfSiaDb/v27XYqrWGCcnv37rUluElMgC4r5zEyOxcAwHlk5uUj5tPCpFLblP3zUvK0Xnqvjqju9BIAAACQjx0/flzuuOMO6du3rw3UmZ51a9eutWW2t91220X7m+w607/OTI01vflMAM4MuTAlsSYbzky1NYMsatasaff/8ccf5ZVXXpGhQ4cmH2PEiBFyyy232NLaAwcO2AxAHx+f5My9du3a2Uy7+++/X15++WWJioqSMWPGpFqHGYxh+va1bNlSihQpYoOBJqOwSpUqZOUBgHIE8/IR82Yhvf55KXlaLz3X/hlOLwEAAAD5mJlka3rWmffQJiAWFxdnS1XNQIzRo0dftL+/v78sWbJEHnvsMenUqZMdcGGy+EwQLyk7znzAvnPnTjtcwwTXXnrpJXnwwQeTj7Fv3z4buDOBxBIlStiAnMm8M9cNb29vmTdvnu2F16RJEwkPD5epU6dKx44dk49RoEAB+eKLL2wg0GQQmqCiuf/JJ5/MdolxXvT+++/bgSbm32HVqlU2cPraa6/Z8uf0grQA4E4E8+DRXHv3iDZeRS6UN2gS7a+r/5PhJ/r6+B07FyvaFAvS13w6fneUaOO3/4Boc/6Yvp/xsFIhok2PZhVFm6qha0STxFv1TW73Kl5ItPE5qa+y4PP1J0Wb3Tv19RbUuKZbBrfM8r4m8GU+NE/5wXlaLpcr1W0TGPrss8/S3XfIkCH2kpmPPvrokusymXlmiEZG6zC9+L7//vtLHic/mjFjhowdO1aGDRsmzz33nCQkJCT3JjQBPYJ5AJxGMM8DZFb6mlG5LAAAAAAg+6ZNmyZvv/22LYV+8cUXk7c3btzYljgDgNMI5nmAzEpfMyqXzU7/vKwGBzX205t0V6DTSwAAAACQh5jS2oYNG6abhZneUBMAcDeCeflURmUAmQUHNfbTS1w9yuklAAAAAMhDTF88M3HYlEOntHjxYjspGACcRjAPnu18jGjjOrpbtAnYu1K0iW3WU7RpV2ataHPe60Ija03877z4k2rHxV/oZaNJcF19/3YFg/T92j+qsFflyfONRJPGxT8VbVynzog2XgH6fr5/+mmXaHM+Ok60CSsX6vQSgFSGDx8ugwYNsskMps/gmjVr5MMPP7QJEe+8847TywMAgnnIWvmtQX8+AAAAAHld//79JSgoyE72PXfunPTs2VPKlCkjU6ZMkbvvvtvp5QEAwTykltkUrsxKcJ3qpzfpNn2TPgEAAAB4tl69etmLCeadOXNGSpYs6fSSACAZwTzkCKf66SUuH+72cwIAAADIHwoUKGAvAKAJwTx4dAmu9/WdRRvXzvWijVflKqKNn3e0aBMVH+70Ei7i76VvYtr5Bf+INgFXlxJtvAr4iTZbth0XbcqUCBZtGpfV1b/r/Hf/ijaBTUuLNr7lQkSb0tGFRZvos/r6VJajZ16y8ePHy/z58+3whSvh5eUl8+bNky5duuTY2vKTw4cPy4gRI2TZsmX27xzTNy+lhAR9vXoB5C8E85DrJbgAAABAXtW7d2+ZPXu2PPjgg/Lmm2+mus8MUXjjjTfk/vvvl4iIiBw5X3h4uOzeffHAtYcfflhef/31HDlHfmf+Tffs2SNPPfWUlC5d2gZHAUATgnnI9ay93DR50s1uPycAAACQUvny5eWjjz6ybWfM4ISkNjRz586VChUq5Oi5fvvtt1SZYf/884/ceOONcscdd+ToefKzn3/+WX766Sdp0KCB00sBgHQRzEOuZ+3lqsRlzpwXAAAA+P8aNWok27dvly+++MIOTTDMdRPIq1Spkr393nvv2Q+/Dxw4IAEBAcmPNaWwISEh8v777ydvM9dNVtjJkyflpptukrffftvuY5QoUSLVuV988UWpUqWKtGrVKtX2gwcP2scuX77cZpdNnDhRbr/99lx9HvJScDZtaS0AaEIwDx7tu30NRR0/fWtq85O+Mugfmt3p9BI8RHHRps6WE6JNyYb6Jsz5lLiQmaHJnc2qOb0Ej7D7lK6ensHbToo63vpKzgIUvg7c0qKi00vAZYqJibGXlEwALmUQLq2+ffvKrFmzkoN57777rvTp08cG0wyTOTd06FD56quvkrPoTD+2hQsXypIlS5KPY4KCpm/eggULbDDvzjvvtAG755577qJzxsbGygcffCDDhw+/qBTUBAPN46ZMmWKDg3fffbf8/fffUqtWrSt8dvK+1157TZ544gn53//+Z8uaAUAbgnmQUaNG2TIAT9T50aecXgIAAADyYNXJhAkTUm0bN26cHVCRkXvuuce+r07qZ7dy5UpbepsUzDPltz179rQBv6RgngnEmey91q1bJx8nMTHR9tdLysS799577SCG9IJ5JugXGRlpe7ylZc7Rv39/e/2ZZ56RpUuXyrRp02wPP2TurrvuknPnztmMRzPJ1s8v9VCrEyf0fbAJIH8hmAcbyDP9PTzRd3v4RQoAAICcZYJyJtstpcyy8pLKXzt37mwDcaZE01wvXjx1hv2AAQPkmmuukf3790vZsmXtviYQlzKrzmSCJQXyDFMiazL40jNz5kxbSlumTJmL7mvWrNlFt690Sm5+YTLzAEAzgnn5UNpMvIzeHHiC4gX8RZv6Mf/1O9HCVaqYaLPj8BnR5vZ6iaLNntP6ymwT4/U9T14VL/4jxmmxyzaKNu3u/FO0eWttFdGmfImCokmRxmGijf815UWbxH36PuCLS9TXc2vSaytFm8eGtRBtLlVSm1mp7eDBg+319CbLNmzYUOrXr2/757Vv3142bNhgy2xTSpsFZgJ9JlsvLZMB+N1339nefMhZZvowAGhGMC8fSpuJ58QUWgAAACCv6dixo+1jZwJwHTp0SHcfU/pqMr9Mdl67du3ssIXLYcp1S5YsaTMA07N69Wq57777Ut02wURkjeldaJ5j89X0HTTP9aJFi2xZdJ06dZxeHoB8jmAeJDAw0GMDevePecbpJQAAAACWj4+PbNq0Kfl6ekzfvBEjRtgJtSZD73KYTD0TaDIZZL6+6f9J9+mnn0rjxo2lZcuWMmfOHFmzZo0ty8WlrVixwpYvt2jRQn788Ufbr9AE8/7880/7HH722WdOLxFAPkcwD7bBr6daf0xfqSYAAADyr0KFCmV6f2hoqHTv3t2W13bp0uWyzmHKa/fs2WPLejNiBniYARwPP/yw7bv34YcfSu3atS/rfPmNmWT77LPP2r6JKfsXtm3bVqZPn+7o2gDAIJgHj9YgcImoE1hKtPFqrG9NoXuz34cmty3dIQqdFW0qligg2iT8uVO02fvVv6JNkceqizahwfreilwTpmvC+4b3tog2jW7X1+Ms8vpRok3UznjRZsCD14o2UdH6nqfsMEMsMmMmzqZlSmx79ep1UV8+My037cTcYcOG2UtKpt+eGbKRkaT7TCAP2ff333/L3LlzL9pusvOOHTvmyJoAICV976BxRSWynjzMAgAAAMjLTp48KcuXL7eXN954w+nlIAOFCxeWgwcPSqVKlVJt/+OPP+wUYgBwGsG8PFYim17vu4ym16bd7okmP3Od00sAAAAAssQMoDABvZdeeklq1Kjh9HKQgbvvvltGjhxp+w4mTRNeuXKl7XWYcqgIADiFYF4+nl6bdrtHOvOF0ysAAAAAsmTXrl1OLwFZ8Pzzz8ugQYPspOGEhATbazA+Pt6WRj/55JNOLw8ACObBs50OqOX0EnCZTp7WlxX6UP0Nos2bf9YRbQIr/NcIWot/Xlsv2lTvWF60ef8vfX2prqtaWLQ5fE5UqXJdmGjjOn5ctCl6fKpoc9KXfmVZcV3V4k4vAUjF39/fThseO3as7Z935swZm1VZrVo1p5cGABbBvCzKyZJU+toBAAAAgE5mim1aq1evtiW3pmd51apV5bbbbpOiRYs6sj4AIJiXRTlZkppeXzsneXLvvKcnPuD0EgAAAADkIWbQxe+//25LbJN6G27dulV8fHykZs2adnjJY489Jj///LMtwQUAdyOYB4/unXc6bpPTSwAAAACQhyRl3c2aNUsKFSpkt506dUr69+8vLVu2lAEDBkjPnj1tksa3337r9HIB5EME8/IhkxpufvHkhXLfPacriDYVQvaINi7xEW0ah+srS1hyqKloU7Z4omhzZvMJ0aZGJ32vBcc3HBNtel+t79/O1+ugaOMbo+v3Y2KnSqKNV4i+3plxK/4SbQ5VUdaAUUQO7I8SbY5F6qsQqdM+ZyfNmvLMefPmSZcuXXL0uMgdL7/8sixdujQ5kGeEhobK+PHjpX379vLII4/YfnrmOgA4wduRs8JRL7zwgs3EK1mypNNLAQAAADze0aNHZeDAgVKhQgUJCAiQsLAw6dChg6xcufKyjhcTEyNjxoyRihUr2uOFh4fLu+++m61jbN++Xbp27SolSpSwQak777xTDh8+fFnryW9MFl56iQ/m3zkq6kJAvHDhwhIbG+vA6gCAzDx4eE++/k89myvHBQAAALKqe/fuNrAze/ZsqVy5sg2aLVu2TI5f5uTnpMDbzJkz7bCFgwcPSmJi1rP1z549a7PG6tevL99//73d9tRTT8ktt9xiBzl4e5PTcaky2759+8qkSZPkmmuusdt+++03GTFiRHJ25Zo1a6R69eoOrxRAfkUwD7keoDOfas2ZMyd3FuL6QbQ5GlNTtCnh9Ydo8+sOf9EmrFgB0ea4wtKj2LNxok1I0SDRxttPX3l7ZGx50cbHS9/PU/GT651egnqus2dFG99y+kp/Qwr4iTYhhQNFG42/f7MjMjJSfvrpJ1m+fLm0atXKbjMZdU2aNMnwMfv27ZPHH3/c9lszWXi1atWS119/Xa699lpZvHixrFixQnbs2JE8LdVk5qVkAkujR4+2gxri4uKkQYMGtvKmUaNG9n6TEbhr1y57f1KpqAk0FilSxAb32rVrl4vPiOf73//+Z9sS3X333RIfH2+3+fr6yv3335/ca9wMwnjnnXccXimA/IpgHnJ9aIa26b0AAABAZkyAzVxSMuWu5pJWcHCwvcyfP1+aNm2a7j4pnTlzxgb9ypYtK1999ZUtyTWTU5My78y2xo0by8SJE+X999+XggULyq233irPPPOMBAVd+PDq9OnTNrA0bdo0cblcNoOsU6dOsm3bNgkJCbFrN336Uq7F9M02GXlmAivBvMyZf8+3337b/r1jgqqGybg025OYACoAOIVgHtw2cCM3TH711lw5LgAAAPJ3j+kJEyak2jZu3Dg7ACEtk7EVERFhJ5y++eabNjvOBOtMVle9evUu2n/u3Lm295rJrkvKvDOltElM8MgE3Mx7aDM049ixY/Lwww/bkl0zXdVo27ZtqmO+9dZbtoebyei7+eabbVDRBAFHjhwpzz//vA34PfHEE5KQkGBLdpE1JniX3r8hADiNYF4+5q6ptubNUK5RWGYLAAAAz28pM3z48FTbMsu4Mz3zOnfubMttTU+6RYsW2cw6U4bZu3fvVPuuX79eGjZsmBzIS8tk6JmsOtOmxkxQNV599VW5/fbb5Y033rDZeaaf3pNPPmlLe817eROkO3funOzZs8fub4ZefPrpp3Yox9SpU21GXo8ePWygkX55AOD5COblY0lBtsyy5jLqjZfbAcCsOnS+rmgTFvCnqJOg703bqTP6pn+dPqdvTRqFVrzQe0eVhKw3BXeXog30TQw/EPdfeZAWIX6nRJvDRW4XTUKWfyba+FbU9+/m5afvd921TUqINofL6HsN1/ieIKOS2kt9UH7jjTfaixk20b9/f5vNlzaYl1Qqm5HSpUvbEtykQJ5heuqZ7DrTa69atWq2xNZk6k2ZMiV54m2zZs1STVc1AzDMRFuT2WeyB03mninpNeWiAADPRjAPl9Ubjz54AAAAQMZq165t++ilZco2TcbeiRMn0s3Oa9Gihc2qM731knq0bd261WbUlStXLnnAhcnSM33yjL1799qgXXqKFy9uv5rBF+YDedN/DwDg2QjmQdXk2+wa+fyYXDkuAAAAkBUmQ+6OO+6Qvn372kCdGUCxdu1aW2Z72223XbS/KXc1fey6dOliK2VMJp6ZOlumTBmbXdezZ0877KJPnz62b58J0pnJt+b4SVl9JjvPDMcwgzKioqLs/Wkz/kx/PZPRZ0puV61aJY888oj9QL5GjRpue24AALmDYB5UTb7NrkPR6X8CCQAAALiDyZ679tpr7ftdU9YaFxcn5cuXtwMxRo8efdH+/v7+smTJEnnsscdsZl18fLzN4nv99deTj7d06VIZMmSIDdYVK1ZM7rzzTnn22WeTjzFz5kx54IEHbA88cy4THBwxYkSq82zZssV+qG4yAMPDw2XMmDFU1wBAHkEwD7k+oTY3++slJPqLOnExos0xV33RpmHVQNHmn50nRJurKqXfHNtJviULiDaJ5+JFG5/Q7PVacodQ/+OiTfGAraLN/nNXiyZeBfxEm8RIfb/rNPbMW7VFR4/hlEoVKyja/L1N4YezDS+Us2aF6VdnMuwyG/pm+t2lZPrcffZZxv0wa9asaQN6GS6vYUM7DTclMyAjpRdffNFeAAB5D8E85PqEWj4BBAAAAAAAyBkE85Bpll1OZNVlJ4svu0Y8OyFXjgsAAAAAAKARwTxkmmWXE0G47GTxZdf+s1G5dmwAAAAAAABtCObBo5U5NVe0cZ09LdqErnhftDne9FXRplmNkqLNjI//FG3a19DXxy/hwBnRxpWYuj+SBiv26uu9ViSwgWhzVfFo0cS3lL4+lb7Vios2icfPijalSwSLNkEBPqJNkUL6+ugCAKCZvk7BAAAAAAAAANJFZh6yzYy4P3/+vGjw6uM1nF4CAAAAAACA2xDMy2PSGzaRE0MsUjKBvMmTJ4sGrgNvOr0EAAAAAAAAtyGYl8ekN2witybJqhAXI+oUDBFtfCqXEm0OK+xtVK90IdHm5g7VRZv4L0+JNgnHzok2vuX0vRbUK6lvTdX8l4o25/3LiybeRYNEncL6Xi996tQUbX74dpdoc/yIvh6j17WqJHmdl5eXzJs3T7p06eL0UgAAeQA985ClTL+Ul5zO9AMAAAA82dGjR2XgwIFSoUIFCQgIkLCwMOnQoYOsXLky28c6ePCg9OzZU6pXry7e3t4ybNiwK1qby+WSm266yQYU58+ff0XHAgDoQGYecjzTz5099V4dFu6W8wAAAAAZ6d69u8TGxsrs2bOlcuXKcvjwYVm2bJkcP34828eKiYmREiVKyJNPPpkjrW1ee+01G8gDAOQdBPOQI335UjKZe3PmzHHPYhKXiTpnI0WdslGiTcFTfqJNVEy8aLN9n76S1tg9+n6e/CuFijZRaw+JNgUeiBNtTvtWE21ciT6iifeJaNHGu5JLtHFt2y7alK/YUbS5plEZ0Wb9hsOiTqus7xoZGSk//fSTLF++XFq1uvDAihUrSpMmTTJ8zL59++Txxx+Xb7/91gbvatWqJa+//rpce+21Eh4eLlOmTLH7vfvuu+k+/rfffpPRo0fLH3/8IXFxcdKgQQMb+GvUqFGq/davXy+TJk2StWvXSunSpbP+TQEAVCOYhxzJ1ss3PfoAAACQ55kAm7mkZMpnzSWt4OBgezElrE2bNk13n5TOnDljg35ly5aVr776ypbk/v7775KYmJjl9Z0+fVruv/9+mTZtmi2jNQG7Tp06ybZt2yQk5EJ/1HPnztlyXRMkNOcAAOQdBPPg9sy9nDR50s1uOQ8AAADy14fXEyZMSLVt3LhxMn78+Iv29fX1lYiICBkwYIC8+eabNjvOBOvuvvtuqVev3kX7z5071/bYM9l1RYsWtduqVq2arfW1bds21e233npLChcuLCtWrJCbb77w/ti8H2/evLncdttt2To2AEA/gnlwe+Zeni+zBQAAgEczPaCHDx+ealtmGXemZ17nzp1tue3q1atl0aJFMnHiRHnnnXekd+/eF5W+NmzYMDmQdzlMTz7TU8+U9poWNwkJCTYTb8+ePfZ+k/H3/fff2zJcAEDeQzAPHs11aJto49q6VbTxqllbtLm1vL4eZ4fO6uqTZTStWUK0Ce5/nagTr6/fYcHTsaJNgP9+p5fgEYJ9lfXvalhRtIlbra8/nUYxNRJEm782HRFt1ry5VrQJ6NPkkuWy6VWn3Hjjjfby1FNPSf/+/W02X9pgXlBQ0BWvz5TYmuEapree6c9n1tqsWTM7hMMwgbzt27fbbL20QcfrrrvOBgEBAJ7L2+kFAAAAAEBeU7t2bTl79uxF203prcnOO3HixGUfe+XKlTJ06FDbJ69OnTo2mHfs2LHk+5944gn566+/7HmSLoYZkjFr1qzLPi8AQAcy83BF5Qfnz593dA2vPl7D0fMDAAAgfzMZcnfccYf07dvXBurMAAozPdaU2abXr65Hjx7y/PPPS5cuXWx7GjNl1pTDlilTxmbXGUnBNzMsw/TXM7f9/f1tgNCoVq2avP/++9K4cWOJioqyk3FTZvyZgRfpDb2oUKGCVKpUKRefDQCAOxDMw2UzgTzz6Z6TXAfedPT8AAAAyN/MJNtrr73Wvi82pa1xcXFSvnx5OxBj9OjRF+1vgnJLliyRxx57zGbWxcfH2yCdmTqbxPTUS7Ju3To7NMOU0+7atctumzlzpjzwwAN22IY5lwkOjhgxwk3fMQDAaQTzcNmZd6bZrtO8SlYWdU5HiTqnLr+MI9cUEnVCA3hJzIqjLy0RbYq0qeD0EjxCbGKwaBPgo+81c/Whi6dPOqlZbX39YX2PKvy9olCAv75erKdORIs2hRuXEU9mSlxNhl1mQ+BcLleq2yYw99lnn2V5/7RMsM9Mw03p9ttvz/QxlzomAMBz8JcrLjvzzoy7BwAAAAAAgPsQzMNlMxO7nA7oTX75JkfPDwAAAAAA4E4E83DZMislcJt4feV+AAAAAAAAuYVgXj6TnQm0GnriXUqiz39Tu7SIqdJBtNHYk0r0teyRQv4nRZvCAfGiTZH+V4s2sb/tFW0CrtM3LfBEfKhoUzT6Z9GmWJCufqyH+78l2oTWKiba+NUoKtr8+vuFYQWahBQOFG0q1S3l9BIAAPAoBPPymexMoHW6hBYAAAAAAACpEcxDjmbzudukV7s5vQQAAAAAAAC3IZiHHM3mc7dE109OLwEAAAAAAMBtCObBo/1+tIFoc/X+SaLNidr9RJvTsQGiTY2Q30SbtcebijbB6/eLNv71w0Sb89/9K9oUbbtTtPHy0tf7tLjPMdGk8B01RBuvYvr6L8b+ou/nO6ign2jTsX010ebtsctEnYeaOb0CAAAyRDAPGQoMDLR98zIahKGh/LbXqKcdPT8AAADyt969e8vs2bPtdT8/P6lQoYLcd999Mnr0aPH1zb0/t8z78IceekjWrVsnmzZtkptvvlnmz5+fa+cDAOhBMA8ZeuGFFzIdhKGh/HbtkdOOnh8AAADo2LGjzJo1S2JiYuSbb76RQYMG2cCe+fA7pdjYWPH398+RcyYkJEhQUJAMHTpUPv/88xw5JgDAMxDMQ5Yz9NLKKGPPnRoHfCPauCpUEW2K+W4WbXwLVBJtTieWE20qhx4XbU6tPSTaFC1VULTxrVBItDkRV1bUyZm/aXPU3sgiokmh07GijW8Dfa+X/o1iRJv90/eKNqF31RNt/IsGiqcLCAiQsLALLR8GDhwo8+bNk6+++kq2bNkikZGRcs0118jrr79u99u5c6e8//77MmXKFHt/wYIFpW3btvLaa69JyZIlk49pHv/YY4/J3r17pVmzZjYD0FxOnjwphQsXto+bMWOG3XflypX2PACA/IFgXj4Lxl1OAC4pQy+tjDL2AAAAAE9mMuzMJSUTiDOXrDAZc8ePX/hAbtmyZVKoUCFZunRp8v1xcXHyzDPPSI0aNez78+HDh9tAncnqM0zA7/bbb5dHHnlE+vfvL3/88YeMGDEiR79HAIDnIpiXD6QMxuVkAC6jjD13mjxe33AAAAAAeP775wkTJqTaNm7cOBk/fnymj3O5XDZ49+2338qQIUPk6NGjNoPunXfeSVVe27dv3+TrlStXlqlTp9rsvTNnzkhwcLD873//s4G+l19+2e5jrv/zzz/y3HPP5fj3CgDwPATzHJDdIJiGctbsZOy51amPnV4BAAAA8hjT685ky6WUWVbeggULbBDOZNwlJiZKz549beDP9M6rW7fuRX3yzNAKc/+ff/5py2bNY4w9e/ZI7dq1bfmtCe6l1KRJkxz9HgEAnotgngcEwZzOftPs8yNtRJvuVf4WbVwnD4g2XoUqijbbIsuLNo2K/SraaPx4wTvET7TZ9PJa0abuw3+KNlEJ+v6/KxF04Y9qLXyvrSPaeIXpew2PXbBatAmpWUy0OXoyWrSp2qyCaJOdklqjTZs2tn+dCdqVKVMm1RRbk5mX0tmzZ6VDhw72MmfOHClRooQN4pnbZkAGAACXQjAPAAAAAK6ACdhVrVo1S/tu3rzZ9tN78cUXpXz5Cx9orF2b+gMgU1ab1D8vyW+//ZaDKwYAeDKCeUi3rOD8+fPiCVo+PMrpJQAAAABZVqFCBZvBN23aNHnooYdsLzwzDCOlBx98UF599VUZOXKk9OvXT9avXy8RERH2Pi8vr+T9Nm7caLP5Tpw4IadPn7b7GQ0aNHDzdwUAcCeCebiICeRNnjxZPMHn2zQW/AEAAADpM2W1JjA3evRoO/iiUaNG8sorr8itt96avE+lSpXks88+k8cee0ymTJkizZo1kzFjxsjAgQNTlf926tRJdu/enXy7YcOGyYM4AAB5F8G8fCYrwze0DtxITwE/H9Fm0R59n4R2TLjwKa0mwUUPizaHz+rrbZRYPHXDbA38C2W9h5C7rBu5SrSp0zVctNH4+tSh4kbR5mxccdHEtXuvqFMgSLTxa1FDtKm9vaxoU7FksGizMUTf77rsSMqYy859PXr0sJeU0gbgTHAvZYDPTLItV66cfT+fZNeuXVewcgCApyKYl89kZfgGAzcAAAAAZ73xxht2om2xYsVk5cqV8vLLL8vgwYOdXhYAQAGCebis7D0t2j/ypNNLAAAAAHLctm3b5Nlnn7X98EyfPVNya3pbAwBAMA+Xlb2nxaJdx51eAgAAAJDjTA9rT+ljDQBwL4J58Gh//KsvmDeq8TrR5u+Eh0SdE6JOx5jXRZsXlnURbf6vdXnRpvHjnUSb2C/19fErG/JfnyUtXlxWSrS5uaGuvpD+H28QffStKahkAdFm4sQ7RZtXf9D33ql/x5pOLwEAAI/i7fQCAAAAAAAAAGQNmXl5hOmfcf78eclvSnQe6PQSAAAAAAAA3IZgXh4J2h05ckTmzJkj+c33e0+KNvGFqog21V1bRZvAuP2izTbfIaJN+yLxos2Jt/4VbUrcrK/MNur3+aLND80PizZNa5QQbWoV0VVC6n2zvt8rXiFBos2uF38RbQY9eUi06X77VaJNeCF97wlEqju9AAAAMkQwz8OYQF56jXA9ZfosAAAAgKwbP368zJ8/X9avX+/0UgAAShDMyyMCAwPzZUDvluFjnV4CAAAA8rHevXvL7Nmz7XU/Pz+pUKGC3HfffTJ69Gjx9eXPLQBAzuO3Sx7xwgsvSH6kscwWAAAA+UvHjh1l1qxZEhMTI998840MGjTIBvZMixwAAHIawTx4tE37T4k2bcudFW1Ox1USbQIDIkWb06cTRJtfdxwXbSoF+Ig6p0+INsVf7iXa+O7XN8S+baElos2JuGtFkw0jfhVtmgypJdqUurqUaPPc8OtEm1Xbj4k276/zF20GNc3e/gEBARIWFmavDxw4UObNmydfffWV3W6CfDt27JCiRYvKLbfcIhMnTpTg4GC7b0REhAwbNsx+ffzxx2Xv3r3SqlUreeedd6R8+fIZZgJGRkZKy5YtZdKkSRIbGyt33323vPbaazaACADI+/S9qwcAAAAAB5kMu6ioqFQXsy2rgoKCbJDN29tbpk6dKhs2bLCluN9//7383//9X6p9z507J88995y89957snLlShuoM8G5zPzwww+yfft2+9Uc1wQDzQUAkD+QmZfPZDQN11NVvesRp5cAAACAPNjCZsKECam2jRs3zg6jyIzL5ZJly5bJt99+K0OGDLFZd0nCw8Pl2WeflYceekjeeOON5O1xcXEyffp0ufbaC1nBJjhXq1YtWbNmjTRp0iTd8xQpUsQ+xsfHR2rWrCmdO3e25x0wYMAVfucAAE9AMC+fyWgarqd6ffUup5cAAACAPPgB+PDhw1NtMyWzGVmwYIEtnTWBucTEROnZs6cN/H333Xc2MLh582ab3RcfH2/fj5tsvAIFCtjHmiEZ11xzTfKxTHCucOHCsmnTpgyDeXXq1LGBvCSlS5eWv//+Owe+cwCAJyCYl8+y8I4cOSJ5ybnoONHmeGxVp5fgEbo8Ey/avPyIvl5wGn/GT+2OEm2KbN/u9BI8wjmXvp8nCSgo2hTyOySa1Lq+mGjj10RfL9Zz760TbeoV+0e0qViorGhz4nxx0cYE7jIL3qXVpk0bmTFjhvj7+0uZMmVsgG7Xrl1y88032x56pozW9Mz7+eefpV+/frYENymYdznS9sbz8vKyQUQAQP5AMC+fZeE9+uijjq0HAAAAyIsKFiwoVaum/kB33bp1NsBmhlSY3nnGJ598ctFjTbbe2rVrk7PwtmzZYvvmmVJbAADSQzAvnwkMDMxTAb0ytz7s9BIAAACAi5jgnim7nTZtmp1ia4ZbvPnmm+lm2Zn+emZQhsnoGzx4sDRt2jTDElsAAAjm5TOmZ0de8vIP25xeAgAAAHCR+vXry6uvviovvfSSbX9z/fXX2/fi9913X6r9TLntyJEjbZ+9/fv3y3XXXSczZ850bN0AAP0I5sGjXV1VX4+VIv76+uMs33eVaDO0T2PRZm+UvknPGn/GK83uI9q4Nmx2egke4bHOunrBGd/va+H0EtQ7NPeYaNOttr4+lZE7I0WbP/bXcHoJHkLf799qoUFZ3jciIiLD+0xFTNqqmHvvvfei/bp162Yv6TGDNFJO0U3vfK+99lqW1wsA8HwXmjcAAAAAAAAAUI/MPFzWVFwtbhk+1uklAAAAAAAAuA3BPFzWVFwtvt970uklAAAAAJeld+/e9gIAQHYQzINHOxuXINqcSygq2mh8nuC5En75XbSJ+1dfYD9w0O2izYKdtUSbNhUOiDY/7Cnl9BLUO7fthGhTvl890aZEaX6+s6JNhcOiTxGnFwAAQIYI5uGSZbRHjhxx63oAAAAAAACQPoJ5uGQZrZnApbVvXtshY5xeAgAAAAAAgNsQzINH9837bo++Mp/VB0qLNq3KHxNt1hws7vQScJli/9BXDhVwdZho4xUYItoEResbYs9r5qWtC/MSbaKP6/uAL1Dha9Pq6/n59tTXgXYVnF4BAAAZ0/euHgAAAADyOC8vL5k/f769vmvXLnt7/fr1Ti8LAOAByMzDJQUGBsqePXuSb2sque386FNOLwEAAAD53KFDh+S5556ThQsXyv79+6VkyZLSoEEDGTZsmNxwww1uW8fQoUNl5cqV8s8//0itWrUIDgJAHkUwD5f0wgsv2L55GktuNZbZAgAAIP8wWXUtWrSQwoULy8svvyx169aVuLg4+fbbb2XQoEGyefNmt66nb9++8uuvv8pff/3l1vMCANyHYB48Wr0ScaJNdEJh0cbb66BocygqRrQJKxQg2kTHJ4o2gR2qizZxfx8QbRLnfiraHL++pWhzQ6V40WbHqTKiSXhNfa9NRaoXEW0O/66vZ96WA1GijbdXCdEm0NezO/88/PDDtkR2zZo1UrBgweTtderUsYE1Y9u2bdKvXz+7T+XKlWXKlCmZHjMhIUEGDBggv/zyiyxZskQqVKhgz/H222/b7D8TKCxbtqxMmjRJbr311uTHTZ061X49evRousG848ePy+DBg+XHH3+UkydPSpUqVWT06NHSo0ePHHxGAAC5jWAeslxqm5Sdd+TIEaeXAwAAAOSamJgYe0kpICDAXlI6ceKELF682JbYpgzkJTHZeomJidKtWzcpVaqUzZg7deqULb/N7NwmuGYy/n766ScpUeK/AOyECRNk4sSJNgNw2rRp0qtXL9m9e7cULVo0S9+XqbC5+uqrZeTIkVKoUCEbGLz33nttUK9JkyZZOgYAwHkE85DlUtskKUtune6hN+r5Jxw5LwAAAPL2e18TOEtp3LhxMn78+FTb/v33X3G5XFKzZs0Mj/Xdd9/ZUluTTVemzIXM3+eff15uuummi/Y9c+aMdO7c2Qb0fvjhBwkNDU11f+/evZOz6MwxTCaeyfbr2LFjlr4vk803YsSI5NtDhgyx6/rkk08I5gGAByGYhyvmZA+9I9H6SmoAAADg2cyH1cOHD0+1LW1WnmECeZeyadMmKV++fHIgz2jWrFm6+5pAXbly5eT777+XoKCgi+6vV69e8nWTCWiy67JTNWPKd00Q0ATvzKCO2NhYGzgsUKBAlo8BAHAewbx8LCmjLrtlsylLbp0uu113WOOP8BnRJ0y0KRYs6sQlXvoNubv5enuJOt76ehv5lgsRbbyKpc6m0KBQkMbXTH0988oFnxJNdh/Q12PUx/eoaBNasZBoUzlM32tTTIK+Xqxtw34Vdfw6phu8S6tatWq2l11ODbno1KmTfPDBB7Jq1Spp27btxcvy80t125zblPFmlSnPNf36XnvtNTuowwQETcmvCeoBADyHxnf1cHNGXdqy2eyU3BrZfTwAAACQF5hedR06dJDXX39dhg4delHfvMjISKlVq5bs3btXDh48KKVLl7bbV69ene7xBg4cKFdddZUdamH62bVq1SpH17ty5Uq57bbb5J577rG3TSBw69atUrt27Rw9DwAgdxHMwxVLm6nnTu0fedKR8wIAAACGCeS1aNHC9px7+umnbSlsfHy8LF26VGbMmCEbNmyQ6tWry/33328z46KiomTMmDEZHs/0sTPlsDfffLMsWrRIWrbM+iRy08PP9N07dOiQREdHy/r16+12E6zz9/e3mYSfffaZnZJbpEgRefXVV+Xw4cME8wDAwxDMwxVLm6nnTot2HXfs3AAAAEDlypXl999/txNtH3vsMZuBZybQmqmxJpjn7e0t8+bNk379+tmAX3h4uB1ckdnQClP6arLmTNmtmZbbvHnzLK2lf//+smLFiuTbDRs2tF937txpz/vkk0/Kjh07bDah6ZP3wAMPSJcuXeyEXQCA5yCYB48WHnpxY2BcrFroX6LNu+v+awKtxZKl/zq9BI9wY9gx0cYVkyDa+ISmLrXSQONrZrFtM0SbhDrtRBPvZiVEG7+g1H27NCh4Y0XRJlZhf7pGYQoHHfheujeddqZ8dvr06faSHpOZ99NPP2U4PMME2tIO0zADOFIO4Uhv2IYp401p+fLllywLnj9//iW+GwCAdvq6mAMAAAAAAABIF5l5uGi6rSd5YOxzTi8BAAAAAADAbQjm4aLptp5k08lzTi8BAAAAAADAbQjmwaP9dei000vwCMH+dUWb0OAzos0dXes4vQSP4PWHvl8dXoH61qRRsP/F/ZYcdz5GtPl8SyXR5NbwUNHm9EZ9A6gOvvCraNNoub7+dL/s0ff7t3mFJqJNeX1tTwEASMZfP5DAwEB59NFH5ciRIx5Xctv0gSecXgIAAAAAAIDbEMyDvPDCC/arCeh5Wsntx5sOO70EAAAAAAAAtyGYB48WE5cg2oQX01dScz5e3+DqRmUKiTZvLt3q9BI8QrdzcaLNrrf/Em2qTL5ZtCkd9LeoU622aBOzR9fvlnPbI0Wb4PIhok3xB64RbbwDfxFtGpVpKdqcj3d6BQAAeBZ9f+EDAAAAAAAASBeZebiod56Rsn9eStp66TXs87jTSwAAAABk1apV0rJlS+nYsaMsXLgwx9vifPHFF7J582YJCgqS5s2by0svvSQ1atTI0fMAADwDwTxc1Dsvbf+8lLT10nvvrwNOLwEAAACQmTNnypAhQ+zXAwcOSJkyZXLs2CtWrJBBgwbJNddcI/Hx8TJ69Ghp3769bNy4UQoWZPQuAOQ3BPPymZSZdRll33mSA8fOiDZXldbXCy6swC7R5vWffESbEkWDnF6CR/CtVF20KdXgkGiTuEPfhw0+AYtEmzP17hZttL2OH/n7mGhT+V59vQ4TD50UbdafaCbaVCus7z3Btsjy4unOnDkjH3/8saxdu1YOHTokERERNuBmLF++XNq0aSMLFiyw78W3bt0qDRo0kHfeeUeuuuqq5GOsXLlSxowZI2vWrJGAgABp0qSJfPTRR1KkSBFZvHhxqvOZ45csWVLWrVsn119/vdu/XwCAswjm5TMpM+syyr5LW3KbUl4IAAIAAACZiYmJsZeUTIDNXNLzySefSM2aNW3Z6z333CPDhg2zgTsvL6/kfR5//HGZMmWKhIWF2UDfLbfcYgN7fn5+sn79ernhhhukb9++dh9fX1/54YcfJCEh/YE8p06dsl+LFi2ao983AMAzEMzDJUtuU0ob4HO6h16pWwY6dm4AAADk3ffCEyZMSLVt3LhxMn78+HT3N6W1JohnmJ55JthmSmNbt26d6vE33nijvT579mwpV66czJs3T+68806ZOHGiNG7cWN54443k/evUqZPuuRITE22wsEWLFqky+wAA+QfBPFwRp3vovfj9VsfODQAAgLzJfGA9fPjwVNsyysrbsmWLLY01gTnDZNXdddddNsCXMpjXrNl/Zdcmo85k8W3atMneNpl5d9xxR5bWZnrn/fPPP/Lzzz9f1vcGAPB8BPOQLWnLb50uu61StrBoUyzIT7Q5EVNZtKlS9kJ5CDzPuTd/EW38ShYQbbyLh4g2XrUaijZ+3nompCfZfuKcaFI5WtSJ33datPGrEybaaPtZMrafKCbaNCmr771TZiW1aZmgnRlKkXLghcvlso+fPn16lo5hJtRmxeDBg23vvR9//NFm9gEA8ieCebii8tvM+u4BAAAAeZkJ4r333nsyadIkO102pS5dusiHH35oe+kZq1evlgoVKtjrJ0+etP3yatWqZW/Xq1dPli1bdlFpb8rgoJmUa7L/zECNSpUq5fr3BgDQi2AerkhGgzLcpflDoxw7NwAAAPI3kyVnAnP9+vWT0NDQVPd1797dZu29/PLL9vbTTz8txYoVk1KlStmptcWLF7cBv6Sy3rp168rDDz8sDz30kPj7+9sBGKb01uxnSmvnzp0rX375pYSEhNiJuYY5Z1az+gAAeQfBPA+X2QAKd5TAZjQow10+3cJ0XQAAADjDBOvatWt3USAvKZhnBlv89ddf9vaLL74ojzzyiGzbtk0aNGggX3/9tQ3aGdWrV5clS5bYKbdNmjSxAbprr71WevToYe+fMWOG/ZqyB58xa9Ys6d27txu+UwCAJgTz8vAAivxQAtu+UpRo8/Gf+vo/aVS77MVvep0266uNTi/BI9wcmrUeQu4U0DxctIlbt0+0+a1eG9GmRcBXok37Srr6Hfr3qCra+JbT9RwZ0d/oG4p1+7hvRRuvAvp+//58spVoUzE4a7/rTEAuIyYoZ8pjTVms0bJlSzu4IiOtWrWSlStXpnufOQ4AAEm8k68BAAAAAAAAUI3MPGRaqqvd+JcGOb0EAAAAAAAAtyGYh0xLdbU7Ffuv00sAAAAAMmT63FEmCwDISQTz4NGW79XX96VUUadX4BmOR8eKNrfeqK8vlUYH++wRbcLb1xFt/OqWEm2qFE4Qbb462FG0uWXLK6JJvMI+lbF/HRVtYqJinF7CRQoUqyDafLm9tuij7z0BAACaEcyDR5fcth482uklAAAAAAAAuA3BPHh0ye2X2/VlBgAAAAAAAOQWgnnwaDsORDm9BI/Qqqa+cr8Vmw+LNjxPWdP5gfqiTex3G0Ub38qFRZsCvidEm/KFwkQb74o3iCZ7Hlso2lSefItoE/ncd6LNrqNXiza3FvpMtPlDbnd6CQAAeBRvpxcA5wQGBsqjjz4qR44ccXopAAAAAC5juMawYcOcXgYAwM3IzPOgoJuRk4G3F154wX5NOrYn9s6r0G2w00sAAABAPta7d2+ZPXu2fW/9xBNPJG+fP3++dO3aNUcm2S5fvlzatGkjJ0+elMKF/8v8/uKLL8TPzy/Lx9m1a5c888wz8v3338uhQ4ekTJkycs8998iYMWPE39//itcJAHAPgnkeICnollHgLT/3zpv803anlwAAAIB8znz4/tJLL8mDDz4oRYoUcdt5ixYtmq39N2/eLImJifK///1PqlatKv/8848MGDBAzp49K6+8omuSNwAgYwTz4NGCAvT9CDevUly0efjhL0WbOTO7izaLNh4SbW6ora+fWPTc/aJNwdtqijZxG/T9PPl5R4s2vt5eoo5P1rNc3KHMdeVEnSPHRBufAB/RpnKovlYme2J7iDY1A7eKPnWztXe7du3k33//tR/CT5w4Md19fv75Z1v9snbtWilevLjN2jP7FyxY0N4fExMjY8eOlblz59pqnPLly9v9b7jhBpuVZyQFCu+//36JiIiwZbYNGjSQ1157zW4PDw+XBx54wK7l008/tfs/+eSTdpvRsWNHe0lSuXJl2bJli8yYMYNgHgB4EHrmAQAAAEAKJrAWFRWV6mK2ZcTHx0eef/55mTZtmuzbt++i+7dv326DaN27d5e//vpLPv74YxvcGzz4v5Yx9913n3z44YcydepU2bRpk82eCw4OtkG9zz//3O5jAm8HDx6UKVOmZLiWSZMmSePGjeWPP/6Qhx9+WAYOHGgfl5FTp05lO8MPAOAsfWlNUN2TT1svvRo9aPgLAACAnGUy5iZMmJBq27hx42T8+PEZPsZk2pksObPfzJkzLzper169kodVVKtWzQbtWrVqZbPi9uzZI5988oksXbrUZvklZc0lSQq2lSxZMlXPvPR06tTJBvGMkSNH2tY5P/zwg9SoUeOifU0GnwlAkpUHAJ6FYB6y1ZNPWy+9N9fsdnoJAAAAyGPMB9jDhw9PtS0gIOCSjzN989q2bSsjRoxItf3PP/+0GXlz5sxJ3mYGY5j+dTt37pS///7bZveZ4N6VqlevXvJ1Ly8vCQsLS/cD+/3799tswTvuuMP2zQMAeA6CebiklNl4OTlNNydUDQsRbUIV9vG7+cHGok1h/72iTeSZRNHmmz/19acbJPrErtP3PPlfW1G0+elAWdGmWlF9r5mJyt4ebfl6l2hT7xp9/TwLVXXf0IGsOhqr7/85jY6ery7aVAwOyFLwLq3rr79eOnToYN8/mym3Sc6cOWOHYwwdOvSix1SoUMFmyOWUtNNtTUDPBA1TOnDggO3D17x5c3nrrbdy7NwAAPfQ9W4VKqXMxnPXNF0AAADAE7344ou23DZlWWujRo1k48aNdoJseurWrWsDbitWrEgus03J39/ffk1ISLji9ZmMPBPIu/rqq2XWrFni7U0bdQDwNATzcNn99TTo/OhTTi8BAAAASBWYM/3xTE+8JKZ3XdOmTe3Ai/79+9sJtia4Z3rkTZ8+3U6hNRNq+/btax9Xv3592b17t62KufPOO6VixYo2w27BggW2J15QUJAdjnE5gTwzAdccz/TJO3r0aPJ9phwXAOAZCObhsvvrafDdnhNOLwEAAABI5emnn7YTa1P2sTNZd2PGjJHrrrvO9surUqWK3HXXXcn7mEEYo0ePtsMrjh8/bstvzW2jbNmydiDHE088IX369LGTbyMiIrK9LhM8NCW95lKuXLlU95k1AQA8A8E8eDR/H31lAcF+uvoKGj5eXqLNvE1Bos2OHftEm1F31RdtAneUEm1c56+87CinRS/cJNrUaJb9/k+5L1602X3mGtGkfKNiok3Mb4dEm8RzcaLN+Xh9vVgL+UeLNucTCognSy+oZjLtYmJiUm275pprZMmSJZlWwLz66qv2kp6nnnrKXlJavnx5qtu7dl3cY3P9+vXJ100fv5S9/AAAnklfJAQAAAAAAABAusjMQ7o98cwneEm98TKbYJty0q0Tuo4Y59i5AQAAAAAA3I1gHtLtide+fftUE2wzCtqZQN+cOXPEKYt2HRdt1hzUVypyVSV9a9Koa7v0J8w5afPxs6JNqU36/r+LPnJOtAkqqe//uzIF/hRtjkTXFG3+Oa6rb1SrsIKiTWDrcNHmWIS+n+8dkfpemxqU1Nd6Q+PvukohgU4vAQCA3A/maZtymtMyy07LD0wgLym4l1Je/jcHAAAAAADIs8E8bVNOcxpBK51B3PaPPOnYuQEAAAAAANyNMlt4dBBXY5ktAAAAAABAbiGYB48WFR3v9BKAfMenTLBoE9qwlGjjOh0r2pyLLyLa/Lg7TrS5s/T3osm5uETRJnrxv04vwSNUL6qv3+GPu0+JNtr+n7ugR44dafny5dKmTRs5efKkFC5cOMeOCwDIv7ydXgA8o5Q2v/cMBAAAADKzatUq8fHxkc6dO2f7sc8995w0b95cChQoQMAPAHBJZOYhS6W0afviZTTd1t2aPvCE00sAAAAAZObMmTJkyBD79cCBA1KmTJksPzY2NlbuuOMOadasmX08AACZIZiHHJ1u624fbzrs9BIAAACQz505c0Y+/vhjWbt2rRw6dEgiIiJk9OjR6e577tw56d69u0RFRcnChQttJt6ECRPsfeZxAABcCsE8eLRDx8+KNg0rFXV6CR7hj50nnF4CLpPrrL4eZ95liok28f/sE22Co9Y6vYSLNK/QTbT5KfIm0aTqbuc/PEsrRGHvzIJh+vrTBQbtEW2aVygr2pz2v1q0KZTN/T/55BOpWbOm1KhRQ+655x4ZNmyYrWTx8vJKtV9kZKQtww0ODpalS5fasloAALKLYB6y3DcvZaktPfQAAACQV8XExNhLSgEBAfaSHlMaYeNo3wAATehJREFUa4J4RseOHeXUqVOyYsUKad26dfI+JmPvrrvukmrVqsncuXPF398/l78LAEBeRTAPWe6bl1LaHnpO9dIL7z7EbecCAABA/nnvm1T6mmTcuHEyfvz4i/bdsmWLrFmzRubNm2dv+/r62qCdCfClDObdeOON0qRJE1uOawZlAABwuQjmwaN76U35eYfbzgUAAID8wXxAPXz48FTbMsvKi4+PTzXwwuVy2f2nT5+evM2U137++eeyceNGqVu3bi6uHgCQ1xHMQ7pKlCiRrbJbp8pvy5XS17Mn0SXqrNt+TLSJjkkQbc5G6+sFFxun73nyUdgrS+LjRRsvP31ZFyeC/8sQ0aKE71bRpnxJXcOVjgbpe7vm7ect2ngF6nueCvnsFW0S/RQ+T777RR2v6hkG71IyQbz33ntPJk2aJO3bt091X5cuXeTDDz+0vfSMF1980fbKu+GGG2T58uVSu3btXFs+ACBv0/fbHCqULFkyW2W3lyq/BQAAAPKaBQsWyMmTJ6Vfv34SGhqa6j4zsdZk7b388svJ21555RVJSEiQtm3b2oBeUqBvz549cuLECfvV3L9+/Xq7vWrVqjYACABASgTzkKMyytjLLS0fHuW2cwEAAAApmWBdu3btLgrkJQXzJk6cKH/99Veq7aYlTcqAXvXq1WXs2LEye/bs5H0aNmxov/7www+p+u4BAGAQzEOOyihjL7d8vo2pugAAAHDG119/neF9ZtiF6Z1nDB06NNV9U6dOtZckERER9gIAQFYQzINHqxga5PQSPEKfpvr+V5+9xku0KaiwL5VG+17fKNqUvbWq00vwCDtPFRZtKofq6wsZEL1ZNAmpXUy0iTt4VrQp2LasaDN5Zbhoc38Tff0O1x5tLNo0zrzjDAAAjuIv13xW8ppbAyrMxC8zydbdeo162u3nBAAAAAAAcArBvHxW8ppb/exMIM/0/3C3tUdOu/2cAAAAAAAATiGYhxzJtsutjL9L2X0q2pHzeprdpwqINhXCnF4BLpd/sL9oc27DcdEmdMxtoo3G18yGJQ6IOsqeJr+qRUSbgz/sEW0CahUVbSrUCRFtQv3/FW1KF9T3bwcAgGYE85Aj2XbunGALAAAAAACQXxHMQ67153OHlg+Pcvs5AQAAAAAAnEIwD7nWn88dPt/mTHkvAAAAAACAEwjmwaMdOHpWtLm2SnHR5tftx0Qbjc9TpdAzos3OU8GiTZHq+vp3+RTX1xdy/6A5os1tC5qJNidiKok2RQp5iyYJu6NEm7DmZUUbb4X9PMsXChJt3lhVWvTR1/d0SPNCTi8BAIAM6Xq3CgAAAAAeaNWqVeLj4yOdO3cWzT2xe/fuLXXr1hVfX1/p0qWL00sCAFwGMvOQYQ+8PXv2ZHuqrbtVvmOo00sAAAAAZObMmTJkyBD79cCBA1KmTJl093O5XJKQkGCDae5mzhsUFCRDhw6Vzz//3O3nBwDkDIJ5yLAHXtJAi+xMtXW3ab/sdHoJAAAAyOfOnDkjH3/8saxdu1YOHTokERERMnr0aHvf8uXLpU2bNvLNN9/Ik08+KX///bcsWbJExo8fbzPkTDbf7Nmzxd/fX5599lnp2bOnDB48WD777DMpVaqUTJs2TW666abkc23YsEFGjhwpP/74ow0MNmjQwJ6vbNmycvXVV0uLFi3krbfesvtu377d3j9lyhTp27evFCxYUGbMmGHvW7lypURGRjr0jAEArgTBPHi08LAQ0eZ8fIJoc03lYqKNr7eXaDN3bbRos+y7v0Wbj/x8RBu/Gvr6+JXw0fczfia+pGhTIn6VaJMQECaaxB/U18/TO0Rff7rYDfr6wyZ2cok2A67VF7z5+l99PRiz65NPPpGaNWtKjRo15J577pFhw4bZ6hYvr/9+FzzxxBPyyiuvSOXKlaVIkQu/t0wQ7//+7/9kzZo1Nhg4cOBAmTdvnnTt2tUGA80H6vfee6+tmClQoIDs379frr/+emndurV8//33UqhQIRuUi4+Pt5U1c+bMkWuvvdaW+t588812LTfeeKMN5AEA8g6CeQAAAACQQkxMjL2kFBAQYC/pMaW1JnBmdOzYUU6dOiUrVqywQbckTz/9tA2spVS/fn2brWeY4N+LL74oxYsXlwEDBthtY8eOtZl0f/31lzRt2lRef/11CQ0NlY8++kj8/PzsPtWrV08+nsnCM9l9/fv3l7vvvlt2794tCxYsyLHnBQCgA8E85Bgneuu1HTLGrecDAABA/mg5M2HChFTbxo0bZ0tj09qyZYvNrDMZdYbphXfXXXfZAF/KYF7jxo0vemy9evWSr5ty22LFitnS2ySmzNY4cuSI/bp+/Xq57rrrkgN56Xnsscdk/vz5Mn36dFm0aJE9JgAgbyGYhwyZVH3TNy/pzcOlONFb7+sd+kpqAAAA4PkfUg8fPjzVtsyy8kyZa8qBF6aXndnfBNSSmH51aaUNypmy3JTbksp0ExMT7VczvOJSzHv3rVu32uDgtm3bbKYgACBvIZiHTD+RNJIGYVwqCy+rQb+cFOKv70f4ly3ufx48UdPq+np31Q8vKtpEXx8u2vj+WUC0iVl3WLTxr1dCtDkbV1y0CY1PXcamgU+Crh51/363T7SpPbyh00vwCOfjLwRgNEl06XvvVKKgvh6MmZXUpmSCeO+9955MmjRJ2rdvn+q+Ll26yIcffmh76eUUk8ln+uzFxcVlmJ1n+uOZ7L5+/frZct127dpJrVq1cmwNAADn6fttDo+QXhZeekE/AAAAIK8y/ehOnjxpA2eml11K3bt3t1l7L7/8co6dz0y5NdNtTT888+G6Oefq1aulSZMmdviG6am3atUq22OvfPnysnDhQunVq5fdx0zLNTZu3CixsbFy4sQJOX36tC3dTeq3BwDwDATzkONlue5022Pj3Ho+AAAAIIkJ1pnMt7SBvKRg3sSJE21gLaeY/ndmiu3jjz8urVq1sqW0JgjXokUL2bx5s91u1mQCecYbb7xhs/meeuopeemll+y2Tp062cEYSRo2bJhcGgwA8AwE85DjZbnutHxfpNvPCQAAABhff/11hveZbLmkANnQoUMvun/58uUXbdu1a9dF29IG2Uxw7ttvv033nOfOnUt1u3DhwrJnz55LngMA4FkI5uGyMu6c6I+XnoNR7p2emxXbth4XbdpdX8npJXiE1Vt1/FyndFvDcqLNsSl7RZsSnSqLNgmHz4o2UbH6fu2XDS4i2iT6XLrBvDuVu1pfr8OEA7r6Chp+VfX9LNUroe9Dx92nq4g21YrEO70EAAA8ir539fCIjDsT3EtvCIa7NRkw0tHzAwAAAAAAuBPBPOToEAx3+3DjIUfPDwAAAAAA4E4E8+DRihW8MJVLk/IVC4s2MXEJok2NoomiTWwVfaVsH6/+r0G1FoOD/Zxegkc4ulTfv12tBy/uz+S4GH3tEryO7xNNQjvoa5XgXbiAaBO/7Zhos/ZQIdHmQORJ0eaq0vqep7IFnV4BAAAZ887kPgAAAAAAAACKkJmHyx6KYSZjOd0376ZhTzl2bgAAAAAAAHcjmIfLHophhmA43Tdvye4Tjp0bAAAAAADA3QjmwaOtVdgfp341fX3XNu3S1x/nTHhR0WbBzxtFmzbXVhBtAgoFiDo+XqJNwVL6Gi6d8ako2hQ89J1o41VSV4+6xBPRok387ijRxq+qvp61ZxX2rO1UXd/b/5jE/Nn5x8vLS+bNmyddunRxeikAAA+TP39zAgAAAEAO6N27d44E5J577jlp3ry5FChQQAoX1hecBgDooe+jOXhc3zzDqd55JToPdPs5AQAAgJwWGxsrd9xxhzRr1kxmzpzp9HIAAIoRzMMV980znOqd9/x3W91+TgAAACAjLpdLqlWrJg899JCMGDEiefv69eulYcOGsm3bNqlatepFj5swYYL9GhER4db1AgA8D8E8D8yGSwqgGUeOHJH87HxMvGjzx5ajTi/BI+w+pa8H1KO31hFtZv24Q7TpWDxItNHYv8s30Ee0Keirr89ofPlmoo3v7pWiScIJ56bGZ8S/ur6+p67z+vrTNSgVItoU8Nsu2hyI1NdjtHzBK+uF17dvX5k1a1aqYJ65ff3116cbyAMAIDsI5nlgNlxKKQN7TgYX83tQEQAAAHlHTEyMvaQUEBBgL1ntozd27FhZs2aNNGnSROLi4mTu3Lnyyiuv5NKKAQD5CcE85EhwMWVQ0Z3980LaDXDLeQAAAJC/3uMmlb0mGTdunIwfPz5Ljy9Tpox07txZ3n33XRvM+/rrr21w0PTEAwDgShHMQ45zZ/+8sQs3uuU8AAAAyD/Mh9PDhw9PtS2rWXlJ+vfvL/fee699X2xKbO+66y47qRYAgCtFMA8ebftWff2frmtRUbSpXrqQ00vwCN9uOizaJCS6RBvXuTjRxquAn2jjHaCvZ15sgr6+VAFn9PXvkvK6+md6h/wk2rgUvjYlntDXi7WQv74+uifOVxZtKhXaL9oE+FfNdvAurU6dOknBggVlxowZsnjxYvnxxx9zbH0AgPyNYB5yfDAH/fMAAACQn5w6dcpOq02pWLFitneeyfIz022bNct84M+ePXvkxIkT9mtCQkLy8czAjODg4FxdPwDAsxDMQ44P5kg7lCNXe+hV6Jo7xwUAAACyaPny5dKwYcNU2/r16yejR4+W559/Xvr06XPJY5iBGbNnz06+nXS8H374QVq3bp0LqwYAeCqCecjVLL2kTL05c+bkyrmeXrxZtKkaFiLaJLr0lUP9slFfBmflcqGiTa1KRUWd5aJOwkn3DN3JDleCvv/vAr31lSGeD64u2gSKrhYOiSf0/Xx7BSh8C+mnr7R97+kiok2RQG/RZmdUmGjToHjW942IiLCX9Pz000/i5+cn991330X3udK8P8vsOAAApKTwnRjyUpZeepl6AAAAQF5mJtcePXrUTr81E2xLlSrl9JIAAHkIwTy4PVMvJxXp8GCuHBcAAAC4XB9++KEts23QoIG89957Ti8HAJDHEMyD2zP18nqZLQAAAPI3M/jCXAAAyA0E8+DRrq5RQrSJUdgnq2XZk6JNTIK+fztkTVBffU24t/efK9oEFQ8SbRLjSju9BI8QI7p6n4a0qCDaeFXUt6Yf2+vLfjrTOUa02X9a35o0alCc6bEAAL30dcAFAAAAAAAAkC4y8/KhUaNGyfnz+ibjXY52Q590egkAAAAAAABuQzAvHzKBvMmTJ0tesHDncaeXAAAAAAAA4DYE8+DROpf5VbTZdE5fP7G4RH2ZmPGJiU4vAZfJtX2naJMQkyDalPlfP9Hmq72hos0th18RbbbUGiealPzwH9FH35qum3GdaPM1v+uyZPU/h0WbzpWK5chxIiIiZNiwYRIZGZkjxwMAwKBnHgAAAABcJjO11svLy178/f2latWq8vTTT0t8fLxjaxo/frzUrFlTChYsKEWKFJF27drJr7/q+xAcAHB5yMzLx/JC77zJL97o9BIAAACQz3Xs2FFmzZolMTEx8s0338igQYPEz89PSpd2Zop49erVZfr06VK5cmWJjo62LXbat28v//77r5QoUcKRNQEAcg7BvHwmMDBQ9uzZk3d658V84/QKAAAAkM8FBARIWFiYvT5w4ECZN2+efPXVV/Lggw8m7+NyueTGG28UHx8fWbx4sc3kO3HihNSrV0/69u1rs/mWL18ubdq0ke+++05GjhwpGzdulAYNGthAYY0aNexxtm/fLsOHD5fVq1fL2bNnpVatWvLCCy/Y7LskPXv2TLW+V199VWbOnCl//fWX3HDDDW57XgAAuYNgXj5jftE/+uijklfE+JYUbf49eU60WfyPvgzMkAL6+ghdVVZfP7HXP/5TtGm3a5doE353TdHm9ISPRZvbJhYRbVwn/EWbyoU2iiZejUqJNq44fa/hXuXLijY1iwWLNusPRok2DarnvUyxoKAgOX489aA2E7ybPXu21K1bV6ZOnSqPPPKIPPTQQ1K2bFkZO3Zsqn3HjBkjkyZNsll0Zh8T7Fu5cqW978yZM9KpUyd57rnnbBDxvffek1tuuUW2bNkiFSpUuGgtsbGx8tZbb0loaKjUr18/l79zAIA7EMwDAAAAgBRMuay5pGQCZ+aSGZN9t2zZMvn2229lyJAhF91vAnf/+9//5L777pNDhw7Zktw//vhDfH1T/1lmAnWtWrWy15944gnp3LmzraoxVTYmIJcyKPfMM88kZwIOHjw4efuCBQvk7rvvlnPnztly36VLl0rx4sUv+zkBAOhBMA8e3VPvxVd6Ob0EAAAA5MFqlgkTJqTaNm7cODtYIj0mcBYcHCxxcXGSmJhoy1zNvp9++ulF+95xxx02+Pbiiy/KjBkzpFq1ahftY0pvkyT13Tty5IjNvDOZeebYCxculIMHD9pBG6YvXlIrnSSmXHf9+vVy7Ngxefvtt+XOO++0QzBKltRX2QIAyB6CeciQJ/TUi0lY6/QSAAAAkAc/1DZ96VLKLCvPBM5MYM5Msy1TpsxFmXYpmUy5devW2d5527ZtS3cfMzwjZXmuYYKExogRI2yW3SuvvGIn55qS3ttvv92W06ZkJtma+82ladOmNmho+uaZ7w0A4NkI5uVDJj3f9M0zn+55uhX7Kos2zcueFW2uKqGvj1B8oku02Xxc37+df4C+l+nApmVEm4QDZ0SbAtc6M8EwM65zp0QbV1zqPz418PfR9VoQve+0aOMVoq/XYcKG7aLNydJxos2mXSdEm727I0Wbu2o1u2RJbXqBs6x47LHHxNvbWxYtWmR735kS2rZt22b5XKZ3Xu/evaVr1672tsnU25WFfrYmGJi2dBgA4Jn0/ZUIt5QNGDk1CMPJctybhj3lyHkBAACA7DKlse+++66sWrVKGjVqJI8//rjcf//9dspskSJZG1JkMuy++OILO/TCZO099dRTyVl7hplwa3ru3XrrrbZE15TZvv7667J//35b4gsA8HwE8+DR5bhLduv7dBkAAABI6+jRo9KvXz/b784E8gzTl2/JkiV2Yu3HH2dtCvqrr75qp9s2b97cDrQYOXKkREX9N6XYlO9u3rzZTs41gbxixYrJNddcIz/99JPUqVMn174/AID7EMyDR6sYGiTanI/X97+Vv7e+Mra9Uf99gqzF9eX0lfmE3FxLtIn/WF/JX+y/+v7tAuuVEHVK1xBtvA7sF20SErNeWucOXgX+652lhU+orufI8C6n7/+55RsPiTZrV6UekqBBmQqFxZNFRERkeJ8phzUXo0SJEnaCbdreeGvX/tcDunXr1nYibkoNGjRItS08PFy+//77VPsMGjQoVUsdk7kHAMi7vJ1eAAAAAAAAAICs0ZdCBLUy6o2XnUEaOd1f76Fxz+fYsQAAAAAAALQjmJePJU21zUjaIF1GvfGyM0gjp/vrbYmMzrFjAQAAAAAAaEcwLx9LmmqbkZyadpubTsXEiza+3v6ijbeXqJPocmYCcma2RZYSbU46NCk6M97B+n7GvXwU/pAr5BWn7wMQl6+PaOPlpaunZ+IZfX1PfYoGijYJWw6INvsK/DeUQIuWrSuJNkdP6HttAgBAM4J5yPUMv8styQUAAAAAAEBqBPOQ6xl+KaUX9LuSPno9nnj6sh4HAAAAAADgiQjmwXFX0kdvzeHTOb4eAAAAAAAArQjmwfGS3CspvT0fnyDa+Pt4izZFAnaJNt5eVUWb2ARdfbKMxBCFfakOnxVtfEIDRBuvEH29BcVbX386SdT3/53Lpex1PMEl2vhWLCTauOL0/Swd2aXvQ8fQwkGizaG9p5xeAgAAHoVgHhwvyfWEQRsAAABAenr37i2zZ8+21319faVo0aJSr1496dGjh73P2zt3PyD48ccf5eWXX5Z169bJwYMHZd68edKlS5dcPScAwFkE8+BRAzTS6jpiXI6vBwAAAMiOjh07yqxZsyQhIUEOHz4sixcvlkceeUQ+++wz+eqrr2yQL7ecPXtW6tevL3379pVu3brl2nkAAHoQzEOWg2y5NYk2OwM00vpxf2SOrgUAAADIroCAAAkLC7PXy5YtK40aNZKmTZvKDTfcIBEREdK/f3/x8vKSt99+WxYuXCjffvut3W/SpEly66232sctX75c2rRpYwOBTzzxhGzevFmaNWsmH330kc26Gz58uOzfv19uvvlmeeedd6RAgQL2cTfddJO9AADyD4J5yHKQTWM57Mb9+nqsbNwv6jSvEi7aHDqjr+/aDgaqZEmjyoVFm8QT0U4vwTPE6nueXEePiTaRseVFE9eRc6KNz6YTok3i6VjRZtJTTUSbhRv0/T/Xql5pyYvatm1rM+a++OILG8wzJkyYIBMnTrRlsdOmTZNevXrJ7t27bWlukvHjx8v06dNtsO7OO++0FxMsnDt3rpw5c0a6du1qHzty5EgHvzsAgJOUdXgGAAAAAGfFxMRIVFRUqovZll01a9aUXbv+G0RmeuiZXnpVq1aV559/3gbn1qxZk+oxzz77rLRo0UIaNmwo/fr1kxUrVsiMGTPs7euuu05uv/12+eGHH3Lk+wQAeCYy85CrRo0aJefPn8+149foMSzXjg0AAID8W6FisuhSGjdunM2ayw6Xy2XLa5OYwRhJChYsKIUKFbqolU3KfUqVKmUz9CpXrpxqW9oAIAAgfyGYh1xlAnmTJ0/OteO/uWZ3rh0bAAAA+fcDadOjLiVT6ppdmzZtkkqVKiXf9vPzS3W/CfQlJiam2pZyH3N/Vh4DAMhfCOYhV7PvcmtoRpJW1UqINpuP6+sFd+hM9stCctuhU7mXsXm54uL1vTE+GaXv305jfzqfUgVFm/OrD4g2p+5oKNoULbZOtCke/6tocrbkhSb3mngF+og6JxJEm92n4kWbUkX1/TxpZAJ3lxO8S+n777+Xv//+W2XfaQCAZyOYh1zNvjOBvtx8A/PA2Ody7dgAAABAVph+eocOHZKEhAQ5fPiwnUhrSnXN5Nn77rsvV89t+u79+++/ybd37twp69evt0M1KlSokKvnBgA4g2Ae3DoRN6dtOqlvwh8AAADyFxO8K126tPj6+kqRIkXsFNupU6fK/fffL97euTtzcO3atdKmTZvk20nlwebcERERuXpuAIAzCObBo0UrLIs8fS5OtNl35oxoc3Pd0qJNsN9J0WbDsSKijWuDvv/vEk/HijZBt9QRbQqcWirqVKkp6vjoensUtfe0aFM49MrKD3ODt8I1bT4YJdr8tTl3W6BcjrBSIaJN+4pFs7yvCZhlJWhmhmGkFRkZmXy9devWF+1jpt+aS0pmCEfKQRzpPQ4AkLfpercK1QIDA9Mtmc3tvngAAAAAAAC4gGAerrhkNinAl9kgjNxy7+hn3Ho+AAAAAAAAJxHMg1sGYeSW34/qKx8FAAAAAADILQTz4NECfLxEG29vfWs6HxMv2pQusEm0+WxTmGhTLDhBtHGd1dcX0qdaMdHm/MKNok1Q266izq9LRJszV/cUTYq2LCvaaOxPF/ndbtEmsJu+t9rlyxQSbUqXCHZ6CQAAeJTcHa0EAAAAAAAAIMfo+7gQHjsYwwzCcHffvP5PPeu2cwEAAAAAADiNYB5ybDCGCei5u2/ehhNn3XYuAAAAAAAApxHMg0fbeuKcaLPvyGnRpm2d0qLNX8dEnYKB7p3GnBXn4xNFHR99HRpi1x8SbbwC9f2KPRNXUrQJ9tb381TwzF+iyfnIGNHGr1KoaBNcpbBoo/E9gUYtKhaR/G78+PEyf/58Wb9+fY4dMyIiQoYNGyaRkZE5dkwAgA763kHDo8ttTaktAAAAkJ8cPXpUBg4cKBUqVJCAgAAJCwuTDh06yMqVK8XTbdiwQbp37y7h4eHi5eUlr732mtNLAoB8T1/aADy63NaU2qaVm330rh80OleOCwAAAGSVCXbFxsbK7NmzpXLlynL48GFZtmyZHD9+XDzduXPn7Pd0xx13pPteHwDgfgTzkOtys4/evH+P5spxAQAAgKwwZaw//fSTLF++XFq1amW3VaxYUZo0aZJqnxEjRsiXX34pMTEx0rhxY/v+uH79+hke95133pFJkybJzp07bVbc0KFD5eGHH7b37dq1SypVqiSff/65TJs2TX799VepVq2avPnmm9KsWbNUxzHlu48//rjs3bvXrs8ct3z58qnKe01W4bPPPmuDjzfffLO8/fbbEhp6oZz+mmuusRfjiSeeyIVnEACQXQTzkCuTbVPKzdLbf/fq6wESVqygaLNTYW/BRmUKiTb+PkGiTWyCS7TxLRci2sSs09czL7BFWdEmyGuvaONK1NcX0itA1+t4wkmF/TxXHxRt/BT2zAvw83F6CR5hzb5Tok2lWoFZ3jc4ONheTFCsadOmtsw2LZPVFhQUJIsWLbJBsv/9739yww03yNatW6Vo0aIX7T9nzhwZO3asTJ8+XRo2bCh//PGHDBgwQAoWLCj3339/8n5jxoyRV155xQbyzPUePXrIv//+K76+vslZdc8995y899574u/vb4OBd999d6ryX7P/J598Il9//bVERUVJv3797H5mDQAAnQjmIVcm26ZEOj4AAAA8icmeM5eUTJAuvUCdCZyZYRMm2GYy4xo1amQz4EzQrF69evLzzz/LmjVr7AfcSY83ATgT/Pvss8/kgQceuOiY48aNs1l53bp1s7dNFt7GjRttEDBlMM9k+3Xu3NlenzBhgtSpU8cG52rWrGm3xcXF2YDgtddea2+bMuBatWrZ9SRlDpoqGhPsK1v2wodQJtPPHNOc3/T+AwDoQzAPjmTr5ZQyt14oNQAAAABy8gNqExxLG2AzZakZ9cwzATBTbrt69WqbgTdx4kRb0nr27Fk5c+aMFCtWLNVjoqOjZfv27Rcdy+xvtpsMORMgTBIfH59c+prEBAuTlC5d2n41QcOkYJ4JNCaVyBpme+HChWXTpk3JwTwztCMpkGeYMt3ExETZsmULwTwAUIpgHhzJ1sspr6/eJdrcX/wrUScwWLT5O/7CJ82arPz3mGhz39XRok3CMX1l2wkxCaLNX+N/FW0OXfuqaNOy0YW+TZqM+uisaDKt33WijeusrufIOqfv9XLYdftEmyHv6yojN26orS9g1GXUKBk+fHiqbell5aX9APvGG2+0l6eeekr69+9vA4CmZNUE2kxPvbRMYC0tE/gzTN+6pIy6JD4+qUu3/fz8kq+bSbOGCcQBAPI2gnkAAAAAkIWS2uyoXbu2LaU1ZbeHDh2yWXJmkMWllCpVSsqUKSM7duyQXr16XdEaTDbf2rVrk7PwTLadGcZhSm2T7NmzRw4cOGDPaZjMQm9vb6lRo8YVnRsAkHsI5iGVUaNG2b4ZnqLqXY84vQQAAADkY2YCrBlw0bdvX1v2GhISYgNopsz2tttuk3bt2tnS1S5dutht1atXt8GzhQsXSteuXe1k27RMia+ZXmvKajt27Gj795ljnjx58qKMwcyYzL0hQ4bI1KlTbTBx8ODBdkhHykm7JqPQ9OEzffzMAAxz3jvvvDO5xDY2Ntb260u6vn//flm/fr0d+lG1atUceQ4BANlDMA+pmEDe5MmTxVNoLLMFAABA/mGCWqYc1ryHNr3uzNCJ8uXL2353o0ePtuWv33zzjZ0226dPHzl69KgNlF1//fU2Cy89pkS3QIEC8vLLL8vjjz9up9jWrVtXhg0blq21mWOMHDlSevbsaYNw1113ncycOTPVPiYgZwZtdOrUSU6cOCE333yzvPHGG8n3m8CjmaibxAT9zMUM+UivdBgAkPu8XC6Xyw3n8XhmgIPGIFfKdeVEVp1pmOtJY+hHzv9btHmpzYVPLlUJKS7ajPyqpGjTst6FxtHIXOuX7xRt4qLjRJvzJ1NPQdQg5oMfRZvwA1NFmwV+fUWTzv+MEm28Gl8t2sR+8K1os+T2t51egke4qoS+3r6VQgIlPzADPUwpsMm0AwB4DjLz8pCcyKrLramzAAAAAAAAuHIE85CK6ZnhSQE9/1a6sicAAAAAAAByE8E8pPLCCy+IJ9FYZgsAAAB4SpmtuQAAPAvBPHi0Hi0riTZvbQ0VbR5ouFm0OXbojGhTvqW+/jihAfpepn1CA0SbXT8dEG2q96kt2hxQ+U7ER7S5pfQvokn0ewdFm8A6Z0Ub/xvqiTbzl2wTbZ69r4Zos/rAadEmv/TMAwB4Jn1/JeKyy2LN8AoAAAAAAADkXQTz8lBZbHq97nJiwq1m9495xuklAAAAAAAAuA3BvDwuJybcarb+mL5STQAAAAAAgNxCMA8erWYRfb1oth0vK9qc9yoh2nRso+950qh0gU2izb4Ve51ewkWq3lZZtInfp68H1KmYeFGnSJhos/5sW9Gk/l0bRZvz7y4TbQLaVBVtOrapItr0HPaDaDP5+Q5OLwEAAI/i7fQCkPP981Je6KMHAAAA5J7evXuLl5eXPPTQQxfdN2jQIHuf2cdo3bq1DBs2LNU+f/75p/To0UPKly8vQUFBUqtWLZkyZYrb1g8A8Dxk5uXR/nmZ9dHLS/30XnjlPqeXAAAAgHzOBOI++ugj297GBOQM8z577ty5UqFChUwfu27dOilZsqR88MEH9ji//PKLPPDAA+Lj4yODBw9203cAAPAkBPPysbzQT+98wh9OLwEAAAD5XKNGjWT79u3yxRdfSK9evew2c90E8ipVqmRvm+y8FStW2EtS5t3OnTulb9++qY5VuXJlWbVqlX08wTwAQHoI5uWT0tv05IUS3G93lRNtTkZFizaB3pGiTY3i1UWbnZHnRJsigbVFmzKtyos2P/1vu2hzza36esE1CP1Z1PEtI9rUP/iBqHLmrGgTNLKPaOM6uFO0KR0SINoknIsTbcIKaqwUCc72I0xQbtasWcnBvHfffVf69Okjy5cvt7dNAG/r1q1y1VVXydNPP223lSiRfl/jU6dOSdGiRa/oOwAA5F0E8/Jh6W1WSnABAACA/ComJsZeUgoICLCXjNxzzz22jc3u3bvt7ZUrV9rS26RgXmhoqPj7+0uBAgUkLCzjD3xMme3HH38sCxcuzLHvBwCQtxDMy8cyy9rzFK0Hj3Z6CQAAAMiDH4hPmDAh1bZx48bJ+PHjM3yMybLr3LmzREREiMvlsteLFy+erfP+888/ctttt9lztW/f/rLXDwDI2wjm5WOZZe15iqi4raJNSOhq0ebPOs+LNv9++aNo063IYtHmdEBT0cb/quz9YeIONy5sJNp8UfQRUWeX0wvwFHVEk26J+qZa/lDjKdGm9fzuos1h79SZXRo8Mq6taLNyX6JoYzLshg8fnmpbZll5KUttk/rcvf7669k658aNG+WGG26wwy+efPLJbK4YAJCfEMwDAAAAgGyU1GakY8eOEhsbK15eXtKhQ4eL7jdltgkJCRdt37Bhg7Rt21buv/9+ee655y573QCA/IFgngOf8pkpssgZEyYOdHoJAAAAgOXj4yObNm1Kvp5WeHi4/Prrr7Jr1y4JDg62Qy5MRp4J5Jngn8kGPHToUPLjMxqQAQDI3wjmuZkJ5E2ePNnpZeQZGstsAQAAkH8VKlQow/tGjBhhs+9q164t0dHRsnPnTvnss8/k6NGj8sEHH9hLkooVK9qgHwAAaRHMg0cLifxFtHGt/Em0qTVAV/8no345fb0F157uKtqUi9fXb2n7c7+JNqWqbxFtKr79hGhTLkTfz9OSbbFOL0E/X31v12rt+Fm0cX07UrQ5GxTv9BI8wpo/D4g23auVzPK+ZuBFZubPn598vXr16rJq1apU95uhGpkN1gAAIC3vi7YAAAAAAAAAUEnfR73IEfmlN9+ro+s7vQQAAAAAAAC3IZiXRYGBgfLoo49e8XGOHDki7pBfevO5jmZe1gAAAAAAAJCXEMzLohdeeCFHjpNRQDCnM+ncFTR0mlfBwqKNy99PtDn160HRZmHbq0WbTtXPOb0Ej1BnUD2nl+ARfEP0/Tx9s1Vf/677A+eINrPP9xJNouem7q+lweL7HhBt7vxpr2iTWMnl9BI8QuN6pZ1eAgAAHoVgnhI5nUmXE1mEAAAAAAAA0IVgXh6VU2XB2k1+rpXTSwAAAAAAAHAbgnl5VE6VBat3br7TKwAAAAAAAHAbgnnwaMcffEW0KdKnsWhT/NX7RJtuJc6KNgHeUaKN/44fRJ1bWog28Yv19RQrvugJ0ca76nOiTo1rRZveZ34VVSY8JNp0C9D3Gh7YVd+E+5kv/ej0EjxCh96NnF6CahERETJs2DCJjIx0eikAACW8nV5Afi1/TXvJLwMrAAAAgLykd+/e0qVLl1TbPvvsM/u+f9KkSW5Zw9ChQ+Xqq6+WgIAAadCgQbr7/PXXX3LdddfZdZUvX14mTpzolrUBAHIemXlKyl8vt79dTk/B9TRPOr0AAAAAIIV33nlHBg0aJG+++ab06dPHbeft27ev/PrrrzZol1ZUVJS0b99e2rVrZ9f1999/2/0LFy4sDzygbzo1ACBzBPM8XE5PwfU0x+9t6fQSAAAAAMtku40bN04++ugj6dq1q92WmJgor7zyirz11luyd+9eKVWqlDz44IMyZswY2bVrl1SqVEk+//xzmTZtmg3GVatWzQbcmjVrlu45jh49KjfddJPNrjPnMdl4U6dOTb4vvWDenDlzJDY2Vt59913x9/eXOnXqyPr16+XVV18lmAcAHohgHjzagT+OiTaFmh8UbeIWbhRtCr1cTbRxbVLWJ0tEvCrUEG0O+rQRbUoPuEq0WXusiWjjfeSMaDPk/YKiTe8Ouvp3XX10mmgTErxXtIn/e49oU/+mu0SbMd0rizbPfb5DtImJibGXlEzQzFwyMnLkSHnjjTdkwYIFcsMNN6SqpHn77bftB/AtW7aUgwcPyubNm1M91gT2TMDPBPLM9R49esi///4rvr6p/1wzwcAbb7xRmjZtKjNnzhQfH58sfT+rVq2S66+/3gbyknTo0EFeeuklOXnypBQpUiRLxwEA6EAwT1kvvexK22svv5Xd9nV6AQAAAMiTrXEmTJiQapvJuBs/fny6+y9atEi+/PJLWbZsmbRt2zZ5++nTp2XKlCkyffp0uf/+++22KlWq2KBeSiNGjJDOnTvb6+a8JnPOBPNq1qyZvM+WLVtsIM9k/L322mvi5eWV5e/n0KFDNgMwJZMhmHQfwTwA8CwE85T30ruUtAHA/FZ2+/dVi5xeAgAAAPIY8wH58OHDU23LLCuvXr16cuzYMRvwa9KkiQQHB9vtmzZtshl+KTP1Mnp8ktKlSyd/aJ8UzIuOjrbDK3r27GkDeQCA/I1gHjxa5bblRJuEo+ecXsJFvEL+K6lQIy516YoG52tc+ERckwCfKNEmbI17JvNlh6tSFdGm0S9zRZuQ1tNFm+419U1zf/+Pk6LJ1SX1lSJHl/0v80iLoOanRJvtHx0VbZ5JdIk2Ozfra5tyqZLatMqWLWsn2LZp00Y6duxoM/VCQkIkKCgoS4/38/NLvp6UcWd67aVcjxleYUp4H3/8cXu+7AgLC5PDhw+n2pZ029wHAPAs3k4vAAAAAAA8XcWKFWXFihW2bNUE9EyJremBZwJ6pvz2Snh7e8v7778vV199tQ0YHjhwIFuPN8M0fvzxR4mLi0vetnTpUqlRowYltgDggcjMy2O99tL20MsKT+6z96zTCwAAAAD+PzNhdvny5TbgZgZMLF682A7G+L//+z87fKJFixZ24uyGDRukX79+2Tq2GXZhptKa4RimL585T1JWnemvd+bMGRtINCW5ZlKtUbt2bXteU55revGZc5r1/PPPP7aXX35qzwMAeQnBvDzWa+9yhmh4cp+9s0PbOb0EAAAAIFm5cuVSBfS+/fZbO5V27NixNqPO9MR76KGHLuvY5jgffvih3HXXXckBvZIlS0r//v1tVmCShg0b2q87d+6U8PBwCQ0NlSVLlsigQYNsdl/x4sXteh544IEc+74BAO7j5XK59DXOwGUzwbzsBuYu5zFaJHzaW7TxCikg6vjr65nnlWI6mxZehUqKNq4off3E4j5eKtr4Visu6nhnfcqguww7OUC0KVw4a/2k3GnbZl19zua0WCzaeNduJNq4Du8SbcZt1deLNTIyWrTp0DxctOlcqZjTSwAAIEP0zAMAAAAAAAA8BGW2ebyHXlak12fPU/rovdLc6RUAAAAAAAC4D8G8PN5DLyvSC/55Sh89jWW2AAAAAAAAuYVgHjxa4qkY0cY7LlG0id8dJdr4175KtNHYn04KhIo2fh2vEXVKlRd1tm8WbYp66evpeTY6TrQpV6mIaOIVECDaJP6zVtRJ1Pf79+qaJUSbo5H6Ki8mvfTf4AYtOr/ZzeklAACQIYJ5SLc0N73SWwAAAAAAADiLYB7SLc1Nr/RWYx+9iXWcXgEAAAAAAID7EMxDlmnsoxf3Tg+nlwAAAAAAAOA2BPPg0eL+jRRtXOf09X+K2ntatAnrra+3kRw+JOqUCxR1vL1FHY1rKlVStKlbuJhoc1rha2bNUiGiin810cY7VN/Pt3jpex34/uN9ok1cbLxoE1RMXz9PAAA0I5iHdNFHDwAAALi03r17y+zZsy/a3qFDB1m8eLF4eXnJvHnzpEuXLpd1/OPHj0v9+vVl//79cvLkSSlcuHAOrBoA4MkI5uGy++hp6KH3nL4BfwAAAMhnOnbsKLNmzUq1LSCHJlH369dP6tWrZ4N5AAAYBPNw2dl6JlNvzpw5jq4pZlI30ca3dAnR54Bo49q9U7TxvrqVaOM6pTAjNlFfibTrr3WijVeJ4qJNrEvfv938LzeKNv83oIlo4uWnr9zedfKgqLN3h2hTpPBdos363/SV/l7TsqJ4OhO4CwsLu2h7eHi4/dq1a1f7tWLFirJkyRKpUaOGbNq0SWrWrJm8r+lNPX36dNm+fXvythkzZkhkZKSMHTtWFi1a5JbvBQCgH8E8XHa2XnoTbwEAAABPFxMTYy9pA3bZzbb77bffpGTJkjZrz2Tv+fj4SIkSJaRx48b2Q/FnnnkmeV9zu2fPnsm3N27cKE8//bT8+uuvsmOHvmA1AMA5BPOQo3313O3Fco6eHgAAAHn0Q+wJEyak2jZu3DgZP358uvsvWLBAgoODU20bPXq0vRimz13KzL1evXrZLLykYN7WrVtl3bp18sEHH9jbJpDYo0cPefnll6VChQoE8wAAqRDMQ4721XM3jWW2AAAA8GymN/Tw4cNTbcssK69Nmza2JDalokWLZrj/3XffLSNGjJDVq1dL06ZNbVZeo0aNkstuzflr1aol99xzzxV/LwCAPMgFwHX+/HnXuHHj7FctWFPWsCbPW4/BmrKGNWUNa8oa1pQ1rClrWNN/7r//ftdtt92W4f3mT6558+ZdtP3GG290DRkyxF6vWrWqa9KkScn31a9f3+Xt7e3y8fGxF3PdHMdcHzt2bC59JwAAT+Fl/uN0QBFwWlRUlISGhsqpU6ekUKFCogFryhrW5HnrMVhT1rCmrGFNWcOasoY1ZQ1r+k/v3r3tkIr58+ene7+/v798+OGH0r1791TbZ8+eLf/3f/9nH9eyZUvZu3evlClTxt5nhmBER0en6r3Xt29f+eWXX6RKlSq2Dx8AIP+izBYAAAAAroDpcXfo0KFU23x9faV48eJ2ou2yZcukRYsWtlS3SJEi9v5u3brJwIED7cWU6SYF8gwTsEvp2LFj9qspvTX99wAA+Zu30wsAAAAAAE+2ePFiKV26dKqLybYzJk2aJEuXLpXy5ctLw4YNkx8TEhIit9xyi/z55592IAYAAFlFMA8AAAAALlNERITpQ37RZfPmzfZ+E7Dbtm2bxMXFya5du1I99uOPP7b79unTJ9NztG7d2u5HVh4AwCCYB/z/6WTjxo3LdEqZu7GmrGFNnrcegzVlDWvKGtaUNawpa1hT1rAmAACcwwAMAAAAAAAAwEOQmQcAAAAAAAB4CIJ5AAAAAAAAgIcgmAcAAAAAAAB4CIJ5AAAAAAAAgIcgmAcAAAAAAAB4CIJ5QAoMdwbcZ+fOnRIfH+/0MtTjOco6XsMB9+E1HAAA5xDMQ74TExMjI0aMkOuvv15eeuklu+3ZZ5+V4OBgCQkJkZ49e0pUVJTb1/Xnn3/KfffdJ5UrV5agoCApWLCg1K1bV5566ilH1rNx40Z5+OGHpWHDhlK6dGl7MdfNNnOfNtu3b5e2bdu6/bwHDx6UDz74QL755huJjY1Ndd/Zs2fl6aefdvuali5dKuPGjZPvv//e3v7xxx/lpptuss/PrFmzRIsaNWrItm3bRIMDBw7Y56xXr1729WHz5s1uX8PixYvl77//ttcTExPlmWeekbJly0pAQICUK1dOXnzxRbcHq2655RZ5//33JTo6WrTgNTzreB33zNdxXsM98zUcAAB38nLxMTbymeHDh8vHH38sPXr0sG/c27RpIwsWLJDnn39evL29ZezYsfZN89SpU922pm+//Va6du0qnTp1sn8EfvHFF9K3b1/7x+Dnn39u/4D/+eefJSwszC3rWbRokXTp0kUaNWokHTp0kFKlStnthw8ftn9krFu3Tr788kt7nxbmD2mz3oSEBLed87fffpP27dvbwEtcXJwNvMyfP1/q1KmT/HyVKVPGrWsyf5D26dNH6tWrJ1u3bpVp06bJo48+Krfffrtdp7l/zpw59ra7dOvWLd3t5mfI/HFqAjCG+bl3lwIFCsju3bulRIkSNqjRvHlze90EOkxAbc+ePbJq1Sr7PLpLzZo15e2335brrrtOXnjhBZk0aZKMGTNGatWqJVu2bLHbzL/lyJEj3bYm85ro4+NjX4vMa2b//v3l6quvFifxGp41vI575us4r+Ge+xoOAIA7EcxDvlOhQgV59913pV27drJjxw6pVq2afQN622232fvNHzkDBgyQXbt2uW1N5s3ngw8+KA899FDyGoYOHSqbNm2yf1yYP0zLly/vtk/k69evb5+PjLIRxo8fb5+zv/76S9zlUn+Y79+/X1555RW3/hF444032n+Xd955x2ZvmCDLJ598Yv/9zL+pE8E8c17zh6D5+Vm2bJnNrHru/7V357FRVe0Dx88MbW0pUNbWQoGiUgHBiIlsbihGYgJ1YxUIiEZjFBVRIxA1Lq/ihriBC0sEgxtqIhK1RkGgSJEgW9ktggLTALJaQKDnl+e8b+ftoLy/yx/eeSbn+0kKc+c26TfYXmdO7znnX/9ybwaFDBB99tlnbmAhLDLAIndRtWnTJuH5mTNnmuLiYtOwYUN3HOYdJ9IUi8VMbm6uG/CQN8nyPZ2WluYey90dhw8fNnPnzg2tKTMz0715l2uU3NElg1L9+/ePn583b565//77Q70TRv6d1q5da0pKStx1s7y83LXJoJ78GzVq1MiEjWt4MFzHU/M6zjU8da/hAACESgbzAJ9kZWXZbdu2xY/T09Pt2rVr48dbt261devWDbUpMzPTfd0a1dXVrmvnzp3ueOHChbZZs2ah9mzYsOG05+WcfE6YIpGIbd68uS0sLPzbDzkXjUZDbWrUqJHduHFjwnPPPvuse37ZsmU2FouF3pSdnW0rKirix/J9tGrVqvjx+vXrbZMmTUJtev/9921BQYGdPn16wvNpaWm2vLzcJoN8P1VWVrrHLVu2dD9jta1YscLm5+eH2iRf74cffnCP8/LyXENtmzZtctevZP07ibKyMnvHHXfYnJwc1zJ48GD77bffhtrENTx4E9fx1LuOcw1P3Ws4AABhYs08eEfu6pCpFzXTayKRiFm2bFn8fFlZmZtmEyb5ejKNrva6QfKb5SZNmrhjWS9LfsMclsLCQncX0OnIudatW5swydd7+eWX3YLbf/fxv3r/SUePHk04fuSRR8y4cePctK0lS5aE3pOenp6w5pOstyZridU+Dnv9s0GDBplFixaZadOmmZtvvtns27fPJJv83MtHzR0eOTk5CeflTpOwO2WaptyBI3cAyR1VkydPTlgjT6bbXXTRRSaZunTpYt566y23PpX0/frrr+7OpjBxDQ+G63hqXse5hqfuNRwAgDClhfrVAAVkGtSIESPclBpZM0im9MiLdlksWV4QTpkyxYwZMybUJlk0XaatyfpY8kJ94sSJbupKRkaGO79y5cq/TG/5J8m0LFlEfsGCBW4qW+21lmTajyzUP3v2bBMmWadL/nsNGDDgb8/Li/qwVw3o2LGje6N36po8svi2vJGXNb3Cdt5557nvZVmYvGbaWs16RjWDDDKwEDYZWJBF3J944gk3/U/Whqt5I5YM8r1SVFTkGmSQRaYa1v7vuGXLllDXNxOy5pv8vMnaed27dzcff/yxm+onndLz+++/u7XZNJD1quQ6Kh8yNThMXMOD4TqemtdxruGpew0HACBMDObBO7LmlKyxInd2yALl8kK9Zn2qqqoqty6NvCELk7wRlbV6ZPdK2alRFiR/5ZVXEu76kDeoYZF1uuRryvpGsj6PrEsj5IWxDDLIm0P5O+w3pvLf53Q6dOjg7uwI+w38999/H18nq7aHH37Yvdl48803Q/9eqr2OWYMGDRLOL1++/LRvpP9pMtAibwTlTi75twtzXaxTnbq2k7yBrm3p0qXuTrkwyZ0lMqggd7/IOk/y5lkGE+QuHblO3XXXXaG/ib/yyivjA1KnI2+ow8Q1PBiu46l5HecanrrXcAAAwsQGGACA0MmdFHKHidyFJncyAQBSB9dwAACSi8E84D9k6pH8OGialqGhSX7zvmfPHvcb+WbNmhkNaAqGpmBoCoam1G06cOBAwp15p64vRpPOJm09NAEAoAcbYMA7su5Uv3793CLqMm1N3njJWkf5+fluSlKPHj3Mrl27vG+ShcivuOIKk52dbZo3b+5eIMuC0sOGDTPbt28PtYUmmmiiiaYzJ+sKytTVxo0bu7/bt28ffyzTuTU01X5Mk86ev2viewkAgORiMA/eeeihh9yug7IWzvr1693ObLIjouzUtnjxYnPixAm3k53PTbNmzXLrUMnOlbIIuKxPJW0TJkxwu1fKIuabN28OrYcmmmiiiaYz88ILL5j77rvP7YwsG16sXbvWlJeXu8c33HCDOyebhyS7ST5o0ttzuia+lwAASDKZZgv4JD8/35aWlrrHsVjMRiIRW1JSEj+/ePFi26JFC6+b2rVrZz/44IP48Y8//mgLCgpsdXW1Ox44cKC98cYbQ+uhiSaaaKLpzLRq1cp++OGHpz0vvS1btqRJWZO2HpoAANCJwTx4p27duvaXX36JH6enp9s1a9bEjysqKmx2drbXTVlZWXbr1q0Jz6WlpdkdO3a4x2VlZbZhw4ah9dBEE0000XRmMjMz7bp16057vry83HXTpKtJWw9NAADoxDRbeKdt27bmiy++cI+//PJLk5mZaUpKSuLnv/76a9OmTRuvmwoLC83y5cvjxytWrHCLuefl5bljWZPm+PHjofXQRBNNNNF0Zi655BI3zVeWaTiVrMv63HPPuc+hSVeTth6aAADQKS3ZAUDYZH264cOHm0mTJrm1jN577z23tkpZWZl78/Xpp5+aiRMnet109913uw04ZN0+GViURaZlEfc6deq489JVVFQUWg9NNNFEE01n5vXXXze9e/d2G3HIxhw1A4uyS/rChQtNRkZGwi+NaNLRpK2HJgAAdIrI7XnJjgDCVlpaapYuXWq6d+/udopdt26d+w1vVVWV6du3rxtY871pypQpblDx2LFj7gXzo48+6t6kClnIXX7z3a5dO5poookmmpQ2HTp0yDXJ/1tisZh7TgY/5P8zt9xyi2nQoEGoPTSlZg9NAADow2AeAAAAAAAAkCKYZgv8h6xnlJ6ebjTR1qStR9AUDE3B0BQMTanXJGuLzZ8/32zfvt2t8dezZ8/4NGCa9DZp66EJAAAd2AAD3vnoo4/Mn3/+mbDuSuvWrd10qKZNm5onn3zS+yZtPTTRRBNNNJ2ZUaNGxTdW+u2330ynTp3MddddZ8aPH++mAXfu3Nns2LGDJmVN2npoAgBAqWRvpwuELRqN2srKSvd4+vTpNjMz0z722GN23rx59umnn7bZ2dn2nXfe8bpJWw9NNNFEE01nJi8vz65Zs8Y9HjBggL3mmmvs7t273fHevXttnz59bL9+/WhS1qSthyYAAHRiMA/eiUQi8TddXbp0sc8//3zC+cmTJ9vOnTt73aSthyaaaKKJpjMjA4oVFRXucUFBgS0rK0s4LwMhTZs2pUlZk7YemgAA0IlptvBSJBJxf1dUVJhrr7024Zwcb9myxfsmbT000UQTTTQFV1RUZJYtW+Ye169f3xw8ePAvO4FWV1fTpKxJWw9NAADoxAYY8NJXX31lcnJy3HpGVVVVCeeOHj0af1Pmc5O2HppoookmmoIbPXq0efDBB01eXp4ZO3asuffee81rr71m2rdvbzZu3Gjuu+8+c9NNN9GkrElbD00AACiV7FsDgbDJdKjaH7KeUW1Tp05NyhQtTU3aemiiiSaaaDpzL730kq1bt67NysqyGRkZbm2/mo8bbrjBHjp0iCaFTdp6aAIAQJ+I/JHsAUVAE9kdLT093e2GpoW2Jm09gqZgaAqGpmBo0t+0f/9+880337jpvzLtMD8/31x66aWmbdu2obfQlLo9NAEAoAuDeQAAAAAAAECKYAMMeOeTTz75y3pGyaatSVuPoCkYmoKhKRiagqEpGJpSr0fQBACAPtyZB+9Eo1G389nAgQPNbbfdZrp27ZrsJHVN2npoookmmmiiyYcmbT00AQCgE3fmwUuyA9ry5ctN9+7dTceOHc2kSZPM3r17aVLcQxNNNNFEE00+NGnroQkAAIWSvQMHEDbZabCystI9Xr58ub3rrrtsw4YN7VlnnWX79+9vS0pKvG/S1kMTTTTRRBNNPjRp66EJAACdGMyDd2q/AKxx5MgRO3PmTNuzZ08bjUZtYWGh103aemiiiSaaaKLJhyZtPTQBAKATg3nwjrzAO/UFYG2bN2+248aN87pJWw9NNNFEE000+dCkrYcmAAB0YgMMeEcWTY7FYiY3N9dooa1JW4+gKRiagqEpGJqCoSkYmlKvR9AEAIA+bIAB72zdutU0a9bMaKKtSVuPoCkYmoKhKRiagqEpGJpSr0fQBACAPtyZBwAAAAAAAKSItGQHAMn2xx9/mI8++shs2bLF5Ofnm8GDB5smTZrQpLiHJppoookmmnxo0tZDEwAASiR70T4gbO3bt7d79+51j7dv3+52O8vJybGXXHKJbdy4sc3NzbUVFRVeN2nroYkmmmiiiSYfmrT10AQAgE4M5sE7kUgkvgPakCFDbI8ePez+/fvd8aFDh+w111xjBw8e7HWTth6aaKKJJppo8qFJWw9NAADoxGAevFP7BeA555xjS0pKEs6Xlpbali1bet2krYcmmmiiiSaafGjS1kMTAAA6sZstvBSJRNzfR48edWur1NaiRQuze/du75u09dBEE0000USTD03aemgCAEAfNsCAl3r16mXS0tLMwYMHzcaNG03Hjh3j57Zt25aURZO1NWnroYkmmmiiiSYfmrT10AQAgD4M5sE7jz/+eMJxvXr1Eo7nzp1rLr/8cq+btPXQRBNNNNFEkw9N2npoAgBAp4jMtU12BAAAAAAAAID/H2vmAQAAAAAAACmCwTx4p1OnTuapp54yv/76q9FCW5O2HkFTMDQFQ1MwNAVDUzA0pV6PoAkAAIWSvZ0uELZIJGKbNGli69SpY3v37m3nzJljjx8/TpPiHppoookmmmjyoUlbD00AAOjEnXnw0urVq82cOXNMRkaGGTRokGnevLl58MEHzfr162lS2kMTTTTRRBNNPjRp66EJAACFkj2aCCTjt7mVlZXx4507d9pnnnnGtm3b1kajUdu9e3c7bdo0r5u09dBEE0000USTD03aemgCAEAnBvPgHXmRV/sFYG3z58+3Q4cOtdnZ2V43aeuhiSaaaKKJJh+atPXQBACAThH5I9l3BwJhikajJhaLmdzc3NN+zsGDB02DBg28bdLWQxNNNNFEE00+NGnroQkAAJ1YMw/eGT58uMnKyvqfnxP2iz9tTdp6BE3B0BQMTcHQFAxNwdCUej2CJgAA9OHOPAAAAAAAACBFcGcevLNnzx6jjbYmbT2CpmBoCoamYGgKhqZgaEq9HkETAAD6MJgH7+Tl5ZlevXqZ2bNnm2PHjhkNtDVp6xE0BUNTMDQFQ1MwNAVDU+r1CJoAANCHwTx4R2aWZ2RkmFtvvdXk5+ebUaNGmZUrV9KkuIcmmmiiiSaafGjS1kMTAABKJXs7XSBskUjEVlZW2t27d9sXX3zRdujQwUajUXvxxRfbyZMn2wMHDnjfpK2HJppoookmmnxo0tZDEwAAOjGYB+/UvACsbcmSJXbkyJG2fv36tm7dunbYsGFeN2nroYkmmmiiiSYfmrT10AQAgE4M5sE78pvbU18A1jh8+LCdOnWq7dGjh9dN2npoookmmmiiyYcmbT00AQCgU0T+SPZUXyBM0WjUxGIxk5uba7TQ1qStR9AUDE3B0BQMTcHQFAxNqdcjaAIAQB82wIB3ZsyYYXJycowm2pq09QiagqEpGJqCoSkYmoKhKfV6BE0AAOjDnXkAAAAAAABAikhLdgAQtu3btwf6vFatWhlfm7T1CJqCoSkYmoKhKRiagqEp9XoETQAA6MOdefBOnTp14o9rvv0jkUjCc3J88uRJb5u09dBEE0000USTD03aemgCAEAn7syDd+TFXUFBgRkxYoTp27evSUtL/o+BtiZtPTTRRBNNNNHkQ5O2HpoAAFAq2dvpAmHbtWuXnTBhgj3//PNtXl6eHTNmjF23bh1NintoookmmmiiyYcmbT00AQCgE4N58NqiRYvsyJEjbf369W3Xrl3t22+/bU+ePEmT4h6aaKKJJppo8qFJWw9NAADowWAeYK2NxWL2qquustFo1O7du9dqoK1JW4+gKRiagqEpGJqCoSkYmlKvR9AEAEByRZM9zRdIpiVLlpjbb7/dFBUVmcOHD5s33njDNGzYkCbFPTTRRBNNNNHkQ5O2HpoAANCD1WLhnV27dpmZM2eaGTNmmH379pkhQ4aY0tJS07FjR5qU9tBEE0000USTD03aemgCAECniNyel+wIIEzp6emmRYsWZvjw4aa4uNgd/50LL7zQ2yZtPTTRRBNNNNHkQ5O2HpoAANCJwTx4Jxr97+zySCTi/j71x0CeP3nypLdN2npoookmmmiiyYcmbT00AQCgE9Ns4Z2tW7cabbQ1aesRNAVDUzA0BUNTMDQFQ1Pq9QiaAADQhzvzAAAAAAAAgBTBbrbwzp49e8y2bdsSnisvLze33nqrGTBggJk9e7b3Tdp6aKKJJppoosmHJm09NAEAoJTcmQf4ZNCgQfaBBx6IH1dWVtpGjRrZCy64wBYXF9v09HQ7c+ZMr5u09dBEE0000USTD03aemgCAEAnBvPgncLCQrtgwYL48QsvvGDPPfdce/z48fhx165dvW7S1kMTTTTRRBNNPjRp66EJAACdmGYL78RiMVNYWBg//u6778xNN91k0tL+vR9McXGx2bx5s9dN2npoookmmmiiyYcmbT00AQCgE4N58E6DBg3M/v3748fLli0zXbt2jR9HIhFz7Ngxr5u09dBEE0000USTD03aemgCAEAnBvPgnW7duplXX33VVFdXmzlz5phDhw6Zq6++On5+06ZNpmXLll43aeuhiSaaaKKJJh+atPXQBACAUsme5wuEbdWqVbZp06Y2IyPDRqNRO378+ITzQ4cOtXfeeafXTdp6aKKJJppoosmHJm09NAEAoFNE/kj2gCIQtj179pjS0lJz9tlnJ0zLEPPmzTMdOnQwbdq08bpJWw9NNNFEE000+dCkrYcmAAD0+fcqsYBHjhw5YpYuXWquv/56dzx27NiEdVXq1KljevXq5XWTth6aaKKJJppo8qFJWw9NAADoxGAevPPuu++639j26dPHHb/++uvmggsuMFlZWe54w4YNpnnz5mb06NHeNmnroYkmmmiiiSYfmrT10AQAgFLJnucLhO2yyy6zn3/+efy4Xr169ueff44fz5o1y3br1s3rJm09NNFEE0000eRDk7YemgAA0IndbOGdLVu2mE6dOsWPMzMzTTT63x+FLl26mHXr1nndpK2HJppoookmmnxo0tZDEwAAOjHNFt7Zv39/wroqu3fvTjhfXV2dcN7HJm09NNFEE0000eRDk7YemgAA0Ik78+CdgoICs3bt2tOeX716tfscn5u09dBEE0000USTD03aemgCAECpZM/zBcJ277332g4dOtgjR4785VxVVZU7J5/jc5O2HppoookmmmjyoUlbD00AAOgUkT+SPaAIhKmystJcdNFFJiMjw9xzzz2mqKjIPb9x40a3G9qJEyfMTz/9ZPLy8rxt0tZDE0000UQTTT40aeuhCQAApZI9mggkQ0VFhe3du7eNRqM2Eom4D3ksz9XeDc3nJm09NNFEE0000eRDk7YemgAA0Ic78+C133//3e2IJs477zzTuHHjZCepa9LWQxNNNNFEE00+NGnroQkAAD0YzAMAAAAAAABSBLvZAgAAAAAAACmCwTwAAAAAAAAgRTCYBwAAAAAAAKQIBvMAAAAAAACAFMFgHgAAAAAAAJAiGMwDAAAAAAAAUgSDeQAAAAAAAECKYDAPAAAAAAAAMKnh/wBuU5L9jvEOIAAAAABJRU5ErkJggg==",
      "text/plain": [
       "<Figure size 1000x1000 with 5 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "✅ Clustermap rendered successfully.\n"
     ]
    }
   ],
   "source": [
    "annotate_clustermap_with_metadata(grapher_cp, metadata_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "315767e8-5769-446d-9c62-8c568c77d780",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv1",
   "language": "python",
   "name": "venv1"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
