{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "77c85962-43b8-4959-bb96-d6a79000cf64",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "Hereâ€™s how you might gather information based on each category for single-cell RNA-seq (scRNA-seq) data analysis:\n",
       "\n",
       "### Preprocessing & Quality Control\n",
       "1. **Cell Ranger** - Filtering low-quality cells and mitochondrial content.\n",
       "   - [Documentation](https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger)\n",
       "   - [GitHub](https://github.com/10XGenomics/cellranger)\n",
       "\n",
       "2. **Scrublet** - Doublet detection.\n",
       "   - [Documentation](https://github.com/swolock/scrublet)\n",
       "   - [GitHub](https://github.com/swolock/scrublet)\n",
       "\n",
       "3. **scater** - Quality control and normalization.\n",
       "   - [Documentation](https://bioconductor.org/packages/release/bioc/html/scater.html)\n",
       "   - [GitHub](https://github.com/dufkot/scater)\n",
       "\n",
       "4. **DropletUtils** - Quality control of droplet-based data.\n",
       "   - [Documentation](https://bioconductor.org/packages/release/bioc/html/DropletUtils.html)\n",
       "   - [GitHub](https://github.com/MarioniLab/DropletUtils)\n",
       "\n",
       "5. **DoubletFinder**- Doublet detection in scRNA-seq.\n",
       "   - [Documentation](https://github.com/chris-mcginnis-ucsf/DoubletFinder)\n",
       "   - [GitHub](https://github.com/chris-mcginnis-ucsf/DoubletFinder)\n",
       "\n",
       "6. **SoupX** - Correcting ambient RNA contamination.\n",
       "   - [Documentation](https://github.com/constantAmateur/SoupX)\n",
       "   - [GitHub](https://github.com/constantAmateur/SoupX)\n",
       "\n",
       "7. **ScDblFinder** - Detection of doublets.\n",
       "   - [Documentation](https://bioconductor.org/packages/release/bioc/html/scDblFinder.html)\n",
       "   - [GitHub](https://github.com/plger/scDblFinder)\n",
       "\n",
       "8. **QCmetrics** - Quality control metrics for scRNA-seq data.\n",
       "   - [GitHub](https://github.com/wmacnair/CGIs)\n",
       "   \n",
       "9. **miQC** - Filtering based on mitochondrial content.\n",
       "   - [Documentation](https://bioconductor.org/packages/release/bioc/html/miQC.html)\n",
       "\n",
       "10. **scanpy** - Preprocessing and visualization.\n",
       "    - [Documentation](https://scanpy.readthedocs.io/)\n",
       "    - [GitHub](https://github.com/theislab/scanpy)\n",
       "\n",
       "### Normalization & Batch Correction\n",
       "1. **SCTransform** - Variance stabilization transformation.\n",
       "   - [Documentation](https://satijalab.org/seurat/articles/sctransform_vignette.html)\n",
       "   - [GitHub](https://github.com/satijalab/seurat)\n",
       "\n",
       "2. **Harmony** - Batch effect correction.\n",
       "   - [Documentation](https://github.com/immunogenomics/harmony)\n",
       "   - [GitHub](https://github.com/immunogenomics/harmony)\n",
       "\n",
       "3. **MNN** - Mutual nearest neighbors for batch correction.\n",
       "   - [Documentation](https://bioconductor.org/packages/release/bioc/html/batchelor.html)\n",
       "   - [GitHub](https://github.com/MarioniLab/batchelor)\n",
       "\n",
       "4. **ComBat-seq** - Counts-based batch effect correction.\n",
       "   - [Documentation](https://bioconductor.org/packages/release/bioc/html/sva.html)\n",
       "\n",
       "5. **scNormalize** - Normalization of scRNA-seq data.\n",
       "   - [GitHub](https://github.com/ewissel/scNormalize)\n",
       "\n",
       "6. **RUVSeq** - Removing unwanted variation.\n",
       "   - [Documentation](https://bioconductor.org/packages/release/bioc/html/RUVSeq.html)\n",
       "   - [GitHub](https://github.com/drisso/ruvseq)\n",
       "\n",
       "7. **Seurat** - Integration across conditions.\n",
       "   - [Documentation](https://satijalab.org/seurat/)\n",
       "   - [GitHub](https://github.com/satijalab/seurat)\n",
       "\n",
       "8. **Liger** - Integrative non-negative matrix factorization.\n",
       "   - [Documentation](https://macoskolab-website.github.io/liger/)\n",
       "   - [GitHub](https://github.com/MacoskoLab/liger)\n",
       "\n",
       "9. **zellkonverter** - Batch correction utilities.\n",
       "   - [GitHub](https://github.com/theislab/zellkonverter)\n",
       "\n",
       "10. **BBKNN** - Fast batch alignment of single-cell genomics data.\n",
       "    - [Documentation](https://github.com/Teichlab/BBKNN)\n",
       "    - [GitHub](https://github.com/Teichlab/BBKNN)\n",
       "\n",
       "### Clustering & Dimensionality Reduction\n",
       "1. **Seurat** - Implement PCA, t-SNE, and UMAP.\n",
       "   - [Documentation](https://satijalab.org/seurat/)\n",
       "   - [GitHub](https://github.com/satijalab/seurat)\n",
       "\n",
       "2. **Scanpy** - Tools for clustering and visualization.\n",
       "   - [Documentation](https://scanpy.readthedocs.io/)\n",
       "   - [GitHub](https://github.com/theislab/scanpy)\n",
       "\n",
       "3. **Monocle** - Clustering and trajectory analysis.\n",
       "   - [Documentation](http://cole-trapnell-lab.github.io/monocle-release)\n",
       "   - [GitHub](https://github.com/cole-trapnell-lab/monocle3)\n",
       "\n",
       "4. **PhenoGraph** - Community detection for clustering.\n",
       "   - [Documentation](https://github.com/jacoblevine/PhenoGraph)\n",
       "   - [GitHub](https://github.com/jacoblevine/PhenoGraph)\n",
       "\n",
       "5. **UMAP** - Manifold learning technique.\n",
       "   - [Documentation](https://umap-learn.readthedocs.io/en/latest/)\n",
       "   - [GitHub](https://github.com/lmcinnes/umap)\n",
       "\n",
       "6. **SC3** - Consensus clustering of single-cell RNA-seq data.\n",
       "   - [Documentation](https://bioconductor.org/packages/release/bioc/html/SC3.html)\n",
       "   - [GitHub](https://github.com/hemberg-lab/SC3)\n",
       "\n",
       "7. **SPRING** - Interactive visualization.\n",
       "   - [Documentation](https://kleintools.hms.harvard.edu/tools/spring.html)\n",
       "   - [GitHub](https://github.com/AllonKleinLab/SPRINGDev)\n",
       "\n",
       "8. **CIDR** - Clustering through imputation and dimensionality reduction.\n",
       "   - [Documentation](https://github.com/VCCRI/CIDR)\n",
       "   - [GitHub](https://github.com/VCCRI/CIDR)\n",
       "\n",
       "9. **pcaReduce** - Hierarchical clustering guided by PCA.\n",
       "   - [GitHub](https://github.com/JustinaZ/pcaReduce)\n",
       "  \n",
       "10. **t-SNE** - Dimensionality reduction.\n",
       "    - [Documentation](https://lvdmaaten.github.io/tsne/)\n",
       "    - [GitHub](https://github.com/lvdmaaten/bhtsne)\n",
       "\n",
       "### Differential Expression Analysis\n",
       "1. **MAST** - For single-cell gene expression.\n",
       "   - [Documentation](https://cran.r-project.org/web/packages/MAST/)\n",
       "   - [GitHub](https://github.com/RGLab/MAST)\n",
       "\n",
       "2. **DESeq2** - For RNA-seq data using shrinkage estimators.\n",
       "   - [Documentation](https://bioconductor.org/packages/release/bioc/html/DESeq2.html)\n",
       "   - [GitHub](https://github.com/mikelove/DESeq2)\n",
       "\n",
       "3. **edgeR** - Empirical analysis of digital gene expression data.\n",
       "   - [Documentation](https://bioconductor.org/packages/release/bioc/html/edgeR.html)\n",
       "\n",
       "4. **limma** - Linear models for differential expression.\n",
       "   - [Documentation](https://bioconductor.org/packages/release/bioc/html/limma.html)\n",
       "   - [GitHub](https://github.com/Bioconductor-mirror/limma)\n",
       "\n",
       "5. **Monocle** - Differential expression along trajectories.\n",
       "   - [Documentation](http://cole-trapnell-lab.github.io/monocle-release)\n",
       "   - [GitHub](https://github.com/cole-trapnell-lab/monocle3)\n",
       "\n",
       "6. **Seurat** - Finds differentially expressed genes.\n",
       "   - [Documentation](https://satijalab.org/seurat/)\n",
       "   - [GitHub](https://github.com/satijalab/seurat)\n",
       "\n",
       "7. **scDD** - Identifying differentially distributed genes.\n",
       "   - [GitHub](https://github.com/kdkorthauer/scDD)\n",
       "\n",
       "8. **SCDE** - Single-cell differential expression.\n",
       "   - [Documentation](https://hms-dbmi.github.io/scde/)\n",
       "   - [GitHub](https://github.com/hms-dbmi/scde)\n",
       "\n",
       "9. **bioconductor-muscat** - Multi-sample multi-group scRNA-seq bioinformatics.\n",
       "   - [Documentation](https://bioconductor.org/packages/release/bioc/html/muscat.html)\n",
       "   - [GitHub](https://github.com/HelenaLC/muscat)\n",
       "\n",
       "10. **DiffusionMap** - Analysis using gene expression distributions.\n",
       "    - [GitHub](https://github.com/theislab/diffusionMap)\n",
       "\n",
       "### Trajectory Inference & Pseudotime Analysis\n",
       "1. **Monocle** - Trajectory inference and visualization.\n",
       "   - [Documentation](http://cole-trapnell-lab.github.io/monocle-release)\n",
       "   - [GitHub](https://github.com/cole-trapnell-lab/monocle3)\n",
       "\n",
       "2. **Slingshot** - Inferring cell lineages.\n",
       "   - [Documentation](https://bioconductor.org/packages/release/bioc/html/slingshot.html)\n",
       "   - [GitHub](https://github.com/kstreet13/slingshot)\n",
       "\n",
       "3. **SCORPIUS** - Trajectory inference framework.\n",
       "   - [Documentation](https://github.com/rcannood/SCORPIUS)\n",
       "   - [GitHub](https://github.com/rcannood/SCORPIUS)\n",
       "\n",
       "4. **paga** - Graph abstraction for trajectories.\n",
       "   - [Documentation](https://scanpy.readthedocs.io/en/stable/)\n",
       "   - [GitHub](https://github.com/theislab/scanpy)\n",
       "\n",
       "5. **velocyto** - RNA velocity of single cells.\n",
       "   - [Documentation](http://velocyto.org/)\n",
       "   - [GitHub](https://github.com/velocyto-team/velocyto.py)\n",
       "\n",
       "6. **TSCAN** - Time-series analysis.\n",
       "   - [Documentation](https://bioconductor.org/packages/release/bioc/html/TSCAN.html)\n",
       "   - [GitHub](https://github.com/zji90/TSCAN)\n",
       "\n",
       "7. **TrajectoryNet** - Learning continuous trajectories.\n",
       "   - [GitHub](https://github.com/ardigen/TrajectoryNet)\n",
       "\n",
       "8. **Dynverse** - Suite for trajectory inference.\n",
       "   - [Documentation](https://dynverse.org)\n",
       "   - [GitHub](https://github.com/dynverse)\n",
       "\n",
       "9. **Wishbone** - Discovering bifurcating developmental trajectories.\n",
       "   - [GitHub](https://github.com/ManuSetty/wishbone)\n",
       "\n",
       "10. **Pseudocell Tracing** - Pseudotime reconstruction.\n",
       "    - [GitHub](https://github.com/cole-trapnell-lab/monocle-release)\n",
       "\n",
       "### Gene Regulatory Networks & Cell-Cell Interactions\n",
       "1. **SCENIC** - Gene regulatory network inference.\n",
       "   - [Documentation](https://pyscenic.readthedocs.io/en/latest/)\n",
       "   - [GitHub](https://github.com/aertslab/pySCENIC)\n",
       "\n",
       "2. **CellChat** - Inference of cell-cell communication.\n",
       "   - [Documentation](https://github.com/sqjin/CellChat)\n",
       "   - [GitHub](https://github.com/sqjin/CellChat)\n",
       "\n",
       "3. **NicheNet** - Ligand-target networks.\n",
       "   - [Documentation](https://github.com/saeyslab/nichenetr)\n",
       "   - [GitHub](https://github.com/saeyslab/nichenetr)\n",
       "\n",
       "4. **PIDC** - Network inference using partial information decomposition.\n",
       "   - [GitHub](https://github.com/Tchanders/NetworkInference.jl)\n",
       "\n",
       "5. **pySCENIC** - Inference of GRNs and cell states.\n",
       "   - [Documentation](https://pyscenic.readthedocs.io/en/latest/)\n",
       "   - [GitHub](https://github.com/aertslab/pySCENIC)\n",
       "\n",
       "6. **GENIE3** - Ensemble-based network inference.\n",
       "   - [GitHub](https://github.com/aertslab/GENIE3)\n",
       "\n",
       "7. **Inferelator** - Network inference.\n",
       "   - [Documentation](https://inferelator.readthedocs.io/en/latest/)\n",
       "   - [GitHub](https://github.com/flatironinstitute/inferelator)\n",
       "\n",
       "8. **SCENIC+** - Enhanced SCENIC workflows.\n",
       "   - [GitHub](https://github.com/aertslab/SCENICPlus)\n",
       "\n",
       "9. **CytoTalk** - Cellular communication analysis.\n",
       "   - [GitHub](https://github.com/smangul1/CytoTalk)\n",
       "\n",
       "10. **CellPhoneDB** - Interaction database.\n",
       "    - [Documentation](https://www.cellphonedb.org/documentation)\n",
       "    - [GitHub](https://github.com/Teichlab/cellphonedb)\n",
       "\n",
       "### Visualization & Interpretation\n",
       "1. **Cellxgene** - Interactive data visualization.\n",
       "   - [Documentation](https://chanzuckerberg.github.io/cellxgene/)\n",
       "   - [GitHub](https://github.com/chanzuckerberg/cellxgene)\n",
       "\n",
       "2. **Loupe Browser** - Visualization of 10x Genomics data.\n",
       "   - [Documentation](https://support.10xgenomics.com/single-cell-gene-expression/software/visualization/latest/what-is-loupe-browser)\n",
       "\n",
       "3. **Cytosplore** - High-dimensional single-cell data exploration.\n",
       "   - [Documentation](https://www.cytosplore.org/)\n",
       "   - [GitHub](https://github.com/Cwseitz/Cytosplore-Feedback)\n",
       "\n",
       "4. **SPRING** - Visualization of single-cell data.\n",
       "   - [Documentation](https://kleintools.hms.harvard.edu/tools/spring.html)\n",
       "   - [GitHub](https://github.com/AllonKleinLab/SPRINGDev)\n",
       "\n",
       "5. **Seurat** - Integrated visualization tools.\n",
       "   - [Documentation](https://satijalab.org/seurat/)\n",
       "   - [GitHub](https://github.com/satijalab/seurat)\n",
       "\n",
       "6. **Stream** - Visualization of trajectories.\n",
       "   - [Documentation](https://hms-dbmi.github.io/steam/)\n",
       "   - [GitHub](https://github.com/pinellolab/STREAM)\n",
       "\n",
       "7. **scVis** - Dimensionality reduction for visualization.\n",
       "   - [GitHub](https://github.com/shahcompbio/scvis)\n",
       "\n",
       "8. **giotto** - Tools for spatial data.\n",
       "   - [Documentation](https://rubd.github.io/Giotto_site/)\n",
       "   - [GitHub](https://github.com/RubD/Giotto)\n",
       "\n",
       "9. **RCy3** - Interface for Cytoscape.\n",
       "   - [Documentation](https://bioconductor.org/packages/release/bioc/html/RCy3.html)\n",
       "   - [GitHub](https://github.com/cytoscape/RCy3)\n",
       "\n",
       "10. **ComplexHeatmap** - Advanced heatmaps.\n",
       "    - [Documentation](https://jokergoo.github.io/ComplexHeatmap-reference/book/)\n",
       "    - [GitHub](https://github.com/jokergoo/ComplexHeatmap)\n",
       "\n",
       "These resources should significantly help in analyzing scRNA-seq data."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from agents import Agent, Runner\n",
    "from IPython.display import display, Markdown\n",
    "import os\n",
    "from openai import OpenAI\n",
    "\n",
    "client = OpenAI()\n",
    "os.environ['OPENAI_API_KEY']=API_KEY\n",
    "import nest_asyncio\n",
    "nest_asyncio.apply()\n",
    "\n",
    "\n",
    "prompts=\"\"\"Category-Specific Deep Search Prompts\n",
    "1. Preprocessing & Quality Control\n",
    "\"Find at least 10 tools for preprocessing and quality control of single-cell RNA-seq (scRNA-seq) data. Include tools for filtering low-quality cells, doublet detection, mitochondrial read filtering, and dataset quality assessment. Provide links to GitHub or official documentation.\"\n",
    "\n",
    "2. Normalization & Batch Correction\n",
    "\"List at least 10 bioinformatics tools for normalization and batch effect correction in single-cell RNA sequencing (scRNA-seq) data. Include methods like SCTransform, Harmony, and MNN. Provide links to documentation and GitHub repositories.\"\n",
    "\n",
    "3. Clustering & Dimensionality Reduction\n",
    "\"Search for at least 10 scRNA-seq tools specialized in clustering and dimensionality reduction. Include methods using PCA, t-SNE, UMAP, and graph-based clustering approaches. Provide official sources and GitHub repositories.\"\n",
    "\n",
    "4. Differential Expression Analysis\n",
    "\"Identify 10+ software packages for performing differential expression analysis in single-cell RNA sequencing (scRNA-seq). Compare statistical models used in each tool. Include links to their GitHub repositories or official documentation.\"\n",
    "\n",
    "5. Trajectory Inference & Pseudotime Analysis\n",
    "\"List at least 10 tools for trajectory inference and pseudotime analysis in single-cell RNA sequencing (scRNA-seq) datasets. Include methods such as Monocle, Slingshot, and SCORPIUS. Provide links to each toolâ€™s documentation and GitHub.\"\n",
    "\n",
    "6. Gene Regulatory Networks & Cell-Cell Interactions\n",
    "\"Find at least 10 computational methods for constructing gene regulatory networks and inferring cell-cell communication from single-cell RNA sequencing (scRNA-seq) data. Include approaches like SCENIC, CellChat, and NicheNet. Provide documentation links.\"\n",
    "\n",
    "7. Visualization & Interpretation\n",
    "\"Gather at least 10 tools designed for visualization and exploration of single-cell RNA sequencing (scRNA-seq) datasets. Include interactive visualization software for gene expression, clustering, and trajectory analysis. Provide GitHub or official tool documentation.\"\"\"\n",
    "\n",
    "response = client.responses.create(\n",
    "    model=\"gpt-4o\",  # or another supported model\n",
    "    input=prompts,\n",
    "    tools=[\n",
    "        {\n",
    "            \"type\": \"web_search\"\n",
    "        }\n",
    "    ]\n",
    ")\n",
    "\n",
    "# import json\n",
    "# print(json.dumps(response.output, default=lambda o: o.__dict__, indent=2))\n",
    "# response.output is a list of ResponseOutputMessage objects\n",
    "\n",
    "# Extract all text from response content correctly\n",
    "output_texts = [content.text for message in response.output for content in message.content]\n",
    "\n",
    "# Combine into one string while maintaining the original format\n",
    "formatted_text = \"\\n\\n\".join(output_texts)\n",
    "\n",
    "# Display as formatted markdown in Jupyter Notebook\n",
    "display(Markdown(formatted_text))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "61472a21-8232-4bda-aa44-b68466ac59ea",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv1",
   "language": "python",
   "name": "venv1"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
